---
title: "FunctionalEnrichmentAnalysis"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction
After DE analysis, functional enrichment analysis is often performed to interpret the results of RNA-seq. 

`DEbPeak` provides three kinds of functional enrichment analysis based on `clusterProfiler`:

* Gene ontology (GO) analysis with DEGs 
* Kyoto Encyclopedia of Genes and Genomes (KEGG) pathway enrichment analysis with DEGs
* Gene Set Enrichment Analysis (GSEA)

<hr />

## Example data
We will used the results output by "Quality Control" vignette.

```{r load_example}
# library
suppressWarnings(suppressMessages(library(DESeq2)))
suppressWarnings(suppressMessages(library(DEbPeak)))
# load data
load(file = "/home/songyabing/R/learn/tmp/DEbPeak/example.RData")
```

<hr />

## Differential Expression Analysis
We will use `DESeq2` as example:
```{r degs_deseq2, warning=FALSE}
# set control level
dds$condition <- relevel(dds$condition, ref = "WT")
# conduct differential expressed genes analysis
dds <- DESeq(dds)
# extract results
dds.results <- results(dds,contrast=c("condition",'KO','WT'))
dds.results.ordered <- dds.results[order(dds.results$log2FoldChange,decreasing = TRUE),]
head(dds.results.ordered)
```

<hr />

## GO enrichment
```{r go, message=FALSE, warning=FALSE}
# save results to working directory
# ConductFE(deres = dds.results.ordered, signif = "pvalue", l2fc.threshold = 0.3,enrich.type = "GO")
all.go.results=ConductFE(deres = dds.results.ordered, signif = "pvalue", l2fc.threshold = 0.3,enrich.type = "GO",str.width = 50, save = F)
```

### Up regulated genes
#### Overview
```{r go_up_overview, warning=FALSE}
up.go.res=all.go.results[["UP"]][["GO"]]
up.go.res.table=up.go.res[["table"]]
head(up.go.res.table)
```

<hr />

#### Visualization
```{r go_up_plot,fig.height = 12, fig.width = 8, fig.align = "center", warning=FALSE}
up.go.res[["plot"]]
```

<hr />

### Down regulated genes
#### Overview
```{r go_down_overview, warning=FALSE}
down.go.res=all.go.results[["DOWN"]][["GO"]]
down.go.res.table = down.go.res[["table"]]
head(down.go.res.table)
```

<hr />

#### Visualization
```{r go_down_plot,fig.height = 12, fig.width = 8, fig.align = "center", warning=FALSE}
down.go.res[["plot"]]
```

<hr />

## KEGG enrichment
```{r kegg, message=FALSE, warning=FALSE}
# save results to working directory
# ConductFE(deres = dds.results.ordered, signif = "pvalue", l2fc.threshold = 0.3,enrich.type = "KEGG")
all.kegg.results=ConductFE(deres = dds.results.ordered, signif = "pvalue", l2fc.threshold = 0.3,enrich.type = "KEGG", str.width = 50, save = F)
```

### Up regulated genes
#### Overview
```{r kegg_up_overview, warning=FALSE}
up.kegg.res=all.kegg.results[["UP"]][["KEGG"]]
up.kegg.res.table=up.kegg.res[["table"]]
head(up.kegg.res.table)
```

<hr />

#### Visualization
```{r kegg_up_plot,fig.height = 6, fig.width = 8, fig.align = "center", warning=FALSE}
up.kegg.res[["plot"]]
```

<hr />

### Down regulated genes
#### Overview
```{r kegg_down_overview, warning=FALSE}
down.kegg.res=all.kegg.results[["DOWN"]][["KEGG"]]
down.kegg.res.table=down.kegg.res[["table"]]
head(down.kegg.res.table)
```

<hr />

#### Visualization
```{r kegg_down_plot,fig.height = 6, fig.width = 8, fig.align = "center", warning=FALSE}
down.kegg.res[["plot"]]
```

<hr />

## GSEA
Gene Set Enrichment Analysis (GSEA) is a computational method that determines whether an a priori defined set of genes shows statistically significant, concordant differences between two biological states(e.g. phenotypes). **Different from GO and KEGG based on DEGs, GSEA is based on whole gene expression matrix without prior gene filtering**.

While GSEA requires priori defined set of genes, for human, we can download gmt file from [MSigDB](http://www.gsea-msigdb.org/gsea/downloads.jsp), for other species, we can obtain gene sets via [msigdbr](https://igordot.github.io/msigdbr/reference/msigdbr.html).

### All available gene sets
```{r msigdbr, warning=FALSE}
library(msigdbr)
# list all possible species
msigdbr_species()
# create gene sets
m_t2g <- msigdbr(species = "Mus musculus", category = "C5", subcategory = "BP") %>% 
  dplyr::select(gs_name, entrez_gene)
head(m_t2g)
```

<hr />

### GSEA
```{r conduct_gsea, message=FALSE, warning=FALSE}
gsea.results=ConductGSEA(deres = dds.results.ordered,gmt.file = NULL,gene.sets = m_t2g,save = F)
```

<hr />

### Overview
```{r gsea_overview, warning=FALSE}
gsea.results.table=gsea.results[["table"]]
head(gsea.results.table)
```

<hr />

### Visualization
```{r gsea_plot, fig.height = 6, fig.width = 8, fig.align = "center", warning=FALSE}
# empty GSEA results, do not visualize
# VisGSEA(gsea.res = gsea.results$gsea, gs.id = c(1,2))
```

<hr />

## Session info
```{r session}
sessionInfo()
```

<hr />
