---
title: "IntegrateATACDE"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction

[Eukaryotic chromatin is tightly packaged into nucleosomes, and the positioning of nucleosomes can regulate gene expression by changing the in vivo availability of binding sites to transcription factors (TFs)](https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/1756-8935-7-33). Assay of Transposase Accessible Chromatin sequencing (ATAC-seq) is a technique used to assess genome-wide chromatin accessibility. It is widely used to identify cis-regulatory regions (e.g. promoters and enhancers). Thus, combining ATAC-seq with RNA-seq can be used to understanding the role of cis-regulatory regions in regulating gene expression.

Different from those in [Integrate RNA-seq with ATAC-seq (intersection)](https://showteeth.github.io/DEbPeak/articles/IntegrateATAC.html), we will integrate RNA-seq and ATAC-seq based on differential expression analysis results. In detail, **integrating differentially expressed genes with differentially accessible peaks**.

**In this vignette, we will focus on promoter regions.** For the enhancer regions, we will talk about in [Integrate RNA-seq with ChIP-seq (differential analysis)](https://showteeth.github.io/DEbPeak/articles/IntegrateChIPDE.html).

<hr />

## Example data

The data used here are from [In vivo CD8+ T cell CRISPR screening reveals control by Fli1 in infection and cancer](https://doi.org/10.1016/j.cell.2021.02.019):

* **RNA-seq data**: the sgCtrl vs sgFli1 RNA sequencing at D8 Cl13 p.i., the raw data are stored in [GSE149838](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149838)
* **ATAC-seq data**: sgCtrl vs sgFli1 ATAC sequencing at D9 Cl13 p.i., the raw data are stored in [GSE149836](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149836)

<hr />

## Load packages

```{r motif_load_example}
# library
suppressWarnings(suppressMessages(library(DEbPeak)))
```

<hr />

## Prepare ATAC-seq counts
The following is an example with count matrix, if you only have bam files and peak files, please refer to [Integrate RNA-seq with ChIP-seq (differential analysis)](https://showteeth.github.io/DEbPeak/articles/IntegrateChIPDE.html).

Format the count matrix:
```{r debatac_atac_prepare, eval=FALSE}
# the GSE149836_combUnionReadsWithLabels.txt is downloaded from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149836
atac.counts = read.table(file = "GSE149836_combUnionReadsWithLabels.txt", sep = "\t", header = T)
# the first three columns should be "CHR", "START", "END"
colnames(atac.counts) = c("CHR", "START", "END", "Fli1KO_1","Fli1KO_2","Fli1KO_3","Fli1KO_4","WT_1","WT_2")
# save file
write.table(x = atac.counts, file = "RA_ATAC_counts_raw.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

Annotate the count matrix:
```{r debatac_atac_prepare2, eval=FALSE}
# this will save count matrix and annotated results to consensus_peak_matrix.txt and consensus_peak_anno.txt
PeakMatrix(meta.file = "RA_ATAC_counts_raw.txt",
           count.matrix = TRUE, species = "Mouse",  seq.style = "UCSC",
           up.dist = 2000, down.dist = 2000)
```

Prepare metadata:
```{r debatac_atac_prepare3, eval=FALSE}
atac.meta = data.frame(SampleName = c("Fli1KO_1","Fli1KO_2","Fli1KO_3","Fli1KO_4","WT_1","WT_2"),
                       condition = c(rep("KO", 4), rep("WT", 2))) %>%
  tibble::column_to_rownames(var = "SampleName")
# save file
write.table(x = atac.meta, file = "RA_ATAC_mata_raw.txt", sep = "\t", quote = FALSE, row.names = TRUE)
```

<hr />

## Differential Analysis 

To simplify the steps of obtaining differential expression analysis results, `DEbPeak` provides `ConductDESeq2` to perfrom all steps of differential expression analysis, including **quality control**, **principal component analysis**, **differential expression analysis** and **functional enrichment analysis**.

### ATAC-seq
With count matrix, we will perform differential analysis:
```{r debatac_atac_da_peak}
# RA_ATAC_count.txt is above consensus_peak_matrix.txt
peak.matrix.file <- system.file("extdata", "RA_ATAC_count.txt", package = "DEbPeak")
# RA_ATAC_meta.txt is above RA_ATAC_mata_raw.txt
peak.meta.file <- system.file("extdata", "RA_ATAC_meta.txt", package = "DEbPeak")
ConductDESeq2(count.matrix.file = peak.matrix.file, meta.file = peak.meta.file, data.type = "ATAC",
              gene.type = "SYMBOL", outlier.detection = F, min.count = 0,
              peak.anno.key = "All", group.key = "condition", ref.group = "WT",
              out.folder = '/home/songyabing/R/learn/tmp/DEbPeak/ATACDE',
              signif = "padj", signif.threshold = 0.05, l2fc.threshold = 0, gmt.file = '')
```


The structure of result directory:
```{bash debatac_atac_da_peak_struc}
tree /home/songyabing/R/learn/tmp/DEbPeak/ATACDE
```

Load the data:
```{r debatac_atac_da_peak_data}
dds.peak.results.ordered = read.csv( "/home/songyabing/R/learn/tmp/DEbPeak/ATACDE/DA/Condition_KO_WT_all.csv", row.names = 1)
head(dds.peak.results.ordered)
```

<hr />

### RNA-seq 

`DEbPeak` has prepared the differential expression analysis results of RNA-seq:
```{r debatac_rna_deg}
rna.diff.file <- system.file("extdata", "RA_RNA_diff.txt", package = "DEbPeak")
rna.diff <- read.table(file = rna.diff.file, header = TRUE, sep = "\t")
```

<hr />

## Integrate ATAC-seq and RNA-seq

In this step, we will integrate ATAC-seq and RNA-seq to explore the role of chromatin structure of **promoter region** in regulating target gene expression.

### Integrate
```{r debatac_integrate}
debatac.res <- DEbPeak(de.res = rna.diff, peak.res = dds.peak.results.ordered, peak.mode = "diff", peak.anno.key = "Promoter",
                       l2fc.threshold = 0, peak.l2fc.threshold = 0, species = "Mouse", merge.key = "SYMBOL")
head(debatac.res)
```

<hr />

### Integrate summary

#### Venn plot

```{r debatac_integrate_plot, fig.height = 4, fig.width = 4, fig.align = "center", warning=FALSE}
# DE and ATAC venn plot
# debatac.plot = PlotDEbPeak(de.peak = debatac.res, peak.type = "ATAC", peak.mode = "diff", show_percentage = FALSE)
# debatac.plot
debatac.plot = InteVenn(inte.res = debatac.res, inte.type = "DEbPeak", peak.mode = "diff",
                        peak.type = "ATAC", show_percentage = FALSE)
debatac.plot
```

#### Quadrant diagram

```{r debatac_integrate_quad_plot, fig.height = 8, fig.width = 6, fig.align = "center", warning=FALSE}
label.df = data.frame(Gene = c("Ccl3", "Ccl5", "Cd28", "Cx3cr1", "Prdm1", "Tcf7", "Slamf6", "Id3", "Cxcr5"))
quad.plot = InteDiffQuad(inte.res = debatac.res, inte.type = "DEbPeak", 
                         label.df = label.df, label.color = "green")
quad.plot
```

<hr />

## Functional enrichment

There are **eight categories** for users to choose to perform functional enrichment: `Down_Up`, `Up_Up`, `Down_Down`, `Up_Down`, `RNAUp`, `RNADown`, `PeakUp`, `PeakDown`. Here, we will use `Up_Up` and `Down_Down` as examples.

### Both up-regulated

```{r debatac_integrate_fe_up}
# debatac.up.fe.results = DEbPeakFE(de.peak = debatac.res, peak.fe.key = "Up_Up", 
#                                   gene.type = "ENTREZID", species="Mouse",save = F)
debatac.up.fe.results = InteFE(inte.res = debatac.res, fe.key = "Up_Up", inte.type = "DEbPeak",
                               gene.type = "ENTREZID", species="Mouse",save = F)
```

The results:
```{r debatac_integrate_fe_up_show, warning=FALSE, fig.height = 20, fig.width = 8, fig.align = "center"}
head(debatac.up.fe.results[["GO"]][["table"]])

debatac.up.fe.results[["GO"]][["plot"]]
```

<hr />

### Both down-regulated

```{r debatac_integrate_fe_down}
# debatac.down.fe.results = DEbPeakFE(de.peak = debatac.res, peak.fe.key = "Down_Down", 
#                                     gene.type = "ENTREZID", species="Mouse",save = F)
debatac.down.fe.results = InteFE(inte.res = debatac.res, fe.key = "Down_Down", inte.type = "DEbPeak",
                                 gene.type = "ENTREZID", species="Mouse",save = F)
```

The results:
```{r debatac_integrate_fe_down_show, warning=FALSE, fig.height = 20, fig.width = 8, fig.align = "center"}
head(debatac.down.fe.results[["GO"]][["table"]])

debatac.down.fe.results[["GO"]][["plot"]]
```

<hr />

## Motif enrichment

[Different from ChIP-seq which profile specific proteinâ€“DNA interactions (depends on the antibody provided), ATAC-seq can proflie global chromatin accessibility](https://link.springer.com/article/10.1186/s43556-020-00009-w). Thus, the peaks of ATAC-seq may originate from hundreds of TFs. To reveal the possible TFs involved, motif enrichment analysis is a good solution.

As pointed above, there are **eight categories** for users to choose to motif enrichment, we will use `Up_Up` as example.
```{r debatac_integrate_motif_up, eval = FALSE}
# debatac.res: similar to debatac.res
# this step is time consuming!
FindMotif(inte.res = debatac.res, peak.anno.res = NULL,
          peak.motif.key = "Up_Up", peak.mode = "diff", homer.motif.path = '~/anaconda3/bin/findMotifsGenome.pl',
          out.folder = "/path/to/out/folder", other.paras = "-len 8,10,12 -size -100,50 -S 25")
```

<hr />

## Session info

```{r session}
sessionInfo()
```

<hr />


