---
title: "QualityControl"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction
`DEbPeak` provides two aspects of Quality Control (QC):

* **QC on count matrix**: 
  - Proportion of genes detected in different samples under **different CPM thresholds**
  - The **saturation of the number of genes detected**.
* **QC on samples**: 
  - **Euclidean distance** and **pearson correlation coefficient** of samples across different conditions
  - **sample similarity** on selected principal components (**check batch information and conduct batch correction**)
  - **outlier detection** with robust PCA

<hr />

## Example data
The data used here are RNA-seq data of the external granule layer in the cerebellum of control and conditional SnoN knockout mice, the raw data are stored in [GSE120279](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120279), they are used in [Robust principal component analysis for accurate outlier sample detection in RNA-seq data](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03608-0).

First, we create `DESeqDataSet` object with above dataset:
```{r loadData, warning=FALSE}
suppressWarnings(suppressMessages(library(DESeq2)))
suppressWarnings(suppressMessages(library(DEbPeak)))
count.file <- system.file("extdata", "snon_count.txt", package = "DEbPeak")
meta.file <- system.file("extdata", "snon_meta.txt", package = "DEbPeak")
count.matrix <- read.table(file = count.file, header = T, sep = "\t")
head(count.matrix)
meta.info <- read.table(file = meta.file, header = T)
head(meta.info)
dds <- DESeq2::DESeqDataSetFromMatrix(countData = count.matrix, colData = meta.info, design = ~condition)
dds
```

<hr />
  
## QC on count matrix
### CPM threashold
This plot shows proportion of genes detected in different samples at different CPM thresholds. Theoretically, all samples should be similarly distributed, so as to avoid false positive results obtained from detection problems.
```{r qc_counts_cpm, warning=FALSE,fig.height = 6, fig.width = 6, fig.align = "center"}
CountQC(deobj = dds, group.key = "condition", type = "cpm")
```

<hr />

### sequencing saturation
This plot shows the proportion of genes detected at different sequencing depths. If the proportion is too low, it indicates that many genes are not detected and therefore some important information may be missing, and increasing the sequencing volume can effectively solve this problem.
```{r qc_counts_saturation, warning=FALSE,fig.height = 6, fig.width = 6, fig.align = "center"}
CountQC(deobj = dds, group.key = "condition", type = "saturation")
```

<hr />

## QC on samples
QC on samples including four aspects:

* Euclidean distance and pearson correlation coefficient: sample similarity on **the entire transcriptome**.
* PCA: sample similarity on **selected principal components**.
* outlier detection with robust PCA
* batch correction

### Euclidean distance and pearson correlation coefficient
```{r qc_samples_sr, warning=FALSE,fig.height = 5, fig.width = 10, fig.align = "center"}
SampleRelation(deobj = dds,transform.method = "rlog",anno.key = "condition")
```

<hr />

### PCA
```{r qc_samples_pca, warning=FALSE,fig.height = 8, fig.width = 6, fig.align = "center"}
qc.pca.res=PCA(deobj = dds,transform.method = "rlog")
PCAtools::biplot(qc.pca.res,x="PC1",y="PC2",colby="condition",legendPosition="bottom")
```

<hr />

### outlier detection with robust PCA
```{r qc_samples_outlier, warning=FALSE,fig.height = 6, fig.width = 7, fig.align = "center"}
outlier.res=OutlierDetection(deobj = dds,var.genes = NULL,transform.method = "rlog")
outlier.res$outlier
outlier.res$plot
```

For batch correction and outlier detection, you can use the following codes:
```{r qc_samples_box, warning=FALSE,eval=FALSE}
# batch effect correction
# this is a demo scripts
batch.res=QCPCA(deobj = dds,var.genes = NULL,remove.sample=NULL,transform.method = "rlog",batch = "cell",
                colby = "dex", outlier.detection = F)
batch.res$plot
# outlier detection
outlier.res=QCPCA(deobj = dds,var.genes = NULL,remove.sample=NULL,transform.method = "rlog",
                  outlier.detection = T,rpca.method = "PcaGrid")
outlier.res$plot
# the final dds
dds=outlier.res$deobj # or outlier.res$plot
```

<hr />

## Save results
Save results for downstream analysis:
```{r save_image}
save.image(file = "/home/songyabing/R/learn/tmp/DEbPeak/example.RData")
```


<hr />

## Session info
```{r session}
sessionInfo()
```

<hr />
