---
title: "<center><div class='report-title'>DEbPeak - Explore, analyze, visualize RNA-seq data and unravel the regulation of gene expression</div></center>"
output: 
  html_document:
    number_sections: TRUE
    toc: true
    toc_depth: 4
    toc_float:
      collapsed: true
      smooth_scroll: true
    theme: readable
    css: /home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/style.css
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

`DEbPeak` is a **web server** and **[R package](https://github.com/showteeth/DEbPeak)**, which provides two main functions:

* **the exploration, analysis, and visualization of RNA-seq data**
* **integration of RNA-seq with ChIP-seq and ATAC-seq** to unravel the detailed regulation network of gene expression.

For **RNA-seq data**, `DEbPeak` provides comprehensive analysis including:

* **Quality Control (QC)**: QC on count matrix and samples.
* **Principal Component Analysis (PCA)**: Visualize the PCA results, analyze the genes with large loadings of specific principal component.
* **Differential Expression Analysis (DEA)**: Get differentially expressed genes (DEGs) and visualize the results interactively or download PDF.
* **Functional Enrichment Analysis (FEA)**: GO enrichment analysis, KEGG enrichment analysis, Gene Set Enrichment Analysis.
* **Utils** : Useful functions when dealing with RNA-seq data, including **gene name conversion (gene symbol, ENSEMBL gene ID, ENTREZ ID)** and **count normalization(DESeq2’s median of ratios, TMM, CPM, TPM, RPKM)**.

For **peak-related data (ChIP-seq and ATAC-seq)**, `DEbPeak` provides comprehensive analysis including:

* **Integrate with ChIP-seq (DEbChIP)**: Get consensus peaks from replicates, visualize peak profile, peak annotation, integrate with RNA-seq, integration summary, GO enrichment on integrated results.
* **Integrate with ATAC-seq (DEbATAC)**: Get consensus peaks from replicates, visualize peak profile, peak annotation, integrate with RNA-seq, integration summary, GO enrichment on integrated results, find motif on integrated results.
* **Integrate with ChIP-seq and ATAC-seq (DEbCA)**: Integrate RNA-seq, ChIP-seq and ATAC-seq based on the peak annotation results of DEbChIP and DEbATAC, integration summary, GO enrichment on integrated results.

# DEbPeak Framework
In general, `DEbPeak` contains five functional modules in web server and R package, including **INPUT**, **Quality Control**, **Analysis**, **Visualization**, **Utils**.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/00_DEbPeak_framework.png"  title="DEbPeak_framework"  alt="DEbPeak_framework" />
</div>

# DEbPeak UI
All above analysis are implemented as sidebar menu items and tabs, and the structure of `DEbPeak` is as follows:

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_000_Structure_sidebar.png"  title="DEbPeak_Structure"  alt="DEbPeak_Structure" />
</div>

Some options for analysis (**Options (Advanced)**) and visualization (**Adjust plots**) are **collapsed by default**, users can **click the `+` on the right to expand**.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_001_DEbPeak.gif"  title="DEbPeak_help"  alt="DEbPeak_help" />
</div>

# Upload data
To perfrom RNA-seq data analysis, users should upload their count matrix and sample metadata, **all these files are tab-separated**. **For count matrix, the rownames are genes, the colnames are samples. For sample metadata, the rownames are samples, the colnames are sample attributes of samples, such as sample conditions and batch information**.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_01_Data.png"  title="DEbPeak_data"  alt="DEbPeak_data" />
</div>

# Quality Control (QC)

`DEbPeak` provides two aspects of Quality Control (QC):

* **QC on count matrix**: 
  - Proportion of genes detected in different samples under **different CPM thresholds**, such as CPM>0, CPM>1, CPM>2, CPM>5, CPM>10. The **horizontal lines** are the corresponding percentage of features with those CPM in at least one of the samples.  In the upper side of the plot, the **sequencing depth of each sample (in million reads)** is given. The output is in <mark>Count QC-CPM</mark>.
  - The **saturation of the number of genes detected**. The **vertical lines** are the current sequencing depth of each sample (in million reads). The output is in <mark>Count QC-Saturation</mark>.
* **QC on samples**: 
  - **Euclidean distance** and **pearson correlation coefficient** of samples across different conditions, the output is in <mark>Sample relationship</mark>.
  - **Sample similarity** on selected principal components to **check batch information and conduct batch correction**. To perform batch effect correction with `ComBat_seq`, specify batch column with `Batch by` (when `Batch by` is the same as `Group by`, do not perform batch effect correction) .The output is in <mark>Sample PCA</mark>
  - **Outlier detection** with robust PCA. Samples in the red lines are considered normal samples, and samples outside the red lines are considered outlier samples.The output is in <mark>Outlier detection</mark>

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_02_QC.png"  title="DEbPeak_QC"  alt="DEbPeak_QC" />
</div>


# Principal Component Analysis (PCA)
Principal component analysis (PCA) is an unsupervised linear dimensionality reduction algorithm, and it can [retain most of the variation in the data set](https://www.nature.com/articles/nbt0308-303). Based on these properties, PCA has two main uses in RNA-seq analysis:

* Quality control: 
  - Evaluate sample similarity across different conditions
  - Check batch information
  - Detect outliers
* Reveal biologically relevant information
  - Find gene expression signatures
  - Reveal underlying population heterogeneity, such as cell differentiation trajectory

For **Quality Control**, we have already covered in the `Quality Control (QC)` sectiion, here, we will focus on the "Reveal biologically relevant information"!

## PCA Basic
This section will include three plots to inspect PCA results:

* **Scree plot**: A Scree Plot is a simple line plot that shows the total amount of variance that can be explained by each individual PC (Y-axis shows explained variance, X-axis shows the number of PCs). It can be used to **determine the number of PCs to be explored for downstream analysis**. Here, we created a **cumulative scree plot** based on [PCAtools](https://github.com/kevinblighe/PCAtools), the red dashed line represents 90% explained variance. The output is in <mark>Scree Plot</mark>.
* **Biplot**: Biplot contains informations from [two aspects](https://blog.bioturing.com/2018/06/18/how-to-read-pca-biplots-and-scree-plots/). The output is in <mark>Biplot</mark>.
    - sample similarity (point)
    - how strongly each gene influences a principal component (vector)
        + the vector will have projections on PCs, and the project values on each PC show how much weight they have on that PC. Positive values indicate that a vector/gene and a principal component are positively correlated whereas negative values indicate a negative correlation.
        + the angle between vectors indicates how vectors/genes correlate with one another: a small angle implies positive correlation, a large one suggests negative correlation, and a 90 degree angle indicates no correlation between two vectors/genes.
* The PC pairs plot (based on [PCAtools](https://github.com/kevinblighe/PCAtools)) will show sample similarity across different PC combination. The output is in <mark>PC pairs plot</mark>.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_03_PCA_basic.png"  title="DEbPeak_PCA_basic"  alt="DEbPeak_PCA_basic" />
</div>

## Loading
PCA loadings are the coefficients of the linear combination of the original variables from which the principal components (PCs) are constructed, which are simply the weights of the original variables. Here, lodding indicates how strongly each gene influences a principal component (Similar to vector in `Biplot`, but here we will evaluate each individual PC). Loadings range from -1 to 1, a high absolute value indicates that the gene strongly influences the component, and the sign of a loading indicates whether a whether a gene and a principal component are positively or negatively correlated. 

The lodding-related analysis in `DEbPeak` contains four aspects: 

* **Loading bar plot**: Visualize the genes with top n positive or negative loaddings of PCs. The output is in <mark>Loading bar plot</mark>.
* **Loading heatmap**: Log2-normalized counts of the genes with top n positive or negative loaddings of PCs across samples. The output is in <mark>Loading heatmap</mark>.
* **GO enrichment**: GO enrichment on the genes with top n positive or negative loaddings of PCs. The support species (specified by `Species (GO)`) are **Human**, **Mouse**, **Rat**, **Fly**, **Arabidopsis**, **Yeast**, **Zebrafish**, **Worm**, **Bovine**, **Pig**, **Chicken**, **Rhesus**, **Canine**, **Xenopus**, **Anopheles**, **Chimp**, **E coli strain Sakai**, **Myxococcus xanthus DK 1622**, the data are from [Bioconductor AnnotationData Packages](https://bioconductor.org/packages/3.16/data/annotation/). The output is in <mark>Positive loading genes (GO table), Negative loading genes (GO table), Positive loading genes (GO plot), Negative loading genes (GO plot)</mark>.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_04_PCA_loadding.png"  title="DEbPeak_PCA_loadding"  alt="DEbPeak_PCA_loadding" />
</div>

## 3D PCA
In this section, `DEbPeak` enables users to visualize three PCs simultaneously, users can change the viewing angle by dragging.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_05_PCA_3D.png"  title="DEbPeak_PCA_3D"  alt="DEbPeak_PCA_3D" />
</div>

# Differential Expression Analysis (DEA)
## Results
`DEbPeak` utilizes [DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html) to perfrom DEA and get differentially expressed genes (DEGs). The support species (specified by `Species`) are **Human**, **Mouse**, **Rat**, **Fly**, **Arabidopsis**, **Yeast**, **Zebrafish**, **Worm**, **Bovine**, **Pig**, **Chicken**, **Rhesus**, **Canine**, **Xenopus**, **Anopheles**, **Chimp**, **E coli strain Sakai**, **Myxococcus xanthus DK 1622**, the data are from [Bioconductor AnnotationData Packages](https://bioconductor.org/packages/3.16/data/annotation/).

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_06_DEA_analyze.png"  title="DEbPeak_DEA_analyze"  alt="DEbPeak_DEA_analyze" />
</div>

## Visualization (interactive)
With these DEGs, `DEbPeak` provides **four differents charts** to visualize these DEGs interactively (To enable users better inspect the results, users can hover over the interested point to view detailed information.):

* **VolcanoPlot**: This plot shows statistical significance (specified by `Signif criterion` in **Results** section) versus magnitude of change (**log2FoldChange**). It enables quick visual identification of genes with large fold changes that are also statistically significant. These may be the most biologically significant genes. The output is in <mark>Volcano Plot</mark>.
* **ScatterPlot**: This plots shows the average expression signal of experimental samples (**log2(mean of normalized counts (RX))**) versus the average expression signal of control samples (**log2(mean of normalized counts (NF))**). The output is in <mark>Scatter Plot</mark>.
* **MAPlot**: This plot shows magnitude of change (**log2FoldChange**, M, on the y-axis) versus the average expression signal (**log2(mean of normalized counts)**, A, on the x-axis). The output is in <mark>MA Plot</mark>.
* **RankPlot**: This plot shows magnitude of change (**log2FoldChange**, M, on the y-axis) versus the order of the change. The output is in <mark>Rank Plot</mark>.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_07_DEA_visualize_interactive.png"  title="DEbPeak_DEA_visualize_interactive"  alt="DEbPeak_DEA_visualize_interactive" />
</div>

## Visualization
After interactively inspect DEGs, `DEbPeak` also provides **six differents charts** to visualize these DEGs (To provide publication-ready plots.), including **VolcanoPlot**, **ScatterPlot**, **MAPlot**, **RankPlot**, **GenePlot** and **DEHeatmap**. To label genes, users can specify genes with `Label gene` in **Options (Advanced)**.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_08_DEA_visualize.png"  title="DEbPeak_DEA_visualize"  alt="DEbPeak_DEA_visualize" />
</div>

# Functional Enrichment Analysis (FEA)
## Gene Ontology (GO) enrichment
[The Gene Ontology (GO) is the most widely used ontology for specifying cellular location, molecular function, and biological process participation of human and model organism genes](https://www.nature.com/articles/s41598-018-23395-2). With DEGs identified, `DEbPeak` enables users to annotate these genes to different GO terms to interpret the results.

Users can specify GO type to enrich with `GO type` (**BP**: biological process; **MF**: molecular function; **CC**: cellular component; **ALL**: all above three).

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_09_FEA_GO.png"  title="DEbPeak_FEA_GO"  alt="DEbPeak_FEA_GO" />
</div>


## KEGG (Kyoto Encyclopedia of Genes and Genomes) enrichment
[KEGG is a database resource for understanding high-level functions and utilities of the biological system, such as the cell, the organism and the ecosystem, from molecular-level information, especially large-scale molecular datasets generated by genome sequencing and other high-throughput experimental technologies](https://www.genome.jp/kegg/). With DEGs identified, `DEbPeak` enables users to annotate these genes to different KEGG pathways to interpret the results.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_10_FEA_KEGG.png"  title="DEbPeak_FEA_KEGG"  alt="DEbPeak_FEA_KEGG" />
</div>

## Gene Set Enrichment Analysis (GSEA)
**GSEA** is a computational method that determines whether an a priori defined set of genes shows statistically significant, concordant differences between two biological states(e.g. phenotypes). **Different from GO and KEGG based on DEGs, GSEA is based on whole gene expression matrix without prior gene filtering**.

To perfrom GSEA, users should upload prepared Gene Matrix Transposed file format (GMT) file, you can download this file from [Molecular Signatures Databases](https://www.gsea-msigdb.org/gsea/msigdb/human/collections.jsp) or crete it yourself.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_11_FEA_GSEA.png"  title="DEbPeak_FEA_GSEA"  alt="DEbPeak_FEA_GSEA" />
</div>

# Integrate with ChIP-seq (DEbChIP)
Gene expression is controlled by transcription factors (TFs), which [binding to DNA recognition sites located near their target genes](https://www.nature.com/articles/s41467-019-14217-8). [Chromatin Immunoprecipitation Sequencing (ChIP-Seq) can be used to profile genome-wide protein–DNA interactions](https://www.nature.com/articles/nrg2641), such as identifying sites of TFs binding. Therefore, ChIP-Seq can be used to [reveal mechanisms involved in differential gene regulation](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4614920/).

`DEbPeak` enables users to integrate RNA-seq and ChIP-seq, which can be used to investigate the direct and indirect targets of a TF, thus revealing the regulation network of gene expression.

## ConsensusPeak
Due to the high noise of ChIP-seq experiments, control samples can be used to remove possible nonspecific interactions. Besides control samples, [replicates are another way to separate actual biological events from variability resulting from random chance](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3962196/). The peaks which have high consistency between replicates are likely to be genuine signals, and we call these peaks consensus peaks. In this step, `DEbPeak` can get consensus peaks with [MSPC](https://genometric.github.io/MSPC/docs/quick_start/) when multiple peak files are available, but when there is only one peak file, we will use it directly (**make sure this peak file contains five columns: "chr", "start", "stop", "name", "score"**).

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_12_DEbChIP_CP.png"  title="DEbPeak_DEbChIP_CP"  alt="DEbPeak_DEbChIP_CP" />
</div>

## PeakProfile
`DEbPeak` provides three differents charts to visualize the ChIP-seq peak binding profile with [ChIPseeker](https://bioconductor.org/packages/release/bioc/html/ChIPseeker.html):

* Heatmap of ChIP-seq peaks binding to TSS regions: The output is in <mark>Binding heatmap</mark>.
* Average Profile of ChIP-seq peaks binding to TSS region: The output is in <mark>Average binding profile</mark>.
* Profile of ChIP-seq peaks binding to gene body regions: The output is in <mark>Region profile</mark>.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_13_DEbChIP_PP.png"  title="DEbPeak_DEbChIP_PP"  alt="DEbPeak_DEbChIP_PP" />
</div>

## PeakAnnotation
In this step, `DEbPeak` can perform peak annotation with [ChIPseeker](https://bioconductor.org/packages/release/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.html). This will include annotating peaks with its nearby genes, assigning genomic region of the peaks, et al.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_14_DEbChIP_PA.png"  title="DEbPeak_DEbChIP_PA"  alt="DEbPeak_DEbChIP_PA" />
</div>

## Integrate
In this step, `DEbPeak` will integrate RNA-seq results (**DEGs**) with ChIP-seq results (peak annotations results with **anno column (last column)** is **Promoter** (specified by `Anno region`)). In detail, this step will select DEGs with peak binds to their **Promoter** region. 

The **Type** column in integration results has **five categories**:

* **UPbPeak**: Up-regulated genes with peak binding to their Promoter region. These genes are highly plausible direct targets of the TF used to perform ChIP-seq.
* **DOWNbPeak**: Down-regulated genes with peak binding to their Promoter region. These genes are highly plausible direct targets of the TF used to perform ChIP-seq.
* **UP**: Up-regulated genes
* **DOWN**: Down-regulated genes
* **Peak**: Genes with peak binding to their Promoter region

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_15_DEbChIP_INTE.png"  title="DEbPeak_DEbChIP_INTE"  alt="DEbPeak_DEbChIP_INTE" />
</div>

## Enrichment
Users may be interested in the functions of direct target genes. `DEbPeak` enables users to perform GO enrichment on these genes. The output of **UPbPeak** related genes is in <mark>UPbChIP table, UPbChIP plot </mark>. The output of **DOWNbPeak** related genes is in <mark>DOWNbChIP table, DOWNbChIP plot </mark>.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_16_DEbChIP_INTE_GO.png"  title="DEbPeak_DEbChIP_INTE_GO"  alt="DEbPeak_DEbChIP_INTE_GO" />
</div>

# Integrate with ATAC-seq (DEbATAC)
[Eukaryotic chromatin is tightly packaged into nucleosomes, and the positioning of nucleosomes can regulate gene expression by changing the in vivo availability of binding sites to transcription factors (TFs)](https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/1756-8935-7-33). Assay of Transposase Accessible Chromatin sequencing (ATAC-seq) is a technique used to assess genome-wide chromatin accessibility. Thus, combining ATAC-seq with RNA-seq can be used to understanding the role of chromatin structure in regulating gene expression.

The ATAC-seq data is also peak data, most of the processing steps are the same as in "Integrate with ChIP-seq (DEbChIP)". However, [different from ChIP-seq which profile specific protein–DNA interactions (depends on the antibody provided), ATAC-seq can proflie global chromatin accessibility](https://link.springer.com/article/10.1186/s43556-020-00009-w). Thus, the peaks of ATAC-seq may originate from hundreds of TFs. To reveal the possible TFs involved, motif enrichment analysis is a good solution. 

The **"FindMotif"** is performed on the peaks binding to **UPbPeak** and **DOWNbPeak** genes:

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_22_DEbATAC_MOTIF.png"  title="DEbPeak_DEbATAC_MOTIF"  alt="DEbPeak_DEbATAC_MOTIF" />
</div>

# Integrate with ChIP-seq and ATAC-seq (DEbCA)
[ATAC-seq and ChIP-seq are highly consistent and complementary](https://doi.org/10.1093/bib/bbac061). [Most TFs require open chromatin for binding to DNA recognition sites, ATAC-seq peaks generally overlap with TF ChIP-seq peaks but are often broader. Thus, TF ChIP-seq and ATAC-seq can mutually validate the quality and reliability of each other within the same experimental system](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-1929-3). To obtain more accurate gene regulation results, `DEbPeak` can integrate RNA-seq with ChIP-seq and ATAC-seq simultaneously.

## Integrate
This step is based on the DEGs of RNA-seq, **PeakAnnotation** results of **"Integrate with ChIP-seq (DEbChIP)"** and **"Integrate with ATAC-seq (DEbATAC)"**. In detail, this step will select DEGs with peaks of ChIP-seq binding to their **Promoter** region and peaks of ATAC-seq indicating the open chromatin of their **Promoter** region (The region is specified by `Anno region`). 

The **Type** column in integration results has eleven categories:

* **UPbPeak**: Up-regulated genes, peaks of ChIP-seq bind to their **Promoter** region, peaks of ATAC-seq indicating the open chromatin of their **Promoter** region
* **DOWNbPeak**: Down-regulated genes, peaks of ChIP-seq bind to their **Promoter** region, peaks of ATAC-seq indicating the open chromatin of their **Promoter** region
* **UPbChIP**: Up-regulated genes and peaks of ChIP-seq bind to their **Promoter** region
* **DOWNbChIP**: Down-regulated genes and peaks of ChIP-seq bind to their **Promoter** region
* **UPbATAC**: Up-regulated genes and peaks of ATAC-seq indicating the open chromatin of their **Promoter** region
* **DOWNbATAC**: Down-regulated genes and peaks of ATAC-seq indicating the open chromatin of their **Promoter** region
* **ChIPbATAC**: Genes with peaks of ChIP-seq bind to their **Promoter** region and peaks of ATAC-seq indicating the open chromatin of their **Promoter** region
* **UP**: Up-regulated genes
* **DOWN**: Down-regulated genes
* **ChIP**: Genes with peaks of ChIP-seq bind to their **Promoter** region
* **ATAC**: Genes with peaks of ATAC-seq indicating the open chromatin of their **Promoter** region

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_23_DEbCA_INTE.png"  title="DEbPeak_DEbCA_INTE"  alt="DEbPeak_DEbCA_INTE" />
</div>

## Enrichment
Users may be interested genes of **UPbPeak** and **DOWNbPeak**, `DEbPeak` enables users to perform GO enrichment on these genes. The output of **UPbPeak** related genes is in <mark>UPbCA table, UPbCA plot </mark>. The output of **DOWNbPeak** related genes is in <mark>DOWNbCA table, DOWNbCA plot </mark>.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_24_DEbCA_INTE_GO.png"  title="DEbPeak_DEbCA_INTE_GO"  alt="DEbPeak_DEbCA_INTE_GO" />
</div>

# Utils
In RNA-seq analysis, it is common that we need to **convert gene IDs in Gene Symbol, ENSEMBL Gene ID, Entrez Gene ID** and **normalize the count matrix** with different methods for subsequent analysis. `DEbPeak` enables users to perform gene ID conversion, and provides five different normalization methods (**CPM**, **TMM**, **DESeq2’s median of ratios**, **RPKM** and **TPM**) to normalize the count matrix.

## IDConversion
To perform gene ID conversion, you should upload you count matrix (the rownames are genes, the colnames are samples), specify the gene ID type of you matrix with `Gene type` and specify the species of the count matrix with `Species` (the support species are **Human**, **Mouse**, **Rat**, **Fly**, **Arabidopsis**, **Yeast**, **Zebrafish**, **Worm**, **Bovine**, **Pig**, **Chicken**, **Rhesus**, **Canine**, **Xenopus**, **Anopheles**, **Chimp**, **E coli strain Sakai**, **Myxococcus xanthus DK 1622**, the data are from [Bioconductor AnnotationData Packages](https://bioconductor.org/packages/3.16/data/annotation/)). After conversion, `DEbPeak` will add two additional columns to the raw count matrix (if the `Gene type` of your count matrix is ENSEMBL, add ENTREZID and SYMBOL).

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_25_Utils_conversion.png"  title="DEbPeak_Utils_conversion"  alt="DEbPeak_Utils_conversion" />
</div>

## Normalization
Here, we provide five different normalization methods: **CPM**, **TMM**, **DESeq2’s median of ratios**, **RPKM** and **TPM**, of which **RPKM** and **TPM** need to provide gtf files to calculate gene length.

<div align="center">
<img src="/home/songyabing/R/learn/tmp/DEbPeak/ManualFigures/DEbPeak_26_Utils_normalization.png"  title="DEbPeak_Utils_normalization"  alt="DEbPeak_Utils_normalization" />
</div>

# R package
The `DEbPeak` package is available on [GitHub](https://github.com/showteeth/DEbPeak), and tha manual is available [here](https://showteeth.github.io/DEbPeak/). `DEbPeak` provides eight vignettes:

* [Quality Control](https://showteeth.github.io/DEbPeak/articles/QualityControl.html)
* [Principal Component Analysis](https://showteeth.github.io/DEbPeak/articles/PrincipalComponentAnalysis.html)
* [Differential Expression Analysis](https://showteeth.github.io/DEbPeak/articles/DifferentialExpressionAnalysis.html)
* [Functional Enrichment Analysis](https://showteeth.github.io/DEbPeak/articles/FunctionalEnrichmentAnalysis.html)
* [Integrate RNA-seq and ChIP-seq](https://showteeth.github.io/DEbPeak/articles/IntegrateChIP.html)
* [Integrate RNA-seq and ATAC-seq](https://showteeth.github.io/DEbPeak/articles/IntegrateATAC.html)
* [Integrate RNA-seq ChIP-seq and ATAC-seq](https://showteeth.github.io/DEbPeak/articles/IntegrateChIPATAC.html)
* [Utils](https://showteeth.github.io/DEbPeak/articles/Utils.html)

# Browser Compatibility






