<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DEbPeak - Analyze and integrate multi-omics to unravel the regulation of gene expression • DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="DEbPeak - Analyze and integrate multi-omics to unravel the regulation of gene expression">
<meta property="og:description" content="DEbPeak aims to explore, visualize, interpret multi-omics data and unravel the regulation of gene expression by 
    combining RNA-seq with peak-related data (eg: ChIP-seq, ATAC-seq, m6a-seq et al.). It contains six functional panels: 
    Quality Control (QC), Principal Component Analysis (PCA), Differential Analysis (DA), Functional Enrichment Analysis (FEA), 
    Integrate RNA-seq with peak-related data (support the simultaneous integration of two kinds of peak-related data), 
    Integrate RNA-seq with RNA-seq, Integrate peak-related data with peak-related data, Utils.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="debpeak---analyze-and-integrate-multi-omics-to-unravel-the-regulation-of-gene-expression" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#debpeak---analyze-and-integrate-multi-omics-to-unravel-the-regulation-of-gene-expression" class="anchor"></a>DEbPeak - Analyze and integrate multi-omics to unravel the regulation of gene expression.</h1></div>
<p><img src="reference/figures/DEbPeak.png" align="right" width="200"></p>
<p><img src="https://img.shields.io/badge/license-MIT-green" alt="License"><a href="https://github.com/showteeth/DEbPeak"><img src="https://img.shields.io/github/languages/code-size/showteeth/DEbPeak.svg" alt="CODE_SIZE"></a></p>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p><code>DEbPeak</code> aims to <strong>explore</strong>, <strong>visualize</strong>, <strong>interpret</strong> multi-omics data and <strong>unravel the regulation of gene expression</strong> by combining RNA-seq with peak-related data (eg: ChIP-seq, ATAC-seq, m6a-seq et al.). It contains <strong>six functional modules</strong>:</p>
<ul>
<li>
<strong>Quality Control (QC)</strong>: QC on count matrix and samples.
<ul>
<li>QC on count matrix: Proportion of genes detected in different samples under different CPM thresholds and the saturation of the number of genes detected.</li>
<li>QC on samples: Euclidean distance and pearson correlation coefficient of samples across different conditions, sample similarity on selected principal components (check batch information and conduct batch correction) and outlier detection with robust PCA.</li>
</ul>
</li>
<li>
<strong>Principal Component Analysis (PCA)</strong>: this module can be divided into three sub modules, basic info, loading related and 3D visualization.
<ul>
<li>Basic info: scree plot (help to select the useful PCs), biplot (sample similarity with corresponding genes with larger loadings) and PC pairs plot (sample similarity under different PC combinations).</li>
<li>Loading related: visualize genes with larger positive and negative loadings on selected PCs, conduct GO enrichment analysis on genes with larger positive and negative loadings on selected PCs.</li>
<li>3D visualization: visualize samples on three selected PCs.</li>
</ul>
</li>
<li>
<strong>Differential Analysis and Visualization</strong>: this module includes six powerful visualization methods (Volcano Plot, Scatter Plot, MA Plot, Rank Plot, Gene/Peak Plot, Heatmap).</li>
<li>
<strong>Functional Enrichment Analysis (FEA)</strong>: GO enrichment analysis, KEGG enrichment analysis, Gene Set Enrichment Analysis (GSEA).
<ul>
<li>GO (Biological Process, Molecular Function, Cellular Component) and KEGG on differential expression genes or accessible/binding peaks.</li>
<li>GSEA on all genes (Notice: GSEA is not available for peak-related data)</li>
</ul>
</li>
<li>
<strong>Integrate RNA-seq with peak-related data</strong>:
<ul>
<li>Get consensus peaks: For multiple peak files, get consensus peaks; for single peak file, use it directly (used in consensus integration mode).</li>
<li>Peak profile plots: Heatmap of peak binding to TSS regions, Average Profile of ChIP peaks binding to TSS region, Profile of ChIP peaks binding to different regions (used in consensus integration mode).</li>
<li>Peak annotaion (used in consensus integration mode).</li>
<li>Integrate RNA-seq with peak-related data (consensus mode): Integrate RNA-seq with peak-related data to find direct targets, including up-regulated and down-regulated.</li>
<li>Integrate RNA-seq with peak-related data (differential mode): Integrate RNA-seq and peak-related data based on differential analysis.</li>
<li>Integration summary: include venn diagram and quadrant diagram (differential mode).</li>
<li>GO enrichment on integrated results.</li>
<li>Find motif on integrated results: Due to the nature of ATAC-seq, we usually need to find motif on integrated results to obtain potential regulatory factors.</li>
</ul>
</li>
<li>
<strong>Integrate RNA-seq with RNA-seq</strong>:
<ul>
<li>Integration summary: include venn diagram and quadrant diagram.</li>
<li>GO enrichment on integrated results.</li>
</ul>
</li>
<li>
<strong>Integrate peak-related data with peak-related data</strong>:
<ul>
<li>Integration summary: include venn diagram and quadrant diagram (differential mode).</li>
<li>GO enrichment on integrated results.</li>
</ul>
</li>
<li>
<strong>Utils</strong>: useful functions when dealing with RNA-seq data, including gene name conversion and count normalization(DESeq2’s median of ratios, TMM, CPM, TPM, RPKM).</li>
</ul>
<p>To enhance the ease of use of the tool, we have also developed an <strong>web server</strong> for <code>DEbPeak</code> that allows users to submit files to the web page and set parameters to get the desired results. Unlike the standalone R package, <em>the web server has built-in <code>DESeq2</code> for differential analysis</em>, while the R package can accept user input results from <code>DESeq2</code> or <code>edgeR</code>, which will be <strong>more flexible</strong>.</p>
<p>By the way, all plots generated are <strong>publication-ready</strong> , and most of them are based on <code>ggplot2</code>, so that users can easily modify them according to their needs. We also provide <strong>various color palettes</strong>, including <strong>discrete</strong> and <strong>continuous</strong>, <strong>color blind friendly</strong> and <strong>multiple categorical variables</strong>.</p>
<hr>
</div>
<div id="framework" class="section level2">
<h2 class="hasAnchor">
<a href="#framework" class="anchor"></a>Framework</h2>
<div align="center">
<p><img src="reference/figures/DEbPeak_outline.png" title="DEbPeak_framework" alt="DEbPeak_framework"></p>
</div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install the package via the Github repository:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.package("remotes")   #In case you have not installed it.</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"showteeth/DEbPeak"</span><span class="op">)</span></code></pre></div>
<p>In general, it is <strong>recommended</strong> to install from <a href="https://github.com/showteeth/DEbPeak">Github repository</a> (update more timely).</p>
<hr>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<div id="vignette" class="section level3">
<h3 class="hasAnchor">
<a href="#vignette" class="anchor"></a>Vignette</h3>
<p>Detailed usage is available in <a href="https://showteeth.github.io/DEbPeak/">here</a>. We divide these vignettes into four categories:</p>
<ul>
<li>For analyzing <strong>RNA-seq</strong>:
<ul>
<li><a href="https://showteeth.github.io/DEbPeak/articles/QualityControl.html">Quality Control</a></li>
<li><a href="https://showteeth.github.io/DEbPeak/articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a></li>
<li><a href="https://showteeth.github.io/DEbPeak/articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a></li>
<li><a href="https://showteeth.github.io/DEbPeak/articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a></li>
<li><a href="https://showteeth.github.io/DEbPeak/articles/Utils.html">Utils</a></li>
</ul>
</li>
<li>For analyzing <strong>peak-related data</strong>:
<ul>
<li><a href="https://showteeth.github.io/DEbPeak/articles/QualityControl.html">Quality Control</a></li>
<li><a href="https://showteeth.github.io/DEbPeak/articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a></li>
<li><a href="https://showteeth.github.io/DEbPeak/articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a></li>
<li><a href="https://showteeth.github.io/DEbPeak/articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a></li>
<li><a href="https://showteeth.github.io/DEbPeak/articles/MotifEnrichment.html">Motif enrichment (Peak-related)</a></li>
</ul>
</li>
<li>
<strong>Integrating RNA-seq (differential expression analysis) with peak-related data (consensus peak)</strong>:
<ul>
<li><a href="https://showteeth.github.io/DEbPeak/articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a></li>
<li><a href="https://showteeth.github.io/DEbPeak/articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a></li>
<li><a href="https://showteeth.github.io/DEbPeak/articles/IntegrateChIPATAC.html">Integrate RNA-seq ChIP-seq and ATAC-seq</a></li>
</ul>
</li>
<li>
<strong>Integrating RNA-seq (differential expression analysis) with peak-related data (differential accessible/binding analysis)</strong>:
<ul>
<li><a href="https://showteeth.github.io/DEbPeak/articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a></li>
<li>Integrate RNA-seq with ChIP-seq (differential analysis) is almost the same as <a href="https://showteeth.github.io/DEbPeak/articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>.</li>
</ul>
</li>
</ul>
</div>
<div id="function-list" class="section level3">
<h3 class="hasAnchor">
<a href="#function-list" class="anchor"></a>Function list</h3>
<table class="table">
<thead><tr>
<th>
Type
</th>
<th>
Function
</th>
<th>
Description
</th>
<th>
Key packages
</th>
</tr></thead>
<tbody>
<tr>
<td rowspan="4">
Quality Control
</td>
<td>
CountQC
</td>
<td>
Quality control on count matrix (gene detection sensitivity and sequencing depth saturation)
</td>
<td>
NOISeq
</td>
</tr>
<tr>
<td>
SampleRelation
</td>
<td>
Quality control on samples (sample clustering based on euclidean distance and pearson correlation coefficient)
</td>
<td>
stats
</td>
</tr>
<tr>
<td>
OutlierDetection
</td>
<td>
Detect outlier with robust PCA
</td>
<td>
rrcov
</td>
</tr>
<tr>
<td>
QCPCA
</td>
<td>
PCA related functions used in quality control (batch detection and correction, outlier detection)
</td>
<td>
stats, sva, rrcov
</td>
</tr>
<tr>
<td rowspan="6">
Principal Component Analysis
</td>
<td>
PCA
</td>
<td>
Conduct principal component analysis
</td>
<td>
stats
</td>
</tr>
<tr>
<td>
PCABasic
</td>
<td>
Generated PCA baisc plots, including screen plot, biplot and pairs plot
</td>
<td>
PCAtools
</td>
</tr>
<tr>
<td>
ExportPCGenes
</td>
<td>
Export genes of selected PCs
</td>
<td>
tidyverse
</td>
</tr>
<tr>
<td>
LoadingPlot
</td>
<td>
PCA loading plot, including bar plot and heatmap
</td>
<td>
ggplot2, ComplexHeatmap
</td>
</tr>
<tr>
<td>
LoadingGO
</td>
<td>
GO enrichment on PC’s loading genes
</td>
<td>
clusterProfiler
</td>
</tr>
<tr>
<td>
PCA3D
</td>
<td>
Create 3D PCA plot
</td>
<td>
plot3D
</td>
</tr>
<tr>
<td rowspan="8">
Differential Axpression Analysis
</td>
<td>
ExtractDA
</td>
<td>
Extract differential analysis results
</td>
<td>
tidyverse
</td>
</tr>
<tr>
<td>
VolcanoPlot
</td>
<td>
VolcanoPlot for differential analysis results
</td>
<td>
ggplot2
</td>
</tr>
<tr>
<td>
ScatterPlot
</td>
<td>
ScatterPlot for differential analysis results
</td>
<td>
ggplot2
</td>
</tr>
<tr>
<td>
MAPlot
</td>
<td>
MA-plot for differential analysis results
</td>
<td>
ggplot2
</td>
</tr>
<tr>
<td>
RankPlot
</td>
<td>
Rank plot for differential analysis results
</td>
<td>
ggplot2
</td>
</tr>
<tr>
<td>
GenePlot
</td>
<td>
Gene expresion or peak accessibility/binding plot
</td>
<td>
ggplot2
</td>
</tr>
<tr>
<td>
DEHeatmap
</td>
<td>
Heatmap for differential analysis results
</td>
<td>
ComplexHeatmap
</td>
</tr>
<tr>
<td>
ConductDESeq2
</td>
<td>
Conduct differential analysis with DESeq2
</td>
<td>
NOISeq, stats, sva, rrcov, PCAtools, DESeq2, ggplot2, ComplexHeatmap, clusterProfiler, plot3D, tidyverse
</td>
</tr>
<tr>
<td rowspan="3">
Functional Enrichment Analysis
</td>
<td>
ConductFE
</td>
<td>
Conduct functional enrichment analysis (GO and KEGG)
</td>
<td>
clusterProfiler
</td>
</tr>
<tr>
<td>
ConductGSEA
</td>
<td>
Conduct gene set enrichment analysis (GSEA)
</td>
<td>
clusterProfiler
</td>
</tr>
<tr>
<td>
VisGSEA
</td>
<td>
Visualize GSEA results
</td>
<td>
enrichplot
</td>
</tr>
<tr>
<td rowspan="6">
Peak-related Analysis<br><br><br><br>
</td>
<td>
PeakMatrix
</td>
<td>
Prepare count matrix and sample metadata for peak-related data
</td>
<td>
DiffBind, ChIPseeker
</td>
</tr>
<tr>
<td>
GetConsensusPeak
</td>
<td>
Get consensus peak from replicates
</td>
<td>
MSPC
</td>
</tr>
<tr>
<td>
PeakProfile
</td>
<td>
Visualize peak accessibility/binding profile
</td>
<td>
ChIPseeker
</td>
</tr>
<tr>
<td>
AnnoPeak
</td>
<td>
Assign peaks with the genomic binding region and nearby genes
</td>
<td>
ChIPseeker
</td>
</tr>
<tr>
<td>
PeakAnnoPie
</td>
<td>
Visualize peak annotation results with pie plot
</td>
<td>
ggpie
</td>
</tr>
<tr>
<td>
MotifEnrich
</td>
<td>
Motif enrichment for differentially accessible/binding peaks
</td>
<td>
HOMER
</td>
</tr>
<tr>
<td rowspan="6">
Integrate RNA-seq with Peak-related Data
</td>
<td>
DEbPeak
</td>
<td>
Integrate differential expression results and peak annotation/differential analysis results.
</td>
<td>
tidyverse
</td>
</tr>
<tr>
<td>
InteVenn
</td>
<td>
Create venn diagram for integration results (support <code>DEbPeak</code>, <code>DEbDE</code>, <code>PeakbPeak</code>)
</td>
<td>
ggvenn
</td>
</tr>
<tr>
<td>
InteDiffQuad
</td>
<td>
Create quadrant diagram for integration results (support <code>DEbPeak</code>, <code>DEbDE</code>, <code>PeakbPeak</code>)
</td>
<td>
ggplot2
</td>
</tr>
<tr>
<td>
InteFE
</td>
<td>
GO enrichment on integration results (support <code>DEbPeak</code>, <code>DEbDE</code>, <code>PeakbPeak</code>)
</td>
<td>
clusterProfiler
</td>
</tr>
<tr>
<td>
FindMotif
</td>
<td>
Find motif on integration results
</td>
<td>
HOMER
</td>
</tr>
<tr>
<td>
DEbCA
</td>
<td>
Integrate differential expression results and peak annotation results (two kinds of peak-related data)
</td>
<td>
tidyverse
</td>
</tr>
<tr>
<td rowspan="2">
Integrate RNA-seq with RNA-seq
</td>
<td>
DEbDE
</td>
<td>
Integrate Two Differential Expression Results
</td>
<td>
tidyverse
</td>
</tr>
<tr>
<td>
DEbDEFE
</td>
<td>
GO Enrichment on Two Differential Expression Integration Results.
</td>
<td>
clusterProfiler
</td>
</tr>
<tr>
<td rowspan="2">
Integrate Peak-related Data with Peak-related Data
</td>
<td>
PeakbPeak
</td>
<td>
Integrate Two Peak Annotation/Differential Analysis Results.
</td>
<td>
tidyverse
</td>
</tr>
<tr>
<td>
PeakbPeakFE
</td>
<td>
GO Enrichment on Two Peak Annotation/Differential Analysis Integration Results.
</td>
<td>
clusterProfiler
</td>
</tr>
<tr>
<td rowspan="3">
Utils
</td>
<td>
IDConversion
</td>
<td>
Gene ID conversion between ENSEMBL ENTREZID SYMBOL
</td>
<td>
clusterProfiler
</td>
</tr>
<tr>
<td>
GetGeneLength
</td>
<td>
Get gene length from GTF
</td>
<td>
GenomicFeatures, GenomicRanges
</td>
</tr>
<tr>
<td>
NormalizedCount
</td>
<td>
Perform counts normalization (DESeq2’s median of ratios, TMM, CPM, RPKM, TPM)
</td>
<td>
DESeq2, edgeR, tidyverse
</td>
</tr>
</tbody>
</table>
<hr>
</div>
</div>
<div id="notice" class="section level2">
<h2 class="hasAnchor">
<a href="#notice" class="anchor"></a>Notice</h2>
<ul>
<li>The <strong>KEGG API</strong> has changed, to perform KEGG enrichment, you’d better update <code>clusterProfiler</code> &gt;= <code>4.7.1</code>.</li>
</ul>
<hr>
</div>
<div id="contact" class="section level2">
<h2 class="hasAnchor">
<a href="#contact" class="anchor"></a>Contact</h2>
<p>For any question, feature request or bug report please write an email to <a href="mailto:songyb0519@gmail.com" class="email">songyb0519@gmail.com</a>.</p>
<hr>
</div>
<div id="code-of-conduct" class="section level2">
<h2 class="hasAnchor">
<a href="#code-of-conduct" class="anchor"></a>Code of Conduct</h2>
<p>Please note that the DEbPeak project is released with a <a href="https://contributor-covenant.org/version/2/0/CODE_OF_CONDUCT.html">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
<hr>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="community">
<h2>Community</h2>
<ul class="list-unstyled">
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Yabing Song <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
