<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DEbPeak User Guide
 • DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="DEbPeak User Guide
">
<meta property="og:description" content="DEbPeak">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.6.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq and ChIP-seq</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><link href="DEbPeak_All_files/font-awesome-5.1.0/css/all.css" rel="stylesheet">
<link href="DEbPeak_All_files/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
<p>DEbPeak User Guide</p>
                        <h4 class="author">Yabing Song</h4>
            <address class="author_afil">
      School of Life Sciences, Tsinghua University<br><a class="author_email" href="mailto:#"></a><a href="mailto:songyb18@mails.tsinghua.edu.cn" class="email">songyb18@mails.tsinghua.edu.cn</a>
      </address>
                  
            <h4 class="date">12 十二月 2022</h4>
      
      
      <div class="hidden name"><code>DEbPeak_All.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The goal of <code>DEbPeak</code> is to explore, visualize, interpret RNA-seq data and decipher gene’s regulation by combining ChIP-seq and ATAC-seq data. It contains six functional modules: Quality Control (QC) on count matrix and samples, Principal Component Analysis (PCA) ,Differential Expression Gene Visualization (Volcano Plot, Scatter Plot, MA Plot, Rank Plot, Gene Plot, Heatmap), Functional Enrichment Analysis (GO enrichment analysis, KEGG enrichment analysis, Gene Set Enrichment Analysis), Integrate with ChIP-seq/ATAC-seq (Get consensus peaks, Peak profile plots, Peak annotaion, Integrate RNA-seq and ChIP-seq/ATAC-seq, Integrate summary, GO enrichment on integrated results, Find motif on integrated results), Utils (useful functions when dealing with RNA-seq data, including gene name conversion and count normalization).</p>
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<p><code>DEbPeak</code> is an R package distributed on <a href="https://github.com/showteeth/DEbPeak">Github</a>. To install the package, start R and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install via Github</span>
<span class="co"># install.package("remotes")   #In case you have not installed it.</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"showteeth/DEbPeak"</span><span class="op">)</span></code></pre></div>
<p>Once <em>DEbPeak</em> is installed, it can be loaded by the following command.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"DEbPeak"</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><img src="../man/figures/DEbPeak.png" align="right" width="200"></p>
<p><code>DEbPeak</code> aims to <strong>explore</strong>, <strong>visualize</strong>, <strong>interpret</strong> RNA-seq data and <strong>unravel the regulation of gene expression</strong> by combining ChIP-seq and ATAC-seq data. It contains <strong>six functional modules</strong>:</p>
<ul>
<li>
<strong>Quality Control (QC)</strong>: QC on count matrix and samples.
<ul>
<li>QC on count matrix: Proportion of genes detected in different samples under different CPM thresholds and the saturation of the number of genes detected.</li>
<li>QC on samples: Euclidean distance and pearson correlation coefficient of samples across different conditions, sample similarity on selected principal components (check batch information and conduct batch correction) and outlier detection with robust PCA.</li>
</ul>
</li>
<li>
<strong>Principal Component Analysis (PCA)</strong>: this module can be divided into three sub modules, basic info, loading related and 3D visualization.
<ul>
<li>Basic info: scree plot (help to select the useful PCs), biplot (sample similarity with corresponding genes with larger loadings) and PC pairs plot (sample similarity under different PC combinations).</li>
<li>Loading related: visualize genes with larger positive and negative loadings on selected PCs, conduct GO enrichment analysis on genes with larger positive and negative loadings on selected PCs.</li>
<li>3D visualization: visualize samples on three selected PCs.</li>
</ul>
</li>
<li>
<strong>Differential Expression Gene Visualization</strong>: this module includes six powerful visualization methods (Volcano Plot, Scatter Plot, MA Plot, Rank Plot, Gene Plot, Heatmap).</li>
<li>
<strong>Functional Enrichment Analysis (FEA)</strong>: GO enrichment analysis, KEGG enrichment analysis, Gene Set Enrichment Analysis (GSEA).
<ul>
<li>GO (Biological Process, Molecular Function, Cellular Component) and KEGG on differential expression genes</li>
<li>GSEA on all genes</li>
</ul>
</li>
<li>
<strong>Integrate with ChIP-seq/ATAC-seq</strong>:
<ul>
<li>Get consensus peaks: For multiple peak files, get consensus peaks; for single peak file, use it directly</li>
<li>Peak profile plots: Heatmap of peak binding to TSS regions, Average Profile of ChIP peaks binding to TSS region, Profile of ChIP peaks binding to different regions.</li>
<li>Peak annotaion</li>
<li>Integrate RNA-seq and ChIP-seq/ATAC-seq: Integrate RNA-seq and ChIP-seq/ATAC-seq to find direct targets, including up-regulated and down-regulated.</li>
<li>Integrate summary: Summary the integrated results, get the overlap number of up-regulated genes and ChIP-seq/ATAC-seq results (UP), down-regulated genes and ChIP-seq/ATAC-seq results.</li>
<li>GO enrichment on integrated results: GO enrichment on up-regulated targets and down-regulated targets.</li>
<li>Find motif on integrated results: This is a step unique to ATAC-seq. Due to the nature of ATAC-seq, we usually need to find motif on integrated results to obtain potential regulatory factors.</li>
</ul>
</li>
<li>
<strong>Utils</strong>: useful functions when dealing with RNA-seq data, including gene name conversion and count normalization(DESeq2’s median of ratios, TMM, CPM, TPM, RPKM).</li>
</ul>
<p>To enhance the ease of use of the tool, we have also developed an <strong>web server</strong> for <code>DEbPeak</code> that allows users to submit files to the web page and set parameters to get the desired results. Unlike the standalone R package, <em>the web server has built-in <code>DESeq2</code> for differential expression analysis</em>, while the R package can accept user input results from <code>DESeq2</code> or <code>edgeR</code>, which will be <strong>more flexible</strong>.</p>
<p>By the way, all plots generated are <strong>publication-ready</strong> , and most of them are based on <code>ggplot2</code>, so that users can easily modify them according to their needs. We also provide <strong>various color palettes</strong>, including <strong>discrete</strong> and <strong>continuous</strong>, <strong>color blind friendly</strong> and <strong>multiple categorical variables</strong>.</p>
<hr>
</div>
<div id="load-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-the-data" class="anchor"></a>Load the data</h1>
<p>The data used here are RNA-seq data of the external granule layer in the cerebellum of control and conditional SnoN knockout mice, the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120279">GSE120279</a>, they are used in <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03608-0">Robust principal component analysis for accurate outlier sample detection in RNA-seq data</a>.</p>
<p>First, we create <code>DESeqDataSet</code> object with above dataset:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2">DESeq2</a></span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DEbPeak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">count.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"snon_count.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">meta.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"snon_meta.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">count.matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">count.file</span>, header <span class="op">=</span> <span class="cn">T</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">count.matrix</span><span class="op">)</span></code></pre></div>
<pre><code>##                        KO1  KO2  KO3  KO4  KO5  KO6  WT1  WT2  WT3  WT4  WT5
## ENSMUSG00000000001.4  4556 4218 3835 3718 5741 3875 4115 5074 2931 4374 5118
## ENSMUSG00000000003.15    0    0    0    0    0    0    0    0    0    0    0
## ENSMUSG00000000028.14  350  579  435  316  432  317  245  621  419  506  545
## ENSMUSG00000000031.16  268  804   66  207   46   66  336  112   60   69  137
## ENSMUSG00000000037.16  262  157  184  162  301  233  311  176   94  139  229
## ENSMUSG00000000049.11    0    0    0    0    0    0    1    0    2    0    1
##                        WT6
## ENSMUSG00000000001.4  3625
## ENSMUSG00000000003.15    0
## ENSMUSG00000000028.14  371
## ENSMUSG00000000031.16  193
## ENSMUSG00000000037.16  186
## ENSMUSG00000000049.11    0</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta.info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">meta.file</span>, header <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">meta.info</span><span class="op">)</span></code></pre></div>
<pre><code>##     condition
## KO1        KO
## KO2        KO
## KO3        KO
## KO4        KO
## KO5        KO
## KO6        KO</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span><span class="op">(</span>countData <span class="op">=</span> <span class="va">count.matrix</span>, colData <span class="op">=</span> <span class="va">meta.info</span>, design <span class="op">=</span> <span class="op">~</span><span class="va">condition</span><span class="op">)</span>
<span class="va">dds</span></code></pre></div>
<pre><code>## class: DESeqDataSet 
## dim: 53379 12 
## metadata(1): version
## assays(1): counts
## rownames(53379): ENSMUSG00000000001.4 ENSMUSG00000000003.15 ...
##   ENSMUSG00000115849.1 ENSMUSG00000115850.1
## rowData names(0):
## colnames(12): KO1 KO2 ... WT5 WT6
## colData names(1): condition</code></pre>
<hr>
</div>
<div id="quality-control" class="section level1">
<h1 class="hasAnchor">
<a href="#quality-control" class="anchor"></a>Quality Control</h1>
<p>Quality control are of vital importance to get realiable results. Here, quality control includes two main aspects: <strong>sequencing saturation (QC on count matrix)</strong> and <strong>sample similarity (QC on samples)</strong>.</p>
<div id="qc-on-count-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#qc-on-count-matrix" class="anchor"></a>QC on count matrix</h2>
<div id="cpm-threshold" class="section level3">
<h3 class="hasAnchor">
<a href="#cpm-threshold" class="anchor"></a>CPM threshold</h3>
<p>This plot shows proportion of genes detected in different samples at different CPM thresholds. Theoretically, all samples should be similarly distributed, so as to avoid false positive results obtained from detection problems.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/CountQC.html">CountQC</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>, group.key <span class="op">=</span> <span class="st">"condition"</span>, type <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## [1] "Warning: 25096 features with 0 counts in all samples are to be removed for this analysis."
## [1] "Count distributions are to be computed for:"
##  [1] "KO1" "KO2" "KO3" "KO4" "KO5" "KO6" "WT1" "WT2" "WT3" "WT4" "WT5" "WT6"</code></pre>
<p><img src="DEbPeak_All_files/figure-html/qc_counts_cpm-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="sequencing-saturation" class="section level3">
<h3 class="hasAnchor">
<a href="#sequencing-saturation" class="anchor"></a>sequencing saturation</h3>
<p>This plot shows the proportion of genes detected at different sequencing depths. If the proportion is too low, it indicates that many genes are not detected and therefore some important information may be missing, and increasing the sequencing volume can effectively solve this problem.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/CountQC.html">CountQC</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>, group.key <span class="op">=</span> <span class="st">"condition"</span>, type <span class="op">=</span> <span class="st">"saturation"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/qc_counts_saturation-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="qc-on-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#qc-on-samples" class="anchor"></a>QC on samples</h2>
<p>QC on samples including four aspects:</p>
<ul>
<li>Euclidean distance and pearson correlation coefficient: sample similarity on <strong>the entire transcriptome</strong>.</li>
<li>PCA: sample similarity on <strong>selected principal components</strong>.</li>
<li>outlier detection with robust PCA</li>
<li>batch correction</li>
</ul>
<div id="euclidean-distance-and-pearson-correlation-coefficient" class="section level3">
<h3 class="hasAnchor">
<a href="#euclidean-distance-and-pearson-correlation-coefficient" class="anchor"></a>Euclidean distance and pearson correlation coefficient</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/SampleRelation.html">SampleRelation</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span>,anno.key <span class="op">=</span> <span class="st">"condition"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/qc_samples_sr-1.png" width="960" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="pca" class="section level3">
<h3 class="hasAnchor">
<a href="#pca" class="anchor"></a>PCA</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">qc.pca.res</span><span class="op">=</span><span class="fu"><a href="../reference/PCA.html">PCA</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Use all genes for PCA!</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">PCAtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/PCAtools/man/biplot.html">biplot</a></span><span class="op">(</span><span class="va">qc.pca.res</span>,x<span class="op">=</span><span class="st">"PC1"</span>,y<span class="op">=</span><span class="st">"PC2"</span>,colby<span class="op">=</span><span class="st">"condition"</span>,legendPosition<span class="op">=</span><span class="st">"bottom"</span><span class="op">)</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/qc_samples_pca-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="outlier-detection-with-robust-pca" class="section level3">
<h3 class="hasAnchor">
<a href="#outlier-detection-with-robust-pca" class="anchor"></a>outlier detection with robust PCA</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outlier.res</span><span class="op">=</span><span class="fu"><a href="../reference/OutlierDetection.html">OutlierDetection</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,var.genes <span class="op">=</span> <span class="cn">NULL</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Use all genes for PCA!</code></pre>
<pre><code>## Detecting 2 outlier(s): KO3,WT1</code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outlier.res</span><span class="op">$</span><span class="va">outlier</span></code></pre></div>
<pre><code>## [1] "KO3" "WT1"</code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outlier.res</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/qc_samples_outlier-1.png" width="672" style="display: block; margin: auto;"></p>
<p>For batch correction and outlier detection, you can use the following codes:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># batch effect correction</span>
<span class="co"># this is a demo scripts</span>
<span class="va">batch.res</span><span class="op">=</span><span class="fu"><a href="../reference/QCPCA.html">QCPCA</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,var.genes <span class="op">=</span> <span class="cn">NULL</span>,remove.sample<span class="op">=</span><span class="cn">NULL</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span>,batch <span class="op">=</span> <span class="st">"cell"</span>,
                colby <span class="op">=</span> <span class="st">"dex"</span>, outlier.detection <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">batch.res</span><span class="op">$</span><span class="va">plot</span>
<span class="co"># outlier detection</span>
<span class="va">outlier.res</span><span class="op">=</span><span class="fu"><a href="../reference/QCPCA.html">QCPCA</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,var.genes <span class="op">=</span> <span class="cn">NULL</span>,remove.sample<span class="op">=</span><span class="cn">NULL</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span>,
                  outlier.detection <span class="op">=</span> <span class="cn">T</span>,rpca.method <span class="op">=</span> <span class="st">"PcaGrid"</span><span class="op">)</span>
<span class="va">outlier.res</span><span class="op">$</span><span class="va">plot</span>
<span class="co"># the final dds</span>
<span class="va">dds</span><span class="op">=</span><span class="va">outlier.res</span><span class="op">$</span><span class="va">deobj</span> <span class="co"># or outlier.res$plot</span></code></pre></div>
<hr>
</div>
</div>
</div>
<div id="principal-component-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#principal-component-analysis" class="anchor"></a>Principal Component Analysis</h1>
<div id="basic-info" class="section level2">
<h2 class="hasAnchor">
<a href="#basic-info" class="anchor"></a>basic info</h2>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># conduct PCA</span>
<span class="va">pca.res</span><span class="op">=</span><span class="fu"><a href="../reference/PCA.html">PCA</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,remove.sample <span class="op">=</span> <span class="va">outlier.res</span><span class="op">$</span><span class="va">outlier</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Use all genes for PCA!</code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get basic plots</span>
<span class="va">basic.plots</span><span class="op">=</span><span class="fu"><a href="../reference/PCABasic.html">PCABasic</a></span><span class="op">(</span><span class="va">pca.res</span>,colby<span class="op">=</span><span class="st">"condition"</span>,legend.pos <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></code></pre></div>
<pre><code>## Coordinate system already present. Adding new coordinate system, which will replace the existing one.</code></pre>
<pre><code>## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.</code></pre>
<div id="screen-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#screen-plot" class="anchor"></a>screen plot</h3>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.plots</span><span class="op">[[</span><span class="st">"screen"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/pca_basic_screen-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="biplot" class="section level3">
<h3 class="hasAnchor">
<a href="#biplot" class="anchor"></a>biplot</h3>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.plots</span><span class="op">[[</span><span class="st">"biplot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/pca_basic_biplot-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div id="pc-pairs-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#pc-pairs-plot" class="anchor"></a>PC pairs plot</h3>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.plots</span><span class="op">[[</span><span class="st">"pairs"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/pca_basic_pairs-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="loading-related" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-related" class="anchor"></a>loading related</h2>
<div id="loading-bar-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-bar-plot" class="anchor"></a>loading bar plot</h3>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># loading bar plot</span>
<span class="fu"><a href="../reference/LoadingPlot.html">LoadingPlot</a></span><span class="op">(</span><span class="va">pca.res</span>,type <span class="op">=</span> <span class="st">"bar"</span><span class="op">)</span></code></pre></div>
<pre><code>## Selecting by Loadding
## Selecting by Loadding</code></pre>
<p><img src="DEbPeak_All_files/figure-html/pca_loading_bar-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="loading-heatmap" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-heatmap" class="anchor"></a>loading heatmap</h3>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># loading heatmap</span>
<span class="fu"><a href="../reference/LoadingPlot.html">LoadingPlot</a></span><span class="op">(</span><span class="va">pca.res</span>,deobj <span class="op">=</span> <span class="va">dds</span>,type <span class="op">=</span> <span class="st">"heat"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Selecting by Loadding
## Selecting by Loadding</code></pre>
<p><img src="DEbPeak_All_files/figure-html/pca_loading_heatmap-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="export-loading-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#export-loading-genes" class="anchor"></a>export loading genes</h3>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># export loading genes on PC1 and PC2 (200 positive genes and 200 negative genes) </span>
<span class="va">loading.gene.df</span><span class="op">=</span><span class="fu"><a href="../reference/ExportPCGenes.html">ExportPCGenes</a></span><span class="op">(</span>pca <span class="op">=</span> <span class="va">pca.res</span>, pc <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, gene.num <span class="op">=</span> <span class="fl">200</span><span class="op">)</span></code></pre></div>
<pre><code>## Selecting by Loadding
## Selecting by Loadding</code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">loading.gene.df</span><span class="op">)</span></code></pre></div>
<pre><code>##    PC                  Gene   Loadding     Type
## 1 PC1 ENSMUSG00000038252.13 0.02543092 Positive
## 2 PC1 ENSMUSG00000027496.15 0.02457406 Positive
## 3 PC1 ENSMUSG00000030795.18 0.02395752 Positive
## 4 PC1 ENSMUSG00000031865.16 0.02373083 Positive
## 5 PC1 ENSMUSG00000027404.15 0.02319317 Positive
## 6 PC1 ENSMUSG00000063511.11 0.02220398 Positive</code></pre>
</div>
<div id="go-enrichment-on-loading-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#go-enrichment-on-loading-genes" class="anchor"></a>GO enrichment on loading genes</h3>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># save results to working folder</span>
<span class="co"># LoadingGO(pca.res,gene.type="ENSEMBL",go.type="BP",padj.method="BH",save = T)</span>
<span class="co"># return list</span>
<span class="va">loading.go.results</span><span class="op">=</span><span class="fu"><a href="../reference/LoadingGO.html">LoadingGO</a></span><span class="op">(</span><span class="va">pca.res</span>,gene.type<span class="op">=</span><span class="st">"ENSEMBL"</span>,go.type<span class="op">=</span><span class="st">"BP"</span>,padj.method<span class="op">=</span><span class="st">"BH"</span>,str.width <span class="op">=</span> <span class="fl">50</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## Selecting by Loadding
## Selecting by Loadding
## 
## 
## Convert ENSEMBL to ENTREZID!
## 
## 'select()' returned 1:many mapping between keys and columns
## 
## conduct BP GO enrichment analysis.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert ENSEMBL to ENTREZID!
## 
## 'select()' returned 1:many mapping between keys and columns
## 
## conduct BP GO enrichment analysis.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<div id="positive-loading-genes-go" class="section level4">
<h4 class="hasAnchor">
<a href="#positive-loading-genes-go" class="anchor"></a>positive loading genes GO</h4>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># positive loading genes</span>
<span class="va">positive_go_results</span><span class="op">=</span><span class="va">loading.go.results</span><span class="op">[[</span><span class="st">"Positive"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">positive_go_results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##                    ID                                       Description
## GO:0022613 GO:0022613              ribonucleoprotein complex biogenesis
## GO:0008380 GO:0008380                                      RNA splicing
## GO:1903311 GO:1903311              regulation of mRNA metabolic process
## GO:0031331 GO:0031331 positive regulation of cellular catabolic process
## GO:0042254 GO:0042254                               ribosome biogenesis
## GO:0006397 GO:0006397                                   mRNA processing
##            GeneRatio   BgRatio       pvalue     p.adjust       qvalue
## GO:0022613    19/194 409/23328 1.667451e-09 4.440423e-06 3.787747e-06
## GO:0008380    17/194 386/23328 2.733412e-08 3.639538e-05 3.104580e-05
## GO:1903311    13/194 266/23328 3.862626e-07 3.428724e-04 2.924753e-04
## GO:0031331    15/194 396/23328 1.243752e-06 6.850558e-04 5.843628e-04
## GO:0042254    13/194 296/23328 1.286248e-06 6.850558e-04 5.843628e-04
## GO:0006397    16/194 462/23328 1.753069e-06 7.115007e-04 6.069208e-04
##                                                                                                                         geneID
## GO:0022613 Snrpb/Sf3a2/Hsp90ab1/Mybbp1a/Ruvbl2/Ddx39b/Ddx47/Ppan/Rcl1/Drosha/Rpl6/Puf60/Ddx54/Ddx27/Mrpl10/Ran/Noc2l/Abt1/Nop2
## GO:0008380              Fus/Snrpb/Snrnp70/Sf3a2/Srsf4/Ddx39b/Khsrp/Scaf1/Ddx47/Txnl4a/Snrpa/Snrpn/Puf60/U2af2/Taf15/Mettl3/Sf1
## GO:1903311                                          Fus/Snrnp70/Srsf4/Khsrp/Upf1/Fto/Supt5/Puf60/U2af2/Taf15/Mettl3/Gtpbp1/Sf1
## GO:0031331                            Aurka/Khsrp/Upf1/Fto/Cdc20/Supt5/Fzr1/Bag6/Mettl3/Bcap31/Rnf185/Gtpbp1/Psen1/Hmox1/Gsk3a
## GO:0042254                                          Mybbp1a/Ddx47/Ppan/Rcl1/Drosha/Rpl6/Ddx54/Ddx27/Mrpl10/Ran/Noc2l/Abt1/Nop2
## GO:0006397                  Snrpb/Snrnp70/Sf3a2/Srsf4/Ddx39b/Khsrp/Scaf1/Ddx47/Txnl4a/Snrpa/Supt5/Snrpn/Puf60/U2af2/Mettl3/Sf1
##            Count
## GO:0022613    19
## GO:0008380    17
## GO:1903311    13
## GO:0031331    15
## GO:0042254    13
## GO:0006397    16</code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">positive_go_results</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/loading_go_positive-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="negative-loading-genes-go" class="section level4">
<h4 class="hasAnchor">
<a href="#negative-loading-genes-go" class="anchor"></a>negative loading genes GO</h4>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># negative loading genes</span>
<span class="va">negative_go_results</span><span class="op">=</span><span class="va">loading.go.results</span><span class="op">[[</span><span class="st">"Negative"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">negative_go_results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##                    ID
## GO:0032535 GO:0032535
## GO:0045666 GO:0045666
## GO:0031346 GO:0031346
## GO:0010976 GO:0010976
## GO:0048813 GO:0048813
## GO:0010769 GO:0010769
##                                                             Description
## GO:0032535                        regulation of cellular component size
## GO:0045666                positive regulation of neuron differentiation
## GO:0031346          positive regulation of cell projection organization
## GO:0010976         positive regulation of neuron projection development
## GO:0048813                                       dendrite morphogenesis
## GO:0010769 regulation of cell morphogenesis involved in differentiation
##            GeneRatio   BgRatio       pvalue    p.adjust      qvalue
## GO:0032535    14/186 413/23328 6.228504e-06 0.006598163 0.005761002
## GO:0045666    15/186 472/23328 6.270266e-06 0.006598163 0.005761002
## GO:0031346    15/186 478/23328 7.290788e-06 0.006598163 0.005761002
## GO:0010976    13/186 372/23328 9.751787e-06 0.006619025 0.005779217
## GO:0048813     9/186 182/23328 1.622507e-05 0.008810210 0.007692389
## GO:0010769    12/186 351/23328 2.677346e-05 0.012114991 0.010577866
##                                                                                            geneID
## GO:0032535    Bmpr2/Map3k1/Sema3d/Dpysl2/Aqp4/Dcx/Pten/Nefl/Arhgap5/Pak3/Wnk3/Cracd/Map2/Tmsb15b1
## GO:0045666 Bmpr2/Cntn1/Fut9/Neurod1/Ube2v2/Cnr1/Dcx/Ptprz1/Tmem106b/Ror1/Pten/Nefl/Pak3/Qk/Elavl4
## GO:0031346   Bmpr2/Dock11/Cntn1/Fut9/Ube2v2/Cnr1/Dcx/Apc/Ptprz1/Tmem106b/Ror1/Nefl/Pak3/Qk/Elavl4
## GO:0010976              Bmpr2/Cntn1/Fut9/Ube2v2/Cnr1/Dcx/Ptprz1/Tmem106b/Ror1/Nefl/Pak3/Qk/Elavl4
## GO:0048813                                Ube3a/Hecw1/Adam10/Dcx/Ptprz1/Tmem106b/Pten/Pak3/Elavl4
## GO:0010769                  Bmpr2/Ube3a/Sema3d/Dpysl2/Hecw1/Adam10/Dcx/Ptprz1/Pten/Nefl/Pak3/Map2
##            Count
## GO:0032535    14
## GO:0045666    15
## GO:0031346    15
## GO:0010976    13
## GO:0048813     9
## GO:0010769    12</code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">negative_go_results</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/loading_go_negative-1.png" width="672" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
<div id="d-visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#d-visualization" class="anchor"></a>3D visualization</h2>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PCA3D.html">PCA3D</a></span><span class="op">(</span>pca <span class="op">=</span> <span class="va">pca.res</span>,color.key <span class="op">=</span> <span class="st">"condition"</span>,main <span class="op">=</span> <span class="st">"3D PCA"</span><span class="op">)</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/pca_3d-1.png" width="480" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="differential-expression-gene-visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-expression-gene-visualization" class="anchor"></a>Differential Expression Gene Visualization</h1>
<div id="differential-expression-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-analysis" class="anchor"></a>Differential Expression Analysis</h2>
<p>First you should get differential expression genes. For <code>DESeq2</code> results:</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set control level</span>
<span class="va">dds</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html">relevel</a></span><span class="op">(</span><span class="va">dds</span><span class="op">$</span><span class="va">condition</span>, ref <span class="op">=</span> <span class="st">"WT"</span><span class="op">)</span>
<span class="co"># conduct differential expressed genes analysis</span>
<span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># extract results</span>
<span class="va">dds.results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds</span>,contrast<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"condition"</span>,<span class="st">'KO'</span>,<span class="st">'WT'</span><span class="op">)</span><span class="op">)</span>
<span class="va">dds.results.ordered</span> <span class="op">&lt;-</span> <span class="va">dds.results</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">dds.results</span><span class="op">$</span><span class="va">log2FoldChange</span>,decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dds.results.ordered</span><span class="op">)</span></code></pre></div>
<pre><code>## log2 fold change (MLE): condition KO vs WT 
## Wald test p-value: condition KO vs WT 
## DataFrame with 6 rows and 6 columns
##                        baseMean log2FoldChange     lfcSE      stat      pvalue
##                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
## ENSMUSG00000025934.15   2.01093        4.46030   3.01742   1.47818          NA
## ENSMUSG00000105597.1    1.64376        4.16931   2.98872   1.39502 0.163011113
## ENSMUSG00000109866.1    2.32064        4.14573   1.22504   3.38415 0.000713987
## ENSMUSG00000085819.7    2.23441        4.07496   1.65939   2.45569 0.014061354
## ENSMUSG00000102474.1    1.52867        4.06781   1.80781   2.25013 0.024440396
## ENSMUSG00000042677.7    1.49781        4.05394   1.71350   2.36588 0.017987495
##                            padj
##                       &lt;numeric&gt;
## ENSMUSG00000025934.15        NA
## ENSMUSG00000105597.1    0.99973
## ENSMUSG00000109866.1    0.99973
## ENSMUSG00000085819.7    0.99973
## ENSMUSG00000102474.1    0.99973
## ENSMUSG00000042677.7    0.99973</code></pre>
<p>For <code>edgeR</code> results:</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span><span class="op">)</span></code></pre></div>
<pre><code>## 载入需要的程辑包：limma</code></pre>
<pre><code>## 
## 载入程辑包：'limma'</code></pre>
<pre><code>## The following object is masked from 'package:DESeq2':
## 
##     plotMA</code></pre>
<pre><code>## The following object is masked from 'package:BiocGenerics':
## 
##     plotMA</code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snon.edgeR</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">count.matrix</span>, group<span class="op">=</span><span class="va">meta.info</span><span class="op">$</span><span class="va">condition</span><span class="op">)</span>
<span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html">filterByExpr</a></span><span class="op">(</span><span class="va">snon.edgeR</span>,min.count<span class="op">=</span><span class="fl">10</span><span class="op">)</span>
<span class="va">snon.edgeR</span> <span class="op">&lt;-</span> <span class="va">snon.edgeR</span><span class="op">[</span><span class="va">keep</span>, , keep.lib.sizes<span class="op">=</span><span class="cn">FALSE</span><span class="op">]</span>
<span class="va">snon.edgeR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span><span class="op">(</span><span class="va">snon.edgeR</span><span class="op">)</span>
<span class="va">snon.edgeR</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html">relevel</a></span><span class="op">(</span><span class="va">snon.edgeR</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span>, ref<span class="op">=</span><span class="st">"WT"</span><span class="op">)</span>
<span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="va">snon.edgeR</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span><span class="op">)</span>
<span class="va">snon.edgeR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateDisp.html">estimateDisp</a></span><span class="op">(</span><span class="va">snon.edgeR</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html">glmQLFit</a></span><span class="op">(</span><span class="va">snon.edgeR</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">qlf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html">glmQLFTest</a></span><span class="op">(</span><span class="va">fit</span>, coef<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
<span class="va">all.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html">topTags</a></span><span class="op">(</span><span class="va">qlf</span>,n<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">snon.edgeR</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">table</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">all.res</span><span class="op">)</span></code></pre></div>
<pre><code>##                            logFC    logCPM        F       PValue         FDR
## ENSMUSG00000039620.17  2.1925630 1.7076072 70.78781 2.686429e-07 0.003577786
## ENSMUSG00000046152.16 -0.7238661 4.9567684 36.89615 1.550592e-05 0.103253915
## ENSMUSG00000022376.8  -1.7356149 0.3243428 21.61086 2.617267e-04 0.998523400
## ENSMUSG00000027660.16 -0.9581197 5.3679317 19.98078 3.791045e-04 0.998523400
## ENSMUSG00000031478.16  0.8734604 2.4708238 18.99645 4.783255e-04 0.998523400
## ENSMUSG00000079469.10 -1.2129059 1.4296852 18.36878 5.568020e-04 0.998523400</code></pre>
<hr>
</div>
<div id="visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h2>
<div id="volcanoplot" class="section level3">
<h3 class="hasAnchor">
<a href="#volcanoplot" class="anchor"></a>VolcanoPlot</h3>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># VolcanoPlot for DESeq2</span>
<span class="fu"><a href="../reference/VolcanoPlot.html">VolcanoPlot</a></span><span class="op">(</span><span class="va">dds.results.ordered</span>,signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,
            point.alpha <span class="op">=</span> <span class="fl">0.8</span>, label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span>,tick.trans <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/volcanoplot-1.png" width="480" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># VolcanoPlot for edgeR</span>
<span class="fu"><a href="../reference/VolcanoPlot.html">VolcanoPlot</a></span><span class="op">(</span><span class="va">all.res</span>,signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,point.alpha <span class="op">=</span> <span class="fl">0.8</span>,
            label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span>,tick.trans <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/volcanoplot-2.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="scatterplot" class="section level3">
<h3 class="hasAnchor">
<a href="#scatterplot" class="anchor"></a>ScatterPlot</h3>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ScatterPlot for DESeq2</span>
<span class="fu"><a href="../reference/ScatterPlot.html">ScatterPlot</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,deres <span class="op">=</span> <span class="va">dds.results.ordered</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
            ref.group <span class="op">=</span> <span class="st">"WT"</span>,signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num <span class="op">=</span> <span class="fl">2</span>,
            point.alpha <span class="op">=</span> <span class="fl">0.8</span>,label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/ScatterPlot-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ScatterPlot for edgeR</span>
<span class="fu"><a href="../reference/ScatterPlot.html">ScatterPlot</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">snon.edgeR</span>,deres <span class="op">=</span> <span class="va">all.res</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
            ref.group <span class="op">=</span> <span class="st">"WT"</span>,signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num <span class="op">=</span> <span class="fl">2</span>,
            point.alpha <span class="op">=</span> <span class="fl">0.8</span>,label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/ScatterPlot-2.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div id="maplot" class="section level3">
<h3 class="hasAnchor">
<a href="#maplot" class="anchor"></a>MAPlot</h3>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># MAPlot for DESeq2</span>
<span class="fu"><a href="../reference/MAPlot.html">MAPlot</a></span><span class="op">(</span><span class="va">dds.results.ordered</span>,signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,
       point.alpha <span class="op">=</span> <span class="fl">0.8</span>, label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/MAPlot-1.png" width="768" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># MAPlot for edgeR</span>
<span class="fu"><a href="../reference/MAPlot.html">MAPlot</a></span><span class="op">(</span><span class="va">all.res</span>,signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,point.alpha <span class="op">=</span> <span class="fl">0.8</span>,
       label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/MAPlot-2.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div id="rankplot" class="section level3">
<h3 class="hasAnchor">
<a href="#rankplot" class="anchor"></a>RankPlot</h3>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># RankPlot for DESeq2</span>
<span class="fu"><a href="../reference/RankPlot.html">RankPlot</a></span><span class="op">(</span><span class="va">dds.results.ordered</span>,signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,
         point.alpha <span class="op">=</span> <span class="fl">0.8</span>, label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/RankPlot-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># RankPlot for edgeR</span>
<span class="fu"><a href="../reference/RankPlot.html">RankPlot</a></span><span class="op">(</span><span class="va">all.res</span>,signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,point.alpha <span class="op">=</span> <span class="fl">0.8</span>,
         label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/RankPlot-2.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div id="geneplot" class="section level3">
<h3 class="hasAnchor">
<a href="#geneplot" class="anchor"></a>GenePlot</h3>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># GenePlot for DESeq2</span>
<span class="fu"><a href="../reference/GenePlot.html">GenePlot</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,deres <span class="op">=</span> <span class="va">dds.results.ordered</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
         ref.group <span class="op">=</span> <span class="st">"WT"</span>,fill.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"blue"</span><span class="op">)</span>, fill.alpha <span class="op">=</span> <span class="fl">0.8</span>,
         gene.num <span class="op">=</span><span class="fl">2</span>,signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/GenePlot-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># GenePlot for edgeR</span>
<span class="fu"><a href="../reference/GenePlot.html">GenePlot</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">snon.edgeR</span>,deres <span class="op">=</span> <span class="va">all.res</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
         ref.group <span class="op">=</span> <span class="st">"WT"</span>,fill.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"blue"</span><span class="op">)</span>,fill.alpha <span class="op">=</span> <span class="fl">0.8</span>,
         gene.num <span class="op">=</span><span class="fl">2</span>,signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/GenePlot-2.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div id="deheatmap" class="section level3">
<h3 class="hasAnchor">
<a href="#deheatmap" class="anchor"></a>DEHeatmap</h3>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DEHeatmap for DESeq2</span>
<span class="fu"><a href="../reference/DEHeatmap.html">DEHeatmap</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,deres <span class="op">=</span> <span class="va">dds.results.ordered</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
          ref.group <span class="op">=</span> <span class="st">"WT"</span>, signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/DEHeatmap-1.png" width="960" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DEHeatmap for edgeR</span>
<span class="fu"><a href="../reference/DEHeatmap.html">DEHeatmap</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">snon.edgeR</span>,deres <span class="op">=</span> <span class="va">all.res</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
          ref.group <span class="op">=</span> <span class="st">"WT"</span>, signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DEbPeak_All_files/figure-html/DEHeatmap-2.png" width="960" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
<div id="functional-enrichment-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#functional-enrichment-analysis" class="anchor"></a>Functional Enrichment Analysis</h1>
<div id="go-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#go-enrichment" class="anchor"></a>GO enrichment</h2>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># save results to working directory</span>
<span class="co"># ConductFE(deres = dds.results.ordered, signif = "pvalue", l2fc.threshold = 0.3,enrich.type = "GO")</span>
<span class="va">all.go.results</span><span class="op">=</span><span class="fu"><a href="../reference/ConductFE.html">ConductFE</a></span><span class="op">(</span>deres <span class="op">=</span> <span class="va">dds.results.ordered</span>, signif <span class="op">=</span> <span class="st">"pvalue"</span>, l2fc.threshold <span class="op">=</span> <span class="fl">0.3</span>,enrich.type <span class="op">=</span> <span class="st">"GO"</span>,str.width <span class="op">=</span> <span class="fl">50</span>, save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Convert ENSEMBL to ENTREZID!</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## conduct ALL GO enrichment analysis on: UP</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## Convert ENSEMBL to ENTREZID!</code></pre>
<pre><code>## 'select()' returned 1:1 mapping between keys and columns</code></pre>
<pre><code>## conduct ALL GO enrichment analysis on: DOWN</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<div id="up-regulated" class="section level3">
<h3 class="hasAnchor">
<a href="#up-regulated" class="anchor"></a>Up regulated</h3>
<div id="overview" class="section level4">
<h4 class="hasAnchor">
<a href="#overview" class="anchor"></a>overview</h4>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.go.res</span><span class="op">=</span><span class="va">all.go.results</span><span class="op">[[</span><span class="st">"UP"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="va">up.go.res.table</span><span class="op">=</span><span class="va">up.go.res</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">up.go.res.table</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                              Description
## GO:0097060       CC GO:0097060                        synaptic membrane
## GO:0099240       CC GO:0099240 intrinsic component of synaptic membrane
## GO:0014069       CC GO:0014069                     postsynaptic density
## GO:0032279       CC GO:0032279                       asymmetric synapse
## GO:0098839       CC GO:0098839            postsynaptic density membrane
## GO:0098984       CC GO:0098984                 neuron to neuron synapse
##            GeneRatio   BgRatio       pvalue   p.adjust     qvalue
## GO:0097060    15/231 470/23271 7.783967e-05 0.01822125 0.01447238
## GO:0099240    10/231 244/23271 1.732798e-04 0.01822125 0.01447238
## GO:0014069    13/231 400/23271 1.957825e-04 0.01822125 0.01447238
## GO:0032279    13/231 405/23271 2.208636e-04 0.01822125 0.01447238
## GO:0098839     6/231  97/23271 4.175021e-04 0.01868255 0.01483877
## GO:0098984    13/231 433/23271 4.183638e-04 0.01868255 0.01483877
##                                                                                                  geneID
## GO:0097060 Slc6a2/Met/Nt5e/Lpar2/Grik1/Tenm2/Tiam1/Abhd6/Rgs7bp/Epha7/Snap25/Atp2b2/Adcy1/Adgrl2/Cacng2
## GO:0099240                             Lpar2/Grik1/Abhd6/Rgs7bp/Epha7/Snap25/Atp2b2/Adcy1/Adgrl2/Cacng2
## GO:0014069              Nefh/Met/Grik1/Rapgef4/Tiam1/Rgs7bp/Epha7/Ctnnd2/Map1b/Atp2b2/Pak2/Adcy1/Cacng2
## GO:0032279              Nefh/Met/Grik1/Rapgef4/Tiam1/Rgs7bp/Epha7/Ctnnd2/Map1b/Atp2b2/Pak2/Adcy1/Cacng2
## GO:0098839                                                       Tiam1/Rgs7bp/Epha7/Atp2b2/Adcy1/Cacng2
## GO:0098984              Nefh/Met/Grik1/Rapgef4/Tiam1/Rgs7bp/Epha7/Ctnnd2/Map1b/Atp2b2/Pak2/Adcy1/Cacng2
##            Count
## GO:0097060    15
## GO:0099240    10
## GO:0014069    13
## GO:0032279    13
## GO:0098839     6
## GO:0098984    13</code></pre>
</div>
<div id="plot" class="section level4">
<h4 class="hasAnchor">
<a href="#plot" class="anchor"></a>plot</h4>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.go.res</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/go_up_plot-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="down-regulated" class="section level3">
<h3 class="hasAnchor">
<a href="#down-regulated" class="anchor"></a>Down regulated</h3>
<div id="overview-1" class="section level4">
<h4 class="hasAnchor">
<a href="#overview-1" class="anchor"></a>overview</h4>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.go.res</span><span class="op">=</span><span class="va">all.go.results</span><span class="op">[[</span><span class="st">"DOWN"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="va">down.go.res.table</span> <span class="op">=</span> <span class="va">down.go.res</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">down.go.res.table</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID       Description GeneRatio   BgRatio
## GO:0005604       CC GO:0005604 basement membrane     7/229 114/23271
##                  pvalue   p.adjust     qvalue
## GO:0005604 0.0001374573 0.04178703 0.04007966
##                                                geneID Count
## GO:0005604 Ntn4/Ntn1/Col4a2/Egflam/Papln/Fras1/Col4a4     7</code></pre>
</div>
<div id="plot-1" class="section level4">
<h4 class="hasAnchor">
<a href="#plot-1" class="anchor"></a>plot</h4>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.go.res</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/go_down_plot-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
<div id="kegg-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#kegg-enrichment" class="anchor"></a>KEGG enrichment</h2>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># save results to working directory</span>
<span class="co"># ConductFE(deres = dds.results.ordered, signif = "pvalue", l2fc.threshold = 0.3,enrich.type = "KEGG")</span>
<span class="va">all.kegg.results</span><span class="op">=</span><span class="fu"><a href="../reference/ConductFE.html">ConductFE</a></span><span class="op">(</span>deres <span class="op">=</span> <span class="va">dds.results.ordered</span>, signif <span class="op">=</span> <span class="st">"pvalue"</span>, l2fc.threshold <span class="op">=</span> <span class="fl">0.3</span>,enrich.type <span class="op">=</span> <span class="st">"KEGG"</span>, str.width <span class="op">=</span> <span class="fl">50</span>, save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Convert ENSEMBL to ENTREZID!</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## conduct KEGG enrichment analysis.</code></pre>
<pre><code>## Reading KEGG annotation online:
## 
## Reading KEGG annotation online:</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## Convert ENSEMBL to ENTREZID!</code></pre>
<pre><code>## 'select()' returned 1:1 mapping between keys and columns</code></pre>
<pre><code>## conduct KEGG enrichment analysis.</code></pre>
<div id="up" class="section level3">
<h3 class="hasAnchor">
<a href="#up" class="anchor"></a>Up</h3>
<div id="overview-2" class="section level4">
<h4 class="hasAnchor">
<a href="#overview-2" class="anchor"></a>overview</h4>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.kegg.res</span><span class="op">=</span><span class="va">all.kegg.results</span><span class="op">[[</span><span class="st">"UP"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"KEGG"</span><span class="op">]</span><span class="op">]</span>
<span class="va">up.kegg.res.table</span><span class="op">=</span><span class="va">up.kegg.res</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">up.kegg.res.table</span><span class="op">)</span></code></pre></div>
<pre><code>##                ID            Description GeneRatio  BgRatio       pvalue
## mmu04911 mmu04911      Insulin secretion     7/101  86/9030 4.633148e-05
## mmu04024 mmu04024 cAMP signaling pathway    10/101 220/9030 1.664655e-04
##            p.adjust      qvalue
## mmu04911 0.01014659 0.008973675
## mmu04024 0.01822797 0.016120868
##                                                                    geneID Count
## mmu04911                  Kcnmb2/Kcnn3/Creb5/Rapgef4/Creb3l2/Snap25/Adcy1     7
## mmu04024 Cftr/Atp2a3/Creb5/Rapgef4/Tiam1/Creb3l2/Sox9/Atp2b2/Crebbp/Adcy1    10</code></pre>
</div>
<div id="plot-2" class="section level4">
<h4 class="hasAnchor">
<a href="#plot-2" class="anchor"></a>plot</h4>
<div class="sourceCode" id="cb128"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.kegg.res</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/kegg_up_plot-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="down" class="section level3">
<h3 class="hasAnchor">
<a href="#down" class="anchor"></a>Down</h3>
<div id="overview-3" class="section level4">
<h4 class="hasAnchor">
<a href="#overview-3" class="anchor"></a>overview</h4>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.kegg.res</span><span class="op">=</span><span class="va">all.kegg.results</span><span class="op">[[</span><span class="st">"DOWN"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"KEGG"</span><span class="op">]</span><span class="op">]</span>
<span class="va">down.kegg.res.table</span><span class="op">=</span><span class="va">down.kegg.res</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">down.kegg.res.table</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] ID          Description GeneRatio   BgRatio     pvalue      p.adjust   
## [7] qvalue      geneID      Count      
## &lt;0 行&gt; (或0-长度的row.names)</code></pre>
</div>
<div id="plot-3" class="section level4">
<h4 class="hasAnchor">
<a href="#plot-3" class="anchor"></a>plot</h4>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.kegg.res</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/kegg_down_plot-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
<div id="gsea" class="section level2">
<h2 class="hasAnchor">
<a href="#gsea" class="anchor"></a>GSEA</h2>
<p>For human, we can download gmt file from <a href="http://www.gsea-msigdb.org/gsea/downloads.jsp">MSigDB</a>. For other species, we can obtain gene sets via <a href="https://igordot.github.io/msigdbr/reference/msigdbr.html">msigdbr</a>.</p>
<div id="msigdbr-gene-sets" class="section level3">
<h3 class="hasAnchor">
<a href="#msigdbr-gene-sets" class="anchor"></a>msigdbr gene sets</h3>
<div class="sourceCode" id="cb132"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/">msigdbr</a></span><span class="op">)</span>
<span class="co"># list all possible species</span>
<span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr_species.html">msigdbr_species</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 20 x 2
##    species_name                species_common_name                              
##    &lt;chr&gt;                       &lt;chr&gt;                                            
##  1 Anolis carolinensis         Carolina anole, green anole                      
##  2 Bos taurus                  bovine, cattle, cow, dairy cow, domestic cattle,…
##  3 Caenorhabditis elegans      roundworm                                        
##  4 Canis lupus familiaris      dog, dogs                                        
##  5 Danio rerio                 leopard danio, zebra danio, zebra fish, zebrafish
##  6 Drosophila melanogaster     fruit fly                                        
##  7 Equus caballus              domestic horse, equine, horse                    
##  8 Felis catus                 cat, cats, domestic cat                          
##  9 Gallus gallus               bantam, chicken, chickens, Gallus domesticus     
## 10 Homo sapiens                human                                            
## 11 Macaca mulatta              rhesus macaque, rhesus macaques, Rhesus monkey, …
## 12 Monodelphis domestica       gray short-tailed opossum                        
## 13 Mus musculus                house mouse, mouse                               
## 14 Ornithorhynchus anatinus    duck-billed platypus, duckbill platypus, platypus
## 15 Pan troglodytes             chimpanzee                                       
## 16 Rattus norvegicus           brown rat, Norway rat, rat, rats                 
## 17 Saccharomyces cerevisiae    baker's yeast, brewer's yeast, S. cerevisiae     
## 18 Schizosaccharomyces pombe … &lt;NA&gt;                                             
## 19 Sus scrofa                  pig, pigs, swine, wild boar                      
## 20 Xenopus tropicalis          tropical clawed frog, western clawed frog</code></pre>
<div class="sourceCode" id="cb134"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create gene sets</span>
<span class="va">m_t2g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Mus musculus"</span>, category <span class="op">=</span> <span class="st">"C5"</span>, subcategory <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">gs_name</span>, <span class="va">entrez_gene</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">m_t2g</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   gs_name                                          entrez_gene
##   &lt;chr&gt;                                                  &lt;int&gt;
## 1 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS       67618
## 2 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS      107747
## 3 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS      216188
## 4 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS      108156
## 5 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS      270685
## 6 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS      665563</code></pre>
<hr>
</div>
<div id="conduct-gsea" class="section level3">
<h3 class="hasAnchor">
<a href="#conduct-gsea" class="anchor"></a>Conduct GSEA</h3>
<div class="sourceCode" id="cb136"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gsea.results</span><span class="op">=</span><span class="fu"><a href="../reference/ConductGSEA.html">ConductGSEA</a></span><span class="op">(</span>deres <span class="op">=</span> <span class="va">dds.results.ordered</span>,gmt.file <span class="op">=</span> <span class="cn">NULL</span>,gene.sets <span class="op">=</span> <span class="va">m_t2g</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Convert ENSEMBL to ENTREZID!</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## conduct GSEA anaysis.</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## no term enriched under specific pvalueCutoff...</code></pre>
<hr>
</div>
<div id="results-overview" class="section level3">
<h3 class="hasAnchor">
<a href="#results-overview" class="anchor"></a>results overview</h3>
<div class="sourceCode" id="cb144"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gsea.results.table</span><span class="op">=</span><span class="va">gsea.results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gsea.results.table</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] ID              Description     setSize         enrichmentScore
## [5] NES             pvalue          p.adjust        qvalues        
## &lt;0 行&gt; (或0-长度的row.names)</code></pre>
<hr>
</div>
<div id="results-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#results-plot" class="anchor"></a>results plot</h3>
<div class="sourceCode" id="cb146"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gsea.results</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<pre><code>## NULL</code></pre>
<hr>
</div>
</div>
</div>
<div id="all-in-one-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#all-in-one-analysis" class="anchor"></a>All-in-one analysis</h1>
<p>To enhance the usability of the tool, we create a all-in-one analysis command, all results will be saved in specific directory:</p>
<div class="sourceCode" id="cb148"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/">msigdbr</a></span><span class="op">)</span>
<span class="va">m_t2g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Mus musculus"</span>, category <span class="op">=</span> <span class="st">"C5"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">gs_name</span>, <span class="va">entrez_gene</span><span class="op">)</span>
<span class="va">count.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"snon_count.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">meta.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"snon_meta.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="fu"><a href="../reference/ConductDESeq2.html">ConductDESeq2</a></span><span class="op">(</span>count.matrix.file <span class="op">=</span> <span class="va">count.file</span>, meta.file <span class="op">=</span> <span class="va">meta.file</span>, out.folder<span class="op">=</span><span class="st">"/home/songyabing/R/learn/tmp/dekit/AllInOne"</span>,
              signif <span class="op">=</span> <span class="st">"pvalue"</span>, l2fc.threshold <span class="op">=</span> <span class="fl">0.3</span>,
              group.key <span class="op">=</span> <span class="st">"condition"</span>, ref.group <span class="op">=</span> <span class="st">"WT"</span>,gmt.file <span class="op">=</span> <span class="cn">NULL</span>, gene.sets <span class="op">=</span> <span class="va">m_t2g</span><span class="op">)</span></code></pre></div>
<p>The structrue of results folder:</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="ex">tree</span> /home/songyabing/R/learn/tmp/dekit/AllInOne</span></code></pre></div>
<pre><code>## /home/songyabing/R/learn/tmp/dekit/AllInOne
## ├── CountQC_CPM.pdf
## ├── CountQC_saturation.pdf
## ├── DEG
## │   ├── Condition_KO_WT_all.csv
## │   ├── Condition_KO_WT_pvalue0.05_FC0.3.csv
## │   ├── DEG_GenePlot.pdf
## │   ├── DEG_Heatmap.pdf
## │   ├── DEG_MAPlot.pdf
## │   ├── DEG_RankPlot.pdf
## │   ├── DEG_ScatterPlot.pdf
## │   ├── DEG_VolcanoPlot.pdf
## │   └── normalized_counts.csv
## ├── DEkit_all_in_one.RData
## ├── FE
## │   ├── DOWN_ALL_GO.csv
## │   ├── DOWN_Biological_Process.png
## │   ├── DOWN_Cellular_Component.png
## │   ├── DOWN_Functional_Enrichment.pdf
## │   ├── DOWN_KEGG.csv
## │   ├── DOWN_KEGG_Enrichment.png
## │   ├── DOWN_Molecular_Function.png
## │   ├── UP_ALL_GO.csv
## │   ├── UP_Biological_Process.png
## │   ├── UP_Cellular_Component.png
## │   ├── UP_Functional_Enrichment.pdf
## │   ├── UP_KEGG.csv
## │   ├── UP_KEGG_Enrichment.png
## │   └── UP_Molecular_Function.png
## ├── GSEA
## │   └── GSEA_enrich_result.csv
## ├── PCA
## │   ├── PC1
## │   │   ├── Negative
## │   │   │   ├── PC1_Negative_ALL_GO.csv
## │   │   │   └── PC1_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC1_Positive_ALL_GO.csv
## │   │       └── PC1_Positive_KEGG.csv
## │   ├── PC1_Negative_Biological_Process.png
## │   ├── PC1_Negative_Cellular_Component.png
## │   ├── PC1_Negative_Functional_Enrichment.pdf
## │   ├── PC1_Negative_KEGG_Enrichment.png
## │   ├── PC1_Negative_Molecular_Function.png
## │   ├── PC1_Positive_Biological_Process.png
## │   ├── PC1_Positive_Cellular_Component.png
## │   ├── PC1_Positive_Functional_Enrichment.pdf
## │   ├── PC1_Positive_KEGG_Enrichment.png
## │   ├── PC1_Positive_Molecular_Function.png
## │   ├── PC2
## │   │   ├── Negative
## │   │   │   ├── PC2_Negative_ALL_GO.csv
## │   │   │   └── PC2_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC2_Positive_ALL_GO.csv
## │   │       └── PC2_Positive_KEGG.csv
## │   ├── PC2_Negative_Biological_Process.png
## │   ├── PC2_Negative_Cellular_Component.png
## │   ├── PC2_Negative_Functional_Enrichment.pdf
## │   ├── PC2_Negative_KEGG_Enrichment.png
## │   ├── PC2_Negative_Molecular_Function.png
## │   ├── PC2_Positive_Biological_Process.png
## │   ├── PC2_Positive_Cellular_Component.png
## │   ├── PC2_Positive_Functional_Enrichment.pdf
## │   ├── PC2_Positive_KEGG_Enrichment.png
## │   ├── PC2_Positive_Molecular_Function.png
## │   ├── PC3
## │   │   ├── Negative
## │   │   │   ├── PC3_Negative_ALL_GO.csv
## │   │   │   └── PC3_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC3_Positive_ALL_GO.csv
## │   │       └── PC3_Positive_KEGG.csv
## │   ├── PC3_Negative_Biological_Process.png
## │   ├── PC3_Negative_Cellular_Component.png
## │   ├── PC3_Negative_Functional_Enrichment.pdf
## │   ├── PC3_Negative_KEGG_Enrichment.png
## │   ├── PC3_Negative_Molecular_Function.png
## │   ├── PC3_Positive_Biological_Process.png
## │   ├── PC3_Positive_Cellular_Component.png
## │   ├── PC3_Positive_Functional_Enrichment.pdf
## │   ├── PC3_Positive_KEGG_Enrichment.png
## │   ├── PC3_Positive_Molecular_Function.png
## │   ├── PC4
## │   │   ├── Negative
## │   │   │   ├── PC4_Negative_ALL_GO.csv
## │   │   │   └── PC4_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC4_Positive_ALL_GO.csv
## │   │       └── PC4_Positive_KEGG.csv
## │   ├── PC4_Negative_Biological_Process.png
## │   ├── PC4_Negative_Cellular_Component.png
## │   ├── PC4_Negative_Functional_Enrichment.pdf
## │   ├── PC4_Negative_KEGG_Enrichment.png
## │   ├── PC4_Negative_Molecular_Function.png
## │   ├── PC4_Positive_Biological_Process.png
## │   ├── PC4_Positive_Cellular_Component.png
## │   ├── PC4_Positive_Functional_Enrichment.pdf
## │   ├── PC4_Positive_KEGG_Enrichment.png
## │   ├── PC4_Positive_Molecular_Function.png
## │   ├── PC5
## │   │   ├── Negative
## │   │   │   ├── PC5_Negative_ALL_GO.csv
## │   │   │   └── PC5_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC5_Positive_ALL_GO.csv
## │   │       └── PC5_Positive_KEGG.csv
## │   ├── PC5_Negative_Biological_Process.png
## │   ├── PC5_Negative_Cellular_Component.png
## │   ├── PC5_Negative_Functional_Enrichment.pdf
## │   ├── PC5_Negative_KEGG_Enrichment.png
## │   ├── PC5_Negative_Molecular_Function.png
## │   ├── PC5_Positive_Biological_Process.png
## │   ├── PC5_Positive_Cellular_Component.png
## │   ├── PC5_Positive_Functional_Enrichment.pdf
## │   ├── PC5_Positive_KEGG_Enrichment.png
## │   ├── PC5_Positive_Molecular_Function.png
## │   ├── PCA_3DPCA.pdf
## │   ├── PCA_biplot.pdf
## │   ├── PCA_loading_bar.pdf
## │   ├── PCA_loading_heat.pdf
## │   ├── PCA_overview.pdf
## │   ├── PCA_pairs_plot.pdf
## │   └── PCA_screen_plot.pdf
## └── SampleQC_dist_pcc.pdf
## 
## 19 directories, 105 files</code></pre>
<hr>
</div>
<div id="integrate-rna-seq-and-chip-seqatac-seq" class="section level1">
<h1 class="hasAnchor">
<a href="#integrate-rna-seq-and-chip-seqatac-seq" class="anchor"></a>Integrate RNA-seq and ChIP-seq/ATAC-seq</h1>
<p>The data used here contains RNA-seq and ChIP-seq datasets from <a href="https://www.science.org/doi/10.1126/sciadv.aau8389">RUNX represses Pmp22 to drive neurofibromagenesis</a>:</p>
<ul>
<li>RNA-seq: two genotypes and three samples per genotype, the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122774">GSE122774</a>
</li>
<li>ChIP-seq: two genotypes and one sample per genotype, the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122775">GSE122775</a>
</li>
<li>ATAC-seq: two genotypes and one sample per genotype, the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122776">GSE122776</a>
</li>
</ul>
<div id="prepare-degs" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-degs" class="anchor"></a>Prepare DEGs</h2>
<div class="sourceCode" id="cb151"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># prepare count matrix and metadata</span>
<span class="va">debchip.count.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"debchip_count.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">debchip.meta.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"debchip_meta.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">debchip.count.matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">debchip.count.file</span>, header <span class="op">=</span> <span class="cn">T</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span>
<span class="va">debchip.meta.info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">debchip.meta.file</span>, header <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="co"># create DESeqDataSet object</span>
<span class="va">debchip.dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span><span class="op">(</span>countData <span class="op">=</span> <span class="va">debchip.count.matrix</span>, 
                                              colData <span class="op">=</span> <span class="va">debchip.meta.info</span>, 
                                              design <span class="op">=</span> <span class="op">~</span><span class="va">condition</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<div class="sourceCode" id="cb153"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set control level</span>
<span class="va">debchip.dds</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html">relevel</a></span><span class="op">(</span><span class="va">debchip.dds</span><span class="op">$</span><span class="va">condition</span>, ref <span class="op">=</span> <span class="st">"NF"</span><span class="op">)</span>
<span class="co"># conduct differential expressed genes analysis</span>
<span class="va">debchip.dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span><span class="op">(</span><span class="va">debchip.dds</span><span class="op">)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb160"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># extract results</span>
<span class="va">debchip.dds.results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">debchip.dds</span>,contrast<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"condition"</span>,<span class="st">'RX'</span>,<span class="st">'NF'</span><span class="op">)</span><span class="op">)</span>
<span class="va">debchip.dds.results.ordered</span> <span class="op">&lt;-</span> <span class="va">debchip.dds.results</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">debchip.dds.results</span><span class="op">$</span><span class="va">log2FoldChange</span>,decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">debchip.dds.results.ordered</span><span class="op">)</span></code></pre></div>
<pre><code>## log2 fold change (MLE): condition RX vs NF 
## Wald test p-value: condition RX vs NF 
## DataFrame with 6 rows and 6 columns
##          baseMean log2FoldChange     lfcSE      stat      pvalue        padj
##         &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
## Sycp1    12.07541        7.15056   3.91075   1.82844 6.74840e-02 1.81006e-01
## Gm16532  10.07020        6.89407   1.74281   3.95573 7.63014e-05 7.17146e-04
## Hs3st4   18.88787        6.82852   1.42952   4.77679 1.78117e-06 2.83848e-05
## Nell1    34.13399        6.80718   1.34257   5.07027 3.97260e-07 7.79916e-06
## Sptssb    9.39376        6.78558   1.52948   4.43651 9.14275e-06 1.16684e-04
## Myo3a     7.19992        6.41077   1.74865   3.66612 2.46259e-04 1.94151e-03</code></pre>
<hr>
</div>
<div id="prepare-chip-seq-data" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-chip-seq-data" class="anchor"></a>Prepare ChIP-seq data</h2>
<div id="consensus-peaks" class="section level3">
<h3 class="hasAnchor">
<a href="#consensus-peaks" class="anchor"></a>Consensus peaks</h3>
<p>In this step, we will get consensus peaks with <a href="https://genometric.github.io/MSPC/docs/quick_start/">MSPC</a> when multiple peak files are available, but when there is only one peak file, we will use it directly (<strong>make sure this peak file contains five columns: “chr”, “start”, “stop”, “name”, “score”</strong>).</p>
<div class="sourceCode" id="cb162"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get consensus peak</span>
<span class="va">peak.file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"debchip_peaks.bed"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">peak.df</span> <span class="op">=</span> <span class="fu"><a href="../reference/GetConsensusPeak.html">GetConsensusPeak</a></span><span class="op">(</span>peak.file <span class="op">=</span> <span class="va">peak.file</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">peak.df</span><span class="op">)</span></code></pre></div>
<pre><code>##     chr     start      stop     name score
## 1 chr13  51519018  51519164  chr13-5  27.6
## 2 chr13  21326999  21327145  chr13-6  25.2
## 3  chr5 151112840 151112986   chr5-2  22.3
## 4 chr11 104361891 104362037 chr11-98  21.8
## 5 chr11  70198551  70198697 chr11-88  21.3
## 6  chr6  29326952  29327098   chr6-4  20.8</code></pre>
<hr>
</div>
<div id="peak-profile" class="section level3">
<h3 class="hasAnchor">
<a href="#peak-profile" class="anchor"></a>Peak profile</h3>
<p>Check the profle of consensus peaks:</p>
<div class="sourceCode" id="cb164"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># peak profile plot</span>
<span class="va">peak.profile</span> <span class="op">=</span> <span class="fu"><a href="../reference/PeakProfile.html">PeakProfile</a></span><span class="op">(</span><span class="va">peak.df</span> ,species<span class="op">=</span><span class="st">"Mouse"</span>, by <span class="op">=</span> <span class="st">"gene"</span>, region.type <span class="op">=</span> <span class="st">"body"</span>, nbin <span class="op">=</span> <span class="fl">800</span><span class="op">)</span></code></pre></div>
<pre><code>## &gt;&gt; preparing promoter regions...  2022-12-12 12时25分49秒 
## &gt;&gt; preparing tag matrix...        2022-12-12 12时25分50秒 
## &gt;&gt; preparing start_site regions by ... 2022-12-12 12时25分50秒
## &gt;&gt; preparing tag matrix...  2022-12-12 12时25分50秒 
## &gt;&gt; generating figure...       2022-12-12 12时25分58秒</code></pre>
<pre><code>## &gt;&gt; done...            2022-12-12 12时25分58秒</code></pre>
<pre><code>## &gt;&gt; binning method is used...2022-12-12 12时25分58秒
## &gt;&gt; preparing start_site regions by gene... 2022-12-12 12时25分58秒
## &gt;&gt; preparing tag matrix by binning...  2022-12-12 12时25分58秒 
## &gt;&gt; Running bootstrapping for tag matrix...        2022-12-12 12时26分03秒 
## &gt;&gt; binning method is used...2022-12-12 12时26分03秒
## &gt;&gt; preparing body regions by gene... 2022-12-12 12时26分03秒
## &gt;&gt; preparing tag matrix by binning...  2022-12-12 12时26分03秒 
## &gt;&gt; preparing matrix with extension from (TSS-20%)~(TTS+20%)... 2022-12-12 12时26分03秒
## &gt;&gt; 1 peaks(0.1536098%), having lengths smaller than 800bp, are filtered... 2022-12-12 12时26分06秒
## &gt;&gt; Running bootstrapping for tag matrix...        2022-12-12 12时26分42秒</code></pre>
<div class="sourceCode" id="cb168"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">peak.profile</span><span class="op">$</span><span class="va">profile.plot</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/debchip_peak_profile-1.png" width="960" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="peak-annotation" class="section level3">
<h3 class="hasAnchor">
<a href="#peak-annotation" class="anchor"></a>Peak annotation</h3>
<p>Peak annotation with <a href="https://bioconductor.org/packages/release/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.html">ChIPseeker</a>:</p>
<div class="sourceCode" id="cb169"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># peak annotation</span>
<span class="va">peak.anno</span> <span class="op">=</span> <span class="fu"><a href="../reference/AnnoPeak.html">AnnoPeak</a></span><span class="op">(</span>peak.df <span class="op">=</span> <span class="va">peak.df</span>,species <span class="op">=</span> <span class="st">"Mouse"</span>,seq.style <span class="op">=</span> <span class="st">"UCSC"</span>,up.dist <span class="op">=</span> <span class="fl">20000</span>,down.dist <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></code></pre></div>
<pre><code>## &gt;&gt; preparing features information...      2022-12-12 12时26分44秒 
## &gt;&gt; identifying nearest features...        2022-12-12 12时26分44秒 
## &gt;&gt; calculating distance from peak to TSS...   2022-12-12 12时26分45秒 
## &gt;&gt; assigning genomic annotation...        2022-12-12 12时26分45秒 
## &gt;&gt; adding gene annotation...          2022-12-12 12时26分56秒</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## &gt;&gt; assigning chromosome lengths           2022-12-12 12时26分56秒 
## &gt;&gt; done...                    2022-12-12 12时26分56秒</code></pre>
<pre><code>## Warning: Removed 6 rows containing non-finite values (stat_count).</code></pre>
<div class="sourceCode" id="cb174"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">peak.anno.df</span> <span class="op">=</span> <span class="va">peak.anno</span><span class="op">$</span><span class="va">df</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">peak.anno.df</span><span class="op">)</span></code></pre></div>
<pre><code>##   seqnames     start       end width strand     name score
## 1    chr13  51519019  51519164   146      *  chr13-5  27.6
## 2    chr13  21327000  21327145   146      *  chr13-6  25.2
## 3     chr5 151112841 151112986   146      *   chr5-2  22.3
## 4    chr11 104361892 104362037   146      * chr11-98  21.8
## 5    chr11  70198552  70198697   146      * chr11-88  21.3
## 6     chr6  29326953  29327098   146      *   chr6-4  20.8
##                                            annotation geneChr geneStart
## 1 Intron (ENSMUST00000021898.5/20418, intron 1 of 11)       1  51431041
## 2                                   Promoter (9-10kb)       1  21317258
## 3                                    Promoter (4-5kb)       2 151095421
## 4 Intron (ENSMUST00000106977.7/76719, intron 4 of 13)       3 104334921
## 5                                  Promoter (14-15kb)       3  70212752
## 6                                    Promoter (7-8kb)       4  29319199
##     geneEnd geneLength geneStrand geneId         transcriptId distanceToTSS
## 1  51567084     136044          2  20418 ENSMUST00000021898.5         47920
## 2  21319624       2367          1  75512 ENSMUST00000136668.1          9742
## 3 151108735      13315          2 243362 ENSMUST00000202866.1         -4106
## 4 104341299       6379          2  76719 ENSMUST00000069188.6        -20593
## 5  70216413       3662          1 216867 ENSMUST00000126388.7        -14055
## 6  29335854      16656          1 330277 ENSMUST00000166462.1          7754
##              ENSEMBL   SYMBOL
## 1 ENSMUSG00000021448     Shc3
## 2 ENSMUSG00000004341     Gpx6
## 3 ENSMUSG00000016128  Stard13
## 4 ENSMUSG00000018412   Kansl1
## 5 ENSMUSG00000040938 Slc16a11
## 6 ENSMUSG00000039742  Fam71f1
##                                                                 GENENAME
## 1               src homology 2 domain-containing transforming protein C3
## 2                                               glutathione peroxidase 6
## 3               StAR-related lipid transfer (START) domain containing 13
## 4                                  KAT8 regulatory NSL complex subunit 1
## 5 solute carrier family 16 (monocarboxylic acid transporters), member 11
## 6                          family with sequence similarity 71, member F1
##       anno
## 1   Intron
## 2 Promoter
## 3 Promoter
## 4   Intron
## 5 Promoter
## 6 Promoter</code></pre>
<div class="sourceCode" id="cb176"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">peak.anno</span><span class="op">$</span><span class="va">plots</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/debchip_peak_annotation_plot-1.png" width="1344" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="integrate-chip-seq-and-rna-seq" class="section level2">
<h2 class="hasAnchor">
<a href="#integrate-chip-seq-and-rna-seq" class="anchor"></a>Integrate ChIP-seq and RNA-seq</h2>
<p>In this step, we will integrate ChIP-seq and RNA-seq to get plausible direct targets of promoter (in this example, Runx). ### Integrate</p>
<div class="sourceCode" id="cb177"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debchip.res</span> <span class="op">=</span> <span class="fu"><a href="../reference/DEbPeak.html">DEbPeak</a></span><span class="op">(</span>de.res <span class="op">=</span> <span class="va">debchip.dds.results.ordered</span>, peak.res <span class="op">=</span> <span class="va">peak.anno.df</span>, peak.anno.key <span class="op">=</span> <span class="st">"Promoter"</span>, merge.key<span class="op">=</span><span class="st">"SYMBOL"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<div class="sourceCode" id="cb179"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">debchip.res</span><span class="op">)</span></code></pre></div>
<pre><code>##          SYMBOL geneId         annotation     anno ENSEMBL
## 1 0610012G03Rik   &lt;NA&gt;               &lt;NA&gt;     &lt;NA&gt;    &lt;NA&gt;
## 2 1110002J07Rik  68488 Promoter (12-13kb) Promoter    &lt;NA&gt;
## 3 1110008P14Rik   &lt;NA&gt;               &lt;NA&gt;     &lt;NA&gt;    &lt;NA&gt;
## 4 1110032F04Rik   &lt;NA&gt;               &lt;NA&gt;     &lt;NA&gt;    &lt;NA&gt;
## 5 1500009C09Rik   &lt;NA&gt;               &lt;NA&gt;     &lt;NA&gt;    &lt;NA&gt;
## 6 1500011B03Rik   &lt;NA&gt;               &lt;NA&gt;     &lt;NA&gt;    &lt;NA&gt;
##                     GENENAME log2FoldChange  abundance   signif   regulation
## 1                       &lt;NA&gt;       1.837668  59.179193 3.073773 Up_regulated
## 2 RIKEN cDNA 1110002J07 gene             NA         NA       NA         &lt;NA&gt;
## 3                       &lt;NA&gt;       1.013546 125.765543 2.277566 Up_regulated
## 4                       &lt;NA&gt;       2.935223  89.091244 4.714179 Up_regulated
## 5                       &lt;NA&gt;       3.183848   9.432253 1.492741 Up_regulated
## 6                       &lt;NA&gt;       1.058984 227.077738 2.656922 Up_regulated
##   Type
## 1   UP
## 2 Peak
## 3   UP
## 4   UP
## 5   UP
## 6   UP</code></pre>
<hr>
<div id="integrate-summary" class="section level3">
<h3 class="hasAnchor">
<a href="#integrate-summary" class="anchor"></a>Integrate summary</h3>
<div class="sourceCode" id="cb181"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DE and ChIP venn plot</span>
<span class="va">debchip.plot</span> <span class="op">=</span> <span class="fu"><a href="../reference/PlotDEbPeak.html">PlotDEbPeak</a></span><span class="op">(</span><span class="va">debchip.res</span>, peak.type <span class="op">=</span> <span class="st">"ChIP"</span>, show_percentage<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="va">debchip.plot</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/debchip_integrate_plot-1.png" width="384" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="functional-enrichment" class="section level3">
<h3 class="hasAnchor">
<a href="#functional-enrichment" class="anchor"></a>Functional enrichment</h3>
<div class="sourceCode" id="cb182"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># functional enrichment on direct targets</span>
<span class="va">debchip.fe.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/DEbPeakFE.html">DEbPeakFE</a></span><span class="op">(</span>de.peak <span class="op">=</span> <span class="va">debchip.res</span>, peak.type <span class="op">=</span> <span class="st">"ChIP"</span>, gene.type <span class="op">=</span> <span class="st">"ENTREZID"</span>, species<span class="op">=</span><span class="st">"Mouse"</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## conduct ALL GO enrichment analysis on: UPbChIP</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## conduct ALL GO enrichment analysis on: DOWNbChIP</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<div id="fe-on-up-regulated-targets" class="section level4">
<h4 class="hasAnchor">
<a href="#fe-on-up-regulated-targets" class="anchor"></a>FE on up-regulated targets</h4>
<div class="sourceCode" id="cb195"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.debchip.fe.results</span><span class="op">=</span><span class="va">debchip.fe.results</span><span class="op">[[</span><span class="st">"UPbChIP"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">up.debchip.fe.results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                                    Description
## GO:0032288       BP GO:0032288                                myelin assembly
## GO:0019233       BP GO:0019233                     sensory perception of pain
## GO:0043951       BP GO:0043951 negative regulation of cAMP-mediated signaling
## GO:0022011       BP GO:0022011       myelination in peripheral nervous system
## GO:0032292       BP GO:0032292    peripheral nervous system axon ensheathment
## GO:0014044       BP GO:0014044                       Schwann cell development
##            GeneRatio   BgRatio       pvalue   p.adjust     qvalue
## GO:0032288      2/21  23/23328 0.0001930551 0.02159856 0.01416299
## GO:0019233      3/21 146/23328 0.0002940462 0.02159856 0.01416299
## GO:0043951      2/21  30/23328 0.0003306741 0.02159856 0.01416299
## GO:0022011      2/21  31/23328 0.0003532875 0.02159856 0.01416299
## GO:0032292      2/21  31/23328 0.0003532875 0.02159856 0.01416299
## GO:0014044      2/21  34/23328 0.0004255310 0.02159856 0.01416299
##                      geneID Count
## GO:0032288        Pmp22/Prx     2
## GO:0019233 Kcnip3/Npy2r/Prx     3
## GO:0043951     Npy2r/Rnf157     2
## GO:0022011        Pmp22/Prx     2
## GO:0032292        Pmp22/Prx     2
## GO:0014044        Pmp22/Prx     2</code></pre>
<div class="sourceCode" id="cb197"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.debchip.fe.results</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/debchip_integrate_fe_up-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div id="fe-on-down-regulated-targets" class="section level4">
<h4 class="hasAnchor">
<a href="#fe-on-down-regulated-targets" class="anchor"></a>FE on down-regulated targets</h4>
<div class="sourceCode" id="cb198"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.debchip.fe.results</span><span class="op">=</span><span class="va">debchip.fe.results</span><span class="op">[[</span><span class="st">"DOWNbChIP"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">down.debchip.fe.results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                                  Description
## GO:0010810       BP GO:0010810        regulation of cell-substrate adhesion
## GO:0045785       BP GO:0045785         positive regulation of cell adhesion
## GO:0007162       BP GO:0007162         negative regulation of cell adhesion
## GO:0030198       BP GO:0030198            extracellular matrix organization
## GO:0043062       BP GO:0043062         extracellular structure organization
## GO:0051250       BP GO:0051250 negative regulation of lymphocyte activation
##            GeneRatio   BgRatio       pvalue   p.adjust      qvalue
## GO:0010810      5/33 214/23328 1.193628e-05 0.01428434 0.008185655
## GO:0045785      6/33 435/23328 2.935433e-05 0.01428434 0.008185655
## GO:0007162      5/33 290/23328 5.117751e-05 0.01428434 0.008185655
## GO:0030198      5/33 302/23328 6.201521e-05 0.01428434 0.008185655
## GO:0043062      5/33 303/23328 6.299252e-05 0.01428434 0.008185655
## GO:0051250      4/33 158/23328 7.112535e-05 0.01428434 0.008185655
##                                             geneID Count
## GO:0010810           Col8a1/Fbln2/Mmp14/Postn/Sdc4     5
## GO:0045785 Col8a1/Fbln2/Il4ra/Ptpn22/Sdc4/Tnfsf13b     6
## GO:0007162           Il4ra/Mmp14/Postn/Ptpn22/Sdc4     5
## GO:0030198           Bcl3/Col8a1/Fbln2/Mmp14/Postn     5
## GO:0043062           Bcl3/Col8a1/Fbln2/Mmp14/Postn     5
## GO:0051250                   Il4ra/Lyn/Ptpn22/Sdc4     4</code></pre>
<div class="sourceCode" id="cb200"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.debchip.fe.results</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/debchip_integrate_fe_down-1.png" width="768" style="display: block; margin: auto;"></p>
<table class="table">
<colgroup><col width="30%"></colgroup>
<tbody>
<tr class="odd">
<td>## Integrate ATAC-seq and RNA-seq All above steps are suitable for integrating ATAC-seq and RNA-seq data, but the following step (<strong>Find motif</strong>) is usually unique to integrate ATAC-seq and RNA-seq data. ### Find motif</td>
</tr>
<tr class="even">
<td><code>r # debatac.res: similar to debchip.res ATAC.motif = FindMotif(inte.res = debatac.res, peak.anno.res = peak.anno.df, gene.key = "SYMBOL", homer.motif.path = '/path/to/findMotifsGenome.pl', genome = '/path/to/genome.fa', out.folder = NULL,other.paras = '-len 8,10,12 -size -100,50 -S 25')</code></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="integrate-rna-seq-chip-seq-and-atac-seq" class="section level2">
<h2 class="hasAnchor">
<a href="#integrate-rna-seq-chip-seq-and-atac-seq" class="anchor"></a>Integrate RNA-seq, ChIP-seq and ATAC-seq</h2>
<p>We have integrated RNA-seq and ChIP-seq data in the above steps, now we will continue to integrate ATAC-seq data based on the previous results. ### Prepare ATAC-seq data #### Consensus peaks</p>
<div class="sourceCode" id="cb201"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get consensus peak</span>
<span class="va">atac.peak.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"debatac_peaks.bed"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">atac.peak.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetConsensusPeak.html">GetConsensusPeak</a></span><span class="op">(</span>peak.file <span class="op">=</span> <span class="va">atac.peak.file</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">atac.peak.df</span><span class="op">)</span></code></pre></div>
<pre><code>##     chr     start      stop     name score
## 1  chr4  52439068  52439142 chr4-383     1
## 2  chr4 139622902 139622976  chr4-99     1
## 3  chr8  84662748  84662822 chr8-119     1
## 4  chr8  70699040  70699114 chr8-115     1
## 5  chr8  69625402  69625476 chr8-202     1
## 6 chr13 119597770 119597844 chr13-92     1</code></pre>
<div id="peak-profile-1" class="section level4">
<h4 class="hasAnchor">
<a href="#peak-profile-1" class="anchor"></a>Peak profile</h4>
<p>Check the profle of consensus peaks:</p>
<div class="sourceCode" id="cb203"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># peak profile plot</span>
<span class="va">atac.peak.profile</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PeakProfile.html">PeakProfile</a></span><span class="op">(</span><span class="va">atac.peak.df</span>, species <span class="op">=</span> <span class="st">"Mouse"</span>, by <span class="op">=</span> <span class="st">"gene"</span>, region.type <span class="op">=</span> <span class="st">"body"</span>, nbin <span class="op">=</span> <span class="fl">800</span><span class="op">)</span></code></pre></div>
<pre><code>## &gt;&gt; preparing promoter regions...  2022-12-12 12时27分45秒 
## &gt;&gt; preparing tag matrix...        2022-12-12 12时27分45秒 
## &gt;&gt; preparing start_site regions by ... 2022-12-12 12时27分45秒
## &gt;&gt; preparing tag matrix...  2022-12-12 12时27分45秒 
## &gt;&gt; generating figure...       2022-12-12 12时27分52秒</code></pre>
<pre><code>## &gt;&gt; done...            2022-12-12 12时27分56秒</code></pre>
<pre><code>## &gt;&gt; binning method is used...2022-12-12 12时27分57秒
## &gt;&gt; preparing start_site regions by gene... 2022-12-12 12时27分57秒
## &gt;&gt; preparing tag matrix by binning...  2022-12-12 12时27分57秒 
## &gt;&gt; Running bootstrapping for tag matrix...        2022-12-12 12时28分40秒 
## &gt;&gt; binning method is used...2022-12-12 12时28分40秒
## &gt;&gt; preparing body regions by gene... 2022-12-12 12时28分40秒
## &gt;&gt; preparing tag matrix by binning...  2022-12-12 12时28分40秒 
## &gt;&gt; preparing matrix with extension from (TSS-20%)~(TTS+20%)... 2022-12-12 12时28分40秒
## &gt;&gt; 16 peaks(0.296077%), having lengths smaller than 800bp, are filtered... 2022-12-12 12时28分45秒
## &gt;&gt; Running bootstrapping for tag matrix...        2022-12-12 12时31分39秒</code></pre>
<div class="sourceCode" id="cb207"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">atac.peak.profile</span><span class="op">$</span><span class="va">profile.plot</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/debchip_atac_peak_profile-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="peak-annotation-1" class="section level4">
<h4 class="hasAnchor">
<a href="#peak-annotation-1" class="anchor"></a>Peak annotation</h4>
<p>Peak annotation with <a href="https://bioconductor.org/packages/release/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.html">ChIPseeker</a>:</p>
<div class="sourceCode" id="cb208"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># peak annotation</span>
<span class="va">atac.peak.anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AnnoPeak.html">AnnoPeak</a></span><span class="op">(</span>
  peak.df <span class="op">=</span> <span class="va">atac.peak.df</span>, species <span class="op">=</span> <span class="st">"Mouse"</span>,
  seq.style <span class="op">=</span> <span class="st">"UCSC"</span>, up.dist <span class="op">=</span> <span class="fl">20000</span>, down.dist <span class="op">=</span> <span class="fl">20000</span>
<span class="op">)</span></code></pre></div>
<pre><code>## &gt;&gt; preparing features information...      2022-12-12 12时31分41秒 
## &gt;&gt; identifying nearest features...        2022-12-12 12时31分41秒 
## &gt;&gt; calculating distance from peak to TSS...   2022-12-12 12时31分41秒 
## &gt;&gt; assigning genomic annotation...        2022-12-12 12时31分41秒 
## &gt;&gt; adding gene annotation...          2022-12-12 12时31分44秒</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## &gt;&gt; assigning chromosome lengths           2022-12-12 12时31分44秒 
## &gt;&gt; done...                    2022-12-12 12时31分44秒</code></pre>
<pre><code>## Warning: Removed 23 rows containing non-finite values (stat_count).</code></pre>
<div class="sourceCode" id="cb213"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">atac.peak.anno.df</span> <span class="op">=</span> <span class="va">atac.peak.anno</span><span class="op">$</span><span class="va">df</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">atac.peak.anno.df</span><span class="op">)</span></code></pre></div>
<pre><code>##   seqnames     start       end width strand     name score       annotation
## 1     chr4  52439069  52439142    74      * chr4-383     1 Promoter (&lt;=1kb)
## 2     chr4 139622903 139622976    74      *  chr4-99     1 Promoter (&lt;=1kb)
## 3     chr8  84662749  84662822    74      * chr8-119     1 Promoter (&lt;=1kb)
## 4     chr8  70699041  70699114    74      * chr8-115     1 Promoter (&lt;=1kb)
## 5     chr8  69625403  69625476    74      * chr8-202     1 Promoter (&lt;=1kb)
## 6    chr13 119597771 119597844    74      * chr13-92     1 Promoter (8-9kb)
##   geneChr geneStart   geneEnd geneLength geneStrand geneId         transcriptId
## 1       5  52439243  52488260      49018          1  14211 ENSMUST00000102915.9
## 2       5 139622894 139649687      26794          1 212647 ENSMUST00000146309.7
## 3       9  84661331  84662854       1524          2  15936 ENSMUST00000060427.5
## 4       9  70698949  70700616       1668          1  16478 ENSMUST00000095267.5
## 5       9  69618869  69625545       6677          2 234362 ENSMUST00000140522.1
## 6       1 119606650 119610458       3809          1 633640 ENSMUST00000178948.1
##   distanceToTSS            ENSEMBL  SYMBOL
## 1          -101 ENSMUSG00000028312    Smc2
## 2             9 ENSMUSG00000028737 Aldh4a1
## 3            32 ENSMUSG00000053560    Ier2
## 4            92 ENSMUSG00000071076    Jund
## 5            69 ENSMUSG00000060427  Zfp868
## 6         -8806 ENSMUSG00000074634 Tmem267
##                                     GENENAME     anno
## 1    structural maintenance of chromosomes 2 Promoter
## 2 aldehyde dehydrogenase 4 family, member A1 Promoter
## 3                 immediate early response 2 Promoter
## 4                       jun D proto-oncogene Promoter
## 5                    zinc finger protein 868 Promoter
## 6                  transmembrane protein 267 Promoter</code></pre>
<div class="sourceCode" id="cb215"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">atac.peak.anno</span><span class="op">$</span><span class="va">plots</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/debchip_atac_peak_annotation_plot-1.png" width="1344" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="integrate-rna-seq-chip-seq-and-atac-seq-1" class="section level3">
<h3 class="hasAnchor">
<a href="#integrate-rna-seq-chip-seq-and-atac-seq-1" class="anchor"></a>Integrate RNA-seq, ChIP-seq and ATAC-seq</h3>
<div id="integrate" class="section level4">
<h4 class="hasAnchor">
<a href="#integrate" class="anchor"></a>Integrate</h4>
<div class="sourceCode" id="cb216"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debpeak.res</span> <span class="op">=</span> <span class="fu"><a href="../reference/DEbCA.html">DEbCA</a></span><span class="op">(</span>de.res <span class="op">=</span> <span class="va">debchip.dds.results.ordered</span>, chip.peak.res <span class="op">=</span> <span class="va">peak.anno.df</span>,
                    atac.peak.res <span class="op">=</span> <span class="va">atac.peak.anno.df</span>,
                    peak.anno.key <span class="op">=</span> <span class="st">"Promoter"</span>, merge.key <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<div class="sourceCode" id="cb218"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">debpeak.res</span><span class="op">)</span></code></pre></div>
<pre><code>##          SYMBOL geneId_ChIP    annotation_ChIP anno_ChIP ENSEMBL_ChIP
## 1 0610009E02Rik        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;         &lt;NA&gt;
## 2 0610012G03Rik        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;         &lt;NA&gt;
## 3 0610039K10Rik        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;         &lt;NA&gt;
## 4 1110002J07Rik       68488 Promoter (12-13kb)  Promoter         &lt;NA&gt;
## 5 1110004F10Rik        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;         &lt;NA&gt;
## 6 1110008P14Rik        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;         &lt;NA&gt;
##                GENENAME_ChIP log2FoldChange abundance   signif   regulation
## 1                       &lt;NA&gt;             NA        NA       NA         &lt;NA&gt;
## 2                       &lt;NA&gt;       1.837668  59.17919 3.073773 Up_regulated
## 3                       &lt;NA&gt;             NA        NA       NA         &lt;NA&gt;
## 4 RIKEN cDNA 1110002J07 gene             NA        NA       NA         &lt;NA&gt;
## 5                       &lt;NA&gt;             NA        NA       NA         &lt;NA&gt;
## 6                       &lt;NA&gt;       1.013546 125.76554 2.277566 Up_regulated
##   Type1 geneId_ATAC    annotation_ATAC anno_ATAC       ENSEMBL_ATAC
## 1  &lt;NA&gt;   100125929   Promoter (&lt;=1kb)  Promoter ENSMUSG00000086714
## 2    UP      106264   Promoter (&lt;=1kb)  Promoter ENSMUSG00000107002
## 3  &lt;NA&gt;       68386   Promoter (&lt;=1kb)  Promoter ENSMUSG00000058812
## 4  ChIP       68488 Promoter (15-16kb)  Promoter               &lt;NA&gt;
## 5  &lt;NA&gt;       56372   Promoter (&lt;=1kb)  Promoter ENSMUSG00000030663
## 6    UP        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;               &lt;NA&gt;
##                GENENAME_ATAC      Type    geneId
## 1 RIKEN cDNA 0610009E02 gene      ATAC 100125929
## 2 RIKEN cDNA 0610012G03 gene   UPbATAC    106264
## 3 RIKEN cDNA 0610039K10 gene      ATAC     68386
## 4 RIKEN cDNA 1110002J07 gene ChIPbATAC     68488
## 5 RIKEN cDNA 1110004F10 gene      ATAC     56372
## 6                       &lt;NA&gt;        UP      &lt;NA&gt;</code></pre>
</div>
<div id="integrate-summary-1" class="section level4">
<h4 class="hasAnchor">
<a href="#integrate-summary-1" class="anchor"></a>Integrate summary</h4>
<div class="sourceCode" id="cb220"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DE and ChIP venn plot</span>
<span class="va">debpeak.plot</span> <span class="op">=</span> <span class="fu"><a href="../reference/PlotDEbPeak.html">PlotDEbPeak</a></span><span class="op">(</span><span class="va">debpeak.res</span>, peak.type <span class="op">=</span> <span class="st">"Peak"</span>, show_percentage<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="va">debpeak.plot</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/debpeak_integrate_plot-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="functional-enrichment-1" class="section level4">
<h4 class="hasAnchor">
<a href="#functional-enrichment-1" class="anchor"></a>Functional enrichment</h4>
<div class="sourceCode" id="cb221"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># functional enrichment on direct targets</span>
<span class="va">debpeak.fe.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/DEbPeakFE.html">DEbPeakFE</a></span><span class="op">(</span>de.peak <span class="op">=</span> <span class="va">debpeak.res</span>, peak.type <span class="op">=</span> <span class="st">"Peak"</span>, gene.type <span class="op">=</span> <span class="st">"ENTREZID"</span>, species<span class="op">=</span><span class="st">"Mouse"</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## conduct ALL GO enrichment analysis on: UPbPeak</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## conduct ALL GO enrichment analysis on: DOWNbPeak</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<div id="fe-on-up-regulated-targets-1" class="section level5">
<h5 class="hasAnchor">
<a href="#fe-on-up-regulated-targets-1" class="anchor"></a>FE on up-regulated targets</h5>
<div class="sourceCode" id="cb232"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.debpeak.fe.results</span><span class="op">=</span><span class="va">debpeak.fe.results</span><span class="op">[[</span><span class="st">"UPbPeak"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">up.debpeak.fe.results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                     Description GeneRatio
## GO:0043218       CC GO:0043218                  compact myelin       2/9
## GO:0043209       CC GO:0043209                   myelin sheath       2/9
## GO:0070382       CC GO:0070382                exocytic vesicle       2/9
## GO:0043256       CC GO:0043256                 laminin complex       1/9
## GO:0019898       CC GO:0019898 extrinsic component of membrane       2/9
## GO:0030133       CC GO:0030133               transport vesicle       2/9
##              BgRatio       pvalue    p.adjust      qvalue    geneID Count
## GO:0043218  19/23271 2.265877e-05 0.001087621 0.000500878 Mbp/Pmp22     2
## GO:0043209 213/23271 2.877487e-03 0.043690148 0.020120463 Mbp/Pmp22     2
## GO:0070382 261/23271 4.282761e-03 0.043690148 0.020120463  Bsn/Syt6     2
## GO:0043256  12/23271 4.632203e-03 0.043690148 0.020120463     Pmp22     1
## GO:0019898 320/23271 6.366508e-03 0.043690148 0.020120463  Bsn/Syt6     2
## GO:0030133 339/23271 7.118989e-03 0.043690148 0.020120463  Bsn/Syt6     2</code></pre>
<div class="sourceCode" id="cb234"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.debpeak.fe.results</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/debpeak_integrate_fe_up-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div id="fe-on-down-regulated-targets-1" class="section level5">
<h5 class="hasAnchor">
<a href="#fe-on-down-regulated-targets-1" class="anchor"></a>FE on down-regulated targets</h5>
<div class="sourceCode" id="cb235"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.debpeak.fe.results</span><span class="op">=</span><span class="va">debpeak.fe.results</span><span class="op">[[</span><span class="st">"DOWNbPeak"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">down.debpeak.fe.results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                                    Description
## GO:0045064       BP GO:0045064                T-helper 2 cell differentiation
## GO:0010811       BP GO:0010811 positive regulation of cell-substrate adhesion
## GO:0045785       BP GO:0045785           positive regulation of cell adhesion
## GO:0045860       BP GO:0045860 positive regulation of protein kinase activity
## GO:0042832       BP GO:0042832                  defense response to protozoan
## GO:0010810       BP GO:0010810          regulation of cell-substrate adhesion
##            GeneRatio   BgRatio       pvalue   p.adjust      qvalue
## GO:0045064      2/13  17/23328 3.880421e-05 0.01212719 0.006013313
## GO:0010811      3/13 128/23328 4.433038e-05 0.01212719 0.006013313
## GO:0045785      4/13 435/23328 7.460788e-05 0.01212719 0.006013313
## GO:0045860      4/13 440/23328 7.798836e-05 0.01212719 0.006013313
## GO:0042832      2/13  36/23328 1.786843e-04 0.01717888 0.008518210
## GO:0010810      3/13 214/23328 2.034275e-04 0.01717888 0.008518210
##                             geneID Count
## GO:0045064              Bcl3/Il4ra     2
## GO:0010811       Col8a1/Fbln2/Sdc4     3
## GO:0045785 Col8a1/Fbln2/Il4ra/Sdc4     4
## GO:0045860  Akap13/Ccnd1/Ern1/Sdc4     4
## GO:0042832              Bcl3/Il4ra     2
## GO:0010810       Col8a1/Fbln2/Sdc4     3</code></pre>
<div class="sourceCode" id="cb237"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.debpeak.fe.results</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="DEbPeak_All_files/figure-html/debpeak_integrate_fe_down-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
</div>
</div>
<div id="utils" class="section level1">
<h1 class="hasAnchor">
<a href="#utils" class="anchor"></a>Utils</h1>
<div id="gene-name-conversion" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-name-conversion" class="anchor"></a>Gene name conversion</h2>
<div id="results-from-de-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#results-from-de-analysis" class="anchor"></a>results from DE analysis</h3>
<div class="sourceCode" id="cb238"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># convert for DESeq2 results</span>
<span class="va">dds.results.ordered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/IDConversion.html">IDConversion</a></span><span class="op">(</span>deres <span class="op">=</span> <span class="va">dds.results.ordered</span>, gene.type <span class="op">=</span> <span class="st">"ENSEMBL"</span>,sort.key <span class="op">=</span> <span class="st">"log2FoldChange"</span><span class="op">)</span></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb240"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dds.results.ordered</span><span class="op">)</span></code></pre></div>
<pre><code>##                           baseMean log2FoldChange     lfcSE       stat
## ENSMUSG00000000001.4  4178.2932143     0.03256101 0.1097033  0.2968099
## ENSMUSG00000000003.15    0.0000000             NA        NA         NA
## ENSMUSG00000000028.14  416.5975492    -0.13238461 0.1709810 -0.7742650
## ENSMUSG00000000031.16  200.5762259     0.57949970 0.7478555  0.7748819
## ENSMUSG00000000037.16  202.0237326     0.13726485 0.2659695  0.5160925
## ENSMUSG00000000049.11    0.3930718    -2.07400983 2.6239287 -0.7904216
##                          pvalue      padj ENTREZID SYMBOL
## ENSMUSG00000000001.4  0.7666117 0.9997304    14679  Gnai3
## ENSMUSG00000000003.15        NA        NA    54192   Pbsn
## ENSMUSG00000000028.14 0.4387741 0.9997304    12544  Cdc45
## ENSMUSG00000000031.16 0.4384095 0.9997304    14955    H19
## ENSMUSG00000000037.16 0.6057898 0.9997304   107815  Scml2
## ENSMUSG00000000049.11 0.4292816 0.9997304    11818   Apoh</code></pre>
<hr>
</div>
<div id="normal-matrix" class="section level3">
<h3 class="hasAnchor">
<a href="#normal-matrix" class="anchor"></a>normal matrix</h3>
<div class="sourceCode" id="cb242"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">count.matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/IDConversion.html">IDConversion</a></span><span class="op">(</span>deres <span class="op">=</span> <span class="va">count.matrix</span>, gene.type <span class="op">=</span> <span class="st">"ENSEMBL"</span>,sort.key<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb244"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">count.matrix</span><span class="op">)</span></code></pre></div>
<pre><code>##                        KO1  KO2  KO3  KO4  KO5  KO6  WT1  WT2  WT3  WT4  WT5
## ENSMUSG00000000001.4  4556 4218 3835 3718 5741 3875 4115 5074 2931 4374 5118
## ENSMUSG00000000003.15    0    0    0    0    0    0    0    0    0    0    0
## ENSMUSG00000000028.14  350  579  435  316  432  317  245  621  419  506  545
## ENSMUSG00000000031.16  268  804   66  207   46   66  336  112   60   69  137
## ENSMUSG00000000037.16  262  157  184  162  301  233  311  176   94  139  229
## ENSMUSG00000000049.11    0    0    0    0    0    0    1    0    2    0    1
##                        WT6 ENTREZID SYMBOL
## ENSMUSG00000000001.4  3625    14679  Gnai3
## ENSMUSG00000000003.15    0    54192   Pbsn
## ENSMUSG00000000028.14  371    12544  Cdc45
## ENSMUSG00000000031.16  193    14955    H19
## ENSMUSG00000000037.16  186   107815  Scml2
## ENSMUSG00000000049.11    0    11818   Apoh</code></pre>
<hr>
</div>
</div>
<div id="count-normalization" class="section level2">
<h2 class="hasAnchor">
<a href="#count-normalization" class="anchor"></a>Count normalization</h2>
<p>Here, we provide five different normalization methods: <strong>CPM</strong>, <strong>TMM</strong>, <strong>DESeq2’s median of ratios</strong>, <strong>RPKM</strong> and <strong>TPM</strong>, of which <strong>RPKM</strong> and <strong>TPM</strong> need to provide gtf files to calculate gene length.</p>
<div id="cpm" class="section level3">
<h3 class="hasAnchor">
<a href="#cpm" class="anchor"></a>CPM</h3>
<p><em>C</em>ounts <em>P</em>er <em>M</em>illion reads mapped (CPM) takes into account the <strong>sequencing depth</strong>. For each feature <code>i</code>, CPM is the count of sequenced fragments mapping to the feature scaled by the total number of reads times one million (to bring it up to a more convenient number).</p>
<p><span class="math display">\[
\mathrm{CPM}=\frac{\text { Number of reads mapped to gene } \times 10^{6}}{\text { Total number of mapped reads }}
\]</span></p>
<div class="sourceCode" id="cb246"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cpm.matrix</span><span class="op">=</span><span class="fu"><a href="../reference/NormalizedCount.html">NormalizedCount</a></span><span class="op">(</span><span class="va">dds</span>, norm.type<span class="op">=</span><span class="st">"CPM"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">cpm.matrix</span><span class="op">)</span></code></pre></div>
<pre><code>##                             KO1       KO2        KO3       KO4        KO5
## ENSMUSG00000000001.4  405.92819 400.64944 389.764529 377.62444 380.625026
## ENSMUSG00000000003.15   0.00000   0.00000   0.000000   0.00000   0.000000
## ENSMUSG00000000028.14  31.18412  54.99669  44.210579  32.09503  28.641354
## ENSMUSG00000000031.16  23.87813  76.36846   6.707812  21.02428   3.049774
## ENSMUSG00000000037.16  23.34354  14.91275  18.700567  16.45378  19.956128
## ENSMUSG00000000049.11   0.00000   0.00000   0.000000   0.00000   0.000000
##                              KO6         WT1        WT2         WT3        WT4
## ENSMUSG00000000001.4  362.974809 424.0865495 343.046563 389.7566993 402.307594
## ENSMUSG00000000003.15   0.000000   0.0000000   0.000000   0.0000000   0.000000
## ENSMUSG00000000028.14  29.693681  25.2493814  41.985005  55.7175220  46.540385
## ENSMUSG00000000031.16   6.182281  34.6277231   7.572175   7.9786428   6.346416
## ENSMUSG00000000037.16  21.825324  32.0512556  11.899132  12.4998737  12.784809
## ENSMUSG00000000049.11   0.000000   0.1030587   0.000000   0.2659548   0.000000
##                                WT5       WT6
## ENSMUSG00000000001.4  383.64870911 355.77287
## ENSMUSG00000000003.15   0.00000000   0.00000
## ENSMUSG00000000028.14  40.85356516  36.41151
## ENSMUSG00000000031.16  10.26961179  18.94184
## ENSMUSG00000000037.16  17.16599343  18.25483
## ENSMUSG00000000049.11   0.07496067   0.00000</code></pre>
<hr>
</div>
<div id="tmm" class="section level3">
<h3 class="hasAnchor">
<a href="#tmm" class="anchor"></a>TMM</h3>
<p>edgeR’s <em>T</em>rimmed <em>M</em>ean of <em>M</em> values (TMM) takes into account the <strong>sequencing depth</strong>, <strong>RNA composition</strong>, and <strong>gene length</strong>. For detailed information, please refer to <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25">A scaling normalization method for differential expression analysis of RNA-seq data</a>.</p>
<div class="sourceCode" id="cb248"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tmm.matrix</span><span class="op">=</span><span class="fu"><a href="../reference/NormalizedCount.html">NormalizedCount</a></span><span class="op">(</span><span class="va">dds</span>, norm.type<span class="op">=</span><span class="st">"TMM"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">tmm.matrix</span><span class="op">)</span></code></pre></div>
<pre><code>##                             KO1       KO2        KO3       KO4        KO5
## ENSMUSG00000000001.4  421.83331 371.93907 371.189992 382.69198 413.745541
## ENSMUSG00000000003.15   0.00000   0.00000   0.000000   0.00000   0.000000
## ENSMUSG00000000028.14  32.40598  51.05565  42.103689  32.52573  31.133613
## ENSMUSG00000000031.16  24.81372  70.89592   6.388146  21.30641   3.315153
## ENSMUSG00000000037.16  24.25819  13.84410  17.809376  16.67458  21.692633
## ENSMUSG00000000049.11   0.00000   0.00000   0.000000   0.00000   0.000000
##                              KO6         WT1        WT2         WT3       WT4
## ENSMUSG00000000001.4  373.041177 486.1604573 334.563934 361.3055744 394.46383
## ENSMUSG00000000003.15   0.000000   0.0000000   0.000000   0.0000000   0.00000
## ENSMUSG00000000028.14  30.517175  28.9451548  40.946828  51.6503022  45.63299
## ENSMUSG00000000031.16   6.353734  39.6962123   7.384935   7.3962247   6.22268
## ENSMUSG00000000037.16  22.430605  36.7426251  11.604898  11.5874186  12.53554
## ENSMUSG00000000049.11   0.000000   0.1181435   0.000000   0.2465408   0.00000
##                                WT5       WT6
## ENSMUSG00000000001.4  371.61142274 347.48647
## ENSMUSG00000000003.15   0.00000000   0.00000
## ENSMUSG00000000028.14  39.57175174  35.56344
## ENSMUSG00000000031.16   9.94739447  18.50066
## ENSMUSG00000000037.16  16.62739660  17.82965
## ENSMUSG00000000049.11   0.07260872   0.00000</code></pre>
<hr>
</div>
<div id="deseq2s-median-of-ratios" class="section level3">
<h3 class="hasAnchor">
<a href="#deseq2s-median-of-ratios" class="anchor"></a>DESeq2’s median of ratios</h3>
<p>DESeq2’s median of ratios takes into account the <strong>sequencing depth</strong> and <strong>RNA composition</strong>. For detailed information, please refer to <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</a>.</p>
<div class="sourceCode" id="cb250"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">deseq2.matrix</span><span class="op">=</span><span class="fu"><a href="../reference/NormalizedCount.html">NormalizedCount</a></span><span class="op">(</span><span class="va">dds</span>, norm.type<span class="op">=</span><span class="st">"DESeq2"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">deseq2.matrix</span><span class="op">)</span></code></pre></div>
<pre><code>##                             KO1       KO2        KO3       KO4        KO5
## ENSMUSG00000000001.4  4606.4055 4014.0575 4037.53736 4145.2746 4476.42605
## ENSMUSG00000000003.15    0.0000    0.0000    0.00000    0.0000    0.00000
## ENSMUSG00000000028.14  353.8722  551.0050  457.97360  352.3149  336.84307
## ENSMUSG00000000031.16  270.9650  765.1262   69.48565  230.7886   35.86755
## ENSMUSG00000000037.16  264.8986  149.4090  193.71757  180.6171  234.69853
## ENSMUSG00000000049.11    0.0000    0.0000    0.00000    0.0000    0.00000
##                              KO6         WT1        WT2         WT3       WT4
## ENSMUSG00000000001.4  4069.51479 5285.859380 3591.55792 3889.896093 4266.4884
## ENSMUSG00000000003.15    0.00000    0.000000    0.00000    0.000000    0.0000
## ENSMUSG00000000028.14  332.91257  314.710947  439.56592  556.078630  493.5627
## ENSMUSG00000000031.16   69.31303  431.603585   79.27759   79.629398   67.3040
## ENSMUSG00000000037.16  244.69599  399.490223  124.57907  124.752724  135.5834
## ENSMUSG00000000049.11    0.00000    1.284534    0.00000    2.654313    0.0000
##                                WT5       WT6
## ENSMUSG00000000001.4  3981.8764585 3774.6246
## ENSMUSG00000000003.15    0.0000000    0.0000
## ENSMUSG00000000028.14  424.0177159  386.3133
## ENSMUSG00000000031.16  106.5879396  200.9662
## ENSMUSG00000000037.16  178.1652421  193.6773
## ENSMUSG00000000049.11    0.7780142    0.0000</code></pre>
<hr>
</div>
<div id="rpkm" class="section level3">
<h3 class="hasAnchor">
<a href="#rpkm" class="anchor"></a>RPKM</h3>
<p><em>R</em>eads <em>P</em>er <em>K</em>ilobase of exon per <em>M</em>illion reads mapped (RPKM) takes into account the <strong>sequencing depth</strong> and <strong>gene length</strong>. For each feature <code>i</code> as the count scaled by the feature’s length times one thousand (to kilobase) and further scaled by the total number of reads times one million.</p>
<p><span class="math display">\[
\text { RPKM }=\frac{\text { Number of reads mapped to gene } \times 10^{3} \times 10^{6}}{\text { Total number of mapped reads } \times \text { gene length }}
\]</span></p>
<div class="sourceCode" id="cb252"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rpkm.matrix</span><span class="op">=</span><span class="fu"><a href="../reference/NormalizedCount.html">NormalizedCount</a></span><span class="op">(</span><span class="va">dds</span>,gtf.file <span class="op">=</span> <span class="st">'/path/to/gtf'</span>, norm.type<span class="op">=</span><span class="st">"RPKM"</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="tpm" class="section level3">
<h3 class="hasAnchor">
<a href="#tpm" class="anchor"></a>TPM</h3>
<p><em>T</em>ranscripts <em>P</em>er kilobase <em>M</em>illion (TPM) takes into account the <strong>sequencing depth</strong> and <strong>gene length</strong>.</p>
<p><span class="math display">\[
\mathrm{TPM}=\frac{\frac{\text { total reads mapped to gene } \times 10^{3}}{\text { gene length }}}{\sum(\frac{\text { total reads mapped to gene } \times 10^{3}}{\text { gene length }})} \times 10^{6}
\]</span></p>
<p>TPM is proportional to RPKM:</p>
<p><span class="math display">\[
\mathrm{TPM}=\frac{R P K M}{\sum(R P K M)} \times 10^{6}
\]</span></p>
<div class="sourceCode" id="cb253"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tpm.matrix</span><span class="op">=</span><span class="fu"><a href="../reference/NormalizedCount.html">NormalizedCount</a></span><span class="op">(</span><span class="va">dds</span>,gtf.file <span class="op">=</span> <span class="st">'/path/to/gtf'</span>, norm.type<span class="op">=</span><span class="st">"TPM"</span><span class="op">)</span></code></pre></div>
<hr>
</div>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb254"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-conda-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /home/softwares/anaconda3/envs/r4.0/lib/libopenblasp-r0.3.12.so
## 
## locale:
##  [1] LC_CTYPE=zh_CN.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=zh_CN.UTF-8        LC_COLLATE=zh_CN.UTF-8    
##  [5] LC_MONETARY=zh_CN.UTF-8    LC_MESSAGES=zh_CN.UTF-8   
##  [7] LC_PAPER=zh_CN.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] TxDb.Mmusculus.UCSC.mm10.knownGene_3.10.0
##  [2] GenomicFeatures_1.42.2                   
##  [3] msigdbr_7.5.1                            
##  [4] edgeR_3.32.1                             
##  [5] limma_3.46.0                             
##  [6] org.Mm.eg.db_3.12.0                      
##  [7] AnnotationDbi_1.52.0                     
##  [8] DESeq2_1.30.1                            
##  [9] SummarizedExperiment_1.20.0              
## [10] Biobase_2.50.0                           
## [11] MatrixGenerics_1.2.1                     
## [12] matrixStats_0.58.0                       
## [13] GenomicRanges_1.42.0                     
## [14] GenomeInfoDb_1.26.7                      
## [15] IRanges_2.24.1                           
## [16] S4Vectors_0.28.1                         
## [17] BiocGenerics_0.42.0                      
## [18] DEbPeak_0.6.0                            
## [19] knitr_1.37                               
## [20] BiocStyle_2.18.1                         
## 
## loaded via a namespace (and not attached):
##   [1] ggvenn_0.1.9                           
##   [2] utf8_1.2.1                             
##   [3] tidyselect_1.1.0                       
##   [4] RSQLite_2.2.5                          
##   [5] grid_4.0.3                             
##   [6] BiocParallel_1.24.1                    
##   [7] scatterpie_0.1.7                       
##   [8] munsell_0.5.0                          
##   [9] codetools_0.2-18                       
##  [10] ragg_0.4.0                             
##  [11] misc3d_0.9-1                           
##  [12] colorspace_2.0-0                       
##  [13] GOSemSim_2.16.1                        
##  [14] highr_0.8                              
##  [15] rstudioapi_0.13                        
##  [16] robustbase_0.95-0                      
##  [17] DOSE_3.16.0                            
##  [18] labeling_0.4.2                         
##  [19] GenomeInfoDbData_1.2.4                 
##  [20] polyclip_1.10-0                        
##  [21] bit64_4.0.5                            
##  [22] farver_2.1.0                           
##  [23] rprojroot_2.0.2                        
##  [24] downloader_0.4                         
##  [25] vctrs_0.4.1                            
##  [26] generics_0.1.0                         
##  [27] xfun_0.30                              
##  [28] BiocFileCache_1.14.0                   
##  [29] R6_2.5.0                               
##  [30] doParallel_1.0.16                      
##  [31] clue_0.3-59                            
##  [32] graphlayouts_0.7.1                     
##  [33] rsvd_1.0.3                             
##  [34] locfit_1.5-9.4                         
##  [35] gridGraphics_0.5-1                     
##  [36] bitops_1.0-6                           
##  [37] cachem_1.0.4                           
##  [38] fgsea_1.16.0                           
##  [39] DelayedArray_0.16.3                    
##  [40] assertthat_0.2.1                       
##  [41] scales_1.1.1                           
##  [42] ggraph_2.0.5                           
##  [43] enrichplot_1.10.2                      
##  [44] gtable_0.3.0                           
##  [45] beachmat_2.6.4                         
##  [46] Cairo_1.5-12.2                         
##  [47] sva_3.38.0                             
##  [48] tidygraph_1.2.0                        
##  [49] rlang_1.0.3                            
##  [50] genefilter_1.72.1                      
##  [51] systemfonts_1.0.1                      
##  [52] GlobalOptions_0.1.2                    
##  [53] splines_4.0.3                          
##  [54] rtracklayer_1.50.0                     
##  [55] checkmate_2.0.0                        
##  [56] BiocManager_1.30.16                    
##  [57] yaml_2.2.1                             
##  [58] reshape2_1.4.4                         
##  [59] backports_1.2.1                        
##  [60] qvalue_2.22.0                          
##  [61] clusterProfiler_3.18.1                 
##  [62] tcltk_4.0.3                            
##  [63] tools_4.0.3                            
##  [64] bookdown_0.26                          
##  [65] ggplotify_0.1.0                        
##  [66] ggplot2_3.3.5                          
##  [67] ellipsis_0.3.2                         
##  [68] gplots_3.1.1                           
##  [69] jquerylib_0.1.3                        
##  [70] RColorBrewer_1.1-2                     
##  [71] Rcpp_1.0.9                             
##  [72] plyr_1.8.6                             
##  [73] sparseMatrixStats_1.2.1                
##  [74] progress_1.2.2                         
##  [75] zlibbioc_1.36.0                        
##  [76] purrr_0.3.4                            
##  [77] RCurl_1.98-1.3                         
##  [78] prettyunits_1.1.1                      
##  [79] openssl_1.4.3                          
##  [80] GetoptLong_1.0.5                       
##  [81] viridis_0.6.1                          
##  [82] cowplot_1.1.1                          
##  [83] ggrepel_0.9.1                          
##  [84] cluster_2.1.1                          
##  [85] fs_1.5.0                               
##  [86] magrittr_2.0.1                         
##  [87] data.table_1.14.2                      
##  [88] DO.db_2.9                              
##  [89] circlize_0.4.15                        
##  [90] mvtnorm_1.1-2                          
##  [91] patchwork_1.0.0                        
##  [92] hms_1.0.0                              
##  [93] evaluate_0.14                          
##  [94] xtable_1.8-4                           
##  [95] XML_3.99-0.6                           
##  [96] gridExtra_2.3                          
##  [97] shape_1.4.6                            
##  [98] ggupset_0.3.0                          
##  [99] compiler_4.0.3                         
## [100] biomaRt_2.46.3                         
## [101] tibble_3.1.0                           
## [102] KernSmooth_2.23-18                     
## [103] crayon_1.4.1                           
## [104] shadowtext_0.0.9                       
## [105] htmltools_0.5.2                        
## [106] mgcv_1.8-34                            
## [107] pcaPP_2.0-1                            
## [108] ggfun_0.0.6                            
## [109] rrcov_1.7-0                            
## [110] tidyr_1.1.3                            
## [111] geneplotter_1.68.0                     
## [112] DBI_1.1.1                              
## [113] tweenr_1.0.2                           
## [114] ChIPseeker_1.33.0.900                  
## [115] dbplyr_2.1.1                           
## [116] ComplexHeatmap_2.13.1                  
## [117] MASS_7.3-53.1                          
## [118] rappdirs_0.3.3                         
## [119] boot_1.3-28                            
## [120] babelgene_21.4                         
## [121] Matrix_1.3-3                           
## [122] cli_3.3.0                              
## [123] parallel_4.0.3                         
## [124] igraph_1.3.4                           
## [125] pkgconfig_2.0.3                        
## [126] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
## [127] pkgdown_1.6.1                          
## [128] rvcheck_0.1.8                          
## [129] GenomicAlignments_1.26.0               
## [130] xml2_1.3.2                             
## [131] foreach_1.5.1                          
## [132] PCAtools_2.2.0                         
## [133] annotate_1.68.0                        
## [134] bslib_0.3.1                            
## [135] dqrng_0.2.1                            
## [136] DEFormats_1.18.0                       
## [137] XVector_0.30.0                         
## [138] yulab.utils_0.0.4                      
## [139] stringr_1.4.0                          
## [140] digest_0.6.27                          
## [141] Biostrings_2.58.0                      
## [142] rmarkdown_2.14                         
## [143] fastmatch_1.1-3                        
## [144] DelayedMatrixStats_1.12.3              
## [145] curl_4.3                               
## [146] Rsamtools_2.6.0                        
## [147] gtools_3.8.2                           
## [148] NOISeq_2.34.0                          
## [149] rjson_0.2.20                           
## [150] nlme_3.1-152                           
## [151] lifecycle_1.0.0                        
## [152] jsonlite_1.7.2                         
## [153] desc_1.3.0                             
## [154] viridisLite_0.4.0                      
## [155] askpass_1.1                            
## [156] fansi_0.4.2                            
## [157] pillar_1.5.1                           
## [158] lattice_0.20-45                        
## [159] DEoptimR_1.0-11                        
## [160] fastmap_1.1.0                          
## [161] httr_1.4.2                             
## [162] plotrix_3.8-2                          
## [163] survival_3.2-10                        
## [164] GO.db_3.12.1                           
## [165] glue_1.6.2                             
## [166] png_0.1-7                              
## [167] iterators_1.0.13                       
## [168] plot3D_1.4                             
## [169] bit_4.0.4                              
## [170] ggforce_0.3.3                          
## [171] stringi_1.5.3                          
## [172] sass_0.4.1                             
## [173] blob_1.2.1                             
## [174] textshaping_0.1.2                      
## [175] BiocSingular_1.6.0                     
## [176] caTools_1.18.2                         
## [177] memoise_2.0.0                          
## [178] dplyr_1.0.5                            
## [179] irlba_2.3.5</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
