<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DifferentialExpressionAnalysisPeak â€¢ DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="DifferentialExpressionAnalysisPeak">
<meta property="og:description" content="DEbPeak">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="DifferentialExpressionAnalysisPeak_files/header-attrs-2.14/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>DifferentialExpressionAnalysisPeak</h1>
            
      
      
      <div class="hidden name"><code>DifferentialExpressionAnalysisPeak.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Here, we will perform differential expression analysis based on <code>DESeq2</code> and <code>edgeR</code> to get differentially accessible (DA) peaks. Similar to those in <a href="https://showteeth.github.io/DEbPeak/articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>, <code>DEbPeak</code> provides <strong>six differents charts</strong> to visualize these DA peaks (<strong>VolcanoPlot</strong>, <strong>ScatterPlot</strong>, <strong>MAPlot</strong>, <strong>RankPlot</strong>, <strong>GenePlot</strong> and <strong>DEHeatmap</strong>).</p>
<hr>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>The data used here are from <a href="https://doi.org/10.1016/j.cell.2021.02.019">In vivo CD8+ T cell CRISPR screening reveals control by Fli1 in infection and cancer</a>:</p>
<ul>
<li>
<strong>RNA-seq data</strong>: the sgCtrl vs sgFli1 RNA sequencing at D8 Cl13 p.i., the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149838">GSE149838</a>
</li>
<li>
<strong>ATAC-seq data</strong>: sgCtrl vs sgFli1 ATAC sequencing at D9 Cl13 p.i., the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149836">GSE149836</a>
</li>
</ul>
<div id="count-matrix" class="section level3">
<h3 class="hasAnchor">
<a href="#count-matrix" class="anchor"></a>Count matrix</h3>
<p>The raw data:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># read the raw data</span>
<span class="va">atac.counts</span> <span class="op">=</span> <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/GSE149836_combUnionReadsWithLabels.txt"</span>,
                                sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">atac.counts</span><span class="op">)</span></code></pre></div>
<pre><code>##    chr   start     end X290_1_S4 X290_2_S5 X360_1_S6 X360_2_S7 R_1_S2 R_2_S3
## 1 chr1 3131803 3131948         2        15         0         3      4      0
## 2 chr1 3210046 3210121         0         1         0         1      0      0
## 3 chr1 3265577 3265648         3         2         0         8      5      2
## 4 chr1 3345206 3345286         3         2         1         0      2      1
## 5 chr1 3505380 3505504         3         4         0        12      4      2
## 6 chr1 3513642 3513745         0         0         2         3     13      2</code></pre>
<p>To perform analysis (including <code>Quality Control</code>), we need to preprocess the data (row: feature, column: sample):</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># read the processed data, the feature is consist of peak region, gene symbol and annotated binding region.</span>
<span class="va">atac.counts.file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RA_ATAC_count.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">atac.counts</span> <span class="op">=</span> <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">atac.counts.file</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">atac.counts</span><span class="op">)</span></code></pre></div>
<pre><code>##                                                    Fli1KO_1 Fli1KO_2 Fli1KO_3
## chr1_GL456211_random:112828-112899|LOC100041034|DI        4        6       14
## chr1_GL456211_random:113055-113190|LOC100041034|DI        7       13       10
## chr1_GL456211_random:147874-147989|LOC100041034|DI       15       16       22
## chr1_GL456211_random:167241-167436|LOC100041034|DI       53       40       43
## chr1_GL456211_random:174677-174747|LOC100041034|I         9        6       17
## chr1_GL456211_random:199244-199444|LOC100041034|DI       23       14       26
##                                                    Fli1KO_4 WT_1 WT_2
## chr1_GL456211_random:112828-112899|LOC100041034|DI       12   10    3
## chr1_GL456211_random:113055-113190|LOC100041034|DI       11   21   31
## chr1_GL456211_random:147874-147989|LOC100041034|DI       14   23   27
## chr1_GL456211_random:167241-167436|LOC100041034|DI       54   39   46
## chr1_GL456211_random:174677-174747|LOC100041034|I         6   14   10
## chr1_GL456211_random:199244-199444|LOC100041034|DI       11   27   46</code></pre>
<p><mark>If you have <strong>raw bam files</strong> of input/control and treatment samples and sample metadata, <code>DEbPeak</code> provides <code>PeakMatrix</code> to prepare the above count matrix.</mark></p>
</div>
<div id="sample-metadata" class="section level3">
<h3 class="hasAnchor">
<a href="#sample-metadata" class="anchor"></a>Sample metadata</h3>
<p>And, we also need the sample metadata:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># read the processed data</span>
<span class="va">atac.meta.file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RA_ATAC_meta.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">atac.meta</span> <span class="op">=</span> <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">atac.meta.file</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">atac.meta</span><span class="op">)</span></code></pre></div>
<pre><code>##          condition
## Fli1KO_1        KO
## Fli1KO_2        KO
## Fli1KO_3        KO
## Fli1KO_4        KO
## WT_1            WT
## WT_2            WT</code></pre>
</div>
<div id="create-deseqdataset" class="section level3">
<h3 class="hasAnchor">
<a href="#create-deseqdataset" class="anchor"></a>Create DESeqDataSet</h3>
<p>With above data, we can create <code>DESeqDataSet</code> object:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2">DESeq2</a></span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co"># create dds</span>
<span class="va">dds.atac</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span><span class="op">(</span>
  countData <span class="op">=</span> <span class="va">atac.counts</span>, colData <span class="op">=</span> <span class="va">atac.meta</span>,
  design <span class="op">=</span> <span class="op">~</span><span class="va">condition</span>
<span class="op">)</span></code></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<hr>
</div>
</div>
<div id="load-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2">DESeq2</a></span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="co"># Already loaded</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DEbPeak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-analysis" class="anchor"></a>Differential Expression Analysis</h2>
<p>To increase flexibility, <code>DEbPeak</code> is compatible with both <code>DESeq2</code> and <code>edgeR</code> results. We provides standard commands to get DA peaks with both <code>DESeq2</code> and <code>edgeR</code>.</p>
<div id="deseq2" class="section level3">
<h3 class="hasAnchor">
<a href="#deseq2" class="anchor"></a>DESeq2</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set control level</span>
<span class="va">dds.atac</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html">relevel</a></span><span class="op">(</span><span class="va">dds.atac</span><span class="op">$</span><span class="va">condition</span>, ref <span class="op">=</span> <span class="st">"WT"</span><span class="op">)</span>
<span class="co"># conduct differential expressed genes analysis</span>
<span class="va">dds.atac</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span><span class="op">(</span><span class="va">dds.atac</span><span class="op">)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## -- note: fitType='parametric', but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType='local' or 'mean' to avoid this message next time.</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># extract results</span>
<span class="va">dds.atac.results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds.atac</span>,contrast<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"condition"</span>,<span class="st">'KO'</span>,<span class="st">'WT'</span><span class="op">)</span><span class="op">)</span>
<span class="va">dds.atac.results.ordered</span> <span class="op">&lt;-</span> <span class="va">dds.atac.results</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">dds.atac.results</span><span class="op">$</span><span class="va">log2FoldChange</span>,decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dds.atac.results.ordered</span><span class="op">)</span></code></pre></div>
<pre><code>## log2 fold change (MLE): condition KO vs WT 
## Wald test p-value: condition KO vs WT 
## DataFrame with 6 rows and 6 columns
##                                           baseMean log2FoldChange     lfcSE
##                                          &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;
## chr8:94825359-94825432|Ciapin1|P          12.10658        6.43921   1.52702
## chr3:105628355-105628501|Ddx20|I          11.18000        6.32317   1.52344
## chr1:118440678-118440831|2900060B14Rik|I   7.83684        5.79091   1.56965
## chr1:165368382-165368517|Dcaf6|I           6.14017        5.43928   1.61963
## chr16:59434969-59435053|Gabrr3|I           5.70677        5.34867   1.63142
## chr11:20780454-20780545|Aftph|DI           5.39047        5.26962   1.63661
##                                               stat      pvalue       padj
##                                          &lt;numeric&gt;   &lt;numeric&gt;  &lt;numeric&gt;
## chr8:94825359-94825432|Ciapin1|P           4.21686 2.47728e-05 0.00126570
## chr3:105628355-105628501|Ddx20|I           4.15058 3.31633e-05 0.00159881
## chr1:118440678-118440831|2900060B14Rik|I   3.68931 2.24864e-04 0.00689256
## chr1:165368382-165368517|Dcaf6|I           3.35834 7.84110e-04 0.01786577
## chr16:59434969-59435053|Gabrr3|I           3.27854 1.04345e-03 0.02191561
## chr11:20780454-20780545|Aftph|DI           3.21985 1.28258e-03 0.02530960</code></pre>
<hr>
</div>
<div id="edger" class="section level3">
<h3 class="hasAnchor">
<a href="#edger" class="anchor"></a>edgeR</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">atac.edgeR</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">atac.counts</span>, group<span class="op">=</span><span class="va">atac.meta</span><span class="op">$</span><span class="va">condition</span><span class="op">)</span>
<span class="va">atac.edgeR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span><span class="op">(</span><span class="va">atac.edgeR</span><span class="op">)</span>
<span class="va">atac.edgeR</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html">relevel</a></span><span class="op">(</span><span class="va">atac.edgeR</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span>, ref<span class="op">=</span><span class="st">"WT"</span><span class="op">)</span>
<span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="va">atac.edgeR</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span><span class="op">)</span>
<span class="va">atac.edgeR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateDisp.html">estimateDisp</a></span><span class="op">(</span><span class="va">atac.edgeR</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html">glmQLFit</a></span><span class="op">(</span><span class="va">atac.edgeR</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">qlf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html">glmQLFTest</a></span><span class="op">(</span><span class="va">fit</span>, coef<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
<span class="va">all.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html">topTags</a></span><span class="op">(</span><span class="va">qlf</span>,n<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">atac.edgeR</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">table</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">all.res</span><span class="op">)</span></code></pre></div>
<pre><code>##                                             logFC    logCPM        F
## chr1:24612887-24616152|Col19a1|E        -0.289687 15.328812 354.3361
## chr2:146197847-146198364|Cfap61|I       -2.240169  5.098303 330.5501
## chr13:74528968-74529954|Zfp825|DI       -1.683876  6.121681 248.1321
## chr8:105874405-105874857|Edc4|5U         3.602722  4.205464 232.7117
## chr16:97401751-97402110|Bace2|I         -2.972395  3.446204 203.3962
## chr17:13552532-13553181|2700054A10Rik|I -1.588283  5.411427 175.7008
##                                               PValue          FDR
## chr1:24612887-24616152|Col19a1|E        2.972132e-49 3.029227e-44
## chr2:146197847-146198364|Cfap61|I       4.119077e-47 2.099102e-42
## chr13:74528968-74529954|Zfp825|DI       6.285243e-39 2.135327e-34
## chr8:105874405-105874857|Edc4|5U        3.039325e-37 7.744275e-33
## chr16:97401751-97402110|Bace2|I         6.970483e-34 1.420877e-29
## chr17:13552532-13553181|2700054A10Rik|I 1.699629e-30 2.887131e-26</code></pre>
<hr>
</div>
</div>
<div id="visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h2>
<div id="volcanoplot" class="section level3">
<h3 class="hasAnchor">
<a href="#volcanoplot" class="anchor"></a>VolcanoPlot</h3>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># VolcanoPlot for DESeq2</span>
<span class="fu"><a href="../reference/VolcanoPlot.html">VolcanoPlot</a></span><span class="op">(</span><span class="va">dds.atac.results.ordered</span>,signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,
            point.alpha <span class="op">=</span> <span class="fl">0.8</span>, label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span>,tick.trans <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_volcanoplot-1.png" width="480" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># VolcanoPlot for edgeR</span>
<span class="fu"><a href="../reference/VolcanoPlot.html">VolcanoPlot</a></span><span class="op">(</span><span class="va">all.res</span>,signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,point.alpha <span class="op">=</span> <span class="fl">0.8</span>,
            label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span>,tick.trans <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_volcanoplot-2.png" width="480" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="scatterplot" class="section level3">
<h3 class="hasAnchor">
<a href="#scatterplot" class="anchor"></a>ScatterPlot</h3>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ScatterPlot for DESeq2</span>
<span class="fu"><a href="../reference/ScatterPlot.html">ScatterPlot</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds.atac</span>,deres <span class="op">=</span> <span class="va">dds.atac.results.ordered</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
            ref.group <span class="op">=</span> <span class="st">"WT"</span>,signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num <span class="op">=</span> <span class="fl">2</span>,
            point.alpha <span class="op">=</span> <span class="fl">0.8</span>,label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_ScatterPlot-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ScatterPlot for edgeR</span>
<span class="fu"><a href="../reference/ScatterPlot.html">ScatterPlot</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">atac.edgeR</span>,deres <span class="op">=</span> <span class="va">all.res</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
            ref.group <span class="op">=</span> <span class="st">"WT"</span>,signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num <span class="op">=</span> <span class="fl">2</span>,
            point.alpha <span class="op">=</span> <span class="fl">0.8</span>,label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_ScatterPlot-2.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="maplot" class="section level3">
<h3 class="hasAnchor">
<a href="#maplot" class="anchor"></a>MAPlot</h3>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># MAPlot for DESeq2</span>
<span class="fu"><a href="../reference/MAPlot.html">MAPlot</a></span><span class="op">(</span><span class="va">dds.atac.results.ordered</span>,signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,
       point.alpha <span class="op">=</span> <span class="fl">0.8</span>, label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_MAPlot-1.png" width="768" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># MAPlot for edgeR</span>
<span class="fu"><a href="../reference/MAPlot.html">MAPlot</a></span><span class="op">(</span><span class="va">all.res</span>,signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,point.alpha <span class="op">=</span> <span class="fl">0.8</span>,
       label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_MAPlot-2.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="rankplot" class="section level3">
<h3 class="hasAnchor">
<a href="#rankplot" class="anchor"></a>RankPlot</h3>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># RankPlot for DESeq2</span>
<span class="fu"><a href="../reference/RankPlot.html">RankPlot</a></span><span class="op">(</span><span class="va">dds.atac.results.ordered</span>,signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,
         point.alpha <span class="op">=</span> <span class="fl">0.8</span>, label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_RankPlot-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># RankPlot for edgeR</span>
<span class="fu"><a href="../reference/RankPlot.html">RankPlot</a></span><span class="op">(</span><span class="va">all.res</span>,signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span>,label.num<span class="op">=</span><span class="fl">2</span>,point.alpha <span class="op">=</span> <span class="fl">0.8</span>,
         label.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"purple"</span>,<span class="st">"green"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_RankPlot-2.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="geneplot" class="section level3">
<h3 class="hasAnchor">
<a href="#geneplot" class="anchor"></a>GenePlot</h3>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># GenePlot for DESeq2</span>
<span class="fu"><a href="../reference/GenePlot.html">GenePlot</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds.atac</span>,deres <span class="op">=</span> <span class="va">dds.atac.results.ordered</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
         ref.group <span class="op">=</span> <span class="st">"WT"</span>,fill.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"blue"</span><span class="op">)</span>, fill.alpha <span class="op">=</span> <span class="fl">0.8</span>,
         gene.num <span class="op">=</span><span class="fl">2</span>,signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_GenePlot-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># GenePlot for edgeR</span>
<span class="fu"><a href="../reference/GenePlot.html">GenePlot</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">atac.edgeR</span>,deres <span class="op">=</span> <span class="va">all.res</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
         ref.group <span class="op">=</span> <span class="st">"WT"</span>,fill.color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"blue"</span><span class="op">)</span>,fill.alpha <span class="op">=</span> <span class="fl">0.8</span>,
         gene.num <span class="op">=</span><span class="fl">2</span>,signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_GenePlot-2.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="deheatmap" class="section level3">
<h3 class="hasAnchor">
<a href="#deheatmap" class="anchor"></a>DEHeatmap</h3>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DEHeatmap for DESeq2</span>
<span class="fu"><a href="../reference/DEHeatmap.html">DEHeatmap</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds.atac</span>,deres <span class="op">=</span> <span class="va">dds.atac.results.ordered</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
          ref.group <span class="op">=</span> <span class="st">"WT"</span>, signif<span class="op">=</span><span class="st">"pvalue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_DEHeatmap-1.png" width="960" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DEHeatmap for edgeR</span>
<span class="fu"><a href="../reference/DEHeatmap.html">DEHeatmap</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">atac.edgeR</span>,deres <span class="op">=</span> <span class="va">all.res</span>,group.key <span class="op">=</span> <span class="st">"condition"</span>,
          ref.group <span class="op">=</span> <span class="st">"WT"</span>, signif<span class="op">=</span><span class="st">"PValue"</span>,l2fc.threshold<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with edgeR!</code></pre>
<p><img src="DifferentialExpressionAnalysisPeak_files/figure-html/atac_DEHeatmap-2.png" width="960" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-conda-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /home/softwares/anaconda3/envs/r4.0/lib/libopenblasp-r0.3.12.so
## 
## locale:
##  [1] LC_CTYPE=zh_CN.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=zh_CN.UTF-8        LC_COLLATE=zh_CN.UTF-8    
##  [5] LC_MONETARY=zh_CN.UTF-8    LC_MESSAGES=zh_CN.UTF-8   
##  [7] LC_PAPER=zh_CN.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] DEbPeak_1.0.0               edgeR_3.32.1               
##  [3] limma_3.46.0                DESeq2_1.30.1              
##  [5] SummarizedExperiment_1.20.0 Biobase_2.50.0             
##  [7] MatrixGenerics_1.2.1        matrixStats_0.58.0         
##  [9] GenomicRanges_1.42.0        GenomeInfoDb_1.26.7        
## [11] IRanges_2.24.1              S4Vectors_0.28.1           
## [13] BiocGenerics_0.42.0        
## 
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                             
##   [2] ggvenn_0.1.9                           
##   [3] apeglm_1.12.0                          
##   [4] Rsamtools_2.6.0                        
##   [5] rsvg_2.1                               
##   [6] foreach_1.5.1                          
##   [7] rprojroot_2.0.2                        
##   [8] crayon_1.4.1                           
##   [9] V8_3.4.2                               
##  [10] MASS_7.3-53.1                          
##  [11] nlme_3.1-152                           
##  [12] backports_1.2.1                        
##  [13] sva_3.38.0                             
##  [14] GOSemSim_2.25.0                        
##  [15] rlang_1.0.3                            
##  [16] XVector_0.30.0                         
##  [17] readxl_1.3.1                           
##  [18] irlba_2.3.5                            
##  [19] GOstats_2.56.0                         
##  [20] BiocParallel_1.24.1                    
##  [21] rjson_0.2.20                           
##  [22] bit64_4.0.5                            
##  [23] glue_1.6.2                             
##  [24] DiffBind_3.0.15                        
##  [25] mixsqp_0.3-43                          
##  [26] pheatmap_1.0.12                        
##  [27] parallel_4.0.3                         
##  [28] AnnotationDbi_1.52.0                   
##  [29] DEFormats_1.18.0                       
##  [30] base64url_1.4                          
##  [31] tcltk_4.0.3                            
##  [32] DOSE_3.23.2                            
##  [33] haven_2.3.1                            
##  [34] tidyselect_1.1.0                       
##  [35] rio_0.5.27                             
##  [36] XML_3.99-0.6                           
##  [37] tidyr_1.1.3                            
##  [38] ggpubr_0.4.0                           
##  [39] GenomicAlignments_1.26.0               
##  [40] xtable_1.8-4                           
##  [41] magrittr_2.0.1                         
##  [42] evaluate_0.14                          
##  [43] ggplot2_3.3.5                          
##  [44] cli_3.3.0                              
##  [45] zlibbioc_1.36.0                        
##  [46] hwriter_1.3.2                          
##  [47] rstudioapi_0.13                        
##  [48] bslib_0.3.1                            
##  [49] GreyListChIP_1.22.0                    
##  [50] fastmatch_1.1-3                        
##  [51] BiocSingular_1.6.0                     
##  [52] xfun_0.30                              
##  [53] askpass_1.1                            
##  [54] clue_0.3-59                            
##  [55] gson_0.0.9                             
##  [56] cluster_2.1.1                          
##  [57] caTools_1.18.2                         
##  [58] tidygraph_1.2.0                        
##  [59] tibble_3.1.0                           
##  [60] ggrepel_0.9.1                          
##  [61] Biostrings_2.58.0                      
##  [62] png_0.1-7                              
##  [63] withr_2.4.1                            
##  [64] bitops_1.0-6                           
##  [65] ggforce_0.3.3                          
##  [66] RBGL_1.66.0                            
##  [67] plyr_1.8.6                             
##  [68] cellranger_1.1.0                       
##  [69] GSEABase_1.52.1                        
##  [70] pcaPP_2.0-1                            
##  [71] dqrng_0.2.1                            
##  [72] coda_0.19-4                            
##  [73] pillar_1.5.1                           
##  [74] gplots_3.1.1                           
##  [75] GlobalOptions_0.1.2                    
##  [76] cachem_1.0.4                           
##  [77] GenomicFeatures_1.42.2                 
##  [78] fs_1.5.0                               
##  [79] GetoptLong_1.0.5                       
##  [80] clusterProfiler_4.7.1                  
##  [81] DelayedMatrixStats_1.12.3              
##  [82] vctrs_0.4.1                            
##  [83] ellipsis_0.3.2                         
##  [84] generics_0.1.0                         
##  [85] plot3D_1.4                             
##  [86] tools_4.0.3                            
##  [87] foreign_0.8-81                         
##  [88] NOISeq_2.34.0                          
##  [89] munsell_0.5.0                          
##  [90] tweenr_1.0.2                           
##  [91] fgsea_1.16.0                           
##  [92] DelayedArray_0.16.3                    
##  [93] fastmap_1.1.0                          
##  [94] compiler_4.0.3                         
##  [95] abind_1.4-5                            
##  [96] rtracklayer_1.50.0                     
##  [97] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
##  [98] GenomeInfoDbData_1.2.4                 
##  [99] gridExtra_2.3                          
## [100] lattice_0.20-45                        
## [101] ggnewscale_0.4.7                       
## [102] AnnotationForge_1.32.0                 
## [103] utf8_1.2.1                             
## [104] dplyr_1.0.5                            
## [105] BiocFileCache_1.14.0                   
## [106] jsonlite_1.7.2                         
## [107] scales_1.1.1                           
## [108] graph_1.68.0                           
## [109] carData_3.0-4                          
## [110] sparseMatrixStats_1.2.1                
## [111] genefilter_1.72.1                      
## [112] car_3.0-11                             
## [113] doParallel_1.0.16                      
## [114] latticeExtra_0.6-29                    
## [115] brew_1.0-6                             
## [116] checkmate_2.0.0                        
## [117] rmarkdown_2.14                         
## [118] openxlsx_4.2.3                         
## [119] pkgdown_1.6.1                          
## [120] cowplot_1.1.1                          
## [121] textshaping_0.1.2                      
## [122] forcats_0.5.1                          
## [123] downloader_0.4                         
## [124] BSgenome_1.58.0                        
## [125] igraph_1.3.4                           
## [126] survival_3.2-10                        
## [127] numDeriv_2016.8-1.1                    
## [128] yaml_2.2.1                             
## [129] plotrix_3.8-2                          
## [130] systemfonts_1.0.1                      
## [131] ashr_2.2-47                            
## [132] SQUAREM_2021.1                         
## [133] htmltools_0.5.2                        
## [134] memoise_2.0.0                          
## [135] VariantAnnotation_1.36.0               
## [136] locfit_1.5-9.4                         
## [137] graphlayouts_0.7.1                     
## [138] batchtools_0.9.15                      
## [139] PCAtools_2.2.0                         
## [140] viridisLite_0.4.0                      
## [141] digest_0.6.27                          
## [142] rrcov_1.7-0                            
## [143] assertthat_0.2.1                       
## [144] rappdirs_0.3.3                         
## [145] emdbook_1.3.12                         
## [146] RSQLite_2.2.5                          
## [147] amap_0.8-18                            
## [148] yulab.utils_0.0.4                      
## [149] misc3d_0.9-1                           
## [150] data.table_1.14.2                      
## [151] blob_1.2.1                             
## [152] ragg_0.4.0                             
## [153] labeling_0.4.2                         
## [154] splines_4.0.3                          
## [155] Cairo_1.5-12.2                         
## [156] ggupset_0.3.0                          
## [157] RCurl_1.98-1.3                         
## [158] broom_0.7.5                            
## [159] hms_1.0.0                              
## [160] colorspace_2.0-0                       
## [161] BiocManager_1.30.16                    
## [162] shape_1.4.6                            
## [163] sass_0.4.1                             
## [164] Rcpp_1.0.9                             
## [165] mvtnorm_1.1-2                          
## [166] circlize_0.4.15                        
## [167] enrichplot_1.10.2                      
## [168] fansi_0.4.2                            
## [169] truncnorm_1.0-8                        
## [170] ChIPseeker_1.33.0.900                  
## [171] R6_2.5.0                               
## [172] grid_4.0.3                             
## [173] lifecycle_1.0.0                        
## [174] ShortRead_1.48.0                       
## [175] zip_2.1.1                              
## [176] curl_4.3                               
## [177] ggsignif_0.6.3                         
## [178] jquerylib_0.1.3                        
## [179] robustbase_0.95-0                      
## [180] DO.db_2.9                              
## [181] Matrix_1.3-3                           
## [182] qvalue_2.22.0                          
## [183] desc_1.3.0                             
## [184] RColorBrewer_1.1-2                     
## [185] iterators_1.0.13                       
## [186] stringr_1.4.0                          
## [187] DOT_0.1                                
## [188] ggpie_0.2.1                            
## [189] beachmat_2.6.4                         
## [190] polyclip_1.10-0                        
## [191] biomaRt_2.46.3                         
## [192] purrr_0.3.4                            
## [193] shadowtext_0.0.9                       
## [194] gridGraphics_0.5-1                     
## [195] mgcv_1.8-34                            
## [196] ComplexHeatmap_2.13.1                  
## [197] openssl_1.4.3                          
## [198] patchwork_1.0.0                        
## [199] bdsmatrix_1.3-4                        
## [200] codetools_0.2-18                       
## [201] invgamma_1.1                           
## [202] GO.db_3.12.1                           
## [203] gtools_3.8.2                           
## [204] prettyunits_1.1.1                      
## [205] dbplyr_2.1.1                           
## [206] gtable_0.3.0                           
## [207] DBI_1.1.1                              
## [208] highr_0.8                              
## [209] ggfun_0.0.6                            
## [210] httr_1.4.2                             
## [211] KernSmooth_2.23-18                     
## [212] stringi_1.5.3                          
## [213] progress_1.2.2                         
## [214] reshape2_1.4.4                         
## [215] farver_2.1.0                           
## [216] annotate_1.68.0                        
## [217] viridis_0.6.1                          
## [218] Rgraphviz_2.34.0                       
## [219] xml2_1.3.2                             
## [220] bbmle_1.0.24                           
## [221] systemPipeR_1.24.3                     
## [222] boot_1.3-28                            
## [223] geneplotter_1.68.0                     
## [224] ggplotify_0.1.0                        
## [225] Category_2.56.0                        
## [226] DEoptimR_1.0-11                        
## [227] bit_4.0.4                              
## [228] scatterpie_0.1.7                       
## [229] jpeg_0.1-8.1                           
## [230] ggraph_2.0.5                           
## [231] pkgconfig_2.0.3                        
## [232] rstatix_0.7.0                          
## [233] knitr_1.37</code></pre>
<hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
