<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>FunctionalEnrichmentAnalysis • DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="FunctionalEnrichmentAnalysis">
<meta property="og:description" content="DEbPeak">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ParseGEO.html">Parse GEO</a>
    </li>
    <li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPDE.html">Integrate RNA-seq with ChIP-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="FunctionalEnrichmentAnalysis_files/header-attrs-2.14/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>FunctionalEnrichmentAnalysis</h1>
            
      
      
      <div class="hidden name"><code>FunctionalEnrichmentAnalysis.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>After DE analysis, functional enrichment analysis is often performed to interpret the results of RNA-seq.</p>
<p><code>DEbPeak</code> provides three kinds of functional enrichment analysis based on <code>clusterProfiler</code>:</p>
<ul>
<li>Gene ontology (GO) analysis with DEGs</li>
<li>Kyoto Encyclopedia of Genes and Genomes (KEGG) pathway enrichment analysis with DEGs</li>
<li>Gene Set Enrichment Analysis (GSEA)</li>
</ul>
<hr>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>We will used the results output by “Quality Control” vignette.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2">DESeq2</a></span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DEbPeak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co"># load data</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/example.RData"</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-analysis" class="anchor"></a>Differential Expression Analysis</h2>
<p>We will use <code>DESeq2</code> as example:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set control level</span>
<span class="va">dds</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html">relevel</a></span><span class="op">(</span><span class="va">dds</span><span class="op">$</span><span class="va">condition</span>, ref <span class="op">=</span> <span class="st">"WT"</span><span class="op">)</span>
<span class="co"># conduct differential expressed genes analysis</span>
<span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># extract results</span>
<span class="va">dds.results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds</span>,contrast<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"condition"</span>,<span class="st">'KO'</span>,<span class="st">'WT'</span><span class="op">)</span><span class="op">)</span>
<span class="va">dds.results.ordered</span> <span class="op">&lt;-</span> <span class="va">dds.results</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">dds.results</span><span class="op">$</span><span class="va">log2FoldChange</span>,decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dds.results.ordered</span><span class="op">)</span></code></pre></div>
<pre><code>## log2 fold change (MLE): condition KO vs WT 
## Wald test p-value: condition KO vs WT 
## DataFrame with 6 rows and 6 columns
##                        baseMean log2FoldChange     lfcSE      stat      pvalue
##                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
## ENSMUSG00000025934.15   2.01093        4.46030   3.01742   1.47818          NA
## ENSMUSG00000105597.1    1.64376        4.16931   2.98872   1.39502 0.163011113
## ENSMUSG00000109866.1    2.32064        4.14573   1.22504   3.38415 0.000713987
## ENSMUSG00000085819.7    2.23441        4.07496   1.65939   2.45569 0.014061354
## ENSMUSG00000102474.1    1.52867        4.06781   1.80781   2.25013 0.024440396
## ENSMUSG00000042677.7    1.49781        4.05394   1.71350   2.36588 0.017987495
##                            padj
##                       &lt;numeric&gt;
## ENSMUSG00000025934.15        NA
## ENSMUSG00000105597.1    0.99973
## ENSMUSG00000109866.1    0.99973
## ENSMUSG00000085819.7    0.99973
## ENSMUSG00000102474.1    0.99973
## ENSMUSG00000042677.7    0.99973</code></pre>
<hr>
</div>
<div id="go-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#go-enrichment" class="anchor"></a>GO enrichment</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># save results to working directory</span>
<span class="co"># ConductFE(deres = dds.results.ordered, signif = "pvalue", l2fc.threshold = 0.3,enrich.type = "GO")</span>
<span class="va">all.go.results</span><span class="op">=</span><span class="fu"><a href="../reference/ConductFE.html">ConductFE</a></span><span class="op">(</span>deres <span class="op">=</span> <span class="va">dds.results.ordered</span>, signif <span class="op">=</span> <span class="st">"pvalue"</span>, l2fc.threshold <span class="op">=</span> <span class="fl">0.3</span>,enrich.type <span class="op">=</span> <span class="st">"GO"</span>,str.width <span class="op">=</span> <span class="fl">50</span>, save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<div id="up-regulated-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#up-regulated-genes" class="anchor"></a>Up regulated genes</h3>
<div id="overview" class="section level4">
<h4 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.go.res</span><span class="op">=</span><span class="va">all.go.results</span><span class="op">[[</span><span class="st">"UP"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="va">up.go.res.table</span><span class="op">=</span><span class="va">up.go.res</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">up.go.res.table</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                              Description
## GO:0097060       CC GO:0097060                        synaptic membrane
## GO:0099240       CC GO:0099240 intrinsic component of synaptic membrane
## GO:0014069       CC GO:0014069                     postsynaptic density
## GO:0032279       CC GO:0032279                       asymmetric synapse
## GO:0098839       CC GO:0098839            postsynaptic density membrane
## GO:0098984       CC GO:0098984                 neuron to neuron synapse
##            GeneRatio   BgRatio       pvalue   p.adjust     qvalue
## GO:0097060    15/231 470/23271 7.783967e-05 0.01822125 0.01447238
## GO:0099240    10/231 244/23271 1.732798e-04 0.01822125 0.01447238
## GO:0014069    13/231 400/23271 1.957825e-04 0.01822125 0.01447238
## GO:0032279    13/231 405/23271 2.208636e-04 0.01822125 0.01447238
## GO:0098839     6/231  97/23271 4.175021e-04 0.01868255 0.01483877
## GO:0098984    13/231 433/23271 4.183638e-04 0.01868255 0.01483877
##                                                                                                  geneID
## GO:0097060 Slc6a2/Met/Nt5e/Lpar2/Grik1/Tenm2/Tiam1/Abhd6/Rgs7bp/Epha7/Snap25/Atp2b2/Adcy1/Adgrl2/Cacng2
## GO:0099240                             Lpar2/Grik1/Abhd6/Rgs7bp/Epha7/Snap25/Atp2b2/Adcy1/Adgrl2/Cacng2
## GO:0014069              Nefh/Met/Grik1/Rapgef4/Tiam1/Rgs7bp/Epha7/Ctnnd2/Map1b/Atp2b2/Pak2/Adcy1/Cacng2
## GO:0032279              Nefh/Met/Grik1/Rapgef4/Tiam1/Rgs7bp/Epha7/Ctnnd2/Map1b/Atp2b2/Pak2/Adcy1/Cacng2
## GO:0098839                                                       Tiam1/Rgs7bp/Epha7/Atp2b2/Adcy1/Cacng2
## GO:0098984              Nefh/Met/Grik1/Rapgef4/Tiam1/Rgs7bp/Epha7/Ctnnd2/Map1b/Atp2b2/Pak2/Adcy1/Cacng2
##            Count
## GO:0097060    15
## GO:0099240    10
## GO:0014069    13
## GO:0032279    13
## GO:0098839     6
## GO:0098984    13</code></pre>
<hr>
</div>
<div id="visualization" class="section level4">
<h4 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h4>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.go.res</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/go_up_plot-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="visualize-selected-terms" class="section level4">
<h4 class="hasAnchor">
<a href="#visualize-selected-terms" class="anchor"></a>Visualize selected terms</h4>
<p>Here, we can visualize selected terms with <code>EnrichPlot</code>.</p>
<p>Extract results:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get all go results</span>
<span class="va">all.up.go.fe.res</span> <span class="op">=</span> <span class="va">all.go.results</span><span class="op">$</span><span class="va">UP</span><span class="op">$</span><span class="va">GO</span><span class="op">$</span><span class="va">table</span>
<span class="co"># extract MF</span>
<span class="va">up.go.mf.fe.res</span> <span class="op">=</span> <span class="va">all.up.go.fe.res</span><span class="op">[</span><span class="va">all.up.go.fe.res</span><span class="op">$</span><span class="va">ONTOLOGY</span><span class="op">==</span><span class="st">"MF"</span>,<span class="op">]</span>
<span class="co"># select terms to plot</span>
<span class="va">up.go.mf.selected</span> <span class="op">=</span> <span class="va">up.go.mf.fe.res</span><span class="op">[</span><span class="va">up.go.mf.fe.res</span><span class="op">$</span><span class="va">ID</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GO:0017016"</span>, <span class="st">"GO:0031267"</span>, <span class="st">"GO:0017048"</span>, <span class="st">"GO:0005085"</span>, <span class="st">"GO:0005088"</span><span class="op">)</span>, <span class="op">]</span></code></pre></div>
<p>Create bar plot:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create bar plot</span>
<span class="fu"><a href="../reference/EnrichPlot.html">EnrichPlot</a></span><span class="op">(</span>fe.res <span class="op">=</span> <span class="va">up.go.mf.selected</span>, grad.col.vec<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/go_up_select_plot_bar-1.png" width="480" style="display: block; margin: auto;"></p>
<p>Create dot plot:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create dot plot</span>
<span class="fu"><a href="../reference/EnrichPlot.html">EnrichPlot</a></span><span class="op">(</span>fe.res <span class="op">=</span> <span class="va">up.go.mf.selected</span>, grad.col.vec<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey"</span>, <span class="st">"red"</span><span class="op">)</span>, plot.type <span class="op">=</span> <span class="st">"dot"</span><span class="op">)</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/go_up_select_plot_dot-1.png" width="480" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="down-regulated-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#down-regulated-genes" class="anchor"></a>Down regulated genes</h3>
<div id="overview-1" class="section level4">
<h4 class="hasAnchor">
<a href="#overview-1" class="anchor"></a>Overview</h4>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.go.res</span><span class="op">=</span><span class="va">all.go.results</span><span class="op">[[</span><span class="st">"DOWN"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="va">down.go.res.table</span> <span class="op">=</span> <span class="va">down.go.res</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">down.go.res.table</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID       Description GeneRatio   BgRatio
## GO:0005604       CC GO:0005604 basement membrane     7/229 114/23271
##                  pvalue   p.adjust     qvalue
## GO:0005604 0.0001374573 0.04178703 0.04007966
##                                                geneID Count
## GO:0005604 Ntn4/Ntn1/Col4a2/Egflam/Papln/Fras1/Col4a4     7</code></pre>
<hr>
</div>
<div id="visualization-1" class="section level4">
<h4 class="hasAnchor">
<a href="#visualization-1" class="anchor"></a>Visualization</h4>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.go.res</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/go_down_plot-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="visualize-selected-terms-1" class="section level4">
<h4 class="hasAnchor">
<a href="#visualize-selected-terms-1" class="anchor"></a>Visualize selected terms</h4>
<p>Here, we can visualize selected terms with <code>EnrichPlot</code>.</p>
<p>Extract results:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get all go results</span>
<span class="va">all.down.go.fe.res</span> <span class="op">=</span> <span class="va">all.go.results</span><span class="op">$</span><span class="va">DOWN</span><span class="op">$</span><span class="va">GO</span><span class="op">$</span><span class="va">table</span>
<span class="co"># extract MF</span>
<span class="va">down.go.cc.fe.res</span> <span class="op">=</span> <span class="va">all.down.go.fe.res</span><span class="op">[</span><span class="va">all.down.go.fe.res</span><span class="op">$</span><span class="va">ONTOLOGY</span><span class="op">==</span><span class="st">"CC"</span>,<span class="op">]</span>
<span class="co"># select terms to plot</span>
<span class="va">down.go.cc.selected</span> <span class="op">=</span> <span class="va">down.go.cc.fe.res</span><span class="op">[</span><span class="va">down.go.cc.fe.res</span><span class="op">$</span><span class="va">ID</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GO:0005604"</span><span class="op">)</span>, <span class="op">]</span></code></pre></div>
<p>Create bar plot:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create bar plot</span>
<span class="fu"><a href="../reference/EnrichPlot.html">EnrichPlot</a></span><span class="op">(</span>fe.res <span class="op">=</span> <span class="va">down.go.cc.selected</span>, grad.col.vec<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/go_down_select_plot_bar-1.png" width="480" style="display: block; margin: auto;"></p>
<p>Create dot plot:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create dot plot</span>
<span class="fu"><a href="../reference/EnrichPlot.html">EnrichPlot</a></span><span class="op">(</span>fe.res <span class="op">=</span> <span class="va">down.go.cc.selected</span>, grad.col.vec<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey"</span>, <span class="st">"red"</span><span class="op">)</span>, plot.type <span class="op">=</span> <span class="st">"dot"</span><span class="op">)</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/go_down_select_plot_dot-1.png" width="480" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
<div id="kegg-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#kegg-enrichment" class="anchor"></a>KEGG enrichment</h2>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># save results to working directory</span>
<span class="co"># ConductFE(deres = dds.results.ordered, signif = "pvalue", l2fc.threshold = 0.3,enrich.type = "KEGG")</span>
<span class="va">all.kegg.results</span><span class="op">=</span><span class="fu"><a href="../reference/ConductFE.html">ConductFE</a></span><span class="op">(</span>deres <span class="op">=</span> <span class="va">dds.results.ordered</span>, signif <span class="op">=</span> <span class="st">"pvalue"</span>, l2fc.threshold <span class="op">=</span> <span class="fl">0.3</span>,enrich.type <span class="op">=</span> <span class="st">"KEGG"</span>, str.width <span class="op">=</span> <span class="fl">50</span>, save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<div id="up-regulated-genes-1" class="section level3">
<h3 class="hasAnchor">
<a href="#up-regulated-genes-1" class="anchor"></a>Up regulated genes</h3>
<div id="overview-2" class="section level4">
<h4 class="hasAnchor">
<a href="#overview-2" class="anchor"></a>Overview</h4>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.kegg.res</span><span class="op">=</span><span class="va">all.kegg.results</span><span class="op">[[</span><span class="st">"UP"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"KEGG"</span><span class="op">]</span><span class="op">]</span>
<span class="va">up.kegg.res.table</span><span class="op">=</span><span class="va">up.kegg.res</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">up.kegg.res.table</span><span class="op">)</span></code></pre></div>
<pre><code>##                ID                                         Description GeneRatio
## mmu04911 mmu04911      Insulin secretion - Mus musculus (house mouse)     7/105
## mmu04024 mmu04024 cAMP signaling pathway - Mus musculus (house mouse)    10/105
##           BgRatio       pvalue   p.adjust     qvalue
## mmu04911  86/9213 5.244736e-05 0.01159087 0.01043426
## mmu04024 224/9213 2.264363e-04 0.02502121 0.02252445
##                                                                    geneID Count
## mmu04911                  Kcnmb2/Kcnn3/Creb5/Rapgef4/Creb3l2/Snap25/Adcy1     7
## mmu04024 Cftr/Atp2a3/Creb5/Rapgef4/Tiam1/Creb3l2/Sox9/Atp2b2/Crebbp/Adcy1    10</code></pre>
<hr>
</div>
<div id="visualization-2" class="section level4">
<h4 class="hasAnchor">
<a href="#visualization-2" class="anchor"></a>Visualization</h4>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">up.kegg.res</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/kegg_up_plot-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="visualize-selected-terms-2" class="section level4">
<h4 class="hasAnchor">
<a href="#visualize-selected-terms-2" class="anchor"></a>Visualize selected terms</h4>
<p>Here, we can visualize selected terms with <code>EnrichPlot</code>.</p>
<p>Extract results:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get all kegg results</span>
<span class="va">all.up.kegg.fe.res</span> <span class="op">=</span> <span class="va">all.kegg.results</span><span class="op">$</span><span class="va">UP</span><span class="op">$</span><span class="va">KEGG</span><span class="op">$</span><span class="va">table</span>
<span class="co"># select terms to plot</span>
<span class="va">up.kegg.selected</span> <span class="op">=</span> <span class="va">all.up.kegg.fe.res</span><span class="op">[</span><span class="va">all.up.kegg.fe.res</span><span class="op">$</span><span class="va">ID</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"mmu04911"</span>, <span class="st">"mmu04024"</span><span class="op">)</span>, <span class="op">]</span></code></pre></div>
<p>Create bar plot:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create bar plot</span>
<span class="fu"><a href="../reference/EnrichPlot.html">EnrichPlot</a></span><span class="op">(</span>fe.res <span class="op">=</span> <span class="va">up.kegg.selected</span>, grad.col.vec<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/kegg_up_select_plot_bar-1.png" width="480" style="display: block; margin: auto;"></p>
<p>Create dot plot:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create dot plot</span>
<span class="fu"><a href="../reference/EnrichPlot.html">EnrichPlot</a></span><span class="op">(</span>fe.res <span class="op">=</span> <span class="va">up.kegg.selected</span>, grad.col.vec<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey"</span>, <span class="st">"red"</span><span class="op">)</span>, plot.type <span class="op">=</span> <span class="st">"dot"</span><span class="op">)</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/kegg_up_select_plot_dot-1.png" width="480" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="down-regulated-genes-1" class="section level3">
<h3 class="hasAnchor">
<a href="#down-regulated-genes-1" class="anchor"></a>Down regulated genes</h3>
<div id="overview-3" class="section level4">
<h4 class="hasAnchor">
<a href="#overview-3" class="anchor"></a>Overview</h4>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.kegg.res</span><span class="op">=</span><span class="va">all.kegg.results</span><span class="op">[[</span><span class="st">"DOWN"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"KEGG"</span><span class="op">]</span><span class="op">]</span>
<span class="va">down.kegg.res.table</span><span class="op">=</span><span class="va">down.kegg.res</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">down.kegg.res.table</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] ID          Description GeneRatio   BgRatio     pvalue      p.adjust   
## [7] qvalue      geneID      Count      
## &lt;0 行&gt; (或0-长度的row.names)</code></pre>
<hr>
</div>
<div id="visualization-3" class="section level4">
<h4 class="hasAnchor">
<a href="#visualization-3" class="anchor"></a>Visualization</h4>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">down.kegg.res</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/kegg_down_plot-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="visualize-selected-terms-3" class="section level4">
<h4 class="hasAnchor">
<a href="#visualize-selected-terms-3" class="anchor"></a>Visualize selected terms</h4>
<p>Here, we can visualize selected terms with <code>EnrichPlot</code>.</p>
<p>Extract results:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get all kegg results</span>
<span class="va">all.down.kegg.fe.res</span> <span class="op">=</span> <span class="va">all.kegg.results</span><span class="op">$</span><span class="va">DOWN</span><span class="op">$</span><span class="va">KEGG</span><span class="op">$</span><span class="va">table</span>
<span class="co"># select terms to plot</span>
<span class="va">down.kegg.selected</span> <span class="op">=</span> <span class="va">all.down.kegg.fe.res</span></code></pre></div>
<p>Create bar plot:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create bar plot</span>
<span class="fu"><a href="../reference/EnrichPlot.html">EnrichPlot</a></span><span class="op">(</span>fe.res <span class="op">=</span> <span class="va">down.kegg.selected</span>, grad.col.vec<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/kegg_down_select_plot_bar-1.png" width="480" style="display: block; margin: auto;"></p>
<p>Create dot plot:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create dot plot</span>
<span class="fu"><a href="../reference/EnrichPlot.html">EnrichPlot</a></span><span class="op">(</span>fe.res <span class="op">=</span> <span class="va">down.kegg.selected</span>, grad.col.vec<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey"</span>, <span class="st">"red"</span><span class="op">)</span>, plot.type <span class="op">=</span> <span class="st">"dot"</span><span class="op">)</span></code></pre></div>
<p><img src="FunctionalEnrichmentAnalysis_files/figure-html/kegg_down_select_plot_dot-1.png" width="480" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
<div id="gsea" class="section level2">
<h2 class="hasAnchor">
<a href="#gsea" class="anchor"></a>GSEA</h2>
<p>Gene Set Enrichment Analysis (GSEA) is a computational method that determines whether an a priori defined set of genes shows statistically significant, concordant differences between two biological states(e.g. phenotypes). <strong>Different from GO and KEGG based on DEGs, GSEA is based on whole gene expression matrix without prior gene filtering</strong>.</p>
<p>While GSEA requires priori defined set of genes, for human, we can download gmt file from <a href="http://www.gsea-msigdb.org/gsea/downloads.jsp">MSigDB</a>, for other species, we can obtain gene sets via <a href="https://igordot.github.io/msigdbr/reference/msigdbr.html">msigdbr</a>.</p>
<div id="all-available-gene-sets" class="section level3">
<h3 class="hasAnchor">
<a href="#all-available-gene-sets" class="anchor"></a>All available gene sets</h3>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/">msigdbr</a></span><span class="op">)</span>
<span class="co"># list all possible species</span>
<span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr_species.html">msigdbr_species</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 20 × 2
##    species_name                    species_common_name                          
##    &lt;chr&gt;                           &lt;chr&gt;                                        
##  1 Anolis carolinensis             Carolina anole, green anole                  
##  2 Bos taurus                      bovine, cattle, cow, dairy cow, domestic cat…
##  3 Caenorhabditis elegans          roundworm                                    
##  4 Canis lupus familiaris          dog, dogs                                    
##  5 Danio rerio                     leopard danio, zebra danio, zebra fish, zebr…
##  6 Drosophila melanogaster         fruit fly                                    
##  7 Equus caballus                  domestic horse, equine, horse                
##  8 Felis catus                     cat, cats, domestic cat                      
##  9 Gallus gallus                   bantam, chicken, chickens, Gallus domesticus 
## 10 Homo sapiens                    human                                        
## 11 Macaca mulatta                  rhesus macaque, rhesus macaques, Rhesus monk…
## 12 Monodelphis domestica           gray short-tailed opossum                    
## 13 Mus musculus                    house mouse, mouse                           
## 14 Ornithorhynchus anatinus        duck-billed platypus, duckbill platypus, pla…
## 15 Pan troglodytes                 chimpanzee                                   
## 16 Rattus norvegicus               brown rat, Norway rat, rat, rats             
## 17 Saccharomyces cerevisiae        baker's yeast, brewer's yeast, S. cerevisiae 
## 18 Schizosaccharomyces pombe 972h- &lt;NA&gt;                                         
## 19 Sus scrofa                      pig, pigs, swine, wild boar                  
## 20 Xenopus tropicalis              tropical clawed frog, western clawed frog</code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create gene sets</span>
<span class="va">m_t2g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Mus musculus"</span>, category <span class="op">=</span> <span class="st">"C5"</span>, subcategory <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">gs_name</span>, <span class="va">entrez_gene</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">m_t2g</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 2
##   gs_name                                          entrez_gene
##   &lt;chr&gt;                                                  &lt;int&gt;
## 1 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS       67618
## 2 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS      107747
## 3 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS      216188
## 4 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS      108156
## 5 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS      270685
## 6 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS      665563</code></pre>
<hr>
</div>
<div id="gsea-1" class="section level3">
<h3 class="hasAnchor">
<a href="#gsea-1" class="anchor"></a>GSEA</h3>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gsea.results</span><span class="op">=</span><span class="fu"><a href="../reference/ConductGSEA.html">ConductGSEA</a></span><span class="op">(</span>deres <span class="op">=</span> <span class="va">dds.results.ordered</span>,gmt.file <span class="op">=</span> <span class="cn">NULL</span>,gene.sets <span class="op">=</span> <span class="va">m_t2g</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="overview-4" class="section level3">
<h3 class="hasAnchor">
<a href="#overview-4" class="anchor"></a>Overview</h3>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gsea.results.table</span><span class="op">=</span><span class="va">gsea.results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gsea.results.table</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] ID              Description     setSize         enrichmentScore
## [5] NES             pvalue          p.adjust        qvalues        
## &lt;0 行&gt; (或0-长度的row.names)</code></pre>
<hr>
</div>
<div id="visualization-4" class="section level3">
<h3 class="hasAnchor">
<a href="#visualization-4" class="anchor"></a>Visualization</h3>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># empty GSEA results, do not visualize</span>
<span class="co"># VisGSEA(gsea.res = gsea.results$gsea, gs.id = c(1,2))</span></code></pre></div>
<hr>
</div>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-conda-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /home/softwares/anaconda3/envs/r4.0/lib/libopenblasp-r0.3.12.so
## 
## locale:
##  [1] LC_CTYPE=zh_CN.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=zh_CN.UTF-8        LC_COLLATE=zh_CN.UTF-8    
##  [5] LC_MONETARY=zh_CN.UTF-8    LC_MESSAGES=zh_CN.UTF-8   
##  [7] LC_PAPER=zh_CN.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] msigdbr_7.5.1               org.Mm.eg.db_3.12.0        
##  [3] AnnotationDbi_1.52.0        DEbPeak_1.2.0              
##  [5] DESeq2_1.30.1               SummarizedExperiment_1.20.0
##  [7] Biobase_2.50.0              MatrixGenerics_1.2.1       
##  [9] matrixStats_0.58.0          GenomicRanges_1.42.0       
## [11] GenomeInfoDb_1.26.7         IRanges_2.24.1             
## [13] S4Vectors_0.28.1            BiocGenerics_0.42.0        
## 
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                             
##   [2] ggvenn_0.1.9                           
##   [3] apeglm_1.12.0                          
##   [4] Rsamtools_2.6.0                        
##   [5] rsvg_2.1                               
##   [6] foreach_1.5.1                          
##   [7] rprojroot_2.0.2                        
##   [8] crayon_1.4.1                           
##   [9] V8_3.4.2                               
##  [10] MASS_7.3-58                            
##  [11] nlme_3.1-152                           
##  [12] backports_1.2.1                        
##  [13] sva_3.38.0                             
##  [14] GOSemSim_2.25.0                        
##  [15] rlang_1.1.0                            
##  [16] XVector_0.30.0                         
##  [17] readxl_1.4.2                           
##  [18] irlba_2.3.5                            
##  [19] limma_3.46.0                           
##  [20] GOstats_2.56.0                         
##  [21] BiocParallel_1.24.1                    
##  [22] rjson_0.2.20                           
##  [23] bit64_4.0.5                            
##  [24] glue_1.6.2                             
##  [25] DiffBind_3.0.15                        
##  [26] mixsqp_0.3-43                          
##  [27] pheatmap_1.0.12                        
##  [28] parallel_4.0.3                         
##  [29] DEFormats_1.18.0                       
##  [30] base64url_1.4                          
##  [31] tcltk_4.0.3                            
##  [32] DOSE_3.23.2                            
##  [33] haven_2.5.2                            
##  [34] tidyselect_1.2.0                       
##  [35] rio_0.5.27                             
##  [36] XML_3.99-0.6                           
##  [37] tidyr_1.3.0                            
##  [38] ggpubr_0.4.0                           
##  [39] GenomicAlignments_1.26.0               
##  [40] xtable_1.8-4                           
##  [41] magrittr_2.0.3                         
##  [42] evaluate_0.14                          
##  [43] ggplot2_3.4.2                          
##  [44] cli_3.6.1                              
##  [45] zlibbioc_1.36.0                        
##  [46] hwriter_1.3.2                          
##  [47] rstudioapi_0.14                        
##  [48] bslib_0.3.1                            
##  [49] GreyListChIP_1.22.0                    
##  [50] fastmatch_1.1-3                        
##  [51] BiocSingular_1.6.0                     
##  [52] xfun_0.30                              
##  [53] askpass_1.1                            
##  [54] clue_0.3-59                            
##  [55] gson_0.0.9                             
##  [56] cluster_2.1.1                          
##  [57] caTools_1.18.2                         
##  [58] tidygraph_1.2.0                        
##  [59] tibble_3.2.1                           
##  [60] ggrepel_0.9.1                          
##  [61] Biostrings_2.58.0                      
##  [62] png_0.1-7                              
##  [63] withr_2.5.0                            
##  [64] bitops_1.0-6                           
##  [65] ggforce_0.3.3                          
##  [66] RBGL_1.66.0                            
##  [67] plyr_1.8.6                             
##  [68] cellranger_1.1.0                       
##  [69] GSEABase_1.52.1                        
##  [70] pcaPP_2.0-1                            
##  [71] dqrng_0.2.1                            
##  [72] coda_0.19-4                            
##  [73] pillar_1.9.0                           
##  [74] gplots_3.1.1                           
##  [75] GlobalOptions_0.1.2                    
##  [76] cachem_1.0.4                           
##  [77] GenomicFeatures_1.42.2                 
##  [78] fs_1.5.0                               
##  [79] GetoptLong_1.0.5                       
##  [80] clusterProfiler_4.7.1                  
##  [81] DelayedMatrixStats_1.12.3              
##  [82] vctrs_0.6.2                            
##  [83] generics_0.1.0                         
##  [84] plot3D_1.4                             
##  [85] tools_4.0.3                            
##  [86] foreign_0.8-81                         
##  [87] NOISeq_2.34.0                          
##  [88] munsell_0.5.0                          
##  [89] tweenr_1.0.2                           
##  [90] fgsea_1.16.0                           
##  [91] DelayedArray_0.16.3                    
##  [92] fastmap_1.1.0                          
##  [93] compiler_4.0.3                         
##  [94] abind_1.4-5                            
##  [95] rtracklayer_1.50.0                     
##  [96] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
##  [97] GenomeInfoDbData_1.2.4                 
##  [98] gridExtra_2.3                          
##  [99] edgeR_3.32.1                           
## [100] lattice_0.20-45                        
## [101] ggnewscale_0.4.7                       
## [102] AnnotationForge_1.32.0                 
## [103] utf8_1.2.1                             
## [104] dplyr_1.1.2                            
## [105] BiocFileCache_1.14.0                   
## [106] jsonlite_1.8.4                         
## [107] scales_1.2.1                           
## [108] graph_1.68.0                           
## [109] carData_3.0-4                          
## [110] sparseMatrixStats_1.2.1                
## [111] genefilter_1.72.1                      
## [112] car_3.0-11                             
## [113] doParallel_1.0.16                      
## [114] latticeExtra_0.6-29                    
## [115] brew_1.0-6                             
## [116] checkmate_2.0.0                        
## [117] rmarkdown_2.14                         
## [118] openxlsx_4.2.3                         
## [119] pkgdown_1.6.1                          
## [120] cowplot_1.1.1                          
## [121] textshaping_0.3.6                      
## [122] forcats_1.0.0                          
## [123] downloader_0.4                         
## [124] BSgenome_1.58.0                        
## [125] igraph_1.3.4                           
## [126] survival_3.2-10                        
## [127] numDeriv_2016.8-1.1                    
## [128] yaml_2.2.1                             
## [129] plotrix_3.8-2                          
## [130] systemfonts_1.0.4                      
## [131] ashr_2.2-47                            
## [132] SQUAREM_2021.1                         
## [133] htmltools_0.5.2                        
## [134] memoise_2.0.0                          
## [135] VariantAnnotation_1.36.0               
## [136] locfit_1.5-9.4                         
## [137] graphlayouts_0.7.1                     
## [138] batchtools_0.9.15                      
## [139] PCAtools_2.2.0                         
## [140] viridisLite_0.4.0                      
## [141] digest_0.6.27                          
## [142] rrcov_1.7-0                            
## [143] assertthat_0.2.1                       
## [144] rappdirs_0.3.3                         
## [145] emdbook_1.3.12                         
## [146] RSQLite_2.2.5                          
## [147] amap_0.8-18                            
## [148] yulab.utils_0.0.4                      
## [149] debugme_1.1.0                          
## [150] misc3d_0.9-1                           
## [151] data.table_1.14.2                      
## [152] blob_1.2.1                             
## [153] ragg_0.4.0                             
## [154] labeling_0.4.2                         
## [155] splines_4.0.3                          
## [156] ggupset_0.3.0                          
## [157] RCurl_1.98-1.3                         
## [158] broom_1.0.4                            
## [159] hms_1.1.3                              
## [160] colorspace_2.0-0                       
## [161] BiocManager_1.30.16                    
## [162] shape_1.4.6                            
## [163] sass_0.4.1                             
## [164] GEOquery_2.58.0                        
## [165] Rcpp_1.0.9                             
## [166] mvtnorm_1.1-2                          
## [167] circlize_0.4.15                        
## [168] enrichplot_1.10.2                      
## [169] fansi_0.4.2                            
## [170] tzdb_0.3.0                             
## [171] truncnorm_1.0-8                        
## [172] ChIPseeker_1.33.0.900                  
## [173] R6_2.5.0                               
## [174] grid_4.0.3                             
## [175] lifecycle_1.0.3                        
## [176] ShortRead_1.48.0                       
## [177] zip_2.1.1                              
## [178] curl_4.3                               
## [179] ggsignif_0.6.3                         
## [180] jquerylib_0.1.3                        
## [181] robustbase_0.95-0                      
## [182] DO.db_2.9                              
## [183] Matrix_1.5-4                           
## [184] qvalue_2.22.0                          
## [185] desc_1.3.0                             
## [186] RColorBrewer_1.1-2                     
## [187] iterators_1.0.13                       
## [188] stringr_1.5.0                          
## [189] DOT_0.1                                
## [190] ggpie_0.2.5                            
## [191] beachmat_2.6.4                         
## [192] polyclip_1.10-0                        
## [193] biomaRt_2.46.3                         
## [194] purrr_1.0.1                            
## [195] shadowtext_0.0.9                       
## [196] gridGraphics_0.5-1                     
## [197] mgcv_1.8-34                            
## [198] ComplexHeatmap_2.13.1                  
## [199] openssl_1.4.3                          
## [200] patchwork_1.0.0                        
## [201] bdsmatrix_1.3-4                        
## [202] codetools_0.2-18                       
## [203] invgamma_1.1                           
## [204] GO.db_3.12.1                           
## [205] gtools_3.8.2                           
## [206] prettyunits_1.1.1                      
## [207] dbplyr_2.3.2                           
## [208] gtable_0.3.0                           
## [209] DBI_1.1.1                              
## [210] highr_0.8                              
## [211] ggfun_0.0.6                            
## [212] httr_1.4.5                             
## [213] KernSmooth_2.23-18                     
## [214] stringi_1.5.3                          
## [215] progress_1.2.2                         
## [216] reshape2_1.4.4                         
## [217] farver_2.1.0                           
## [218] annotate_1.68.0                        
## [219] viridis_0.6.1                          
## [220] Rgraphviz_2.34.0                       
## [221] xml2_1.3.4                             
## [222] bbmle_1.0.24                           
## [223] systemPipeR_1.24.3                     
## [224] boot_1.3-28                            
## [225] readr_2.1.4                            
## [226] geneplotter_1.68.0                     
## [227] ggplotify_0.1.0                        
## [228] Category_2.56.0                        
## [229] DEoptimR_1.0-11                        
## [230] bit_4.0.4                              
## [231] scatterpie_0.1.7                       
## [232] jpeg_0.1-8.1                           
## [233] ggraph_2.0.5                           
## [234] babelgene_21.4                         
## [235] pkgconfig_2.0.3                        
## [236] rstatix_0.7.0                          
## [237] knitr_1.37</code></pre>
<hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
