<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>IntegrateATACDE • DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="IntegrateATACDE">
<meta property="og:description" content="DEbPeak">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="IntegrateATACDE_files/header-attrs-2.14/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>IntegrateATACDE</h1>
            
      
      
      <div class="hidden name"><code>IntegrateATACDE.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p><a href="https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/1756-8935-7-33">Eukaryotic chromatin is tightly packaged into nucleosomes, and the positioning of nucleosomes can regulate gene expression by changing the in vivo availability of binding sites to transcription factors (TFs)</a>. Assay of Transposase Accessible Chromatin sequencing (ATAC-seq) is a technique used to assess genome-wide chromatin accessibility. Thus, combining ATAC-seq with RNA-seq can be used to understanding the role of chromatin structure in regulating gene expression.</p>
<p>Different from those in <a href="https://showteeth.github.io/DEbPeak/articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (intersection)</a>, we will integrate RNA-seq and ATAC-seq based on differential expression analysis results. In detail, <strong>integrating differentially expressed genes with differentially accessible peaks</strong>.</p>
<hr>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>The data used here are from <a href="https://doi.org/10.1016/j.cell.2021.02.019">In vivo CD8+ T cell CRISPR screening reveals control by Fli1 in infection and cancer</a>:</p>
<ul>
<li>
<strong>RNA-seq data</strong>: the sgCtrl vs sgFli1 RNA sequencing at D8 Cl13 p.i., the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149838">GSE149838</a>
</li>
<li>
<strong>ATAC-seq data</strong>: sgCtrl vs sgFli1 ATAC sequencing at D9 Cl13 p.i., the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149836">GSE149836</a>
</li>
</ul>
<hr>
</div>
<div id="load-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DEbPeak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-analysis" class="anchor"></a>Differential Expression Analysis</h2>
<p>To simplify the steps of obtaining differential expression analysis results, <code>DEbPeak</code> provides <code>ConductDESeq2</code> to perfrom all steps of differential expression analysis, including <strong>quality control</strong>, <strong>principal component analysis</strong>, <strong>differential expression analysis</strong> and <strong>functional enrichment analysis</strong>.</p>
<div id="atac-seq" class="section level3">
<h3 class="hasAnchor">
<a href="#atac-seq" class="anchor"></a>ATAC-seq</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">peak.matrix.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RA_ATAC_count.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">peak.meta.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RA_ATAC_meta.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="fu"><a href="../reference/ConductDESeq2.html">ConductDESeq2</a></span><span class="op">(</span>count.matrix.file <span class="op">=</span> <span class="va">peak.matrix.file</span>, meta.file <span class="op">=</span> <span class="va">peak.meta.file</span>, data.type <span class="op">=</span> <span class="st">"ATAC"</span>,
              gene.type <span class="op">=</span> <span class="st">"SYMBOL"</span>, outlier.detection <span class="op">=</span> <span class="cn">F</span>, min.count <span class="op">=</span> <span class="fl">0</span>,
              peak.anno.key <span class="op">=</span> <span class="st">"All"</span>, group.key <span class="op">=</span> <span class="st">"condition"</span>, ref.group <span class="op">=</span> <span class="st">"WT"</span>,
              out.folder <span class="op">=</span> <span class="st">'/home/songyabing/R/learn/tmp/DEbPeak/ATACDE'</span>,
              signif <span class="op">=</span> <span class="st">"padj"</span>, signif.threshold <span class="op">=</span> <span class="fl">0.05</span>, l2fc.threshold <span class="op">=</span> <span class="fl">0</span>, gmt.file <span class="op">=</span> <span class="st">''</span><span class="op">)</span></code></pre></div>
<pre><code>## Create DESeqDataSet from count matrix!</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## Conduct Quality Control on Counts!</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## [1] "Warning: 7 features with 0 counts in all samples are to be removed for this analysis."
## [1] "Count distributions are to be computed for:"
## [1] "Fli1KO_1" "Fli1KO_2" "Fli1KO_3" "Fli1KO_4" "WT_1"     "WT_2"</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Raw gene number: 101921</code></pre>
<pre><code>## Gene number after filtered with counts bigger than 0 : 101921</code></pre>
<pre><code>## Conduct Quality Control on Samples!</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Conduct PCA analysis!
## Conduct PCA analysis!</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Use all genes for PCA!</code></pre>
<pre><code>## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.</code></pre>
<pre><code>## Selecting by Loadding
## Selecting by Loadding
## Differential expression analysis with DESeq2!
## 
## Selecting by Loadding
## Selecting by Loadding
## 
## 载入程辑包：'BiocGenerics'
## 
## 
## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs
## 
## 
## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min
## 
## 
## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.
## 
## 
## 
## 载入程辑包：'S4Vectors'
## 
## 
## The following object is masked from 'package:base':
## 
##     expand.grid
## 
## 
## 
## 
## Conduct Functional Enrichment on PC Loading genes!
## 
## Selecting by Loadding
## Selecting by Loadding
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC1_Positive
## 
## conduct KEGG enrichment analysis: PC1_Positive
## 
## Reading KEGG annotation online: "https://rest.kegg.jp/link/mmu/pathway"...
## 
## Reading KEGG annotation online: "https://rest.kegg.jp/list/pathway/mmu"...
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC1_Negative
## 
## conduct KEGG enrichment analysis: PC1_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC2_Positive
## 
## conduct KEGG enrichment analysis: PC2_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC2_Negative
## 
## conduct KEGG enrichment analysis: PC2_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC3_Positive
## 
## conduct KEGG enrichment analysis: PC3_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC3_Negative
## 
## conduct KEGG enrichment analysis: PC3_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC4_Positive
## 
## conduct KEGG enrichment analysis: PC4_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC4_Negative
## 
## conduct KEGG enrichment analysis: PC4_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC5_Positive
## 
## conduct KEGG enrichment analysis: PC5_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC5_Negative
## 
## conduct KEGG enrichment analysis: PC5_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Conduct Differential Expression Analysis!
## 
## estimating size factors
## 
## estimating dispersions
## 
## gene-wise dispersion estimates
## 
## mean-dispersion relationship
## 
## -- note: fitType='parametric', but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType='local' or 'mean' to avoid this message next time.
## 
## final dispersion estimates
## 
## fitting model and testing
## 
## Conduct Gene ID Conversion!
## 
## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in clusterProfiler::bitr(de.df[, "Gene2Convert"], fromType =
## gene.type, : 1.63% of input gene IDs are fail to map...</code></pre>
<pre><code>## Visualize Differential Analysis Results!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Conduct Functional Enrichment Analysis!
## Differential expression analysis with DESeq2!
## Convert SYMBOL to ENTREZID!
## 'select()' returned 1:1 mapping between keys and columns
## conduct ALL GO enrichment analysis on: UP
## conduct KEGG enrichment analysis: UP
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## Convert SYMBOL to ENTREZID!
## 'select()' returned 1:1 mapping between keys and columns
## conduct ALL GO enrichment analysis on: DOWN
## conduct KEGG enrichment analysis: DOWN
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## Gene Set Enrichment Analysis is not available for ChIP-seq or ATAC-seq data!
## All Analysis Done!</code></pre>
<p>The structure of result directory:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">tree</span> /home/songyabing/R/learn/tmp/DEbPeak/ATACDE</span></code></pre></div>
<pre><code>## /home/songyabing/R/learn/tmp/DEbPeak/ATACDE
## ├── CountQC_CPM.pdf
## ├── CountQC_saturation.pdf
## ├── DA
## │   ├── Condition_KO_WT_all.csv
## │   ├── Condition_KO_WT_padj0.05_FC0.csv
## │   ├── DA_GenePlot.pdf
## │   ├── DA_Heatmap.pdf
## │   ├── DA_MAPlot.pdf
## │   ├── DA_RankPlot.pdf
## │   ├── DA_ScatterPlot.pdf
## │   ├── DA_VolcanoPlot.pdf
## │   └── normalized_counts.csv
## ├── DEG
## │   ├── Condition_KO_WT_all.csv
## │   ├── Condition_KO_WT_padj0.05_FC0.csv
## │   ├── DEG_GenePlot.pdf
## │   ├── DEG_Heatmap.pdf
## │   ├── DEG_MAPlot.pdf
## │   ├── DEG_RankPlot.pdf
## │   ├── DEG_ScatterPlot.pdf
## │   ├── DEG_VolcanoPlot.pdf
## │   └── normalized_counts.csv
## ├── FE
## │   ├── DOWN_ALL_GO.csv
## │   ├── DOWN_Biological_Process.png
## │   ├── DOWN_Cellular_Component.png
## │   ├── DOWN_Functional_Enrichment.pdf
## │   ├── DOWN_KEGG.csv
## │   ├── DOWN_KEGG_Enrichment.png
## │   ├── DOWN_Molecular_Function.png
## │   ├── UP_ALL_GO.csv
## │   ├── UP_Biological_Process.png
## │   ├── UP_Cellular_Component.png
## │   ├── UP_Functional_Enrichment.pdf
## │   ├── UP_KEGG.csv
## │   ├── UP_KEGG_Enrichment.png
## │   └── UP_Molecular_Function.png
## ├── PCA
## │   ├── PC1
## │   │   ├── Negative
## │   │   │   ├── PC1_Negative_ALL_GO.csv
## │   │   │   └── PC1_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC1_Positive_ALL_GO.csv
## │   │       └── PC1_Positive_KEGG.csv
## │   ├── PC1_Negative_Biological_Process.png
## │   ├── PC1_Negative_Cellular_Component.png
## │   ├── PC1_Negative_Functional_Enrichment.pdf
## │   ├── PC1_Negative_KEGG_Enrichment.png
## │   ├── PC1_Negative_Molecular_Function.png
## │   ├── PC1_Positive_Biological_Process.png
## │   ├── PC1_Positive_Cellular_Component.png
## │   ├── PC1_Positive_Functional_Enrichment.pdf
## │   ├── PC1_Positive_KEGG_Enrichment.png
## │   ├── PC1_Positive_Molecular_Function.png
## │   ├── PC2
## │   │   ├── Negative
## │   │   │   ├── PC2_Negative_ALL_GO.csv
## │   │   │   └── PC2_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC2_Positive_ALL_GO.csv
## │   │       └── PC2_Positive_KEGG.csv
## │   ├── PC2_Negative_Biological_Process.png
## │   ├── PC2_Negative_Cellular_Component.png
## │   ├── PC2_Negative_Functional_Enrichment.pdf
## │   ├── PC2_Negative_KEGG_Enrichment.png
## │   ├── PC2_Negative_Molecular_Function.png
## │   ├── PC2_Positive_Biological_Process.png
## │   ├── PC2_Positive_Cellular_Component.png
## │   ├── PC2_Positive_Functional_Enrichment.pdf
## │   ├── PC2_Positive_KEGG_Enrichment.png
## │   ├── PC2_Positive_Molecular_Function.png
## │   ├── PC3
## │   │   ├── Negative
## │   │   │   ├── PC3_Negative_ALL_GO.csv
## │   │   │   └── PC3_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC3_Positive_ALL_GO.csv
## │   │       └── PC3_Positive_KEGG.csv
## │   ├── PC3_Negative_Biological_Process.png
## │   ├── PC3_Negative_Cellular_Component.png
## │   ├── PC3_Negative_Functional_Enrichment.pdf
## │   ├── PC3_Negative_KEGG_Enrichment.png
## │   ├── PC3_Negative_Molecular_Function.png
## │   ├── PC3_Positive_Biological_Process.png
## │   ├── PC3_Positive_Cellular_Component.png
## │   ├── PC3_Positive_Functional_Enrichment.pdf
## │   ├── PC3_Positive_KEGG_Enrichment.png
## │   ├── PC3_Positive_Molecular_Function.png
## │   ├── PC4
## │   │   ├── Negative
## │   │   │   ├── PC4_Negative_ALL_GO.csv
## │   │   │   └── PC4_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC4_Positive_ALL_GO.csv
## │   │       └── PC4_Positive_KEGG.csv
## │   ├── PC4_Negative_Biological_Process.png
## │   ├── PC4_Negative_Cellular_Component.png
## │   ├── PC4_Negative_Functional_Enrichment.pdf
## │   ├── PC4_Negative_KEGG_Enrichment.png
## │   ├── PC4_Negative_Molecular_Function.png
## │   ├── PC4_Positive_Biological_Process.png
## │   ├── PC4_Positive_Cellular_Component.png
## │   ├── PC4_Positive_Functional_Enrichment.pdf
## │   ├── PC4_Positive_KEGG_Enrichment.png
## │   ├── PC4_Positive_Molecular_Function.png
## │   ├── PC5
## │   │   ├── Negative
## │   │   │   ├── PC5_Negative_ALL_GO.csv
## │   │   │   └── PC5_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC5_Positive_ALL_GO.csv
## │   │       └── PC5_Positive_KEGG.csv
## │   ├── PC5_Negative_Biological_Process.png
## │   ├── PC5_Negative_Cellular_Component.png
## │   ├── PC5_Negative_Functional_Enrichment.pdf
## │   ├── PC5_Negative_KEGG_Enrichment.png
## │   ├── PC5_Negative_Molecular_Function.png
## │   ├── PC5_Positive_Biological_Process.png
## │   ├── PC5_Positive_Cellular_Component.png
## │   ├── PC5_Positive_Functional_Enrichment.pdf
## │   ├── PC5_Positive_KEGG_Enrichment.png
## │   ├── PC5_Positive_Molecular_Function.png
## │   ├── PCA_3DPCA.pdf
## │   ├── PCA_biplot.pdf
## │   ├── PCA_loading_bar.pdf
## │   ├── PCA_loading_heat.pdf
## │   ├── PCA_overview.pdf
## │   ├── PCA_pairs_plot.pdf
## │   └── PCA_screen_plot.pdf
## └── SampleQC_dist_pcc.pdf
## 
## 19 directories, 112 files</code></pre>
<p>Load the data:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds.peak.results.ordered</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/ATACDE/DA/Condition_KO_WT_all.csv"</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dds.peak.results.ordered</span><span class="op">)</span></code></pre></div>
<pre><code>##                                           baseMean log2FoldChange    lfcSE
## chr8:94825359-94825432|Ciapin1|P         12.106577       6.439216 1.527019
## chr3:105628355-105628501|Ddx20|I         11.180004       6.323179 1.523447
## chr1:118440678-118440831|2900060B14Rik|I  7.836836       5.790918 1.569651
## chr1:165368382-165368517|Dcaf6|I          6.140173       5.439287 1.619636
## chr16:59434969-59435053|Gabrr3|I          5.706767       5.348676 1.631421
## chr11:20780454-20780545|Aftph|DI          5.390470       5.269629 1.636609
##                                              stat       pvalue        padj
## chr8:94825359-94825432|Ciapin1|P         4.216854 2.477343e-05 0.001265734
## chr3:105628355-105628501|Ddx20|I         4.150575 3.316417e-05 0.001598853
## chr1:118440678-118440831|2900060B14Rik|I 3.689304 2.248687e-04 0.006892699
## chr1:165368382-165368517|Dcaf6|I         3.358339 7.841226e-04 0.017866061
## chr16:59434969-59435053|Gabrr3|I         3.278539 1.043461e-03 0.021915949
## chr11:20780454-20780545|Aftph|DI         3.219846 1.282594e-03 0.025309971
##                                                 SYMBOL            ENSEMBL
## chr8:94825359-94825432|Ciapin1|P               Ciapin1 ENSMUSG00000031781
## chr3:105628355-105628501|Ddx20|I                 Ddx20 ENSMUSG00000027905
## chr1:118440678-118440831|2900060B14Rik|I 2900060B14Rik ENSMUSG00000107722
## chr1:165368382-165368517|Dcaf6|I                 Dcaf6 ENSMUSG00000026571
## chr16:59434969-59435053|Gabrr3|I                Gabrr3 ENSMUSG00000074991
## chr11:20780454-20780545|Aftph|DI                 Aftph ENSMUSG00000049659
##                                          ENTREZID
## chr8:94825359-94825432|Ciapin1|P           109006
## chr3:105628355-105628501|Ddx20|I            53975
## chr1:118440678-118440831|2900060B14Rik|I    68204
## chr1:165368382-165368517|Dcaf6|I            74106
## chr16:59434969-59435053|Gabrr3|I           328699
## chr11:20780454-20780545|Aftph|DI           216549</code></pre>
<hr>
</div>
<div id="rna-seq" class="section level3">
<h3 class="hasAnchor">
<a href="#rna-seq" class="anchor"></a>RNA-seq</h3>
<p><code>DEbPeak</code> has prepared the differential expression analysis results of RNA-seq:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rna.diff.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RA_RNA_diff.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">rna.diff</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">rna.diff.file</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></code></pre></div>
<hr>
</div>
</div>
<div id="integrate-atac-seq-and-rna-seq" class="section level2">
<h2 class="hasAnchor">
<a href="#integrate-atac-seq-and-rna-seq" class="anchor"></a>Integrate ATAC-seq and RNA-seq</h2>
<p>In this step, we will integrate ATAC-seq and RNA-seq to explore the role of chromatin structure of promoter region in regulating target gene expression. ### Integrate</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debatac.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DEbPeak.html">DEbPeak</a></span><span class="op">(</span>de.res <span class="op">=</span> <span class="va">rna.diff</span>, peak.res <span class="op">=</span> <span class="va">dds.peak.results.ordered</span>, peak.mode <span class="op">=</span> <span class="st">"diff"</span>, peak.anno.key <span class="op">=</span> <span class="st">"All"</span>,
                       l2fc.threshold <span class="op">=</span> <span class="fl">0</span>, peak.l2fc.threshold <span class="op">=</span> <span class="fl">0</span>, org.db <span class="op">=</span> <span class="st">"org.Mm.eg.db"</span>, merge.key <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in clusterProfiler::bitr(de.df[, "Gene2Convert"], fromType =
## gene.type, : 1.63% of input gene IDs are fail to map...</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Warning: The `.data` argument of `add_column()` must have unique names as of tibble 3.0.0.
## Use `.name_repair = "minimal"`.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in clusterProfiler::bitr(de.df[, "Gene2Convert"], fromType =
## gene.type, : 13.86% of input gene IDs are fail to map...</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">debatac.res</span><span class="op">)</span></code></pre></div>
<pre><code>##     Peak_SYMBOL                                 Peak_Gene Peak_log2FoldChange
## 1 0610012G03Rik                                      &lt;NA&gt;                  NA
## 2 0610038B21Rik    chr8:77506599-77506784|0610038B21Rik|I           2.6356218
## 3 0610038B21Rik    chr8:77433472-77433594|0610038B21Rik|I           1.5644152
## 4 0610038B21Rik    chr8:77461252-77461753|0610038B21Rik|I           0.8012059
## 5 0610038B21Rik    chr8:77328578-77328867|0610038B21Rik|I          -1.4598963
## 6 0610040F04Rik chr6:108665492-108665967|0610040F04Rik|3U          -0.3895841
##   Peak_abundance Peak_signif Peak_regulation RNA_log2FoldChange RNA_abundance
## 1             NA          NA            &lt;NA&gt;         -0.9030355      755.4293
## 2       33.32172    5.098303    Up_regulated                 NA            NA
## 3       36.83648    2.753478    Up_regulated                 NA            NA
## 4      122.95199    2.261173    Up_regulated                 NA            NA
## 5       24.64638    2.101107  Down_regulated                 NA            NA
## 6      354.09944    1.672098  Down_regulated                 NA            NA
##   RNA_signif RNA_regulation        RNA_ENSEMBL geneId     Type
## 1    4.75213 Down_regulated ENSMUSG00000107002 106264  RNADown
## 2         NA           &lt;NA&gt;               &lt;NA&gt;   &lt;NA&gt;   PeakUp
## 3         NA           &lt;NA&gt;               &lt;NA&gt;   &lt;NA&gt;   PeakUp
## 4         NA           &lt;NA&gt;               &lt;NA&gt;   &lt;NA&gt;   PeakUp
## 5         NA           &lt;NA&gt;               &lt;NA&gt;   &lt;NA&gt; PeakDown
## 6         NA           &lt;NA&gt;               &lt;NA&gt;   &lt;NA&gt; PeakDown</code></pre>
<hr>
<div id="integrate-summary" class="section level3">
<h3 class="hasAnchor">
<a href="#integrate-summary" class="anchor"></a>Integrate summary</h3>
<div id="venn-plot" class="section level4">
<h4 class="hasAnchor">
<a href="#venn-plot" class="anchor"></a>Venn plot</h4>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DE and ATAC venn plot</span>
<span class="va">debatac.plot</span> <span class="op">=</span> <span class="fu"><a href="../reference/PlotDEbPeak.html">PlotDEbPeak</a></span><span class="op">(</span>de.peak <span class="op">=</span> <span class="va">debatac.res</span>, peak.type <span class="op">=</span> <span class="st">"ATAC"</span>, peak.mode <span class="op">=</span> <span class="st">"diff"</span>, show_percentage <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">debatac.plot</span></code></pre></div>
<p><img src="IntegrateATACDE_files/figure-html/debatac_integrate_plot-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="quadrant-diagram" class="section level4">
<h4 class="hasAnchor">
<a href="#quadrant-diagram" class="anchor"></a>Quadrant diagram</h4>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">label.df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Ccl3"</span>, <span class="st">"Ccl5"</span>, <span class="st">"Cd28"</span>, <span class="st">"Cx3cr1"</span>, <span class="st">"Prdm1"</span>, <span class="st">"Tcf7"</span>, <span class="st">"Slamf6"</span>, <span class="st">"Id3"</span>, <span class="st">"Cxcr5"</span><span class="op">)</span><span class="op">)</span>
<span class="va">quad.plot</span> <span class="op">=</span> <span class="fu"><a href="../reference/InteDiffQuad.html">InteDiffQuad</a></span><span class="op">(</span>de.peak <span class="op">=</span> <span class="va">debatac.res</span>, peak.type <span class="op">=</span> <span class="st">"ATAC"</span>, label.df <span class="op">=</span> <span class="va">label.df</span>, label.color <span class="op">=</span> <span class="st">"green"</span><span class="op">)</span>
<span class="va">quad.plot</span></code></pre></div>
<p><img src="IntegrateATACDE_files/figure-html/debatac_integrate_quad_plot-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
<div id="functional-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#functional-enrichment" class="anchor"></a>Functional enrichment</h2>
<p>There are <strong>eight categories</strong> for users to choose to perform functional enrichment: <code>Down_Up</code>, <code>Up_Up</code>, <code>Down_Down</code>, <code>Up_Down</code>, <code>RNAUp</code>, <code>RNADown</code>, <code>PeakUp</code>, <code>PeakDown</code>. Here, we will use <code>Up_Up</code> and <code>Down_Down</code> as examples. ### Both up-regulated</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debatac.up.fe.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/DEbPeakFE.html">DEbPeakFE</a></span><span class="op">(</span>de.peak <span class="op">=</span> <span class="va">debatac.res</span>, peak.fe.key <span class="op">=</span> <span class="st">"Up_Up"</span>, 
                                  gene.type <span class="op">=</span> <span class="st">"ENTREZID"</span>, species<span class="op">=</span><span class="st">"Mouse"</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## conduct ALL GO enrichment analysis on: Up_Up</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<p>The results:</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">debatac.up.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                                  Description
## GO:0022407       BP GO:0022407             regulation of cell-cell adhesion
## GO:0002683       BP GO:0002683 negative regulation of immune system process
## GO:0042098       BP GO:0042098                         T cell proliferation
## GO:0042129       BP GO:0042129           regulation of T cell proliferation
## GO:0007159       BP GO:0007159                 leukocyte cell-cell adhesion
## GO:0046651       BP GO:0046651                     lymphocyte proliferation
##            GeneRatio   BgRatio       pvalue     p.adjust       qvalue
## GO:0022407    24/230 427/23328 7.408694e-12 2.547109e-08 1.973832e-08
## GO:0002683    24/230 462/23328 3.819413e-11 5.035183e-08 3.901916e-08
## GO:0042098    17/230 213/23328 4.393702e-11 5.035183e-08 3.901916e-08
## GO:0042129    15/230 175/23328 2.314904e-10 1.797886e-07 1.393236e-07
## GO:0007159    20/230 345/23328 2.614726e-10 1.797886e-07 1.393236e-07
## GO:0046651    19/230 317/23328 4.264371e-10 2.262048e-07 1.752929e-07
##                                                                                                                                                          geneID
## GO:0022407     Adam19/Casp3/Cblb/Ccl5/Ccr2/Cd28/Cd44/Cd47/Ctla4/Cyld/Ets1/Havcr2/Il21/Lgals3/Malt1/Nckap1l/Notch1/Piezo1/Ptprc/Rdx/Slamf1/Tnfaip3/Twsg1/Ubash3b
## GO:0002683 Abr/Casp3/Cblb/Ccr2/Cd44/Cd47/Cdk6/Ctla4/Cyld/Havcr2/Lgals3/Nlrc3/Notch1/Padi2/Plcl2/Prdm1/Ptprc/Samhd1/Samsn1/Serpinb9/Slamf1/Tnfaip3/Twsg1/Ubash3b
## GO:0042098                                              Casp3/Cblb/Ccl5/Ccr2/Cd28/Cd44/Ctla4/Dock2/Havcr2/Il21/Lgals3/Malt1/Nckap1l/Ptprc/Slamf1/Tnfrsf1b/Twsg1
## GO:0042129                                                          Casp3/Cblb/Ccl5/Ccr2/Cd28/Cd44/Ctla4/Havcr2/Il21/Lgals3/Nckap1l/Ptprc/Slamf1/Tnfrsf1b/Twsg1
## GO:0007159                                 Casp3/Cblb/Ccl5/Ccr2/Cd28/Cd44/Cd47/Ctla4/Cyld/Ets1/Havcr2/Il21/Lgals3/Malt1/Nckap1l/Ptprc/Sema4d/Slamf1/Stk10/Twsg1
## GO:0046651                                  Casp3/Cblb/Ccl5/Ccr2/Cd28/Cd44/Ctla4/Dock2/Havcr2/Il21/Lgals3/Malt1/Nckap1l/Plcl2/Prdm1/Ptprc/Slamf1/Tnfrsf1b/Twsg1
##            Count
## GO:0022407    24
## GO:0002683    24
## GO:0042098    17
## GO:0042129    15
## GO:0007159    20
## GO:0046651    19</code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debatac.up.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="IntegrateATACDE_files/figure-html/debatac_integrate_fe_up_show-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
<div id="both-down-regulated" class="section level3">
<h3 class="hasAnchor">
<a href="#both-down-regulated" class="anchor"></a>Both down-regulated</h3>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debatac.down.fe.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/DEbPeakFE.html">DEbPeakFE</a></span><span class="op">(</span>de.peak <span class="op">=</span> <span class="va">debatac.res</span>, peak.fe.key <span class="op">=</span> <span class="st">"Down_Down"</span>, 
                                    gene.type <span class="op">=</span> <span class="st">"ENTREZID"</span>, species<span class="op">=</span><span class="st">"Mouse"</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## conduct ALL GO enrichment analysis on: Down_Down</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<p>The results:</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">debatac.down.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                             Description
## GO:0030217       BP GO:0030217                  T cell differentiation
## GO:0007159       BP GO:0007159            leukocyte cell-cell adhesion
## GO:0030098       BP GO:0030098              lymphocyte differentiation
## GO:0046631       BP GO:0046631            alpha-beta T cell activation
## GO:0050900       BP GO:0050900                     leukocyte migration
## GO:1902105       BP GO:1902105 regulation of leukocyte differentiation
##            GeneRatio   BgRatio       pvalue     p.adjust       qvalue
## GO:0030217    15/126 282/23328 3.673777e-11 9.496714e-08 7.676260e-08
## GO:0007159    15/126 345/23328 6.108900e-10 6.315386e-07 5.104771e-07
## GO:0030098    16/126 409/23328 7.329268e-10 6.315386e-07 5.104771e-07
## GO:0046631    11/126 166/23328 1.702958e-09 1.100537e-06 8.895715e-07
## GO:0050900    14/126 360/23328 9.779100e-09 5.055795e-06 4.086634e-06
## GO:1902105    13/126 309/23328 1.344419e-08 5.160997e-06 4.171671e-06
##                                                                                                geneID
## GO:0030217        Ccr6/Ccr7/Cd74/Cd83/Crtam/H2-Aa/Kit/Relb/Runx1/Slamf6/Syk/Tcf7/Tespa1/Tnfsf4/Tnfsf8
## GO:0007159           Adk/Bmp7/Btla/Ccr7/Cd74/Cd83/Crtam/H2-Aa/Itgb1/Nt5e/Runx1/Sell/Syk/Tespa1/Tnfsf4
## GO:0030098 Ccr6/Ccr7/Cd74/Cd83/Crtam/H2-Aa/Kit/Relb/Runx1/Slamf6/Syk/Tcf7/Tespa1/Tnfsf4/Tnfsf8/Zbtb7a
## GO:0046631                              Btla/Ccr7/Cd83/Crtam/Relb/Runx1/Slamf6/Syk/Tcf7/Tnfsf4/Tnfsf8
## GO:0050900                 Ccr6/Ccr7/Cd200/Cd74/Cd9/Crtam/Cxcl10/Cxcr5/Il16/Itgb1/Kit/Sell/Syk/Tnfsf4
## GO:1902105                    Car2/Ccr7/Cd74/Cd83/Crtam/H2-Aa/Jun/Runx1/Syk/Tcf7/Tespa1/Tnfsf4/Zbtb7a
##            Count
## GO:0030217    15
## GO:0007159    15
## GO:0030098    16
## GO:0046631    11
## GO:0050900    14
## GO:1902105    13</code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debatac.down.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="IntegrateATACDE_files/figure-html/debatac_integrate_fe_down_show-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="motif-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#motif-enrichment" class="anchor"></a>Motif enrichment</h2>
<p><a href="https://link.springer.com/article/10.1186/s43556-020-00009-w">Different from ChIP-seq which profile specific protein–DNA interactions (depends on the antibody provided), ATAC-seq can proflie global chromatin accessibility</a>. Thus, the peaks of ATAC-seq may originate from hundreds of TFs. To reveal the possible TFs involved, motif enrichment analysis is a good solution.</p>
<p>As pointed above, there are <strong>eight categories</strong> for users to choose to motif enrichment, we will use <code>Up_Up</code> as example.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># debatac.res: similar to debatac.res</span>
<span class="co"># this step is time consuming!</span>
<span class="fu"><a href="../reference/FindMotif.html">FindMotif</a></span><span class="op">(</span>inte.res <span class="op">=</span> <span class="va">debatac.res</span>, peak.anno.res <span class="op">=</span> <span class="cn">NULL</span>,
          peak.motif.key <span class="op">=</span> <span class="st">"Up_Up"</span>, peak.mode <span class="op">=</span> <span class="st">"diff"</span>, homer.motif.path <span class="op">=</span> <span class="st">'~/anaconda3/bin/findMotifsGenome.pl'</span>,
          out.folder <span class="op">=</span> <span class="st">"/path/to/out/folder"</span>, other.paras <span class="op">=</span> <span class="st">"-len 8,10,12 -size -100,50 -S 25"</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-conda-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /home/softwares/anaconda3/envs/r4.0/lib/libopenblasp-r0.3.12.so
## 
## locale:
##  [1] LC_CTYPE=zh_CN.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=zh_CN.UTF-8        LC_COLLATE=zh_CN.UTF-8    
##  [5] LC_MONETARY=zh_CN.UTF-8    LC_MESSAGES=zh_CN.UTF-8   
##  [7] LC_PAPER=zh_CN.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
## [1] org.Mm.eg.db_3.12.0  AnnotationDbi_1.52.0 IRanges_2.24.1      
## [4] S4Vectors_0.28.1     Biobase_2.50.0       BiocGenerics_0.42.0 
## [7] DEbPeak_0.9.0       
## 
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                             
##   [2] ggvenn_0.1.9                           
##   [3] apeglm_1.12.0                          
##   [4] Rsamtools_2.6.0                        
##   [5] rsvg_2.1                               
##   [6] foreach_1.5.1                          
##   [7] rprojroot_2.0.2                        
##   [8] crayon_1.4.1                           
##   [9] V8_3.4.2                               
##  [10] MASS_7.3-53.1                          
##  [11] nlme_3.1-152                           
##  [12] backports_1.2.1                        
##  [13] sva_3.38.0                             
##  [14] GOSemSim_2.25.0                        
##  [15] rlang_1.0.3                            
##  [16] XVector_0.30.0                         
##  [17] readxl_1.3.1                           
##  [18] irlba_2.3.5                            
##  [19] limma_3.46.0                           
##  [20] GOstats_2.56.0                         
##  [21] BiocParallel_1.24.1                    
##  [22] rjson_0.2.20                           
##  [23] bit64_4.0.5                            
##  [24] glue_1.6.2                             
##  [25] DiffBind_3.0.15                        
##  [26] mixsqp_0.3-43                          
##  [27] pheatmap_1.0.12                        
##  [28] parallel_4.0.3                         
##  [29] DEFormats_1.18.0                       
##  [30] base64url_1.4                          
##  [31] tcltk_4.0.3                            
##  [32] DOSE_3.23.2                            
##  [33] haven_2.3.1                            
##  [34] tidyselect_1.1.0                       
##  [35] SummarizedExperiment_1.20.0            
##  [36] rio_0.5.27                             
##  [37] XML_3.99-0.6                           
##  [38] tidyr_1.1.3                            
##  [39] ggpubr_0.4.0                           
##  [40] GenomicAlignments_1.26.0               
##  [41] xtable_1.8-4                           
##  [42] magrittr_2.0.1                         
##  [43] evaluate_0.14                          
##  [44] ggplot2_3.3.5                          
##  [45] cli_3.3.0                              
##  [46] zlibbioc_1.36.0                        
##  [47] hwriter_1.3.2                          
##  [48] rstudioapi_0.13                        
##  [49] bslib_0.3.1                            
##  [50] GreyListChIP_1.22.0                    
##  [51] fastmatch_1.1-3                        
##  [52] BiocSingular_1.6.0                     
##  [53] xfun_0.30                              
##  [54] askpass_1.1                            
##  [55] clue_0.3-59                            
##  [56] gson_0.0.9                             
##  [57] cluster_2.1.1                          
##  [58] caTools_1.18.2                         
##  [59] tidygraph_1.2.0                        
##  [60] tibble_3.1.0                           
##  [61] ggrepel_0.9.1                          
##  [62] Biostrings_2.58.0                      
##  [63] png_0.1-7                              
##  [64] withr_2.4.1                            
##  [65] bitops_1.0-6                           
##  [66] ggforce_0.3.3                          
##  [67] RBGL_1.66.0                            
##  [68] plyr_1.8.6                             
##  [69] cellranger_1.1.0                       
##  [70] GSEABase_1.52.1                        
##  [71] pcaPP_2.0-1                            
##  [72] dqrng_0.2.1                            
##  [73] coda_0.19-4                            
##  [74] pillar_1.5.1                           
##  [75] gplots_3.1.1                           
##  [76] GlobalOptions_0.1.2                    
##  [77] cachem_1.0.4                           
##  [78] GenomicFeatures_1.42.2                 
##  [79] fs_1.5.0                               
##  [80] GetoptLong_1.0.5                       
##  [81] clusterProfiler_4.7.1                  
##  [82] DelayedMatrixStats_1.12.3              
##  [83] vctrs_0.4.1                            
##  [84] ellipsis_0.3.2                         
##  [85] generics_0.1.0                         
##  [86] plot3D_1.4                             
##  [87] tools_4.0.3                            
##  [88] foreign_0.8-81                         
##  [89] NOISeq_2.34.0                          
##  [90] munsell_0.5.0                          
##  [91] tweenr_1.0.2                           
##  [92] fgsea_1.16.0                           
##  [93] DelayedArray_0.16.3                    
##  [94] fastmap_1.1.0                          
##  [95] compiler_4.0.3                         
##  [96] abind_1.4-5                            
##  [97] rtracklayer_1.50.0                     
##  [98] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
##  [99] GenomeInfoDbData_1.2.4                 
## [100] gridExtra_2.3                          
## [101] edgeR_3.32.1                           
## [102] lattice_0.20-45                        
## [103] ggnewscale_0.4.7                       
## [104] AnnotationForge_1.32.0                 
## [105] utf8_1.2.1                             
## [106] dplyr_1.0.5                            
## [107] BiocFileCache_1.14.0                   
## [108] jsonlite_1.7.2                         
## [109] scales_1.1.1                           
## [110] graph_1.68.0                           
## [111] carData_3.0-4                          
## [112] sparseMatrixStats_1.2.1                
## [113] genefilter_1.72.1                      
## [114] car_3.0-11                             
## [115] doParallel_1.0.16                      
## [116] latticeExtra_0.6-29                    
## [117] brew_1.0-6                             
## [118] checkmate_2.0.0                        
## [119] rmarkdown_2.14                         
## [120] openxlsx_4.2.3                         
## [121] pkgdown_1.6.1                          
## [122] cowplot_1.1.1                          
## [123] textshaping_0.1.2                      
## [124] forcats_0.5.1                          
## [125] downloader_0.4                         
## [126] BSgenome_1.58.0                        
## [127] igraph_1.3.4                           
## [128] survival_3.2-10                        
## [129] numDeriv_2016.8-1.1                    
## [130] yaml_2.2.1                             
## [131] plotrix_3.8-2                          
## [132] systemfonts_1.0.1                      
## [133] ashr_2.2-47                            
## [134] SQUAREM_2021.1                         
## [135] htmltools_0.5.2                        
## [136] memoise_2.0.0                          
## [137] VariantAnnotation_1.36.0               
## [138] locfit_1.5-9.4                         
## [139] graphlayouts_0.7.1                     
## [140] batchtools_0.9.15                      
## [141] PCAtools_2.2.0                         
## [142] viridisLite_0.4.0                      
## [143] digest_0.6.27                          
## [144] rrcov_1.7-0                            
## [145] assertthat_0.2.1                       
## [146] rappdirs_0.3.3                         
## [147] emdbook_1.3.12                         
## [148] RSQLite_2.2.5                          
## [149] amap_0.8-18                            
## [150] yulab.utils_0.0.4                      
## [151] misc3d_0.9-1                           
## [152] data.table_1.14.2                      
## [153] blob_1.2.1                             
## [154] ragg_0.4.0                             
## [155] labeling_0.4.2                         
## [156] splines_4.0.3                          
## [157] Cairo_1.5-12.2                         
## [158] ggupset_0.3.0                          
## [159] RCurl_1.98-1.3                         
## [160] broom_0.7.5                            
## [161] hms_1.0.0                              
## [162] colorspace_2.0-0                       
## [163] BiocManager_1.30.16                    
## [164] GenomicRanges_1.42.0                   
## [165] shape_1.4.6                            
## [166] sass_0.4.1                             
## [167] Rcpp_1.0.9                             
## [168] mvtnorm_1.1-2                          
## [169] circlize_0.4.15                        
## [170] enrichplot_1.10.2                      
## [171] fansi_0.4.2                            
## [172] truncnorm_1.0-8                        
## [173] ChIPseeker_1.33.0.900                  
## [174] R6_2.5.0                               
## [175] grid_4.0.3                             
## [176] lifecycle_1.0.0                        
## [177] ShortRead_1.48.0                       
## [178] zip_2.1.1                              
## [179] curl_4.3                               
## [180] ggsignif_0.6.3                         
## [181] jquerylib_0.1.3                        
## [182] robustbase_0.95-0                      
## [183] DO.db_2.9                              
## [184] Matrix_1.3-3                           
## [185] qvalue_2.22.0                          
## [186] desc_1.3.0                             
## [187] RColorBrewer_1.1-2                     
## [188] iterators_1.0.13                       
## [189] stringr_1.4.0                          
## [190] DOT_0.1                                
## [191] ggpie_0.2.1                            
## [192] beachmat_2.6.4                         
## [193] polyclip_1.10-0                        
## [194] biomaRt_2.46.3                         
## [195] purrr_0.3.4                            
## [196] shadowtext_0.0.9                       
## [197] gridGraphics_0.5-1                     
## [198] mgcv_1.8-34                            
## [199] ComplexHeatmap_2.13.1                  
## [200] openssl_1.4.3                          
## [201] patchwork_1.0.0                        
## [202] bdsmatrix_1.3-4                        
## [203] codetools_0.2-18                       
## [204] matrixStats_0.58.0                     
## [205] invgamma_1.1                           
## [206] GO.db_3.12.1                           
## [207] gtools_3.8.2                           
## [208] prettyunits_1.1.1                      
## [209] dbplyr_2.1.1                           
## [210] GenomeInfoDb_1.26.7                    
## [211] gtable_0.3.0                           
## [212] DBI_1.1.1                              
## [213] highr_0.8                              
## [214] ggfun_0.0.6                            
## [215] httr_1.4.2                             
## [216] KernSmooth_2.23-18                     
## [217] stringi_1.5.3                          
## [218] progress_1.2.2                         
## [219] reshape2_1.4.4                         
## [220] farver_2.1.0                           
## [221] annotate_1.68.0                        
## [222] viridis_0.6.1                          
## [223] Rgraphviz_2.34.0                       
## [224] xml2_1.3.2                             
## [225] bbmle_1.0.24                           
## [226] systemPipeR_1.24.3                     
## [227] boot_1.3-28                            
## [228] geneplotter_1.68.0                     
## [229] ggplotify_0.1.0                        
## [230] Category_2.56.0                        
## [231] DESeq2_1.30.1                          
## [232] DEoptimR_1.0-11                        
## [233] bit_4.0.4                              
## [234] scatterpie_0.1.7                       
## [235] jpeg_0.1-8.1                           
## [236] MatrixGenerics_1.2.1                   
## [237] ggraph_2.0.5                           
## [238] pkgconfig_2.0.3                        
## [239] rstatix_0.7.0                          
## [240] knitr_1.37</code></pre>
<hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
