<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>IntegrateChIPDE • DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="IntegrateChIPDE">
<meta property="og:description" content="DEbPeak">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ParseGEO.html">Parse GEO</a>
    </li>
    <li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPDE.html">Integrate RNA-seq with ChIP-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="IntegrateChIPDE_files/header-attrs-2.14/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>IntegrateChIPDE</h1>
            
      
      
      <div class="hidden name"><code>IntegrateChIPDE.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The gene expression can also be regulated by <em>cis</em>-regulatory elements, such as enhancers. Different from promoters which are typically located directly upstream or downstream of the transcription initiation sites (TSSs), <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02322-1">enhancers can act independent of orientation, distance, and location with respect to the target gene and can be located over as much as a million base pairs away</a>. This characteristic determines that annotating peak to its nearest gene for analysis is inaccurate. Here, we adopted the following strategies:</p>
<ul>
<li>Step1: Extract differential binding peaks and differential expressed genes (both are based on hard-threshold)</li>
<li>Step2: Annotate differential binding peaks to genes within a given distance threshold</li>
<li>Step3: Merge the annotated genes with differential expressed genes</li>
</ul>
<hr>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>The data used here are from <a href="https://doi.org/10.1016/j.celrep.2016.04.063">Persistent Alterations in Microglial Enhancers in a Model of Chronic Pain</a>:</p>
<ul>
<li>
<strong>RNA-seq data</strong>: RNA sequencing of the spinal microglia after SNL vs sham surgery, the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE71133">GSE71133</a>
</li>
<li>
<strong>ChIP-seq data</strong>: Genome-wide H3K4me1 enrichment profiles (ChIP-seq) of spinal cord microglia after SNL vs sham surgery, the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE71134">GSE71134</a>
</li>
</ul>
<p><a href="https://faseb.onlinelibrary.wiley.com/doi/epdf/10.1096/fj.202100488R">Histone H3K4me1 and H3K27ac are enhancer-specific modifications and are required for enhancers to activate transcription of target genes</a>.</p>
<hr>
</div>
<div id="load-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DEbPeak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="prepare-chip-seq-counts" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-chip-seq-counts" class="anchor"></a>Prepare ChIP-seq counts</h2>
<p>To perform differential binding analysis, we need to prepare count matrix of ChIP-seq samples. <code>DEbPeak</code> provides <code>PeakMatrix</code> to do this, it contains two steps:</p>
<ul>
<li>get read counts in each bin with <code>DiffBind</code>
</li>
<li>annotate the peak to its nearest gene with <code>ChIPseeker</code> (in the downstream analysis, we will not use this information)</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">chip.peak.meta.file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"enhancer_ChIP_peak_meta.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="co"># this will save count matrix, annotated results and peak sample metadata to consensus_peak_matrix.txt, consensus_peak_anno.txt and peak_metadata.txt</span>
<span class="va">peak.matrix.list</span> <span class="op">=</span> <span class="fu"><a href="../reference/PeakMatrix.html">PeakMatrix</a></span><span class="op">(</span>meta.file <span class="op">=</span> <span class="va">chip.peak.meta.file</span>, species <span class="op">=</span> <span class="st">"Mouse"</span>,  seq.style <span class="op">=</span> <span class="st">"Ensembl"</span>,
                              summits <span class="op">=</span> <span class="fl">0</span>, up.dist <span class="op">=</span> <span class="fl">250000</span>, down.dist <span class="op">=</span> <span class="fl">250000</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-analysis" class="anchor"></a>Differential Expression Analysis</h2>
<p>To simplify the steps of obtaining differential expression analysis results, <code>DEbPeak</code> provides <code>ConductDESeq2</code> to perfrom all steps of differential expression analysis, including <strong>quality control</strong>, <strong>principal component analysis</strong>, <strong>differential expression analysis</strong> and <strong>functional enrichment analysis</strong>.</p>
<div id="rna-seq" class="section level3">
<h3 class="hasAnchor">
<a href="#rna-seq" class="anchor"></a>RNA-seq</h3>
<p>In this step, we will perform differetial expression analysis for RNA-seq:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">count.matrix.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"enhancer_RNA_count.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">meta.info.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"enhancer_RNA_meta.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">gmt.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"m5.go.bp.v2022.1.Mm.entrez.gmt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="fu"><a href="../reference/ConductDESeq2.html">ConductDESeq2</a></span><span class="op">(</span>count.matrix.file <span class="op">=</span> <span class="va">count.matrix.file</span>, meta.file <span class="op">=</span> <span class="va">meta.info.file</span>, data.type <span class="op">=</span> <span class="st">"RNA"</span>,
              gene.type <span class="op">=</span> <span class="st">"ENSEMBL"</span>, outlier.detection <span class="op">=</span> <span class="cn">F</span>, min.count <span class="op">=</span> <span class="fl">10</span>,
              group.key <span class="op">=</span> <span class="st">"condition"</span>, ref.group <span class="op">=</span> <span class="st">"SHAM"</span>,
              out.folder <span class="op">=</span> <span class="st">'/home/songyabing/R/learn/tmp/DEbPeak/enhancer/RNA-seq'</span>,
              signif <span class="op">=</span> <span class="st">"padj"</span>, signif.threshold <span class="op">=</span> <span class="fl">0.05</span>, l2fc.threshold <span class="op">=</span> <span class="fl">0</span>, gmt.file <span class="op">=</span> <span class="va">gmt.file</span><span class="op">)</span></code></pre></div>
<p>The structure of result directory:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">tree</span> /home/songyabing/R/learn/tmp/DEbPeak/enhancer/RNA-seq</span></code></pre></div>
<pre><code>## /home/songyabing/R/learn/tmp/DEbPeak/enhancer/RNA-seq
## ├── CountQC_CPM.pdf
## ├── CountQC_saturation.pdf
## ├── DA
## │   ├── Condition_SNL_SHAM_all.csv
## │   ├── Condition_SNL_SHAM_padj0.05_FC0.csv
## │   ├── DA_GenePlot.pdf
## │   ├── DA_Heatmap.pdf
## │   ├── DA_MAPlot.pdf
## │   ├── DA_RankPlot.pdf
## │   ├── DA_ScatterPlot.pdf
## │   ├── DA_VolcanoPlot.pdf
## │   └── normalized_counts.csv
## ├── FE
## │   ├── DOWN_ALL_GO.csv
## │   ├── DOWN_Biological_Process.png
## │   ├── DOWN_Cellular_Component.png
## │   ├── DOWN_Functional_Enrichment.pdf
## │   ├── DOWN_KEGG.csv
## │   ├── DOWN_KEGG_Enrichment.png
## │   ├── DOWN_Molecular_Function.png
## │   ├── UP_ALL_GO.csv
## │   ├── UP_Biological_Process.png
## │   ├── UP_Cellular_Component.png
## │   ├── UP_Functional_Enrichment.pdf
## │   ├── UP_KEGG.csv
## │   ├── UP_KEGG_Enrichment.png
## │   └── UP_Molecular_Function.png
## ├── GSEA
## │   └── GSEA_enrich_result.csv
## ├── PCA
## │   ├── PC1
## │   │   ├── Negative
## │   │   │   ├── PC1_Negative_ALL_GO.csv
## │   │   │   └── PC1_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC1_Positive_ALL_GO.csv
## │   │       └── PC1_Positive_KEGG.csv
## │   ├── PC1_Negative_Biological_Process.png
## │   ├── PC1_Negative_Cellular_Component.png
## │   ├── PC1_Negative_Functional_Enrichment.pdf
## │   ├── PC1_Negative_KEGG_Enrichment.png
## │   ├── PC1_Negative_Molecular_Function.png
## │   ├── PC1_Positive_Biological_Process.png
## │   ├── PC1_Positive_Cellular_Component.png
## │   ├── PC1_Positive_Functional_Enrichment.pdf
## │   ├── PC1_Positive_KEGG_Enrichment.png
## │   ├── PC1_Positive_Molecular_Function.png
## │   ├── PC2
## │   │   ├── Negative
## │   │   │   ├── PC2_Negative_ALL_GO.csv
## │   │   │   └── PC2_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC2_Positive_ALL_GO.csv
## │   │       └── PC2_Positive_KEGG.csv
## │   ├── PC2_Negative_Biological_Process.png
## │   ├── PC2_Negative_Cellular_Component.png
## │   ├── PC2_Negative_Functional_Enrichment.pdf
## │   ├── PC2_Negative_KEGG_Enrichment.png
## │   ├── PC2_Negative_Molecular_Function.png
## │   ├── PC2_Positive_Biological_Process.png
## │   ├── PC2_Positive_Cellular_Component.png
## │   ├── PC2_Positive_Functional_Enrichment.pdf
## │   ├── PC2_Positive_KEGG_Enrichment.png
## │   ├── PC2_Positive_Molecular_Function.png
## │   ├── PC3
## │   │   ├── Negative
## │   │   │   ├── PC3_Negative_ALL_GO.csv
## │   │   │   └── PC3_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC3_Positive_ALL_GO.csv
## │   │       └── PC3_Positive_KEGG.csv
## │   ├── PC3_Negative_Biological_Process.png
## │   ├── PC3_Negative_Cellular_Component.png
## │   ├── PC3_Negative_Functional_Enrichment.pdf
## │   ├── PC3_Negative_KEGG_Enrichment.png
## │   ├── PC3_Negative_Molecular_Function.png
## │   ├── PC3_Positive_Biological_Process.png
## │   ├── PC3_Positive_Cellular_Component.png
## │   ├── PC3_Positive_Functional_Enrichment.pdf
## │   ├── PC3_Positive_KEGG_Enrichment.png
## │   ├── PC3_Positive_Molecular_Function.png
## │   ├── PC4
## │   │   ├── Negative
## │   │   │   ├── PC4_Negative_ALL_GO.csv
## │   │   │   └── PC4_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC4_Positive_ALL_GO.csv
## │   │       └── PC4_Positive_KEGG.csv
## │   ├── PC4_Negative_Biological_Process.png
## │   ├── PC4_Negative_Cellular_Component.png
## │   ├── PC4_Negative_Functional_Enrichment.pdf
## │   ├── PC4_Negative_KEGG_Enrichment.png
## │   ├── PC4_Negative_Molecular_Function.png
## │   ├── PC4_Positive_Biological_Process.png
## │   ├── PC4_Positive_Cellular_Component.png
## │   ├── PC4_Positive_Functional_Enrichment.pdf
## │   ├── PC4_Positive_KEGG_Enrichment.png
## │   ├── PC4_Positive_Molecular_Function.png
## │   ├── PC5
## │   │   ├── Negative
## │   │   │   ├── PC5_Negative_ALL_GO.csv
## │   │   │   └── PC5_Negative_KEGG.csv
## │   │   └── Positive
## │   │       ├── PC5_Positive_ALL_GO.csv
## │   │       └── PC5_Positive_KEGG.csv
## │   ├── PC5_Negative_Biological_Process.png
## │   ├── PC5_Negative_Cellular_Component.png
## │   ├── PC5_Negative_Functional_Enrichment.pdf
## │   ├── PC5_Negative_KEGG_Enrichment.png
## │   ├── PC5_Negative_Molecular_Function.png
## │   ├── PC5_Positive_Biological_Process.png
## │   ├── PC5_Positive_Cellular_Component.png
## │   ├── PC5_Positive_Functional_Enrichment.pdf
## │   ├── PC5_Positive_KEGG_Enrichment.png
## │   ├── PC5_Positive_Molecular_Function.png
## │   ├── PCA_3DPCA.pdf
## │   ├── PCA_biplot.pdf
## │   ├── PCA_loading_bar.pdf
## │   ├── PCA_loading_heat.pdf
## │   ├── PCA_overview.pdf
## │   ├── PCA_pairs_plot.pdf
## │   └── PCA_scree_plot.pdf
## └── SampleQC_dist_pcc.pdf
## 
## 19 directories, 104 files</code></pre>
<p>Load the data:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">enhancer.rna.res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/enhancer/RNA-seq/DA/Condition_SNL_SHAM_all.csv"</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">enhancer.rna.res</span><span class="op">)</span></code></pre></div>
<pre><code>##                     baseMean log2FoldChange    lfcSE     stat       pvalue
## ENSMUSG00000018924 39.788217       8.727338 1.377682 6.334800 2.376495e-10
## ENSMUSG00000020672  7.050370       6.224094 1.458727 4.266798 1.982987e-05
## ENSMUSG00000030247  6.768719       6.177329 2.031739 3.040415 2.362524e-03
## ENSMUSG00000009350 10.166003       6.021441 1.354274 4.446250 8.738245e-06
## ENSMUSG00000068744  7.864203       5.652519 1.710464 3.304671 9.508810e-04
## ENSMUSG00000022431  4.571048       5.611394 1.674608 3.350870 8.055816e-04
##                            padj ENTREZID SYMBOL
## ENSMUSG00000018924 1.354186e-06    11687 Alox15
## ENSMUSG00000020672 2.152296e-02   268534  Sntg2
## ENSMUSG00000030247 6.205017e-01    16523  Kcnj8
## ENSMUSG00000009350 1.171593e-02    17523    Mpo
## ENSMUSG00000068744 4.017459e-01    56742  Psrc1
## ENSMUSG00000022431 3.906728e-01    67747  Ribc2</code></pre>
<hr>
</div>
<div id="chip-seq" class="section level3">
<h3 class="hasAnchor">
<a href="#chip-seq" class="anchor"></a>ChIP-seq</h3>
<p>With the count matrix and sample metadata generated above, we will perform differential binding analysis:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">peak.matrix.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"consensus_peak_matrix.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">peak.meta.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"peak_metadata.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="fu"><a href="../reference/ConductDESeq2.html">ConductDESeq2</a></span><span class="op">(</span>count.matrix.file <span class="op">=</span> <span class="va">peak.matrix.file</span>, meta.file <span class="op">=</span> <span class="va">peak.meta.file</span>, data.type <span class="op">=</span> <span class="st">"ATAC"</span>,
              gene.type <span class="op">=</span> <span class="st">"SYMBOL"</span>, outlier.detection <span class="op">=</span> <span class="cn">F</span>, min.count <span class="op">=</span> <span class="fl">0</span>, loding.pc <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,
              peak.anno.key <span class="op">=</span> <span class="st">"All"</span>, group.key <span class="op">=</span> <span class="st">"Condition"</span>, ref.group <span class="op">=</span> <span class="st">"SHAM"</span>,
              out.folder <span class="op">=</span> <span class="st">'/home/songyabing/R/learn/tmp/DEbPeak/enhancer/ChIP-seq'</span>,
              signif <span class="op">=</span> <span class="st">"padj"</span>, signif.threshold <span class="op">=</span> <span class="fl">0.05</span>, l2fc.threshold <span class="op">=</span> <span class="fl">0</span>, gmt.file <span class="op">=</span> <span class="st">''</span><span class="op">)</span></code></pre></div>
<p>The output structure is same as above.</p>
<p>Load the data:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all.enhancer.res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/enhancer/ChIP-seq/DA/Condition_SNL_SHAM_all.csv"</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">all.enhancer.res</span><span class="op">)</span></code></pre></div>
<pre><code>##                                        baseMean log2FoldChange    lfcSE
## 14:8053761-8054509|Rpp14|P             23.47939       7.875103 1.655338
## 7:120873156-120873435|Eef2k|P          17.54738       7.451267 1.875877
## 16:62855534-62855880|Pros1|P           14.54948       7.183562 2.108218
## 16:76762872-76763240|Gm30790|P         12.75539       6.993929 2.371687
## 8:115645132-115645520|Maf|P            10.34494       6.692268 3.023559
## 13:113371428-113371848|4921509O07Rik|P 19.04070       6.095864 1.744785
##                                            stat       pvalue       padj
## 14:8053761-8054509|Rpp14|P             4.757400 1.961024e-06 0.01239410
## 7:120873156-120873435|Eef2k|P          3.972151 7.122640e-05 0.08268367
## 16:62855534-62855880|Pros1|P           3.407409 6.558284e-04 0.22472791
## 16:76762872-76763240|Gm30790|P         2.948926 3.188808e-03 0.39764476
## 8:115645132-115645520|Maf|P            2.213374 2.687187e-02 0.68991582
## 13:113371428-113371848|4921509O07Rik|P 3.493762 4.762648e-04 0.20072596
##                                               SYMBOL            ENSEMBL
## 14:8053761-8054509|Rpp14|P                     Rpp14 ENSMUSG00000023156
## 7:120873156-120873435|Eef2k|P                  Eef2k ENSMUSG00000035064
## 16:62855534-62855880|Pros1|P                   Pros1 ENSMUSG00000022912
## 16:76762872-76763240|Gm30790|P               Gm30790 ENSMUSG00000109333
## 8:115645132-115645520|Maf|P                      Maf ENSMUSG00000055435
## 13:113371428-113371848|4921509O07Rik|P 4921509O07Rik ENSMUSG00000114790
##                                         ENTREZID
## 14:8053761-8054509|Rpp14|P                 67053
## 7:120873156-120873435|Eef2k|P              13631
## 16:62855534-62855880|Pros1|P               19128
## 16:76762872-76763240|Gm30790|P         102632814
## 8:115645132-115645520|Maf|P                17132
## 13:113371428-113371848|4921509O07Rik|P     70917</code></pre>
<hr>
</div>
</div>
<div id="integrate-chip-seq-and-rna-seq" class="section level2">
<h2 class="hasAnchor">
<a href="#integrate-chip-seq-and-rna-seq" class="anchor"></a>Integrate ChIP-seq and RNA-seq</h2>
<p>Here, we will integrate RNA-seq with <strong>H3K4me1 ChIP-seq</strong> data to explore the enhancers and their candidate targets.</p>
<div id="integrate" class="section level3">
<h3 class="hasAnchor">
<a href="#integrate" class="anchor"></a>Integrate</h3>
<p><a href="https://doi.org/10.1016/j.celrep.2016.04.063">Previous work on latent enhancers indicates that 80% of binding sites emerge at this distance, and many are within 250 kb or less</a>. Here, we will use 250kb as a threshold, user can change this value with <code>dis.threshold</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set enhancer = TRUE to use enhancer mode</span>
<span class="va">debchip.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DEbPeak.html">DEbPeak</a></span><span class="op">(</span>
    de.res <span class="op">=</span> <span class="va">enhancer.rna.res</span>, peak.res <span class="op">=</span> <span class="va">all.enhancer.res</span>, peak.mode <span class="op">=</span> <span class="st">"diff"</span>,  enhancer <span class="op">=</span> <span class="cn">TRUE</span>,
    signif <span class="op">=</span> <span class="st">"pvalue"</span>, signif.threshold <span class="op">=</span> <span class="fl">0.05</span>, l2fc.threshold <span class="op">=</span> <span class="fl">0</span>, peak.signif <span class="op">=</span> <span class="st">"pvalue"</span>, label.key <span class="op">=</span> <span class="cn">NULL</span>,
    peak.signif.threshold <span class="op">=</span> <span class="fl">0.05</span>, peak.l2fc.threshold <span class="op">=</span> <span class="fl">0</span>, species <span class="op">=</span> <span class="st">"Mouse"</span>, merge.key <span class="op">=</span> <span class="st">"SYMBOL"</span>, n.cores <span class="op">=</span> <span class="fl">5</span>,
    seq.style <span class="op">=</span> <span class="st">"Ensembl"</span>, gtf.file <span class="op">=</span> <span class="cn">NULL</span>, dis.threshold <span class="op">=</span> <span class="fl">250000</span>
  <span class="op">)</span></code></pre></div>
<pre><code>## 
## 载入程辑包：'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<pre><code>## 
## 载入程辑包：'S4Vectors'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## </code></pre>
<pre><code>## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!</code></pre>
<pre><code>##   42 genes were dropped because they have exons located on both strands
##   of the same reference sequence or on more than one reference sequence,
##   so cannot be represented by a single genomic range.
##   Use 'single.strand.genes.only=FALSE' to get all the genes in a
##   GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in clusterProfiler::bitr(de.df[, "Gene2Convert"], fromType =
## gene.type, : 1.74% of input gene IDs are fail to map...</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in clusterProfiler::bitr(de.df[, "Gene2Convert"], fromType =
## gene.type, : 0.01% of input gene IDs are fail to map...</code></pre>
<pre><code>## Warning in DEbPeak(de.res = enhancer.rna.res, peak.res = all.enhancer.res, :
## 强制改变过程中产生了NA</code></pre>
<hr>
</div>
<div id="integrate-summary" class="section level3">
<h3 class="hasAnchor">
<a href="#integrate-summary" class="anchor"></a>Integrate summary</h3>
<div id="venn-plot" class="section level4">
<h4 class="hasAnchor">
<a href="#venn-plot" class="anchor"></a>Venn plot</h4>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debchip.plot</span> <span class="op">=</span> <span class="fu"><a href="../reference/InteVenn.html">InteVenn</a></span><span class="op">(</span>inte.res <span class="op">=</span> <span class="va">debchip.res</span>, inte.type <span class="op">=</span> <span class="st">"DEbPeak"</span>, peak.mode <span class="op">=</span> <span class="st">"diff"</span>,
                        peak.type <span class="op">=</span> <span class="st">"ChIP"</span>, show_percentage <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">debchip.plot</span></code></pre></div>
<p><img src="IntegrateChIPDE_files/figure-html/debchip_de_integrate_plot-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="quadrant-diagram" class="section level4">
<h4 class="hasAnchor">
<a href="#quadrant-diagram" class="anchor"></a>Quadrant diagram</h4>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">label.df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Ripor2"</span>, <span class="st">"C4b"</span>, <span class="st">"Ccl12"</span>, <span class="st">"Fcgr2b"</span>, <span class="st">"Ly86"</span>, <span class="st">"Ccl5"</span>, <span class="st">"Ccr5"</span><span class="op">)</span><span class="op">)</span>
<span class="va">quad.plot</span> <span class="op">=</span> <span class="fu"><a href="../reference/InteDiffQuad.html">InteDiffQuad</a></span><span class="op">(</span>inte.res <span class="op">=</span> <span class="va">debchip.res</span>, inte.type <span class="op">=</span> <span class="st">"DEbPeak"</span>,
                         label.df <span class="op">=</span> <span class="va">label.df</span>, label.color <span class="op">=</span> <span class="st">"green"</span><span class="op">)</span>
<span class="va">quad.plot</span></code></pre></div>
<p><img src="IntegrateChIPDE_files/figure-html/debchip_de_integrate_quad_plot-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
<div id="functional-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#functional-enrichment" class="anchor"></a>Functional enrichment</h2>
<p>There are <strong>eight categories</strong> for users to choose to perform functional enrichment: <code>Down_Up</code>, <code>Up_Up</code>, <code>Down_Down</code>, <code>Up_Down</code>, <code>RNAUp</code>, <code>RNADown</code>, <code>PeakUp</code>, <code>PeakDown</code>. Here, we will use <code>Up_Up</code> and <code>Down_Down</code> as examples.</p>
<div id="both-up-regulated" class="section level3">
<h3 class="hasAnchor">
<a href="#both-up-regulated" class="anchor"></a>Both up-regulated</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debchip.up.fe.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/InteFE.html">InteFE</a></span><span class="op">(</span>inte.res <span class="op">=</span> <span class="va">debchip.res</span>, fe.key <span class="op">=</span> <span class="st">"Up_Up"</span>, inte.type <span class="op">=</span> <span class="st">"DEbPeak"</span>,
                               gene.type <span class="op">=</span> <span class="st">"ENTREZID"</span>, species<span class="op">=</span><span class="st">"Mouse"</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## conduct ALL GO enrichment analysis on: Up_Up</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.</code></pre>
<p>The results:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">debchip.up.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID
## GO:0051607       BP GO:0051607
## GO:0009615       BP GO:0009615
## GO:0044403       BP GO:0044403
## GO:0002274       BP GO:0002274
## GO:0019882       BP GO:0019882
## GO:0032103       BP GO:0032103
##                                                     Description GeneRatio
## GO:0051607                            defense response to virus    14/184
## GO:0009615                                    response to virus    15/184
## GO:0044403                                    symbiotic process    15/184
## GO:0002274                         myeloid leukocyte activation    12/184
## GO:0019882                  antigen processing and presentation     9/184
## GO:0032103 positive regulation of response to external stimulus    15/184
##              BgRatio       pvalue     p.adjust       qvalue
## GO:0051607 239/23328 7.233705e-09 1.367222e-05 1.128361e-05
## GO:0009615 286/23328 9.213089e-09 1.367222e-05 1.128361e-05
## GO:0044403 374/23328 3.085012e-07 2.128616e-04 1.756735e-04
## GO:0002274 231/23328 3.255523e-07 2.128616e-04 1.756735e-04
## GO:0019882 116/23328 3.585944e-07 2.128616e-04 1.756735e-04
## GO:0032103 418/23328 1.248090e-06 5.483178e-04 4.525236e-04
##                                                                                                      geneID
## GO:0051607        Ifit1bl1/Ifit3b/Abcf3/Ifit3/Il12b/Ccl5/Ifit2/Fgl2/Ifitm3/Irf7/Polr3c/Ifitm2/Cxcl9/Apobec1
## GO:0009615   Ifit1bl1/Lcn2/Ifit3b/Abcf3/Ifit3/Il12b/Ccl5/Ifit2/Fgl2/Ifitm3/Irf7/Polr3c/Ifitm2/Cxcl9/Apobec1
## GO:0044403 Ddb1/Ccl5/Tmem250-ps/Trim30b/Camp/Gbp2/Irgm1/Hspa8/Trim30d/Spint1/Tpcn2/Ccr5/Ifitm3/Ifitm2/Eif3l
## GO:0002274                         Mfhas1/Adora3/Ccl5/Rac2/Cd244a/Sbno2/Tlr1/Itgam/Fcgr3/Cst7/Cd300lf/Psen1
## GO:0019882                                                 Ctss/Rab8b/B2m/Fcgr2b/Ctsl/Fgl2/Psap/Fcgr3/Ifi30
## GO:0032103              Ctss/Ripor2/Ly86/C3/Adora3/Il12b/Ccl5/Rac2/Ccr1/Irgm1/Fcgr3/Ccr5/Irf7/Polr3c/Ifi209
##            Count
## GO:0051607    14
## GO:0009615    15
## GO:0044403    15
## GO:0002274    12
## GO:0019882     9
## GO:0032103    15</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debchip.up.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="IntegrateChIPDE_files/figure-html/debchip_de_integrate_fe_up_show-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="both-down-regulated" class="section level3">
<h3 class="hasAnchor">
<a href="#both-down-regulated" class="anchor"></a>Both down-regulated</h3>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debchip.down.fe.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/InteFE.html">InteFE</a></span><span class="op">(</span>inte.res <span class="op">=</span> <span class="va">debchip.res</span>, fe.key <span class="op">=</span> <span class="st">"Down_Down"</span>, inte.type <span class="op">=</span> <span class="st">"DEbPeak"</span>,
                                 gene.type <span class="op">=</span> <span class="st">"ENTREZID"</span>, species<span class="op">=</span><span class="st">"Mouse"</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## conduct ALL GO enrichment analysis on: Down_Down</code></pre>
<p>The results:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">debchip.down.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] ONTOLOGY    ID          Description GeneRatio   BgRatio     pvalue     
## [7] p.adjust    qvalue      Count      
## &lt;0 行&gt; (或0-长度的row.names)</code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debchip.down.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="IntegrateChIPDE_files/figure-html/debchip_de_integrate_fe_down_show-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="motif-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#motif-enrichment" class="anchor"></a>Motif enrichment</h2>
<p><a href="https://link.springer.com/article/10.1186/s43556-020-00009-w">Different from ChIP-seq which profile specific protein–DNA interactions (depends on the antibody provided), ATAC-seq can proflie global chromatin accessibility</a>. Thus, the peaks of ATAC-seq may originate from hundreds of TFs. To reveal the possible TFs involved, motif enrichment analysis is a good solution.</p>
<p>As pointed above, there are <strong>eight categories</strong> for users to choose to motif enrichment, we will use <code>Up_Up</code> as example.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># debchip.res: similar to debchip.res</span>
<span class="co"># this step is time consuming!</span>
<span class="fu"><a href="../reference/FindMotif.html">FindMotif</a></span><span class="op">(</span>inte.res <span class="op">=</span> <span class="va">debchip.res</span>, peak.anno.res <span class="op">=</span> <span class="cn">NULL</span>,
          peak.motif.key <span class="op">=</span> <span class="st">"Up_Up"</span>, peak.mode <span class="op">=</span> <span class="st">"diff"</span>, homer.motif.path <span class="op">=</span> <span class="st">'~/anaconda3/bin/findMotifsGenome.pl'</span>,
          out.folder <span class="op">=</span> <span class="st">"/path/to/out/folder"</span>, other.paras <span class="op">=</span> <span class="st">"-len 8,10,12 -size -100,50 -S 25"</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="enhancer-gene-network" class="section level2">
<h2 class="hasAnchor">
<a href="#enhancer-gene-network" class="anchor"></a>Enhancer-gene network</h2>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6061867/">Multiple enhancers often regulate a single gene, and multiple genes can be regulated by a single enhancer</a>. Here, we will create the regulatory network.</p>
<div id="whole-category" class="section level3">
<h3 class="hasAnchor">
<a href="#whole-category" class="anchor"></a>Whole category</h3>
<p>Here, we will show the whole enhancer-gene network of <strong>Up_Up</strong> category:</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/">igraph</a></span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/NetViz.html">NetViz</a></span><span class="op">(</span>inte.res <span class="op">=</span> <span class="va">debchip.res</span>, whole <span class="op">=</span> <span class="cn">TRUE</span>, type <span class="op">=</span> <span class="st">"Up_Up"</span>,
       labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Ripor2"</span>, <span class="st">"C4b"</span>, <span class="st">"Ccl12"</span>, <span class="st">"Fcgr2b"</span>, <span class="st">"Ly86"</span>, <span class="st">"Ccl5"</span>, <span class="st">"Ccr5"</span><span class="op">)</span>, seed <span class="op">=</span> <span class="fl">1001</span><span class="op">)</span></code></pre></div>
<pre><code>## Visualize the whole network!</code></pre>
<p><img src="IntegrateChIPDE_files/figure-html/debchip_de_integrate_whole_network-1.png" width="1056" style="display: block; margin: auto;"></p>
<p>We can extract subset of the whole network:</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/NetViz.html">NetViz</a></span><span class="op">(</span>inte.res <span class="op">=</span> <span class="va">debchip.res</span>, whole <span class="op">=</span> <span class="cn">FALSE</span>, type <span class="op">=</span> <span class="st">"Up_Up"</span>,
       gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Ripor2"</span>, <span class="st">"C4b"</span>, <span class="st">"Ccl12"</span>, <span class="st">"Fcgr2b"</span>, <span class="st">"Ly86"</span>, <span class="st">"Ccl5"</span>, <span class="st">"Ccr5"</span><span class="op">)</span>,
       peak <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"15:74931074-74931694|Ly6e|P"</span>, <span class="st">"9:124012191-124012495|Ccr3|P"</span>, <span class="st">"7:141122254-141122774|Ptdss2|P"</span>, <span class="st">"1:170985190-170986301|Fcgr2b|P"</span><span class="op">)</span>,
       show.all.labels <span class="op">=</span> <span class="cn">TRUE</span>, seed <span class="op">=</span> <span class="fl">1001</span><span class="op">)</span></code></pre></div>
<pre><code>## Visualize the subset network with genes and peaks!</code></pre>
<p><img src="IntegrateChIPDE_files/figure-html/debchip_de_integrate_subset_network-1.png" width="960" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="gene-based" class="section level3">
<h3 class="hasAnchor">
<a href="#gene-based" class="anchor"></a>Gene-based</h3>
<p>Here, we will show enhancer-specific gene network of <strong>Up_Up</strong> category:</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/NetViz.html">NetViz</a></span><span class="op">(</span>inte.res <span class="op">=</span> <span class="va">debchip.res</span>, whole <span class="op">=</span> <span class="cn">FALSE</span>, type <span class="op">=</span> <span class="st">"Up_Up"</span>,
       gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Ripor2"</span>, <span class="st">"C4b"</span>, <span class="st">"Ccl12"</span>, <span class="st">"Fcgr2b"</span>, <span class="st">"Ly86"</span>, <span class="st">"Ccl5"</span>, <span class="st">"Ccr5"</span><span class="op">)</span>,
       show.all.labels <span class="op">=</span> <span class="cn">TRUE</span>, seed <span class="op">=</span> <span class="fl">1001</span><span class="op">)</span></code></pre></div>
<pre><code>## Visualize the subset network with peak!</code></pre>
<p><img src="IntegrateChIPDE_files/figure-html/debchip_de_integrate_gene_network-1.png" width="960" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="peak-based" class="section level3">
<h3 class="hasAnchor">
<a href="#peak-based" class="anchor"></a>Peak-based</h3>
<p>Here, we will show specific enhancer-gene network of <strong>Up_Up</strong> category:</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/NetViz.html">NetViz</a></span><span class="op">(</span>inte.res <span class="op">=</span> <span class="va">debchip.res</span>, whole <span class="op">=</span> <span class="cn">FALSE</span>, type <span class="op">=</span> <span class="st">"Up_Up"</span>,
       peak <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"15:74931074-74931694|Ly6e|P"</span>, <span class="st">"9:124012191-124012495|Ccr3|P"</span>, <span class="st">"7:141122254-141122774|Ptdss2|P"</span>, <span class="st">"1:170985190-170986301|Fcgr2b|P"</span><span class="op">)</span>,
       show.all.labels <span class="op">=</span> <span class="cn">TRUE</span>, seed <span class="op">=</span> <span class="fl">1001</span><span class="op">)</span></code></pre></div>
<pre><code>## Visualize the subset network with peaks!</code></pre>
<p><img src="IntegrateChIPDE_files/figure-html/debchip_de_integrate_peak_network-1.png" width="960" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-conda-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /home/softwares/anaconda3/envs/r4.0/lib/libopenblasp-r0.3.12.so
## 
## locale:
##  [1] LC_CTYPE=zh_CN.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=zh_CN.UTF-8        LC_COLLATE=zh_CN.UTF-8    
##  [5] LC_MONETARY=zh_CN.UTF-8    LC_MESSAGES=zh_CN.UTF-8   
##  [7] LC_PAPER=zh_CN.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] igraph_1.4.99.9024                       
##  [2] TxDb.Mmusculus.UCSC.mm10.knownGene_3.10.0
##  [3] GenomicFeatures_1.42.2                   
##  [4] GenomicRanges_1.42.0                     
##  [5] GenomeInfoDb_1.26.7                      
##  [6] org.Mm.eg.db_3.12.0                      
##  [7] AnnotationDbi_1.52.0                     
##  [8] IRanges_2.24.1                           
##  [9] S4Vectors_0.28.1                         
## [10] Biobase_2.50.0                           
## [11] BiocGenerics_0.42.0                      
## [12] DEbPeak_1.2.0                            
## 
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                             
##   [2] ggvenn_0.1.9                           
##   [3] apeglm_1.12.0                          
##   [4] Rsamtools_2.6.0                        
##   [5] rsvg_2.1                               
##   [6] foreach_1.5.1                          
##   [7] rprojroot_2.0.2                        
##   [8] crayon_1.4.1                           
##   [9] V8_3.4.2                               
##  [10] MASS_7.3-58                            
##  [11] nlme_3.1-152                           
##  [12] backports_1.2.1                        
##  [13] sva_3.38.0                             
##  [14] GOSemSim_2.25.0                        
##  [15] rlang_1.1.0                            
##  [16] XVector_0.30.0                         
##  [17] readxl_1.4.2                           
##  [18] irlba_2.3.5                            
##  [19] limma_3.46.0                           
##  [20] GOstats_2.56.0                         
##  [21] BiocParallel_1.24.1                    
##  [22] rjson_0.2.20                           
##  [23] bit64_4.0.5                            
##  [24] glue_1.6.2                             
##  [25] DiffBind_3.0.15                        
##  [26] mixsqp_0.3-43                          
##  [27] pheatmap_1.0.12                        
##  [28] parallel_4.0.3                         
##  [29] DEFormats_1.18.0                       
##  [30] base64url_1.4                          
##  [31] tcltk_4.0.3                            
##  [32] DOSE_3.23.2                            
##  [33] haven_2.5.2                            
##  [34] tidyselect_1.2.0                       
##  [35] SummarizedExperiment_1.20.0            
##  [36] rio_0.5.27                             
##  [37] XML_3.99-0.6                           
##  [38] tidyr_1.3.0                            
##  [39] ggpubr_0.4.0                           
##  [40] GenomicAlignments_1.26.0               
##  [41] xtable_1.8-4                           
##  [42] ggnetwork_0.5.12                       
##  [43] magrittr_2.0.3                         
##  [44] evaluate_0.14                          
##  [45] ggplot2_3.4.2                          
##  [46] cli_3.6.1                              
##  [47] zlibbioc_1.36.0                        
##  [48] hwriter_1.3.2                          
##  [49] rstudioapi_0.14                        
##  [50] bslib_0.3.1                            
##  [51] GreyListChIP_1.22.0                    
##  [52] fastmatch_1.1-3                        
##  [53] BiocSingular_1.6.0                     
##  [54] xfun_0.30                              
##  [55] askpass_1.1                            
##  [56] clue_0.3-59                            
##  [57] gson_0.0.9                             
##  [58] cluster_2.1.1                          
##  [59] caTools_1.18.2                         
##  [60] tidygraph_1.2.0                        
##  [61] tibble_3.2.1                           
##  [62] ggrepel_0.9.1                          
##  [63] Biostrings_2.58.0                      
##  [64] png_0.1-7                              
##  [65] withr_2.5.0                            
##  [66] bitops_1.0-6                           
##  [67] ggforce_0.3.3                          
##  [68] RBGL_1.66.0                            
##  [69] plyr_1.8.6                             
##  [70] cellranger_1.1.0                       
##  [71] GSEABase_1.52.1                        
##  [72] pcaPP_2.0-1                            
##  [73] dqrng_0.2.1                            
##  [74] coda_0.19-4                            
##  [75] pillar_1.9.0                           
##  [76] gplots_3.1.1                           
##  [77] GlobalOptions_0.1.2                    
##  [78] cachem_1.0.4                           
##  [79] fs_1.5.0                               
##  [80] GetoptLong_1.0.5                       
##  [81] clusterProfiler_4.7.1                  
##  [82] DelayedMatrixStats_1.12.3              
##  [83] vctrs_0.6.2                            
##  [84] generics_0.1.0                         
##  [85] plot3D_1.4                             
##  [86] tools_4.0.3                            
##  [87] foreign_0.8-81                         
##  [88] NOISeq_2.34.0                          
##  [89] munsell_0.5.0                          
##  [90] tweenr_1.0.2                           
##  [91] fgsea_1.16.0                           
##  [92] DelayedArray_0.16.3                    
##  [93] fastmap_1.1.0                          
##  [94] compiler_4.0.3                         
##  [95] abind_1.4-5                            
##  [96] rtracklayer_1.50.0                     
##  [97] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
##  [98] GenomeInfoDbData_1.2.4                 
##  [99] gridExtra_2.3                          
## [100] edgeR_3.32.1                           
## [101] lattice_0.20-45                        
## [102] ggnewscale_0.4.7                       
## [103] AnnotationForge_1.32.0                 
## [104] utf8_1.2.1                             
## [105] dplyr_1.1.2                            
## [106] BiocFileCache_1.14.0                   
## [107] jsonlite_1.8.4                         
## [108] scales_1.2.1                           
## [109] graph_1.68.0                           
## [110] carData_3.0-4                          
## [111] sparseMatrixStats_1.2.1                
## [112] genefilter_1.72.1                      
## [113] car_3.0-11                             
## [114] doParallel_1.0.16                      
## [115] latticeExtra_0.6-29                    
## [116] brew_1.0-6                             
## [117] checkmate_2.0.0                        
## [118] rmarkdown_2.14                         
## [119] openxlsx_4.2.3                         
## [120] pkgdown_1.6.1                          
## [121] cowplot_1.1.1                          
## [122] textshaping_0.3.6                      
## [123] forcats_1.0.0                          
## [124] downloader_0.4                         
## [125] BSgenome_1.58.0                        
## [126] survival_3.2-10                        
## [127] numDeriv_2016.8-1.1                    
## [128] yaml_2.2.1                             
## [129] plotrix_3.8-2                          
## [130] systemfonts_1.0.4                      
## [131] ashr_2.2-47                            
## [132] SQUAREM_2021.1                         
## [133] htmltools_0.5.2                        
## [134] memoise_2.0.0                          
## [135] VariantAnnotation_1.36.0               
## [136] locfit_1.5-9.4                         
## [137] graphlayouts_0.7.1                     
## [138] batchtools_0.9.15                      
## [139] PCAtools_2.2.0                         
## [140] viridisLite_0.4.0                      
## [141] digest_0.6.27                          
## [142] rrcov_1.7-0                            
## [143] assertthat_0.2.1                       
## [144] rappdirs_0.3.3                         
## [145] emdbook_1.3.12                         
## [146] RSQLite_2.2.5                          
## [147] amap_0.8-18                            
## [148] yulab.utils_0.0.4                      
## [149] debugme_1.1.0                          
## [150] misc3d_0.9-1                           
## [151] data.table_1.14.2                      
## [152] blob_1.2.1                             
## [153] ragg_0.4.0                             
## [154] labeling_0.4.2                         
## [155] splines_4.0.3                          
## [156] ggupset_0.3.0                          
## [157] RCurl_1.98-1.3                         
## [158] broom_1.0.4                            
## [159] hms_1.1.3                              
## [160] colorspace_2.0-0                       
## [161] BiocManager_1.30.16                    
## [162] shape_1.4.6                            
## [163] sass_0.4.1                             
## [164] GEOquery_2.58.0                        
## [165] Rcpp_1.0.9                             
## [166] mvtnorm_1.1-2                          
## [167] circlize_0.4.15                        
## [168] enrichplot_1.10.2                      
## [169] fansi_0.4.2                            
## [170] tzdb_0.3.0                             
## [171] truncnorm_1.0-8                        
## [172] ChIPseeker_1.33.0.900                  
## [173] R6_2.5.0                               
## [174] grid_4.0.3                             
## [175] lifecycle_1.0.3                        
## [176] ShortRead_1.48.0                       
## [177] zip_2.1.1                              
## [178] curl_4.3                               
## [179] ggsignif_0.6.3                         
## [180] jquerylib_0.1.3                        
## [181] robustbase_0.95-0                      
## [182] DO.db_2.9                              
## [183] Matrix_1.5-4                           
## [184] qvalue_2.22.0                          
## [185] desc_1.3.0                             
## [186] RColorBrewer_1.1-2                     
## [187] iterators_1.0.13                       
## [188] stringr_1.5.0                          
## [189] DOT_0.1                                
## [190] ggpie_0.2.5                            
## [191] beachmat_2.6.4                         
## [192] polyclip_1.10-0                        
## [193] biomaRt_2.46.3                         
## [194] purrr_1.0.1                            
## [195] shadowtext_0.0.9                       
## [196] gridGraphics_0.5-1                     
## [197] mgcv_1.8-34                            
## [198] ComplexHeatmap_2.13.1                  
## [199] openssl_1.4.3                          
## [200] patchwork_1.0.0                        
## [201] bdsmatrix_1.3-4                        
## [202] codetools_0.2-18                       
## [203] matrixStats_0.58.0                     
## [204] invgamma_1.1                           
## [205] GO.db_3.12.1                           
## [206] gtools_3.8.2                           
## [207] prettyunits_1.1.1                      
## [208] dbplyr_2.3.2                           
## [209] gtable_0.3.0                           
## [210] DBI_1.1.1                              
## [211] highr_0.8                              
## [212] ggfun_0.0.6                            
## [213] httr_1.4.5                             
## [214] KernSmooth_2.23-18                     
## [215] stringi_1.5.3                          
## [216] progress_1.2.2                         
## [217] reshape2_1.4.4                         
## [218] farver_2.1.0                           
## [219] annotate_1.68.0                        
## [220] viridis_0.6.1                          
## [221] Rgraphviz_2.34.0                       
## [222] xml2_1.3.4                             
## [223] bbmle_1.0.24                           
## [224] systemPipeR_1.24.3                     
## [225] boot_1.3-28                            
## [226] readr_2.1.4                            
## [227] geneplotter_1.68.0                     
## [228] ggplotify_0.1.0                        
## [229] Category_2.56.0                        
## [230] DEoptimR_1.0-11                        
## [231] DESeq2_1.30.1                          
## [232] bit_4.0.4                              
## [233] scatterpie_0.1.7                       
## [234] jpeg_0.1-8.1                           
## [235] MatrixGenerics_1.2.1                   
## [236] ggraph_2.0.5                           
## [237] pkgconfig_2.0.3                        
## [238] rstatix_0.7.0                          
## [239] knitr_1.37</code></pre>
<hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
