<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>IntegrateChIPATAC • DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="IntegrateChIPATAC">
<meta property="og:description" content="DEbPeak">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="IntegrateChIPATAC_files/header-attrs-2.14/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>IntegrateChIPATAC</h1>
            
      
      
      <div class="hidden name"><code>IntegrateChIPATAC.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>In previous vignettes, we have introduced intregrating RNA-seq with ATAC-seq and ChIP-seq separately. But we should keep in mind that <a href="https://doi.org/10.1093/bib/bbac061">ATAC-seq and ChIP-seq are highly consistent and complementary</a>. <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-1929-3">Most TFs require open chromatin for binding to DNA recognition sites, ATAC-seq peaks generally overlap with TF ChIP-seq peaks but are often broader. Thus, TF ChIP-seq and ATAC-seq can mutually validate the quality and reliability of each other within the same experimental system</a>.</p>
<p>To obtain more accurate gene regulation results, here we will integrate RNA-seq with ChIP-seq and ATAC-seq simultaneously.</p>
<hr>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>The data used here contains RNA-seq and ChIP-seq datasets from <a href="https://www.science.org/doi/10.1126/sciadv.aau8389">RUNX represses Pmp22 to drive neurofibromagenesis</a>:</p>
<ul>
<li>RNA-seq: two genotypes and three samples per genotype, the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122774">GSE122774</a>
</li>
<li>ChIP-seq: two genotypes and one sample per genotype, the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122775">GSE122775</a>
</li>
<li>ATAC-seq: two genotypes and one sample per genotype, the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122776">GSE122776</a>
</li>
</ul>
<hr>
</div>
<div id="load-results" class="section level2">
<h2 class="hasAnchor">
<a href="#load-results" class="anchor"></a>Load results</h2>
<p>The integration is based on results of “IntegrateChIP” vignette and “IntegrateATAC” vignette, load the results first.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2">DESeq2</a></span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DEbPeak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co"># load RNA-seq and ChIP-seq integration results</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/RNAandChIP.RData"</span><span class="op">)</span>
<span class="co"># load RNA-seq and ATAC-seq integration results</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/RNAandATAC.RData"</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="integrate-rna-seq-chip-seq-and-atac-seq" class="section level2">
<h2 class="hasAnchor">
<a href="#integrate-rna-seq-chip-seq-and-atac-seq" class="anchor"></a>Integrate RNA-seq, ChIP-seq and ATAC-seq</h2>
<div id="integrate" class="section level3">
<h3 class="hasAnchor">
<a href="#integrate" class="anchor"></a>Integrate</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">debpeak.res</span> <span class="op">=</span> <span class="fu"><a href="../reference/DEbCA.html">DEbCA</a></span><span class="op">(</span>de.res <span class="op">=</span> <span class="va">debchip.dds.results.ordered</span>, chip.peak.res <span class="op">=</span> <span class="va">peak.anno.df</span>,
                    atac.peak.res <span class="op">=</span> <span class="va">atac.peak.anno.df</span>,
                    peak.anno.key <span class="op">=</span> <span class="st">"Promoter"</span>, merge.key <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">debpeak.res</span><span class="op">)</span></code></pre></div>
<pre><code>##          SYMBOL geneId_ChIP    annotation_ChIP anno_ChIP ENSEMBL_ChIP
## 1 0610009E02Rik        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;         &lt;NA&gt;
## 2 0610012G03Rik        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;         &lt;NA&gt;
## 3 0610039K10Rik        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;         &lt;NA&gt;
## 4 1110002J07Rik       68488 Promoter (12-13kb)  Promoter         &lt;NA&gt;
## 5 1110004F10Rik        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;         &lt;NA&gt;
## 6 1110008P14Rik        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;         &lt;NA&gt;
##                GENENAME_ChIP log2FoldChange abundance   signif   regulation
## 1                       &lt;NA&gt;             NA        NA       NA         &lt;NA&gt;
## 2                       &lt;NA&gt;       1.837668  59.17919 3.073773 Up_regulated
## 3                       &lt;NA&gt;             NA        NA       NA         &lt;NA&gt;
## 4 RIKEN cDNA 1110002J07 gene             NA        NA       NA         &lt;NA&gt;
## 5                       &lt;NA&gt;             NA        NA       NA         &lt;NA&gt;
## 6                       &lt;NA&gt;       1.013546 125.76554 2.277566 Up_regulated
##   Type1 geneId_ATAC    annotation_ATAC anno_ATAC       ENSEMBL_ATAC
## 1  &lt;NA&gt;   100125929   Promoter (&lt;=1kb)  Promoter ENSMUSG00000086714
## 2    UP      106264   Promoter (&lt;=1kb)  Promoter ENSMUSG00000107002
## 3  &lt;NA&gt;       68386   Promoter (&lt;=1kb)  Promoter ENSMUSG00000058812
## 4  ChIP       68488 Promoter (15-16kb)  Promoter               &lt;NA&gt;
## 5  &lt;NA&gt;       56372   Promoter (&lt;=1kb)  Promoter ENSMUSG00000030663
## 6    UP        &lt;NA&gt;               &lt;NA&gt;      &lt;NA&gt;               &lt;NA&gt;
##                GENENAME_ATAC      Type    geneId
## 1 RIKEN cDNA 0610009E02 gene      ATAC 100125929
## 2 RIKEN cDNA 0610012G03 gene   UPbATAC    106264
## 3 RIKEN cDNA 0610039K10 gene      ATAC     68386
## 4 RIKEN cDNA 1110002J07 gene ChIPbATAC     68488
## 5 RIKEN cDNA 1110004F10 gene      ATAC     56372
## 6                       &lt;NA&gt;        UP      &lt;NA&gt;</code></pre>
<hr>
</div>
<div id="integrate-summary" class="section level3">
<h3 class="hasAnchor">
<a href="#integrate-summary" class="anchor"></a>Integrate summary</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DE and ChIP venn plot</span>
<span class="va">debpeak.plot</span> <span class="op">=</span> <span class="fu"><a href="../reference/PlotDEbPeak.html">PlotDEbPeak</a></span><span class="op">(</span><span class="va">debpeak.res</span>, peak.type <span class="op">=</span> <span class="st">"Peak"</span>, show_percentage<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="va">debpeak.plot</span></code></pre></div>
<p><img src="IntegrateChIPATAC_files/figure-html/debpeak_integrate_plot-1.png" width="384" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="functional-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#functional-enrichment" class="anchor"></a>Functional enrichment</h2>
<p>There are <strong>five categories</strong> for users to choose to perform functional enrichment: <code>ChIP</code>, <code>ATAC</code>, <code>UP</code>, <code>DOWN</code>, <code>ChIPbATAC</code>, <code>UPbChIP</code>, <code>DOWNbChIP</code>, <code>UPbPeak</code>, <code>DOWNbPeak</code>, <code>UPbATAC</code>, <code>DOWNbATAC</code>. Here, we will use <code>DOWNbPeak</code> and <code>UPbPeak</code> as examples.</p>
<div id="upbpeak" class="section level3">
<h3 class="hasAnchor">
<a href="#upbpeak" class="anchor"></a>UPbPeak</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># functional enrichment on direct targets</span>
<span class="va">debpeak.up.fe.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/DEbPeakFE.html">DEbPeakFE</a></span><span class="op">(</span>de.peak <span class="op">=</span> <span class="va">debpeak.res</span>, peak.fe.key <span class="op">=</span> <span class="st">"UPbPeak"</span>, gene.type <span class="op">=</span> <span class="st">"ENTREZID"</span>, species<span class="op">=</span><span class="st">"Mouse"</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## </code></pre>
<pre><code>## conduct ALL GO enrichment analysis on: UPbPeak</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<p>The results:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># the result table</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">debpeak.up.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                     Description GeneRatio
## GO:0043218       CC GO:0043218                  compact myelin       2/9
## GO:0043209       CC GO:0043209                   myelin sheath       2/9
## GO:0070382       CC GO:0070382                exocytic vesicle       2/9
## GO:0043256       CC GO:0043256                 laminin complex       1/9
## GO:0019898       CC GO:0019898 extrinsic component of membrane       2/9
## GO:0030133       CC GO:0030133               transport vesicle       2/9
##              BgRatio       pvalue    p.adjust      qvalue    geneID Count
## GO:0043218  19/23271 2.265877e-05 0.001087621 0.000500878 Mbp/Pmp22     2
## GO:0043209 213/23271 2.877487e-03 0.043690148 0.020120463 Mbp/Pmp22     2
## GO:0070382 261/23271 4.282761e-03 0.043690148 0.020120463  Bsn/Syt6     2
## GO:0043256  12/23271 4.632203e-03 0.043690148 0.020120463     Pmp22     1
## GO:0019898 320/23271 6.366508e-03 0.043690148 0.020120463  Bsn/Syt6     2
## GO:0030133 339/23271 7.118989e-03 0.043690148 0.020120463  Bsn/Syt6     2</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># the result plot</span>
<span class="va">debpeak.up.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="IntegrateChIPATAC_files/figure-html/debpeak_integrate_fe_up_show-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="downbpeak" class="section level3">
<h3 class="hasAnchor">
<a href="#downbpeak" class="anchor"></a>DOWNbPeak</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># functional enrichment on direct targets</span>
<span class="va">debpeak.down.fe.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/DEbPeakFE.html">DEbPeakFE</a></span><span class="op">(</span>de.peak <span class="op">=</span> <span class="va">debpeak.res</span>, peak.fe.key <span class="op">=</span> <span class="st">"DOWNbPeak"</span>, 
                                    gene.type <span class="op">=</span> <span class="st">"ENTREZID"</span>, species<span class="op">=</span><span class="st">"Mouse"</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code>## conduct ALL GO enrichment analysis on: DOWNbPeak</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.</code></pre>
<p>The results:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># the result table</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">debpeak.down.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                                    Description
## GO:0045064       BP GO:0045064                T-helper 2 cell differentiation
## GO:0010811       BP GO:0010811 positive regulation of cell-substrate adhesion
## GO:0045785       BP GO:0045785           positive regulation of cell adhesion
## GO:0045860       BP GO:0045860 positive regulation of protein kinase activity
## GO:0042832       BP GO:0042832                  defense response to protozoan
## GO:0010810       BP GO:0010810          regulation of cell-substrate adhesion
##            GeneRatio   BgRatio       pvalue   p.adjust      qvalue
## GO:0045064      2/13  17/23328 3.880421e-05 0.01212719 0.006013313
## GO:0010811      3/13 128/23328 4.433038e-05 0.01212719 0.006013313
## GO:0045785      4/13 435/23328 7.460788e-05 0.01212719 0.006013313
## GO:0045860      4/13 440/23328 7.798836e-05 0.01212719 0.006013313
## GO:0042832      2/13  36/23328 1.786843e-04 0.01717888 0.008518210
## GO:0010810      3/13 214/23328 2.034275e-04 0.01717888 0.008518210
##                             geneID Count
## GO:0045064              Bcl3/Il4ra     2
## GO:0010811       Col8a1/Fbln2/Sdc4     3
## GO:0045785 Col8a1/Fbln2/Il4ra/Sdc4     4
## GO:0045860  Akap13/Ccnd1/Ern1/Sdc4     4
## GO:0042832              Bcl3/Il4ra     2
## GO:0010810       Col8a1/Fbln2/Sdc4     3</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># the result plot</span>
<span class="va">debpeak.down.fe.results</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="IntegrateChIPATAC_files/figure-html/debpeak_integrate_fe_down_show-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-conda-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /home/softwares/anaconda3/envs/r4.0/lib/libopenblasp-r0.3.12.so
## 
## locale:
##  [1] LC_CTYPE=zh_CN.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=zh_CN.UTF-8        LC_COLLATE=zh_CN.UTF-8    
##  [5] LC_MONETARY=zh_CN.UTF-8    LC_MESSAGES=zh_CN.UTF-8   
##  [7] LC_PAPER=zh_CN.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] org.Mm.eg.db_3.12.0         AnnotationDbi_1.52.0       
##  [3] DEbPeak_0.9.0               DESeq2_1.30.1              
##  [5] SummarizedExperiment_1.20.0 Biobase_2.50.0             
##  [7] MatrixGenerics_1.2.1        matrixStats_0.58.0         
##  [9] GenomicRanges_1.42.0        GenomeInfoDb_1.26.7        
## [11] IRanges_2.24.1              S4Vectors_0.28.1           
## [13] BiocGenerics_0.42.0        
## 
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                             
##   [2] ggvenn_0.1.9                           
##   [3] apeglm_1.12.0                          
##   [4] Rsamtools_2.6.0                        
##   [5] rsvg_2.1                               
##   [6] foreach_1.5.1                          
##   [7] rprojroot_2.0.2                        
##   [8] crayon_1.4.1                           
##   [9] V8_3.4.2                               
##  [10] MASS_7.3-53.1                          
##  [11] nlme_3.1-152                           
##  [12] backports_1.2.1                        
##  [13] sva_3.38.0                             
##  [14] GOSemSim_2.25.0                        
##  [15] rlang_1.0.3                            
##  [16] XVector_0.30.0                         
##  [17] readxl_1.3.1                           
##  [18] irlba_2.3.5                            
##  [19] limma_3.46.0                           
##  [20] GOstats_2.56.0                         
##  [21] BiocParallel_1.24.1                    
##  [22] rjson_0.2.20                           
##  [23] bit64_4.0.5                            
##  [24] glue_1.6.2                             
##  [25] DiffBind_3.0.15                        
##  [26] mixsqp_0.3-43                          
##  [27] pheatmap_1.0.12                        
##  [28] parallel_4.0.3                         
##  [29] DEFormats_1.18.0                       
##  [30] base64url_1.4                          
##  [31] tcltk_4.0.3                            
##  [32] DOSE_3.23.2                            
##  [33] haven_2.3.1                            
##  [34] tidyselect_1.1.0                       
##  [35] rio_0.5.27                             
##  [36] XML_3.99-0.6                           
##  [37] tidyr_1.1.3                            
##  [38] ggpubr_0.4.0                           
##  [39] GenomicAlignments_1.26.0               
##  [40] xtable_1.8-4                           
##  [41] magrittr_2.0.1                         
##  [42] evaluate_0.14                          
##  [43] ggplot2_3.3.5                          
##  [44] cli_3.3.0                              
##  [45] zlibbioc_1.36.0                        
##  [46] hwriter_1.3.2                          
##  [47] rstudioapi_0.13                        
##  [48] bslib_0.3.1                            
##  [49] GreyListChIP_1.22.0                    
##  [50] fastmatch_1.1-3                        
##  [51] BiocSingular_1.6.0                     
##  [52] xfun_0.30                              
##  [53] askpass_1.1                            
##  [54] clue_0.3-59                            
##  [55] gson_0.0.9                             
##  [56] cluster_2.1.1                          
##  [57] caTools_1.18.2                         
##  [58] tidygraph_1.2.0                        
##  [59] tibble_3.1.0                           
##  [60] ggrepel_0.9.1                          
##  [61] Biostrings_2.58.0                      
##  [62] png_0.1-7                              
##  [63] withr_2.4.1                            
##  [64] bitops_1.0-6                           
##  [65] ggforce_0.3.3                          
##  [66] RBGL_1.66.0                            
##  [67] plyr_1.8.6                             
##  [68] cellranger_1.1.0                       
##  [69] GSEABase_1.52.1                        
##  [70] pcaPP_2.0-1                            
##  [71] dqrng_0.2.1                            
##  [72] coda_0.19-4                            
##  [73] pillar_1.5.1                           
##  [74] gplots_3.1.1                           
##  [75] GlobalOptions_0.1.2                    
##  [76] cachem_1.0.4                           
##  [77] GenomicFeatures_1.42.2                 
##  [78] fs_1.5.0                               
##  [79] GetoptLong_1.0.5                       
##  [80] clusterProfiler_4.7.1                  
##  [81] DelayedMatrixStats_1.12.3              
##  [82] vctrs_0.4.1                            
##  [83] ellipsis_0.3.2                         
##  [84] generics_0.1.0                         
##  [85] plot3D_1.4                             
##  [86] tools_4.0.3                            
##  [87] foreign_0.8-81                         
##  [88] NOISeq_2.34.0                          
##  [89] munsell_0.5.0                          
##  [90] tweenr_1.0.2                           
##  [91] fgsea_1.16.0                           
##  [92] DelayedArray_0.16.3                    
##  [93] fastmap_1.1.0                          
##  [94] compiler_4.0.3                         
##  [95] abind_1.4-5                            
##  [96] rtracklayer_1.50.0                     
##  [97] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
##  [98] GenomeInfoDbData_1.2.4                 
##  [99] gridExtra_2.3                          
## [100] edgeR_3.32.1                           
## [101] lattice_0.20-45                        
## [102] ggnewscale_0.4.7                       
## [103] AnnotationForge_1.32.0                 
## [104] utf8_1.2.1                             
## [105] dplyr_1.0.5                            
## [106] BiocFileCache_1.14.0                   
## [107] jsonlite_1.7.2                         
## [108] scales_1.1.1                           
## [109] graph_1.68.0                           
## [110] carData_3.0-4                          
## [111] sparseMatrixStats_1.2.1                
## [112] genefilter_1.72.1                      
## [113] car_3.0-11                             
## [114] doParallel_1.0.16                      
## [115] latticeExtra_0.6-29                    
## [116] brew_1.0-6                             
## [117] checkmate_2.0.0                        
## [118] rmarkdown_2.14                         
## [119] openxlsx_4.2.3                         
## [120] pkgdown_1.6.1                          
## [121] cowplot_1.1.1                          
## [122] textshaping_0.1.2                      
## [123] forcats_0.5.1                          
## [124] downloader_0.4                         
## [125] BSgenome_1.58.0                        
## [126] igraph_1.3.4                           
## [127] survival_3.2-10                        
## [128] numDeriv_2016.8-1.1                    
## [129] yaml_2.2.1                             
## [130] plotrix_3.8-2                          
## [131] systemfonts_1.0.1                      
## [132] ashr_2.2-47                            
## [133] SQUAREM_2021.1                         
## [134] htmltools_0.5.2                        
## [135] memoise_2.0.0                          
## [136] VariantAnnotation_1.36.0               
## [137] locfit_1.5-9.4                         
## [138] graphlayouts_0.7.1                     
## [139] batchtools_0.9.15                      
## [140] PCAtools_2.2.0                         
## [141] viridisLite_0.4.0                      
## [142] digest_0.6.27                          
## [143] rrcov_1.7-0                            
## [144] assertthat_0.2.1                       
## [145] rappdirs_0.3.3                         
## [146] emdbook_1.3.12                         
## [147] RSQLite_2.2.5                          
## [148] amap_0.8-18                            
## [149] yulab.utils_0.0.4                      
## [150] misc3d_0.9-1                           
## [151] data.table_1.14.2                      
## [152] blob_1.2.1                             
## [153] ragg_0.4.0                             
## [154] labeling_0.4.2                         
## [155] splines_4.0.3                          
## [156] ggupset_0.3.0                          
## [157] RCurl_1.98-1.3                         
## [158] broom_0.7.5                            
## [159] hms_1.0.0                              
## [160] colorspace_2.0-0                       
## [161] BiocManager_1.30.16                    
## [162] shape_1.4.6                            
## [163] sass_0.4.1                             
## [164] Rcpp_1.0.9                             
## [165] mvtnorm_1.1-2                          
## [166] circlize_0.4.15                        
## [167] enrichplot_1.10.2                      
## [168] fansi_0.4.2                            
## [169] truncnorm_1.0-8                        
## [170] ChIPseeker_1.33.0.900                  
## [171] R6_2.5.0                               
## [172] grid_4.0.3                             
## [173] lifecycle_1.0.0                        
## [174] ShortRead_1.48.0                       
## [175] zip_2.1.1                              
## [176] curl_4.3                               
## [177] ggsignif_0.6.3                         
## [178] jquerylib_0.1.3                        
## [179] robustbase_0.95-0                      
## [180] DO.db_2.9                              
## [181] Matrix_1.3-3                           
## [182] qvalue_2.22.0                          
## [183] desc_1.3.0                             
## [184] RColorBrewer_1.1-2                     
## [185] iterators_1.0.13                       
## [186] stringr_1.4.0                          
## [187] DOT_0.1                                
## [188] ggpie_0.2.1                            
## [189] beachmat_2.6.4                         
## [190] polyclip_1.10-0                        
## [191] biomaRt_2.46.3                         
## [192] purrr_0.3.4                            
## [193] shadowtext_0.0.9                       
## [194] gridGraphics_0.5-1                     
## [195] mgcv_1.8-34                            
## [196] ComplexHeatmap_2.13.1                  
## [197] openssl_1.4.3                          
## [198] patchwork_1.0.0                        
## [199] bdsmatrix_1.3-4                        
## [200] codetools_0.2-18                       
## [201] invgamma_1.1                           
## [202] GO.db_3.12.1                           
## [203] gtools_3.8.2                           
## [204] prettyunits_1.1.1                      
## [205] dbplyr_2.1.1                           
## [206] gtable_0.3.0                           
## [207] DBI_1.1.1                              
## [208] highr_0.8                              
## [209] ggfun_0.0.6                            
## [210] httr_1.4.2                             
## [211] KernSmooth_2.23-18                     
## [212] stringi_1.5.3                          
## [213] progress_1.2.2                         
## [214] reshape2_1.4.4                         
## [215] farver_2.1.0                           
## [216] annotate_1.68.0                        
## [217] viridis_0.6.1                          
## [218] Rgraphviz_2.34.0                       
## [219] xml2_1.3.2                             
## [220] bbmle_1.0.24                           
## [221] systemPipeR_1.24.3                     
## [222] boot_1.3-28                            
## [223] geneplotter_1.68.0                     
## [224] ggplotify_0.1.0                        
## [225] Category_2.56.0                        
## [226] DEoptimR_1.0-11                        
## [227] bit_4.0.4                              
## [228] scatterpie_0.1.7                       
## [229] jpeg_0.1-8.1                           
## [230] ggraph_2.0.5                           
## [231] pkgconfig_2.0.3                        
## [232] rstatix_0.7.0                          
## [233] knitr_1.37</code></pre>
<hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
