<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MotifEnrichment â€¢ DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MotifEnrichment">
<meta property="og:description" content="DEbPeak">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="MotifEnrichment_files/header-attrs-2.14/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MotifEnrichment</h1>
            
      
      
      <div class="hidden name"><code>MotifEnrichment.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>For peak-related data, motif enrichment can be used to find the TFs involved in the gene regulation, especially for ATAC-seq (ATAC-seq can proflie global chromatin accessibility. Thus, the peaks of ATAC-seq may originate from hundreds of TFs.).</p>
<p>Here, we will perform motif enrichment on differentially accessible (DA) peaks.</p>
<hr>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>The data used here are from <a href="https://doi.org/10.1016/j.cell.2021.02.019">In vivo CD8+ T cell CRISPR screening reveals control by Fli1 in infection and cancer</a>:</p>
<ul>
<li>
<strong>RNA-seq data</strong>: the sgCtrl vs sgFli1 RNA sequencing at D8 Cl13 p.i., the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149838">GSE149838</a>
</li>
<li>
<strong>ATAC-seq data</strong>: sgCtrl vs sgFli1 ATAC sequencing at D9 Cl13 p.i., the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149836">GSE149836</a>
</li>
</ul>
<hr>
</div>
<div id="load-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DEbPeak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-analysis" class="anchor"></a>Differential Expression Analysis</h2>
<p>To simplify the steps of obtaining differential expression analysis results, <code>DEbPeak</code> provides <code>ConductDESeq2</code> to perfrom all steps of differential expression analysis, including <strong>quality control</strong>, <strong>principal component analysis</strong>, <strong>differential expression analysis</strong> and <strong>functional enrichment analysis</strong>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">peak.matrix.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RA_ATAC_count.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">peak.meta.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RA_ATAC_meta.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="fu"><a href="../reference/ConductDESeq2.html">ConductDESeq2</a></span><span class="op">(</span>count.matrix.file <span class="op">=</span> <span class="va">peak.matrix.file</span>, meta.file <span class="op">=</span> <span class="va">peak.meta.file</span>, data.type <span class="op">=</span> <span class="st">"ATAC"</span>,
              gene.type <span class="op">=</span> <span class="st">"SYMBOL"</span>, outlier.detection <span class="op">=</span> <span class="cn">F</span>, min.count <span class="op">=</span> <span class="fl">0</span>,
              peak.anno.key <span class="op">=</span> <span class="st">"All"</span>, group.key <span class="op">=</span> <span class="st">"condition"</span>, ref.group <span class="op">=</span> <span class="st">"WT"</span>,
              out.folder <span class="op">=</span> <span class="st">'/home/songyabing/R/learn/tmp/DEbPeak/ATACDE'</span>,
              signif <span class="op">=</span> <span class="st">"padj"</span>, signif.threshold <span class="op">=</span> <span class="fl">0.05</span>, l2fc.threshold <span class="op">=</span> <span class="fl">0</span>, gmt.file <span class="op">=</span> <span class="st">''</span><span class="op">)</span></code></pre></div>
<pre><code>## Create DESeqDataSet from count matrix!</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## Conduct Quality Control on Counts!</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## [1] "Warning: 7 features with 0 counts in all samples are to be removed for this analysis."
## [1] "Count distributions are to be computed for:"
## [1] "Fli1KO_1" "Fli1KO_2" "Fli1KO_3" "Fli1KO_4" "WT_1"     "WT_2"</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Raw gene number: 101921</code></pre>
<pre><code>## Gene number after filtered with counts bigger than 0 : 101921</code></pre>
<pre><code>## Conduct Quality Control on Samples!</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Conduct PCA analysis!
## Conduct PCA analysis!</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Use all genes for PCA!</code></pre>
<pre><code>## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.</code></pre>
<pre><code>## Selecting by Loadding
## Selecting by Loadding
## Differential expression analysis with DESeq2!
## 
## Selecting by Loadding
## Selecting by Loadding
## 
## è½½å…¥ç¨‹è¾‘åŒ…ï¼š'BiocGenerics'
## 
## 
## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs
## 
## 
## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min
## 
## 
## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.
## 
## 
## 
## è½½å…¥ç¨‹è¾‘åŒ…ï¼š'S4Vectors'
## 
## 
## The following object is masked from 'package:base':
## 
##     expand.grid
## 
## 
## 
## 
## Conduct Functional Enrichment on PC Loading genes!
## 
## Selecting by Loadding
## Selecting by Loadding
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC1_Positive
## 
## conduct KEGG enrichment analysis: PC1_Positive
## 
## Reading KEGG annotation online: "https://rest.kegg.jp/link/mmu/pathway"...
## 
## Reading KEGG annotation online: "https://rest.kegg.jp/list/pathway/mmu"...
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC1_Negative
## 
## conduct KEGG enrichment analysis: PC1_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC2_Positive
## 
## conduct KEGG enrichment analysis: PC2_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC2_Negative
## 
## conduct KEGG enrichment analysis: PC2_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC3_Positive
## 
## conduct KEGG enrichment analysis: PC3_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC3_Negative
## 
## conduct KEGG enrichment analysis: PC3_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC4_Positive
## 
## conduct KEGG enrichment analysis: PC4_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC4_Negative
## 
## conduct KEGG enrichment analysis: PC4_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC5_Positive
## 
## conduct KEGG enrichment analysis: PC5_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC5_Negative
## 
## conduct KEGG enrichment analysis: PC5_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## 
## Conduct Differential Expression Analysis!
## 
## estimating size factors
## 
## estimating dispersions
## 
## gene-wise dispersion estimates
## 
## mean-dispersion relationship
## 
## -- note: fitType='parametric', but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType='local' or 'mean' to avoid this message next time.
## 
## final dispersion estimates
## 
## fitting model and testing
## 
## Conduct Gene ID Conversion!
## 
## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in clusterProfiler::bitr(de.df[, "Gene2Convert"], fromType =
## gene.type, : 1.63% of input gene IDs are fail to map...</code></pre>
<pre><code>## Visualize Differential Analysis Results!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Conduct Functional Enrichment Analysis!
## Differential expression analysis with DESeq2!
## Convert SYMBOL to ENTREZID!
## 'select()' returned 1:1 mapping between keys and columns
## conduct ALL GO enrichment analysis on: UP
## conduct KEGG enrichment analysis: UP
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## Convert SYMBOL to ENTREZID!
## 'select()' returned 1:1 mapping between keys and columns
## conduct ALL GO enrichment analysis on: DOWN
## conduct KEGG enrichment analysis: DOWN
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for 'y' is already present. Adding another scale for 'y', which will
## replace the existing scale.
## Gene Set Enrichment Analysis is not available for ChIP-seq or ATAC-seq data!
## All Analysis Done!</code></pre>
<p>The structure of result directory:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">tree</span> /home/songyabing/R/learn/tmp/DEbPeak/ATACDE</span></code></pre></div>
<pre><code>## /home/songyabing/R/learn/tmp/DEbPeak/ATACDE
## â”œâ”€â”€ CountQC_CPM.pdf
## â”œâ”€â”€ CountQC_saturation.pdf
## â”œâ”€â”€ DA
## â”‚Â Â  â”œâ”€â”€ Condition_KO_WT_all.csv
## â”‚Â Â  â”œâ”€â”€ Condition_KO_WT_padj0.05_FC0.csv
## â”‚Â Â  â”œâ”€â”€ DA_GenePlot.pdf
## â”‚Â Â  â”œâ”€â”€ DA_Heatmap.pdf
## â”‚Â Â  â”œâ”€â”€ DA_MAPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DA_RankPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DA_ScatterPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DA_VolcanoPlot.pdf
## â”‚Â Â  â””â”€â”€ normalized_counts.csv
## â”œâ”€â”€ DEG
## â”‚Â Â  â”œâ”€â”€ Condition_KO_WT_all.csv
## â”‚Â Â  â”œâ”€â”€ Condition_KO_WT_padj0.05_FC0.csv
## â”‚Â Â  â”œâ”€â”€ DEG_GenePlot.pdf
## â”‚Â Â  â”œâ”€â”€ DEG_Heatmap.pdf
## â”‚Â Â  â”œâ”€â”€ DEG_MAPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DEG_RankPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DEG_ScatterPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DEG_VolcanoPlot.pdf
## â”‚Â Â  â””â”€â”€ normalized_counts.csv
## â”œâ”€â”€ FE
## â”‚Â Â  â”œâ”€â”€ DOWN_ALL_GO.csv
## â”‚Â Â  â”œâ”€â”€ DOWN_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ DOWN_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ DOWN_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ DOWN_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ DOWN_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ DOWN_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ UP_ALL_GO.csv
## â”‚Â Â  â”œâ”€â”€ UP_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ UP_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ UP_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ UP_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ UP_KEGG_Enrichment.png
## â”‚Â Â  â””â”€â”€ UP_Molecular_Function.png
## â”œâ”€â”€ PCA
## â”‚Â Â  â”œâ”€â”€ PC1
## â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Negative
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â”œâ”€â”€ PC1_Negative_ALL_GO.csv
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â””â”€â”€ PC1_Negative_KEGG.csv
## â”‚Â Â  â”‚Â Â  â””â”€â”€ Positive
## â”‚Â Â  â”‚Â Â      â”œâ”€â”€ PC1_Positive_ALL_GO.csv
## â”‚Â Â  â”‚Â Â      â””â”€â”€ PC1_Positive_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ PC1_Negative_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC1_Negative_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC1_Negative_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC1_Negative_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC1_Negative_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC1_Positive_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC1_Positive_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC1_Positive_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC1_Positive_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC1_Positive_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC2
## â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Negative
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â”œâ”€â”€ PC2_Negative_ALL_GO.csv
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â””â”€â”€ PC2_Negative_KEGG.csv
## â”‚Â Â  â”‚Â Â  â””â”€â”€ Positive
## â”‚Â Â  â”‚Â Â      â”œâ”€â”€ PC2_Positive_ALL_GO.csv
## â”‚Â Â  â”‚Â Â      â””â”€â”€ PC2_Positive_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ PC2_Negative_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC2_Negative_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC2_Negative_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC2_Negative_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC2_Negative_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC2_Positive_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC2_Positive_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC2_Positive_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC2_Positive_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC2_Positive_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC3
## â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Negative
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â”œâ”€â”€ PC3_Negative_ALL_GO.csv
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â””â”€â”€ PC3_Negative_KEGG.csv
## â”‚Â Â  â”‚Â Â  â””â”€â”€ Positive
## â”‚Â Â  â”‚Â Â      â”œâ”€â”€ PC3_Positive_ALL_GO.csv
## â”‚Â Â  â”‚Â Â      â””â”€â”€ PC3_Positive_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ PC3_Negative_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC3_Negative_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC3_Negative_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC3_Negative_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC3_Negative_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC3_Positive_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC3_Positive_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC3_Positive_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC3_Positive_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC3_Positive_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC4
## â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Negative
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â”œâ”€â”€ PC4_Negative_ALL_GO.csv
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â””â”€â”€ PC4_Negative_KEGG.csv
## â”‚Â Â  â”‚Â Â  â””â”€â”€ Positive
## â”‚Â Â  â”‚Â Â      â”œâ”€â”€ PC4_Positive_ALL_GO.csv
## â”‚Â Â  â”‚Â Â      â””â”€â”€ PC4_Positive_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ PC4_Negative_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC4_Negative_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC4_Negative_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC4_Negative_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC4_Negative_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC4_Positive_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC4_Positive_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC4_Positive_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC4_Positive_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC4_Positive_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC5
## â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Negative
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â”œâ”€â”€ PC5_Negative_ALL_GO.csv
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â””â”€â”€ PC5_Negative_KEGG.csv
## â”‚Â Â  â”‚Â Â  â””â”€â”€ Positive
## â”‚Â Â  â”‚Â Â      â”œâ”€â”€ PC5_Positive_ALL_GO.csv
## â”‚Â Â  â”‚Â Â      â””â”€â”€ PC5_Positive_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ PC5_Negative_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC5_Negative_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC5_Negative_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC5_Negative_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC5_Negative_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC5_Positive_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC5_Positive_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC5_Positive_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC5_Positive_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC5_Positive_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PCA_3DPCA.pdf
## â”‚Â Â  â”œâ”€â”€ PCA_biplot.pdf
## â”‚Â Â  â”œâ”€â”€ PCA_loading_bar.pdf
## â”‚Â Â  â”œâ”€â”€ PCA_loading_heat.pdf
## â”‚Â Â  â”œâ”€â”€ PCA_overview.pdf
## â”‚Â Â  â”œâ”€â”€ PCA_pairs_plot.pdf
## â”‚Â Â  â””â”€â”€ PCA_screen_plot.pdf
## â””â”€â”€ SampleQC_dist_pcc.pdf
## 
## 19 directories, 112 files</code></pre>
<p>Read the results:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds.peak.results.ordered</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/ATACDE/DEG/Condition_KO_WT_all.csv"</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dds.peak.results.ordered</span><span class="op">)</span></code></pre></div>
<pre><code>##                                           baseMean log2FoldChange    lfcSE
## chr8:94825359-94825432|Ciapin1|P         12.106577       6.439216 1.527019
## chr3:105628355-105628501|Ddx20|I         11.180004       6.323179 1.523447
## chr1:118440678-118440831|2900060B14Rik|I  7.836836       5.790918 1.569651
## chr1:165368382-165368517|Dcaf6|I          6.140173       5.439287 1.619636
## chr16:59434969-59435053|Gabrr3|I          5.706767       5.348676 1.631421
## chr11:20780454-20780545|Aftph|DI          5.390470       5.269629 1.636609
##                                              stat       pvalue        padj
## chr8:94825359-94825432|Ciapin1|P         4.216854 2.477343e-05 0.001265734
## chr3:105628355-105628501|Ddx20|I         4.150575 3.316417e-05 0.001598853
## chr1:118440678-118440831|2900060B14Rik|I 3.689304 2.248687e-04 0.006892699
## chr1:165368382-165368517|Dcaf6|I         3.358339 7.841226e-04 0.017866061
## chr16:59434969-59435053|Gabrr3|I         3.278539 1.043461e-03 0.021915949
## chr11:20780454-20780545|Aftph|DI         3.219846 1.282594e-03 0.025309971
##                                                 SYMBOL            ENSEMBL
## chr8:94825359-94825432|Ciapin1|P               Ciapin1 ENSMUSG00000031781
## chr3:105628355-105628501|Ddx20|I                 Ddx20 ENSMUSG00000027905
## chr1:118440678-118440831|2900060B14Rik|I 2900060B14Rik ENSMUSG00000107722
## chr1:165368382-165368517|Dcaf6|I                 Dcaf6 ENSMUSG00000026571
## chr16:59434969-59435053|Gabrr3|I                Gabrr3 ENSMUSG00000074991
## chr11:20780454-20780545|Aftph|DI                 Aftph ENSMUSG00000049659
##                                          ENTREZID
## chr8:94825359-94825432|Ciapin1|P           109006
## chr3:105628355-105628501|Ddx20|I            53975
## chr1:118440678-118440831|2900060B14Rik|I    68204
## chr1:165368382-165368517|Dcaf6|I            74106
## chr16:59434969-59435053|Gabrr3|I           328699
## chr11:20780454-20780545|Aftph|DI           216549</code></pre>
<hr>
</div>
<div id="motif-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#motif-enrichment" class="anchor"></a>Motif enrichment</h2>
<p>There are <strong>three categories</strong> for users to choose to perform functional enrichment: <code>Up_regulated</code>, <code>Down_regulated</code>, <code>Not_regulated</code>. Here, we will use <code>Up_regulated</code> as example.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># this step is time consuming!</span>
<span class="va">da.peak.motif</span> <span class="op">=</span> <span class="fu"><a href="../reference/MotifEnrich.html">MotifEnrich</a></span><span class="op">(</span>de.res <span class="op">=</span> <span class="va">dds.peak.results.ordered</span>, reg.type <span class="op">=</span> <span class="st">"Up_regulated"</span>,
                            homer.motif.path <span class="op">=</span> <span class="st">'~/anaconda3/bin/findMotifsGenome.pl'</span>,
                            out.folder <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/DAmotif"</span>,
                            other.paras <span class="op">=</span> <span class="st">"-len 8,10,12 -size -100,50 -S 25"</span><span class="op">)</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
