<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MotifAnalysis â€¢ DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MotifAnalysis">
<meta property="og:description" content="DEbPeak">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.4.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ParseGEO.html">Parse GEO</a>
    </li>
    <li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifAnalysis.html">Motif Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IdentifyRegulator.html">Identify regulators (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPDE.html">Integrate RNA-seq with ChIP-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="MotifAnalysis_files/header-attrs-2.14/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MotifAnalysis</h1>
            
      
      
      <div class="hidden name"><code>MotifAnalysis.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>For peak-related data, there are two levels of motif analysis:</p>
<ul>
<li>
<strong><em>de novo</em> motif discovery</strong>: this can be used to find the specific binding motif of regulator.</li>
<li>
<strong>motif enrichment</strong>: this can be used to find the TFs involved in the gene regulation, especially for ATAC-seq (ATAC-seq can proflie global chromatin accessibility. Thus, the peaks of ATAC-seq may originate from hundreds of TFs.).</li>
</ul>
<p><code>DEbPeak</code> provides <code>MotifDiscovery</code> to perfrom <em>de novo</em> motif discovery with <a href="https://meme-suite.org/meme/tools/streme">STREME</a> and <code>MotifEnrich</code> to perform <em>de novo</em> motif discovery and motif enrichment with <a href="http://homer.ucsd.edu/homer/ngs/peakMotifs.html">Homer</a>. While <a href="http://homer.ucsd.edu/homer/ngs/peakMotifs.html">Homer</a> only supports human, mouse, rat, frog, zebrafish, fly, worm, yeast and arabidopsis, <code>DEbPeak</code> provides the <code>MotifCompare</code> to map motifs provided by users against <em>de novo</em> motif discovered by <code>MotifDiscovery</code>.</p>
<hr>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>The data used here are from <a href="https://doi.org/10.1016/j.cell.2021.02.019">In vivo CD8+ T cell CRISPR screening reveals control by Fli1 in infection and cancer</a>:</p>
<ul>
<li>
<strong>RNA-seq data</strong>: the sgCtrl vs sgFli1 RNA sequencing at D8 Cl13 p.i., the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149838">GSE149838</a>
</li>
<li>
<strong>ATAC-seq data</strong>: sgCtrl vs sgFli1 ATAC sequencing at D9 Cl13 p.i., the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149836">GSE149836</a>
</li>
</ul>
<hr>
</div>
<div id="load-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DEbPeak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-analysis" class="anchor"></a>Differential Expression Analysis</h2>
<p>To simplify the steps of obtaining differential expression analysis results, <code>DEbPeak</code> provides <code>ConductDESeq2</code> to perfrom all steps of differential expression analysis, including <strong>quality control</strong>, <strong>principal component analysis</strong>, <strong>differential expression analysis</strong> and <strong>functional enrichment analysis</strong>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">peak.matrix.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RA_ATAC_count.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">peak.meta.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RA_ATAC_meta.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="fu"><a href="../reference/ConductDESeq2.html">ConductDESeq2</a></span><span class="op">(</span>count.matrix.file <span class="op">=</span> <span class="va">peak.matrix.file</span>, meta.file <span class="op">=</span> <span class="va">peak.meta.file</span>, data.type <span class="op">=</span> <span class="st">"ATAC"</span>,
              gene.type <span class="op">=</span> <span class="st">"SYMBOL"</span>, outlier.detection <span class="op">=</span> <span class="cn">F</span>, min.count <span class="op">=</span> <span class="fl">0</span>,
              peak.anno.key <span class="op">=</span> <span class="st">"All"</span>, group.key <span class="op">=</span> <span class="st">"condition"</span>, ref.group <span class="op">=</span> <span class="st">"WT"</span>,
              out.folder <span class="op">=</span> <span class="st">'/home/songyabing/R/learn/tmp/DEbPeak/ATACDE'</span>,
              signif <span class="op">=</span> <span class="st">"padj"</span>, signif.threshold <span class="op">=</span> <span class="fl">0.05</span>, l2fc.threshold <span class="op">=</span> <span class="fl">0</span>, gmt.file <span class="op">=</span> <span class="st">''</span><span class="op">)</span></code></pre></div>
<pre><code>## Create DESeqDataSet from count matrix!</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## Conduct Quality Control on Counts!</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## [1] "Warning: 7 features with 0 counts in all samples are to be removed for this analysis."
## [1] "Count distributions are to be computed for:"
## [1] "Fli1KO_1" "Fli1KO_2" "Fli1KO_3" "Fli1KO_4" "WT_1"     "WT_2"</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Raw gene number: 101921</code></pre>
<pre><code>## Gene number after filtered with counts bigger than 0 : 101921</code></pre>
<pre><code>## Conduct Quality Control on Samples!</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Conduct PCA analysis!
## Conduct PCA analysis!</code></pre>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Use all genes for PCA!</code></pre>
<pre><code>## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will
## replace the existing one.
## Selecting by Loadding
## Selecting by Loadding
## Differential expression analysis with DESeq2!
## 
## Selecting by Loadding
## Selecting by Loadding
## 
## è½½å…¥ç¨‹è¾‘åŒ…ï¼š'BiocGenerics'
## 
## 
## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs
## 
## 
## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min
## 
## 
## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.
## 
## 
## 
## è½½å…¥ç¨‹è¾‘åŒ…ï¼š'S4Vectors'
## 
## 
## The following object is masked from 'package:base':
## 
##     expand.grid
## 
## 
## 
## 
## Conduct Functional Enrichment on PC Loading genes!
## 
## Selecting by Loadding
## Selecting by Loadding
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC1_Positive
## 
## conduct KEGG enrichment analysis: PC1_Positive
## 
## Reading KEGG annotation online: "https://rest.kegg.jp/link/mmu/pathway"...
## 
## Reading KEGG annotation online: "https://rest.kegg.jp/list/pathway/mmu"...
## 
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC1_Negative
## 
## conduct KEGG enrichment analysis: PC1_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC2_Positive
## 
## conduct KEGG enrichment analysis: PC2_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC2_Negative
## 
## conduct KEGG enrichment analysis: PC2_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC3_Positive
## 
## conduct KEGG enrichment analysis: PC3_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC3_Negative
## 
## conduct KEGG enrichment analysis: PC3_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC4_Positive
## 
## conduct KEGG enrichment analysis: PC4_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC4_Negative
## 
## conduct KEGG enrichment analysis: PC4_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC5_Positive
## 
## conduct KEGG enrichment analysis: PC5_Positive
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Convert SYMBOL to ENTREZID!
## 
## 'select()' returned 1:1 mapping between keys and columns
## 
## conduct ALL GO enrichment analysis on: PC5_Negative
## 
## conduct KEGG enrichment analysis: PC5_Negative
## 
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## wrong orderBy parameter; set to default `orderBy = "x"`
## 
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Conduct Differential Expression Analysis!
## 
## estimating size factors
## 
## estimating dispersions
## 
## gene-wise dispersion estimates
## 
## mean-dispersion relationship
## 
## -- note: fitType='parametric', but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType='local' or 'mean' to avoid this message next time.
## 
## final dispersion estimates
## 
## fitting model and testing
## 
## Conduct Gene ID Conversion!
## 
## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in clusterProfiler::bitr(de.df[, "Gene2Convert"], fromType =
## gene.type, : 1.63% of input gene IDs are fail to map...</code></pre>
<pre><code>## Visualize Differential Analysis Results!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Differential expression analysis with DESeq2!
## Conduct Functional Enrichment Analysis!
## Differential expression analysis with DESeq2!
## Convert SYMBOL to ENTREZID!
## 'select()' returned 1:1 mapping between keys and columns
## conduct ALL GO enrichment analysis on: UP
## conduct KEGG enrichment analysis: UP
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.Convert SYMBOL to ENTREZID!
## 'select()' returned 1:1 mapping between keys and columns
## conduct ALL GO enrichment analysis on: DOWN
## conduct KEGG enrichment analysis: DOWN
## wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.wrong orderBy parameter; set to default `orderBy = "x"`
## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.Gene Set Enrichment Analysis is not available for ChIP-seq or ATAC-seq data!
## All Analysis Done!</code></pre>
<p>The structure of result directory:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">tree</span> /home/songyabing/R/learn/tmp/DEbPeak/ATACDE</span></code></pre></div>
<pre><code>## /home/songyabing/R/learn/tmp/DEbPeak/ATACDE
## â”œâ”€â”€ CountQC_CPM.pdf
## â”œâ”€â”€ CountQC_saturation.pdf
## â”œâ”€â”€ DA
## â”‚Â Â  â”œâ”€â”€ Condition_KO_WT_all.csv
## â”‚Â Â  â”œâ”€â”€ Condition_KO_WT_padj0.05_FC0.csv
## â”‚Â Â  â”œâ”€â”€ DA_GenePlot.pdf
## â”‚Â Â  â”œâ”€â”€ DA_Heatmap.pdf
## â”‚Â Â  â”œâ”€â”€ DA_MAPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DA_RankPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DA_ScatterPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DA_VolcanoPlot.pdf
## â”‚Â Â  â””â”€â”€ normalized_counts.csv
## â”œâ”€â”€ DEG
## â”‚Â Â  â”œâ”€â”€ Condition_KO_WT_all.csv
## â”‚Â Â  â”œâ”€â”€ Condition_KO_WT_padj0.05_FC0.csv
## â”‚Â Â  â”œâ”€â”€ DEG_GenePlot.pdf
## â”‚Â Â  â”œâ”€â”€ DEG_Heatmap.pdf
## â”‚Â Â  â”œâ”€â”€ DEG_MAPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DEG_RankPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DEG_ScatterPlot.pdf
## â”‚Â Â  â”œâ”€â”€ DEG_VolcanoPlot.pdf
## â”‚Â Â  â””â”€â”€ normalized_counts.csv
## â”œâ”€â”€ FE
## â”‚Â Â  â”œâ”€â”€ DOWN_ALL_GO.csv
## â”‚Â Â  â”œâ”€â”€ DOWN_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ DOWN_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ DOWN_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ DOWN_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ DOWN_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ DOWN_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ UP_ALL_GO.csv
## â”‚Â Â  â”œâ”€â”€ UP_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ UP_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ UP_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ UP_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ UP_KEGG_Enrichment.png
## â”‚Â Â  â””â”€â”€ UP_Molecular_Function.png
## â”œâ”€â”€ PCA
## â”‚Â Â  â”œâ”€â”€ PC1
## â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Negative
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â”œâ”€â”€ PC1_Negative_ALL_GO.csv
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â””â”€â”€ PC1_Negative_KEGG.csv
## â”‚Â Â  â”‚Â Â  â””â”€â”€ Positive
## â”‚Â Â  â”‚Â Â      â”œâ”€â”€ PC1_Positive_ALL_GO.csv
## â”‚Â Â  â”‚Â Â      â””â”€â”€ PC1_Positive_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ PC1_Negative_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC1_Negative_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC1_Negative_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC1_Negative_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC1_Negative_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC1_Positive_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC1_Positive_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC1_Positive_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC1_Positive_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC1_Positive_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC2
## â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Negative
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â”œâ”€â”€ PC2_Negative_ALL_GO.csv
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â””â”€â”€ PC2_Negative_KEGG.csv
## â”‚Â Â  â”‚Â Â  â””â”€â”€ Positive
## â”‚Â Â  â”‚Â Â      â”œâ”€â”€ PC2_Positive_ALL_GO.csv
## â”‚Â Â  â”‚Â Â      â””â”€â”€ PC2_Positive_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ PC2_Negative_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC2_Negative_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC2_Negative_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC2_Negative_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC2_Negative_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC2_Positive_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC2_Positive_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC2_Positive_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC2_Positive_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC2_Positive_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC3
## â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Negative
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â”œâ”€â”€ PC3_Negative_ALL_GO.csv
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â””â”€â”€ PC3_Negative_KEGG.csv
## â”‚Â Â  â”‚Â Â  â””â”€â”€ Positive
## â”‚Â Â  â”‚Â Â      â”œâ”€â”€ PC3_Positive_ALL_GO.csv
## â”‚Â Â  â”‚Â Â      â””â”€â”€ PC3_Positive_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ PC3_Negative_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC3_Negative_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC3_Negative_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC3_Negative_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC3_Negative_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC3_Positive_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC3_Positive_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC3_Positive_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC3_Positive_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC3_Positive_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC4
## â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Negative
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â”œâ”€â”€ PC4_Negative_ALL_GO.csv
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â””â”€â”€ PC4_Negative_KEGG.csv
## â”‚Â Â  â”‚Â Â  â””â”€â”€ Positive
## â”‚Â Â  â”‚Â Â      â”œâ”€â”€ PC4_Positive_ALL_GO.csv
## â”‚Â Â  â”‚Â Â      â””â”€â”€ PC4_Positive_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ PC4_Negative_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC4_Negative_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC4_Negative_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC4_Negative_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC4_Negative_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC4_Positive_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC4_Positive_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC4_Positive_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC4_Positive_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC4_Positive_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC5
## â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Negative
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â”œâ”€â”€ PC5_Negative_ALL_GO.csv
## â”‚Â Â  â”‚Â Â  â”‚Â Â  â””â”€â”€ PC5_Negative_KEGG.csv
## â”‚Â Â  â”‚Â Â  â””â”€â”€ Positive
## â”‚Â Â  â”‚Â Â      â”œâ”€â”€ PC5_Positive_ALL_GO.csv
## â”‚Â Â  â”‚Â Â      â””â”€â”€ PC5_Positive_KEGG.csv
## â”‚Â Â  â”œâ”€â”€ PC5_Negative_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC5_Negative_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC5_Negative_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC5_Negative_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC5_Negative_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PC5_Positive_Biological_Process.png
## â”‚Â Â  â”œâ”€â”€ PC5_Positive_Cellular_Component.png
## â”‚Â Â  â”œâ”€â”€ PC5_Positive_Functional_Enrichment.pdf
## â”‚Â Â  â”œâ”€â”€ PC5_Positive_KEGG_Enrichment.png
## â”‚Â Â  â”œâ”€â”€ PC5_Positive_Molecular_Function.png
## â”‚Â Â  â”œâ”€â”€ PCA_3DPCA.pdf
## â”‚Â Â  â”œâ”€â”€ PCA_biplot.pdf
## â”‚Â Â  â”œâ”€â”€ PCA_loading_bar.pdf
## â”‚Â Â  â”œâ”€â”€ PCA_loading_heat.pdf
## â”‚Â Â  â”œâ”€â”€ PCA_overview.pdf
## â”‚Â Â  â”œâ”€â”€ PCA_pairs_plot.pdf
## â”‚Â Â  â”œâ”€â”€ PCA_screen_plot.pdf
## â”‚Â Â  â””â”€â”€ PCA_scree_plot.pdf
## â””â”€â”€ SampleQC_dist_pcc.pdf
## 
## 19 directories, 113 files</code></pre>
<p>Read the results:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds.peak.results.ordered</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/ATACDE/DEG/Condition_KO_WT_all.csv"</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dds.peak.results.ordered</span><span class="op">)</span></code></pre></div>
<pre><code>##                                           baseMean log2FoldChange    lfcSE
## chr8:94825359-94825432|Ciapin1|P         12.106577       6.439216 1.527019
## chr3:105628355-105628501|Ddx20|I         11.180004       6.323179 1.523447
## chr1:118440678-118440831|2900060B14Rik|I  7.836836       5.790918 1.569651
## chr1:165368382-165368517|Dcaf6|I          6.140173       5.439287 1.619636
## chr16:59434969-59435053|Gabrr3|I          5.706767       5.348676 1.631421
## chr11:20780454-20780545|Aftph|DI          5.390470       5.269629 1.636609
##                                              stat       pvalue        padj
## chr8:94825359-94825432|Ciapin1|P         4.216854 2.477343e-05 0.001265734
## chr3:105628355-105628501|Ddx20|I         4.150575 3.316417e-05 0.001598853
## chr1:118440678-118440831|2900060B14Rik|I 3.689304 2.248687e-04 0.006892699
## chr1:165368382-165368517|Dcaf6|I         3.358339 7.841226e-04 0.017866061
## chr16:59434969-59435053|Gabrr3|I         3.278539 1.043461e-03 0.021915949
## chr11:20780454-20780545|Aftph|DI         3.219846 1.282594e-03 0.025309971
##                                                 SYMBOL            ENSEMBL
## chr8:94825359-94825432|Ciapin1|P               Ciapin1 ENSMUSG00000031781
## chr3:105628355-105628501|Ddx20|I                 Ddx20 ENSMUSG00000027905
## chr1:118440678-118440831|2900060B14Rik|I 2900060B14Rik ENSMUSG00000107722
## chr1:165368382-165368517|Dcaf6|I                 Dcaf6 ENSMUSG00000026571
## chr16:59434969-59435053|Gabrr3|I                Gabrr3 ENSMUSG00000074991
## chr11:20780454-20780545|Aftph|DI                 Aftph ENSMUSG00000049659
##                                          ENTREZID
## chr8:94825359-94825432|Ciapin1|P           109006
## chr3:105628355-105628501|Ddx20|I            53975
## chr1:118440678-118440831|2900060B14Rik|I    68204
## chr1:165368382-165368517|Dcaf6|I            74106
## chr16:59434969-59435053|Gabrr3|I           328699
## chr11:20780454-20780545|Aftph|DI           216549</code></pre>
<hr>
</div>
<div id="motif-analysis-with-meme" class="section level2">
<h2 class="hasAnchor">
<a href="#motif-analysis-with-meme" class="anchor"></a>Motif analysis with MEME</h2>
<div id="de-novo-motif-discovery-with-streme" class="section level3">
<h3 class="hasAnchor">
<a href="#de-novo-motif-discovery-with-streme" class="anchor"></a><em>de novo</em> motif discovery with STREME</h3>
<p>Here, we will perform <em>de novo</em> motif discovery with STREME. There are <strong>three categories</strong> for users to choose to perform <em>de novo</em> motif discovery: <code>Up_regulated</code>, <code>Down_regulated</code>, <code>Not_regulated</code>. Here, we will use <code>Up_regulated</code> as example.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># this step is time consuming!</span>
<span class="fu"><a href="../reference/MotifDiscovery.html">MotifDiscovery</a></span><span class="op">(</span>diff.peak <span class="op">=</span> <span class="va">dds.peak.results.ordered</span>, peak.motif.key <span class="op">=</span> <span class="st">"Up_regulated"</span>,
               genome <span class="op">=</span> <span class="st">'/path/to/genome.fa'</span>, streme.path <span class="op">=</span> <span class="st">"~/data/software/meme-5.5.2/bin/streme"</span>,
               samtools.path <span class="op">=</span> <span class="st">"/home/songyabing/data/software/samtools-1.17/samtools"</span>,
               out.folder <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/DAdenovo"</span>, show.html <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># after running, there will be a browser pop-up window to show the results.</span></code></pre></div>
<hr>
</div>
<div id="motif-comparison" class="section level3">
<h3 class="hasAnchor">
<a href="#motif-comparison" class="anchor"></a>Motif comparison</h3>
<p>In this step, we will map <em>de novo</em> motifs against motifs provided by users (known motif database).</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">verteb.motifs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"homer_verteb.motifs.meme"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">motif.cmp</span> <span class="op">=</span> <span class="fu"><a href="../reference/MotifCompare.html">MotifCompare</a></span><span class="op">(</span>motif.folder <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/DAdenovo"</span>,
                         known.motif <span class="op">=</span> <span class="va">verteb.motifs</span>,
                         tomtom.path <span class="op">=</span> <span class="st">'~/data/software/meme-5.5.2/bin/tomtom'</span>,
                         out.folder <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/DAdenovo"</span>, show.html <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># after running, there will be a browser pop-up window to show the results.</span></code></pre></div>
<hr>
</div>
</div>
<div id="motif-analysis-with-homer" class="section level2">
<h2 class="hasAnchor">
<a href="#motif-analysis-with-homer" class="anchor"></a>Motif analysis with Homer</h2>
<p>There are <strong>three categories</strong> for users to choose to perform motif enrichment: <code>Up_regulated</code>, <code>Down_regulated</code>, <code>Not_regulated</code>. Here, we will use <code>Up_regulated</code> as example.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># this step is time consuming!</span>
<span class="va">da.peak.motif</span> <span class="op">=</span> <span class="fu"><a href="../reference/MotifEnrich.html">MotifEnrich</a></span><span class="op">(</span>de.res <span class="op">=</span> <span class="va">dds.peak.results.ordered</span>, reg.type <span class="op">=</span> <span class="st">"Up_regulated"</span>,
                            homer.motif.path <span class="op">=</span> <span class="st">'~/anaconda3/bin/findMotifsGenome.pl'</span>,
                            out.folder <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/DAmotif"</span>,
                            other.paras <span class="op">=</span> <span class="st">"-len 8,10,12 -size -100,50 -S 25"</span><span class="op">)</span></code></pre></div>
<p>The results will contains <strong>de novo</strong> motif results and known motif enrichment analysis results.</p>
<hr>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-conda-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /home/softwares/anaconda3/envs/r4.0/lib/libopenblasp-r0.3.12.so
## 
## locale:
##  [1] LC_CTYPE=zh_CN.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=zh_CN.UTF-8        LC_COLLATE=zh_CN.UTF-8    
##  [5] LC_MONETARY=zh_CN.UTF-8    LC_MESSAGES=zh_CN.UTF-8   
##  [7] LC_PAPER=zh_CN.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
## [1] org.Mm.eg.db_3.12.0  AnnotationDbi_1.52.0 IRanges_2.24.1      
## [4] S4Vectors_0.28.1     Biobase_2.50.0       BiocGenerics_0.42.0 
## [7] DEbPeak_1.4.0       
## 
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                             
##   [2] ggvenn_0.1.9                           
##   [3] apeglm_1.12.0                          
##   [4] Rsamtools_2.6.0                        
##   [5] rsvg_2.1                               
##   [6] foreach_1.5.1                          
##   [7] rprojroot_2.0.2                        
##   [8] crayon_1.4.1                           
##   [9] V8_3.4.2                               
##  [10] MASS_7.3-58                            
##  [11] nlme_3.1-152                           
##  [12] backports_1.2.1                        
##  [13] sva_3.38.0                             
##  [14] GOSemSim_2.25.0                        
##  [15] rlang_1.1.0                            
##  [16] XVector_0.30.0                         
##  [17] readxl_1.4.2                           
##  [18] irlba_2.3.5                            
##  [19] limma_3.46.0                           
##  [20] GOstats_2.56.0                         
##  [21] BiocParallel_1.24.1                    
##  [22] rjson_0.2.20                           
##  [23] bit64_4.0.5                            
##  [24] glue_1.6.2                             
##  [25] DiffBind_3.0.15                        
##  [26] mixsqp_0.3-43                          
##  [27] pheatmap_1.0.12                        
##  [28] parallel_4.0.3                         
##  [29] DEFormats_1.18.0                       
##  [30] base64url_1.4                          
##  [31] tcltk_4.0.3                            
##  [32] DOSE_3.23.2                            
##  [33] haven_2.5.2                            
##  [34] tidyselect_1.2.0                       
##  [35] SummarizedExperiment_1.20.0            
##  [36] rio_0.5.27                             
##  [37] XML_3.99-0.6                           
##  [38] tidyr_1.3.0                            
##  [39] ggpubr_0.4.0                           
##  [40] GenomicAlignments_1.26.0               
##  [41] xtable_1.8-4                           
##  [42] ggnetwork_0.5.12                       
##  [43] magrittr_2.0.3                         
##  [44] evaluate_0.14                          
##  [45] ggplot2_3.4.2                          
##  [46] cli_3.6.1                              
##  [47] zlibbioc_1.36.0                        
##  [48] hwriter_1.3.2                          
##  [49] rstudioapi_0.14                        
##  [50] bslib_0.3.1                            
##  [51] GreyListChIP_1.22.0                    
##  [52] fastmatch_1.1-3                        
##  [53] BiocSingular_1.6.0                     
##  [54] xfun_0.30                              
##  [55] askpass_1.1                            
##  [56] clue_0.3-59                            
##  [57] gson_0.0.9                             
##  [58] cluster_2.1.1                          
##  [59] caTools_1.18.2                         
##  [60] tidygraph_1.2.0                        
##  [61] tibble_3.2.1                           
##  [62] ggrepel_0.9.1                          
##  [63] Biostrings_2.58.0                      
##  [64] png_0.1-7                              
##  [65] withr_2.5.0                            
##  [66] bitops_1.0-6                           
##  [67] ggforce_0.3.3                          
##  [68] RBGL_1.66.0                            
##  [69] plyr_1.8.6                             
##  [70] cellranger_1.1.0                       
##  [71] GSEABase_1.52.1                        
##  [72] pcaPP_2.0-1                            
##  [73] dqrng_0.2.1                            
##  [74] coda_0.19-4                            
##  [75] pillar_1.9.0                           
##  [76] gplots_3.1.1                           
##  [77] GlobalOptions_0.1.2                    
##  [78] cachem_1.0.4                           
##  [79] GenomicFeatures_1.42.2                 
##  [80] fs_1.5.0                               
##  [81] GetoptLong_1.0.5                       
##  [82] clusterProfiler_4.7.1                  
##  [83] DelayedMatrixStats_1.12.3              
##  [84] vctrs_0.6.2                            
##  [85] generics_0.1.0                         
##  [86] plot3D_1.4                             
##  [87] tools_4.0.3                            
##  [88] foreign_0.8-81                         
##  [89] NOISeq_2.34.0                          
##  [90] munsell_0.5.0                          
##  [91] tweenr_1.0.2                           
##  [92] fgsea_1.16.0                           
##  [93] DelayedArray_0.16.3                    
##  [94] fastmap_1.1.0                          
##  [95] compiler_4.0.3                         
##  [96] abind_1.4-5                            
##  [97] rtracklayer_1.50.0                     
##  [98] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
##  [99] GenomeInfoDbData_1.2.4                 
## [100] gridExtra_2.3                          
## [101] edgeR_3.32.1                           
## [102] lattice_0.20-45                        
## [103] ggnewscale_0.4.7                       
## [104] AnnotationForge_1.32.0                 
## [105] utf8_1.2.1                             
## [106] dplyr_1.1.2                            
## [107] BiocFileCache_1.14.0                   
## [108] jsonlite_1.8.4                         
## [109] scales_1.2.1                           
## [110] graph_1.68.0                           
## [111] carData_3.0-4                          
## [112] sparseMatrixStats_1.2.1                
## [113] TFEA.ChIP_1.10.0                       
## [114] genefilter_1.72.1                      
## [115] car_3.0-11                             
## [116] doParallel_1.0.16                      
## [117] latticeExtra_0.6-29                    
## [118] R.utils_2.12.0                         
## [119] brew_1.0-6                             
## [120] checkmate_2.0.0                        
## [121] rmarkdown_2.14                         
## [122] openxlsx_4.2.3                         
## [123] pkgdown_1.6.1                          
## [124] cowplot_1.1.1                          
## [125] textshaping_0.3.6                      
## [126] forcats_1.0.0                          
## [127] downloader_0.4                         
## [128] BSgenome_1.58.0                        
## [129] igraph_1.4.99.9024                     
## [130] survival_3.2-10                        
## [131] numDeriv_2016.8-1.1                    
## [132] yaml_2.2.1                             
## [133] plotrix_3.8-2                          
## [134] systemfonts_1.0.4                      
## [135] ashr_2.2-47                            
## [136] SQUAREM_2021.1                         
## [137] htmltools_0.5.2                        
## [138] memoise_2.0.0                          
## [139] VariantAnnotation_1.36.0               
## [140] locfit_1.5-9.4                         
## [141] graphlayouts_0.7.1                     
## [142] batchtools_0.9.15                      
## [143] PCAtools_2.2.0                         
## [144] viridisLite_0.4.0                      
## [145] rrcov_1.7-0                            
## [146] digest_0.6.27                          
## [147] assertthat_0.2.1                       
## [148] rappdirs_0.3.3                         
## [149] emdbook_1.3.12                         
## [150] RSQLite_2.2.5                          
## [151] amap_0.8-18                            
## [152] yulab.utils_0.0.4                      
## [153] debugme_1.1.0                          
## [154] misc3d_0.9-1                           
## [155] data.table_1.14.2                      
## [156] blob_1.2.1                             
## [157] R.oo_1.24.0                            
## [158] ragg_0.4.0                             
## [159] labeling_0.4.2                         
## [160] splines_4.0.3                          
## [161] Cairo_1.5-12.2                         
## [162] ggupset_0.3.0                          
## [163] RCurl_1.98-1.3                         
## [164] broom_1.0.4                            
## [165] hms_1.1.3                              
## [166] colorspace_2.0-0                       
## [167] BiocManager_1.30.16                    
## [168] GenomicRanges_1.42.0                   
## [169] shape_1.4.6                            
## [170] sass_0.4.1                             
## [171] GEOquery_2.58.0                        
## [172] Rcpp_1.0.9                             
## [173] mvtnorm_1.1-2                          
## [174] circlize_0.4.15                        
## [175] enrichplot_1.10.2                      
## [176] fansi_0.4.2                            
## [177] tzdb_0.3.0                             
## [178] truncnorm_1.0-8                        
## [179] ChIPseeker_1.33.0.900                  
## [180] R6_2.5.0                               
## [181] grid_4.0.3                             
## [182] lifecycle_1.0.3                        
## [183] ShortRead_1.48.0                       
## [184] zip_2.1.1                              
## [185] curl_4.3                               
## [186] ggsignif_0.6.3                         
## [187] jquerylib_0.1.3                        
## [188] robustbase_0.95-0                      
## [189] DO.db_2.9                              
## [190] Matrix_1.5-4                           
## [191] qvalue_2.22.0                          
## [192] desc_1.3.0                             
## [193] org.Hs.eg.db_3.12.0                    
## [194] RColorBrewer_1.1-2                     
## [195] iterators_1.0.13                       
## [196] stringr_1.5.0                          
## [197] DOT_0.1                                
## [198] ggpie_0.2.5                            
## [199] beachmat_2.6.4                         
## [200] polyclip_1.10-0                        
## [201] biomaRt_2.46.3                         
## [202] purrr_1.0.1                            
## [203] shadowtext_0.0.9                       
## [204] gridGraphics_0.5-1                     
## [205] mgcv_1.8-34                            
## [206] ComplexHeatmap_2.13.1                  
## [207] openssl_1.4.3                          
## [208] patchwork_1.0.0                        
## [209] bdsmatrix_1.3-4                        
## [210] codetools_0.2-18                       
## [211] matrixStats_0.58.0                     
## [212] invgamma_1.1                           
## [213] GO.db_3.12.1                           
## [214] gtools_3.8.2                           
## [215] prettyunits_1.1.1                      
## [216] dbplyr_2.3.2                           
## [217] R.methodsS3_1.8.1                      
## [218] GenomeInfoDb_1.26.7                    
## [219] gtable_0.3.0                           
## [220] DBI_1.1.1                              
## [221] ggfun_0.0.6                            
## [222] httr_1.4.5                             
## [223] KernSmooth_2.23-18                     
## [224] stringi_1.5.3                          
## [225] progress_1.2.2                         
## [226] reshape2_1.4.4                         
## [227] farver_2.1.0                           
## [228] annotate_1.68.0                        
## [229] viridis_0.6.1                          
## [230] Rgraphviz_2.34.0                       
## [231] xml2_1.3.4                             
## [232] bbmle_1.0.24                           
## [233] systemPipeR_1.24.3                     
## [234] boot_1.3-28                            
## [235] readr_2.1.4                            
## [236] geneplotter_1.68.0                     
## [237] ggplotify_0.1.0                        
## [238] Category_2.56.0                        
## [239] DEoptimR_1.0-11                        
## [240] DESeq2_1.30.1                          
## [241] bit_4.0.4                              
## [242] scatterpie_0.1.7                       
## [243] jpeg_0.1-8.1                           
## [244] MatrixGenerics_1.2.1                   
## [245] ggraph_2.0.5                           
## [246] pkgconfig_2.0.3                        
## [247] rstatix_0.7.0                          
## [248] knitr_1.37</code></pre>
<hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
