<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>PrincipalComponentAnalysis • DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="PrincipalComponentAnalysis">
<meta property="og:description" content="DEbPeak">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="PrincipalComponentAnalysis_files/header-attrs-2.14/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>PrincipalComponentAnalysis</h1>
            
      
      
      <div class="hidden name"><code>PrincipalComponentAnalysis.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Principal component analysis (PCA) is an unsupervised linear dimensionality reduction algorithm, and it can <a href="https://www.nature.com/articles/nbt0308-303">retain most of the variation in the data set</a>. Based on these properties, PCA has two main uses in RNA-seq analysis:</p>
<ul>
<li>Quality control:
<ul>
<li>Evaluate sample similarity across different conditions</li>
<li>Check batch information</li>
<li>Detect outliers</li>
</ul>
</li>
<li>Reveal biologically relevant information
<ul>
<li>Find gene expression signatures (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5317117/">ref1</a>, <a href="https://link.springer.com/article/10.1007/s13238-019-0648-6">ref2</a>, <a href="https://www.biorxiv.org/content/10.1101/2020.10.26.355750v3.full">ref3</a>)</li>
<li>Reveal underlying population heterogeneity, such as cell differentiation trajectory (<a href="https://insight.jci.org/articles/view/159093">ref1</a>, <a href="https://www.pnas.org/doi/10.1073/pnas.1908180116">ref2</a>).</li>
</ul>
</li>
</ul>
<p>For Quality Control, We have already covered in the previous vignette, here, we will focus on the “Reveal biologically relevant information”!</p>
<hr>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>This vignette follows the previous “Quality Control” vignette, so we will used the results output by “Quality Control” vignette.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DEbPeak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co"># load data</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/example.RData"</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="pca" class="section level2">
<h2 class="hasAnchor">
<a href="#pca" class="anchor"></a>PCA</h2>
<p>We will perform PCA with <code>prcomp</code>, and remove outlier samples detected by previous <code>QCPCA</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># conduct PCA</span>
<span class="va">pca.res</span><span class="op">=</span><span class="fu"><a href="../reference/PCA.html">PCA</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,remove.sample <span class="op">=</span> <span class="va">outlier.res</span><span class="op">$</span><span class="va">outlier</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span><span class="op">)</span>
<span class="co"># get basic plots</span>
<span class="va">basic.plots</span><span class="op">=</span><span class="fu"><a href="../reference/PCABasic.html">PCABasic</a></span><span class="op">(</span><span class="va">pca.res</span>,colby<span class="op">=</span><span class="st">"condition"</span>,legend.pos <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></code></pre></div>
<div id="scree-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#scree-plot" class="anchor"></a>Scree plot</h3>
<p>A Scree Plot is a simple line plot that shows the total amount of variance that can be explained by each individual PC (Y-axis shows explained variance, X-axis shows the number of PCs). It can be used to <strong>determine the number of PCs to be explored for downstream analysis</strong>. Here, we created a <strong>cumulative scree plot</strong> based on <a href="https://github.com/kevinblighe/PCAtools">PCAtools</a>, the red dashed line represents 90% explained variance.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.plots</span><span class="op">[[</span><span class="st">"screen"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="PrincipalComponentAnalysis_files/figure-html/pca_basic_screen-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="biplot" class="section level3">
<h3 class="hasAnchor">
<a href="#biplot" class="anchor"></a>Biplot</h3>
<p>Biplot contains informations from <a href="https://blog.bioturing.com/2018/06/18/how-to-read-pca-biplots-and-scree-plots/">two aspects</a>:</p>
<ul>
<li>sample similarity (point)</li>
<li>how strongly each gene influences a principal component (vector)
<ul>
<li>the vector will have projections on PCs, and the project values on each PC show how much weight they have on that PC. Positive values indicate that a vector/gene and a principal component are positively correlated whereas negative values indicate a negative correlation.</li>
<li>the angle between vectors indicates how vectors/genes correlate with one another: a small angle implies positive correlation, a large one suggests negative correlation, and a 90 degree angle indicates no correlation between two vectors/genes.</li>
</ul>
</li>
</ul>
<p>The biplot is created based on <a href="https://github.com/kevinblighe/PCAtools">PCAtools</a>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.plots</span><span class="op">[[</span><span class="st">"biplot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="PrincipalComponentAnalysis_files/figure-html/pca_basic_biplot-1.png" width="480" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="pc-pairs-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#pc-pairs-plot" class="anchor"></a>PC pairs plot</h3>
<p>The PC pairs plot (based on <a href="https://github.com/kevinblighe/PCAtools">PCAtools</a>) will show sample similarity across different PC combination:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.plots</span><span class="op">[[</span><span class="st">"pairs"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="PrincipalComponentAnalysis_files/figure-html/pca_basic_pairs-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="lodding-related-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#lodding-related-analysis" class="anchor"></a>Lodding-related analysis</h2>
<p>PCA loadings are the coefficients of the linear combination of the original variables from which the principal components (PCs) are constructed, which are simply the weights of the original variables. Here, lodding indicates how strongly each gene influences a principal component (Similar to vector in <code>Biplot</code>, but here we will evaluate each individual PC). Loadings range from -1 to 1, a high absolute value indicates that the gene strongly influences the component, and the sign of a loading indicates whether a whether a gene and a principal component are positively or negatively correlated.</p>
<p>The lodding-related analysis in <code>DEbPeak</code> contains four aspects:</p>
<ul>
<li>
<strong>Loading bar plot</strong>: Visualize the genes with top n positive or negative loaddings of PCs</li>
<li>
<strong>Loading heatmap</strong>: Log2-normalized counts of the genes with top n positive or negative loaddings of PCs across samples</li>
<li>
<strong>Export genes</strong>: Export the genes with top n positive or negative loaddings of PCs</li>
<li>
<strong>GO enrichment</strong>: GO enrichment on the genes with top n positive or negative loaddings of PCs</li>
</ul>
<div id="loading-bar-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-bar-plot" class="anchor"></a>Loading bar plot</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># loading bar plot</span>
<span class="fu"><a href="../reference/LoadingPlot.html">LoadingPlot</a></span><span class="op">(</span><span class="va">pca.res</span>,type <span class="op">=</span> <span class="st">"bar"</span><span class="op">)</span></code></pre></div>
<p><img src="PrincipalComponentAnalysis_files/figure-html/pca_loading_bar-1.png" width="960" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="loading-heatmap" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-heatmap" class="anchor"></a>Loading heatmap</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># loading heatmap</span>
<span class="fu"><a href="../reference/LoadingPlot.html">LoadingPlot</a></span><span class="op">(</span><span class="va">pca.res</span>,deobj <span class="op">=</span> <span class="va">dds</span>,type <span class="op">=</span> <span class="st">"heat"</span><span class="op">)</span></code></pre></div>
<p><img src="PrincipalComponentAnalysis_files/figure-html/pca_loading_heatmap-1.png" width="960" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="export-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#export-genes" class="anchor"></a>Export genes</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># export loading genes on PC1 and PC2 (200 positive genes and 200 negative genes) </span>
<span class="va">loading.gene.df</span><span class="op">=</span><span class="fu"><a href="../reference/ExportPCGenes.html">ExportPCGenes</a></span><span class="op">(</span>pca <span class="op">=</span> <span class="va">pca.res</span>, pc <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, gene.num <span class="op">=</span> <span class="fl">200</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">loading.gene.df</span><span class="op">)</span></code></pre></div>
<pre><code>##    PC                  Gene   Loadding     Type
## 1 PC1 ENSMUSG00000038252.13 0.02543092 Positive
## 2 PC1 ENSMUSG00000027496.15 0.02457406 Positive
## 3 PC1 ENSMUSG00000030795.18 0.02395752 Positive
## 4 PC1 ENSMUSG00000031865.16 0.02373083 Positive
## 5 PC1 ENSMUSG00000027404.15 0.02319317 Positive
## 6 PC1 ENSMUSG00000063511.11 0.02220398 Positive</code></pre>
<hr>
</div>
<div id="go-enrichment" class="section level3">
<h3 class="hasAnchor">
<a href="#go-enrichment" class="anchor"></a>GO enrichment</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># save results to working folder</span>
<span class="co"># LoadingGO(pca.res,gene.type="ENSEMBL",go.type="BP",padj.method="BH",save = T)</span>
<span class="co"># return list</span>
<span class="va">loading.go.results</span><span class="op">=</span><span class="fu"><a href="../reference/LoadingGO.html">LoadingGO</a></span><span class="op">(</span><span class="va">pca.res</span>,gene.type<span class="op">=</span><span class="st">"ENSEMBL"</span>,go.type<span class="op">=</span><span class="st">"BP"</span>,padj.method<span class="op">=</span><span class="st">"BH"</span>,str.width <span class="op">=</span> <span class="fl">50</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<hr>
<div id="genes-with-positive-loaddings" class="section level4">
<h4 class="hasAnchor">
<a href="#genes-with-positive-loaddings" class="anchor"></a>Genes with positive loaddings</h4>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># positive loading genes</span>
<span class="va">positive_go_results</span><span class="op">=</span><span class="va">loading.go.results</span><span class="op">[[</span><span class="st">"Positive"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">positive_go_results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##                    ID                                       Description
## GO:0022613 GO:0022613              ribonucleoprotein complex biogenesis
## GO:0008380 GO:0008380                                      RNA splicing
## GO:1903311 GO:1903311              regulation of mRNA metabolic process
## GO:0031331 GO:0031331 positive regulation of cellular catabolic process
## GO:0042254 GO:0042254                               ribosome biogenesis
## GO:0006397 GO:0006397                                   mRNA processing
##            GeneRatio   BgRatio       pvalue     p.adjust       qvalue
## GO:0022613    19/194 409/23328 1.667451e-09 4.440423e-06 3.787747e-06
## GO:0008380    17/194 386/23328 2.733412e-08 3.639538e-05 3.104580e-05
## GO:1903311    13/194 266/23328 3.862626e-07 3.428724e-04 2.924753e-04
## GO:0031331    15/194 396/23328 1.243752e-06 6.850558e-04 5.843628e-04
## GO:0042254    13/194 296/23328 1.286248e-06 6.850558e-04 5.843628e-04
## GO:0006397    16/194 462/23328 1.753069e-06 7.115007e-04 6.069208e-04
##                                                                                                                         geneID
## GO:0022613 Snrpb/Sf3a2/Hsp90ab1/Mybbp1a/Ruvbl2/Ddx39b/Ddx47/Ppan/Rcl1/Drosha/Rpl6/Puf60/Ddx54/Ddx27/Mrpl10/Ran/Noc2l/Abt1/Nop2
## GO:0008380              Fus/Snrpb/Snrnp70/Sf3a2/Srsf4/Ddx39b/Khsrp/Scaf1/Ddx47/Txnl4a/Snrpa/Snrpn/Puf60/U2af2/Taf15/Mettl3/Sf1
## GO:1903311                                          Fus/Snrnp70/Srsf4/Khsrp/Upf1/Fto/Supt5/Puf60/U2af2/Taf15/Mettl3/Gtpbp1/Sf1
## GO:0031331                            Aurka/Khsrp/Upf1/Fto/Cdc20/Supt5/Fzr1/Bag6/Mettl3/Bcap31/Rnf185/Gtpbp1/Psen1/Hmox1/Gsk3a
## GO:0042254                                          Mybbp1a/Ddx47/Ppan/Rcl1/Drosha/Rpl6/Ddx54/Ddx27/Mrpl10/Ran/Noc2l/Abt1/Nop2
## GO:0006397                  Snrpb/Snrnp70/Sf3a2/Srsf4/Ddx39b/Khsrp/Scaf1/Ddx47/Txnl4a/Snrpa/Supt5/Snrpn/Puf60/U2af2/Mettl3/Sf1
##            Count
## GO:0022613    19
## GO:0008380    17
## GO:1903311    13
## GO:0031331    15
## GO:0042254    13
## GO:0006397    16</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">positive_go_results</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="PrincipalComponentAnalysis_files/figure-html/loading_go_positive-1.png" width="672" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="genes-with-negative-loaddings" class="section level4">
<h4 class="hasAnchor">
<a href="#genes-with-negative-loaddings" class="anchor"></a>Genes with negative loaddings</h4>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># negative loading genes</span>
<span class="va">negative_go_results</span><span class="op">=</span><span class="va">loading.go.results</span><span class="op">[[</span><span class="st">"Negative"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">negative_go_results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##                    ID
## GO:0032535 GO:0032535
## GO:0045666 GO:0045666
## GO:0031346 GO:0031346
## GO:0010976 GO:0010976
## GO:0048813 GO:0048813
## GO:0010769 GO:0010769
##                                                             Description
## GO:0032535                        regulation of cellular component size
## GO:0045666                positive regulation of neuron differentiation
## GO:0031346          positive regulation of cell projection organization
## GO:0010976         positive regulation of neuron projection development
## GO:0048813                                       dendrite morphogenesis
## GO:0010769 regulation of cell morphogenesis involved in differentiation
##            GeneRatio   BgRatio       pvalue    p.adjust      qvalue
## GO:0032535    14/186 413/23328 6.228504e-06 0.006598163 0.005761002
## GO:0045666    15/186 472/23328 6.270266e-06 0.006598163 0.005761002
## GO:0031346    15/186 478/23328 7.290788e-06 0.006598163 0.005761002
## GO:0010976    13/186 372/23328 9.751787e-06 0.006619025 0.005779217
## GO:0048813     9/186 182/23328 1.622507e-05 0.008810210 0.007692389
## GO:0010769    12/186 351/23328 2.677346e-05 0.012114991 0.010577866
##                                                                                            geneID
## GO:0032535    Bmpr2/Map3k1/Sema3d/Dpysl2/Aqp4/Dcx/Pten/Nefl/Arhgap5/Pak3/Wnk3/Cracd/Map2/Tmsb15b1
## GO:0045666 Bmpr2/Cntn1/Fut9/Neurod1/Ube2v2/Cnr1/Dcx/Ptprz1/Tmem106b/Ror1/Pten/Nefl/Pak3/Qk/Elavl4
## GO:0031346   Bmpr2/Dock11/Cntn1/Fut9/Ube2v2/Cnr1/Dcx/Apc/Ptprz1/Tmem106b/Ror1/Nefl/Pak3/Qk/Elavl4
## GO:0010976              Bmpr2/Cntn1/Fut9/Ube2v2/Cnr1/Dcx/Ptprz1/Tmem106b/Ror1/Nefl/Pak3/Qk/Elavl4
## GO:0048813                                Ube3a/Hecw1/Adam10/Dcx/Ptprz1/Tmem106b/Pten/Pak3/Elavl4
## GO:0010769                  Bmpr2/Ube3a/Sema3d/Dpysl2/Hecw1/Adam10/Dcx/Ptprz1/Pten/Nefl/Pak3/Map2
##            Count
## GO:0032535    14
## GO:0045666    15
## GO:0031346    15
## GO:0010976    13
## GO:0048813     9
## GO:0010769    12</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">negative_go_results</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="PrincipalComponentAnalysis_files/figure-html/loading_go_negative-1.png" width="672" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
<div id="d-visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#d-visualization" class="anchor"></a>3D visualization</h2>
<p>To visualize three PCs simultaneously, <code>DEbPeak</code> provides <code>PCA3D</code> to create 3D PCA plot:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PCA3D.html">PCA3D</a></span><span class="op">(</span>pca <span class="op">=</span> <span class="va">pca.res</span>,color.key <span class="op">=</span> <span class="st">"condition"</span>,main <span class="op">=</span> <span class="st">"3D PCA"</span><span class="op">)</span></code></pre></div>
<p><img src="PrincipalComponentAnalysis_files/figure-html/pca_3d-1.png" width="480" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-conda-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /home/softwares/anaconda3/envs/r4.0/lib/libopenblasp-r0.3.12.so
## 
## locale:
##  [1] LC_CTYPE=zh_CN.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=zh_CN.UTF-8        LC_COLLATE=zh_CN.UTF-8    
##  [5] LC_MONETARY=zh_CN.UTF-8    LC_MESSAGES=zh_CN.UTF-8   
##  [7] LC_PAPER=zh_CN.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
## [1] org.Mm.eg.db_3.12.0  AnnotationDbi_1.52.0 IRanges_2.24.1      
## [4] S4Vectors_0.28.1     Biobase_2.50.0       BiocGenerics_0.42.0 
## [7] DEbPeak_0.9.0       
## 
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                             
##   [2] ggvenn_0.1.9                           
##   [3] apeglm_1.12.0                          
##   [4] Rsamtools_2.6.0                        
##   [5] rsvg_2.1                               
##   [6] foreach_1.5.1                          
##   [7] rprojroot_2.0.2                        
##   [8] crayon_1.4.1                           
##   [9] V8_3.4.2                               
##  [10] MASS_7.3-53.1                          
##  [11] nlme_3.1-152                           
##  [12] backports_1.2.1                        
##  [13] sva_3.38.0                             
##  [14] GOSemSim_2.25.0                        
##  [15] rlang_1.0.3                            
##  [16] XVector_0.30.0                         
##  [17] readxl_1.3.1                           
##  [18] irlba_2.3.5                            
##  [19] limma_3.46.0                           
##  [20] GOstats_2.56.0                         
##  [21] BiocParallel_1.24.1                    
##  [22] rjson_0.2.20                           
##  [23] bit64_4.0.5                            
##  [24] glue_1.6.2                             
##  [25] DiffBind_3.0.15                        
##  [26] mixsqp_0.3-43                          
##  [27] pheatmap_1.0.12                        
##  [28] parallel_4.0.3                         
##  [29] DEFormats_1.18.0                       
##  [30] base64url_1.4                          
##  [31] tcltk_4.0.3                            
##  [32] DOSE_3.23.2                            
##  [33] haven_2.3.1                            
##  [34] tidyselect_1.1.0                       
##  [35] SummarizedExperiment_1.20.0            
##  [36] rio_0.5.27                             
##  [37] XML_3.99-0.6                           
##  [38] tidyr_1.1.3                            
##  [39] ggpubr_0.4.0                           
##  [40] GenomicAlignments_1.26.0               
##  [41] xtable_1.8-4                           
##  [42] magrittr_2.0.1                         
##  [43] evaluate_0.14                          
##  [44] ggplot2_3.3.5                          
##  [45] cli_3.3.0                              
##  [46] zlibbioc_1.36.0                        
##  [47] hwriter_1.3.2                          
##  [48] rstudioapi_0.13                        
##  [49] bslib_0.3.1                            
##  [50] GreyListChIP_1.22.0                    
##  [51] fastmatch_1.1-3                        
##  [52] BiocSingular_1.6.0                     
##  [53] xfun_0.30                              
##  [54] askpass_1.1                            
##  [55] clue_0.3-59                            
##  [56] gson_0.0.9                             
##  [57] cluster_2.1.1                          
##  [58] caTools_1.18.2                         
##  [59] tidygraph_1.2.0                        
##  [60] tibble_3.1.0                           
##  [61] ggrepel_0.9.1                          
##  [62] Biostrings_2.58.0                      
##  [63] png_0.1-7                              
##  [64] withr_2.4.1                            
##  [65] bitops_1.0-6                           
##  [66] ggforce_0.3.3                          
##  [67] RBGL_1.66.0                            
##  [68] plyr_1.8.6                             
##  [69] cellranger_1.1.0                       
##  [70] GSEABase_1.52.1                        
##  [71] pcaPP_2.0-1                            
##  [72] dqrng_0.2.1                            
##  [73] coda_0.19-4                            
##  [74] pillar_1.5.1                           
##  [75] gplots_3.1.1                           
##  [76] GlobalOptions_0.1.2                    
##  [77] cachem_1.0.4                           
##  [78] GenomicFeatures_1.42.2                 
##  [79] fs_1.5.0                               
##  [80] GetoptLong_1.0.5                       
##  [81] clusterProfiler_4.7.1                  
##  [82] DelayedMatrixStats_1.12.3              
##  [83] vctrs_0.4.1                            
##  [84] ellipsis_0.3.2                         
##  [85] generics_0.1.0                         
##  [86] plot3D_1.4                             
##  [87] tools_4.0.3                            
##  [88] foreign_0.8-81                         
##  [89] NOISeq_2.34.0                          
##  [90] munsell_0.5.0                          
##  [91] tweenr_1.0.2                           
##  [92] fgsea_1.16.0                           
##  [93] DelayedArray_0.16.3                    
##  [94] fastmap_1.1.0                          
##  [95] compiler_4.0.3                         
##  [96] abind_1.4-5                            
##  [97] rtracklayer_1.50.0                     
##  [98] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
##  [99] GenomeInfoDbData_1.2.4                 
## [100] gridExtra_2.3                          
## [101] edgeR_3.32.1                           
## [102] lattice_0.20-45                        
## [103] ggnewscale_0.4.7                       
## [104] AnnotationForge_1.32.0                 
## [105] utf8_1.2.1                             
## [106] dplyr_1.0.5                            
## [107] BiocFileCache_1.14.0                   
## [108] jsonlite_1.7.2                         
## [109] scales_1.1.1                           
## [110] graph_1.68.0                           
## [111] carData_3.0-4                          
## [112] sparseMatrixStats_1.2.1                
## [113] genefilter_1.72.1                      
## [114] car_3.0-11                             
## [115] doParallel_1.0.16                      
## [116] latticeExtra_0.6-29                    
## [117] brew_1.0-6                             
## [118] checkmate_2.0.0                        
## [119] rmarkdown_2.14                         
## [120] openxlsx_4.2.3                         
## [121] pkgdown_1.6.1                          
## [122] cowplot_1.1.1                          
## [123] textshaping_0.1.2                      
## [124] forcats_0.5.1                          
## [125] downloader_0.4                         
## [126] BSgenome_1.58.0                        
## [127] igraph_1.3.4                           
## [128] survival_3.2-10                        
## [129] numDeriv_2016.8-1.1                    
## [130] yaml_2.2.1                             
## [131] plotrix_3.8-2                          
## [132] systemfonts_1.0.1                      
## [133] ashr_2.2-47                            
## [134] SQUAREM_2021.1                         
## [135] htmltools_0.5.2                        
## [136] memoise_2.0.0                          
## [137] VariantAnnotation_1.36.0               
## [138] locfit_1.5-9.4                         
## [139] graphlayouts_0.7.1                     
## [140] batchtools_0.9.15                      
## [141] PCAtools_2.2.0                         
## [142] viridisLite_0.4.0                      
## [143] digest_0.6.27                          
## [144] rrcov_1.7-0                            
## [145] assertthat_0.2.1                       
## [146] rappdirs_0.3.3                         
## [147] emdbook_1.3.12                         
## [148] RSQLite_2.2.5                          
## [149] amap_0.8-18                            
## [150] yulab.utils_0.0.4                      
## [151] misc3d_0.9-1                           
## [152] data.table_1.14.2                      
## [153] blob_1.2.1                             
## [154] ragg_0.4.0                             
## [155] labeling_0.4.2                         
## [156] splines_4.0.3                          
## [157] Cairo_1.5-12.2                         
## [158] ggupset_0.3.0                          
## [159] RCurl_1.98-1.3                         
## [160] broom_0.7.5                            
## [161] hms_1.0.0                              
## [162] colorspace_2.0-0                       
## [163] BiocManager_1.30.16                    
## [164] GenomicRanges_1.42.0                   
## [165] shape_1.4.6                            
## [166] sass_0.4.1                             
## [167] Rcpp_1.0.9                             
## [168] mvtnorm_1.1-2                          
## [169] circlize_0.4.15                        
## [170] enrichplot_1.10.2                      
## [171] fansi_0.4.2                            
## [172] truncnorm_1.0-8                        
## [173] ChIPseeker_1.33.0.900                  
## [174] R6_2.5.0                               
## [175] grid_4.0.3                             
## [176] lifecycle_1.0.0                        
## [177] ShortRead_1.48.0                       
## [178] zip_2.1.1                              
## [179] curl_4.3                               
## [180] ggsignif_0.6.3                         
## [181] jquerylib_0.1.3                        
## [182] robustbase_0.95-0                      
## [183] DO.db_2.9                              
## [184] Matrix_1.3-3                           
## [185] qvalue_2.22.0                          
## [186] desc_1.3.0                             
## [187] RColorBrewer_1.1-2                     
## [188] iterators_1.0.13                       
## [189] stringr_1.4.0                          
## [190] DOT_0.1                                
## [191] ggpie_0.2.1                            
## [192] beachmat_2.6.4                         
## [193] polyclip_1.10-0                        
## [194] biomaRt_2.46.3                         
## [195] purrr_0.3.4                            
## [196] shadowtext_0.0.9                       
## [197] gridGraphics_0.5-1                     
## [198] mgcv_1.8-34                            
## [199] ComplexHeatmap_2.13.1                  
## [200] openssl_1.4.3                          
## [201] patchwork_1.0.0                        
## [202] bdsmatrix_1.3-4                        
## [203] codetools_0.2-18                       
## [204] matrixStats_0.58.0                     
## [205] invgamma_1.1                           
## [206] GO.db_3.12.1                           
## [207] gtools_3.8.2                           
## [208] prettyunits_1.1.1                      
## [209] dbplyr_2.1.1                           
## [210] GenomeInfoDb_1.26.7                    
## [211] gtable_0.3.0                           
## [212] DBI_1.1.1                              
## [213] highr_0.8                              
## [214] ggfun_0.0.6                            
## [215] httr_1.4.2                             
## [216] KernSmooth_2.23-18                     
## [217] stringi_1.5.3                          
## [218] progress_1.2.2                         
## [219] reshape2_1.4.4                         
## [220] farver_2.1.0                           
## [221] annotate_1.68.0                        
## [222] viridis_0.6.1                          
## [223] Rgraphviz_2.34.0                       
## [224] xml2_1.3.2                             
## [225] bbmle_1.0.24                           
## [226] systemPipeR_1.24.3                     
## [227] boot_1.3-28                            
## [228] geneplotter_1.68.0                     
## [229] ggplotify_0.1.0                        
## [230] Category_2.56.0                        
## [231] DESeq2_1.30.1                          
## [232] DEoptimR_1.0-11                        
## [233] bit_4.0.4                              
## [234] scatterpie_0.1.7                       
## [235] jpeg_0.1-8.1                           
## [236] MatrixGenerics_1.2.1                   
## [237] ggraph_2.0.5                           
## [238] pkgconfig_2.0.3                        
## [239] rstatix_0.7.0                          
## [240] knitr_1.37</code></pre>
<hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
