<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>PrincipalComponentAnalysisPeak â€¢ DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="PrincipalComponentAnalysisPeak">
<meta property="og:description" content="DEbPeak">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="PrincipalComponentAnalysisPeak_files/header-attrs-2.14/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>PrincipalComponentAnalysisPeak</h1>
            
      
      
      <div class="hidden name"><code>PrincipalComponentAnalysisPeak.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Here, we will perfrom principal component analysis (PCA) on <strong>peak-related data</strong> (eg: <code>ChIP-seq</code>, <code>ATAC-seq</code>, <code>m6a-seq</code> et al.)</p>
<hr>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>The data used here are from <a href="https://doi.org/10.1016/j.cell.2021.02.019">In vivo CD8+ T cell CRISPR screening reveals control by Fli1 in infection and cancer</a>:</p>
<ul>
<li>
<strong>RNA-seq data</strong>: the sgCtrl vs sgFli1 RNA sequencing at D8 Cl13 p.i., the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149838">GSE149838</a>
</li>
<li>
<strong>ATAC-seq data</strong>: sgCtrl vs sgFli1 ATAC sequencing at D9 Cl13 p.i., the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149836">GSE149836</a>
</li>
</ul>
<div id="count-matrix" class="section level3">
<h3 class="hasAnchor">
<a href="#count-matrix" class="anchor"></a>Count matrix</h3>
<p>The raw data:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># read the raw data</span>
<span class="va">atac.counts</span> <span class="op">=</span> <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/GSE149836_combUnionReadsWithLabels.txt"</span>,
                                sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">atac.counts</span><span class="op">)</span></code></pre></div>
<pre><code>##    chr   start     end X290_1_S4 X290_2_S5 X360_1_S6 X360_2_S7 R_1_S2 R_2_S3
## 1 chr1 3131803 3131948         2        15         0         3      4      0
## 2 chr1 3210046 3210121         0         1         0         1      0      0
## 3 chr1 3265577 3265648         3         2         0         8      5      2
## 4 chr1 3345206 3345286         3         2         1         0      2      1
## 5 chr1 3505380 3505504         3         4         0        12      4      2
## 6 chr1 3513642 3513745         0         0         2         3     13      2</code></pre>
<p>To perform analysis (including <code>Quality Control</code>), we need to preprocess the data (row: feature, column: sample):</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># read the processed data, the feature is consist of peak region, gene symbol and annotated binding region.</span>
<span class="va">atac.counts.file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RA_ATAC_count.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">atac.counts</span> <span class="op">=</span> <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">atac.counts.file</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">atac.counts</span><span class="op">)</span></code></pre></div>
<pre><code>##                                                    Fli1KO_1 Fli1KO_2 Fli1KO_3
## chr1_GL456211_random:112828-112899|LOC100041034|DI        4        6       14
## chr1_GL456211_random:113055-113190|LOC100041034|DI        7       13       10
## chr1_GL456211_random:147874-147989|LOC100041034|DI       15       16       22
## chr1_GL456211_random:167241-167436|LOC100041034|DI       53       40       43
## chr1_GL456211_random:174677-174747|LOC100041034|I         9        6       17
## chr1_GL456211_random:199244-199444|LOC100041034|DI       23       14       26
##                                                    Fli1KO_4 WT_1 WT_2
## chr1_GL456211_random:112828-112899|LOC100041034|DI       12   10    3
## chr1_GL456211_random:113055-113190|LOC100041034|DI       11   21   31
## chr1_GL456211_random:147874-147989|LOC100041034|DI       14   23   27
## chr1_GL456211_random:167241-167436|LOC100041034|DI       54   39   46
## chr1_GL456211_random:174677-174747|LOC100041034|I         6   14   10
## chr1_GL456211_random:199244-199444|LOC100041034|DI       11   27   46</code></pre>
<p><mark>If you have <strong>raw bam files</strong> of input/control and treatment samples and sample metadata, <code>DEbPeak</code> provides <code>PeakMatrix</code> to prepare the above count matrix.</mark></p>
</div>
<div id="sample-metadata" class="section level3">
<h3 class="hasAnchor">
<a href="#sample-metadata" class="anchor"></a>Sample metadata</h3>
<p>And, we also need the sample metadata:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># read the processed data</span>
<span class="va">atac.meta.file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RA_ATAC_meta.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">atac.meta</span> <span class="op">=</span> <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">atac.meta.file</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">atac.meta</span><span class="op">)</span></code></pre></div>
<pre><code>##          condition
## Fli1KO_1        KO
## Fli1KO_2        KO
## Fli1KO_3        KO
## Fli1KO_4        KO
## WT_1            WT
## WT_2            WT</code></pre>
</div>
<div id="create-deseqdataset" class="section level3">
<h3 class="hasAnchor">
<a href="#create-deseqdataset" class="anchor"></a>Create DESeqDataSet</h3>
<p>With above data, we can create <code>DESeqDataSet</code> object:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2">DESeq2</a></span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co"># create dds</span>
<span class="va">dds.atac</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span><span class="op">(</span>
  countData <span class="op">=</span> <span class="va">atac.counts</span>, colData <span class="op">=</span> <span class="va">atac.meta</span>,
  design <span class="op">=</span> <span class="op">~</span><span class="va">condition</span>
<span class="op">)</span></code></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<hr>
</div>
</div>
<div id="load-package" class="section level2">
<h2 class="hasAnchor">
<a href="#load-package" class="anchor"></a>Load package</h2>
<p>Load <code>DEbPeak</code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DEbPeak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="pca" class="section level2">
<h2 class="hasAnchor">
<a href="#pca" class="anchor"></a>PCA</h2>
<p>We will perform PCA with <code>prcomp</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># conduct PCA</span>
<span class="va">pca.res</span><span class="op">=</span><span class="fu"><a href="../reference/PCA.html">PCA</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds.atac</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span><span class="op">)</span>
<span class="co"># get basic plots</span>
<span class="va">basic.plots</span><span class="op">=</span><span class="fu"><a href="../reference/PCABasic.html">PCABasic</a></span><span class="op">(</span><span class="va">pca.res</span>,colby<span class="op">=</span><span class="st">"condition"</span>,legend.pos <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></code></pre></div>
<div id="scree-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#scree-plot" class="anchor"></a>Scree plot</h3>
<p>A Scree Plot is a simple line plot that shows the total amount of variance that can be explained by each individual PC (Y-axis shows explained variance, X-axis shows the number of PCs). It can be used to <strong>determine the number of PCs to be explored for downstream analysis</strong>. Here, we created a <strong>cumulative scree plot</strong> based on <a href="https://github.com/kevinblighe/PCAtools">PCAtools</a>, the red dashed line represents 90% explained variance.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.plots</span><span class="op">[[</span><span class="st">"screen"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="PrincipalComponentAnalysisPeak_files/figure-html/atac_pca_basic_screen-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="biplot" class="section level3">
<h3 class="hasAnchor">
<a href="#biplot" class="anchor"></a>Biplot</h3>
<p>Biplot contains informations from <a href="https://blog.bioturing.com/2018/06/18/how-to-read-pca-biplots-and-scree-plots/">two aspects</a>:</p>
<ul>
<li>sample similarity (point)</li>
<li>how strongly each gene influences a principal component (vector)
<ul>
<li>the vector will have projections on PCs, and the project values on each PC show how much weight they have on that PC. Positive values indicate that a vector/gene and a principal component are positively correlated whereas negative values indicate a negative correlation.</li>
<li>the angle between vectors indicates how vectors/genes correlate with one another: a small angle implies positive correlation, a large one suggests negative correlation, and a 90 degree angle indicates no correlation between two vectors/genes.</li>
</ul>
</li>
</ul>
<p>The biplot is created based on <a href="https://github.com/kevinblighe/PCAtools">PCAtools</a>:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.plots</span><span class="op">[[</span><span class="st">"biplot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="PrincipalComponentAnalysisPeak_files/figure-html/atac_pca_basic_biplot-1.png" width="480" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="pc-pairs-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#pc-pairs-plot" class="anchor"></a>PC pairs plot</h3>
<p>The PC pairs plot (based on <a href="https://github.com/kevinblighe/PCAtools">PCAtools</a>) will show sample similarity across different PC combination:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.plots</span><span class="op">[[</span><span class="st">"pairs"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="PrincipalComponentAnalysisPeak_files/figure-html/atac_pca_basic_pairs-1.png" width="768" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="lodding-related-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#lodding-related-analysis" class="anchor"></a>Lodding-related analysis</h2>
<p>PCA loadings are the coefficients of the linear combination of the original variables from which the principal components (PCs) are constructed, which are simply the weights of the original variables. Here, lodding indicates how strongly each gene influences a principal component (Similar to vector in <code>Biplot</code>, but here we will evaluate each individual PC). Loadings range from -1 to 1, a high absolute value indicates that the gene strongly influences the component, and the sign of a loading indicates whether a whether a gene and a principal component are positively or negatively correlated.</p>
<p>The lodding-related analysis in <code>DEbPeak</code> contains four aspects:</p>
<ul>
<li>
<strong>Loading bar plot</strong>: Visualize the genes with top n positive or negative loaddings of PCs</li>
<li>
<strong>Loading heatmap</strong>: Log2-normalized counts of the genes with top n positive or negative loaddings of PCs across samples</li>
<li>
<strong>Export genes</strong>: Export the genes with top n positive or negative loaddings of PCs</li>
<li>
<strong>GO enrichment</strong>: GO enrichment on the genes with top n positive or negative loaddings of PCs</li>
</ul>
<div id="loading-bar-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-bar-plot" class="anchor"></a>Loading bar plot</h3>
<p>Different from those in RNA-seq <a href="https://showteeth.github.io/DEbPeak/articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>, we need to use <code>data.type</code> to indicate that the data are from peak-related data, and <code>peak.anno.key</code> to specify the peak binding to which region are used for analysis, available <code>peak.anno.key</code> values are: <code>Promoter</code>, <code>5' UTR</code>, <code>3' UTR</code>, <code>Exon</code>, <code>Intron</code>, <code>Downstream</code>, <code>Distal Intergenic</code>, <code>All</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># loading bar plot</span>
<span class="fu"><a href="../reference/LoadingPlot.html">LoadingPlot</a></span><span class="op">(</span><span class="va">pca.res</span>, data.type <span class="op">=</span> <span class="st">"ATAC"</span>, peak.anno.key <span class="op">=</span> <span class="st">"All"</span>, type <span class="op">=</span> <span class="st">"bar"</span><span class="op">)</span></code></pre></div>
<p><img src="PrincipalComponentAnalysisPeak_files/figure-html/atac_pca_loading_bar-1.png" width="960" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="loading-heatmap" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-heatmap" class="anchor"></a>Loading heatmap</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># loading heatmap</span>
<span class="fu"><a href="../reference/LoadingPlot.html">LoadingPlot</a></span><span class="op">(</span><span class="va">pca.res</span>, deobj <span class="op">=</span> <span class="va">dds.atac</span>, data.type <span class="op">=</span> <span class="st">"ATAC"</span>, peak.anno.key <span class="op">=</span> <span class="st">"All"</span>, type <span class="op">=</span> <span class="st">"heat"</span><span class="op">)</span></code></pre></div>
<p><img src="PrincipalComponentAnalysisPeak_files/figure-html/atac_pca_loading_heatmap-1.png" width="960" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="export-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#export-genes" class="anchor"></a>Export genes</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># export loading genes on PC1 and PC2 (200 positive genes and 200 negative genes) </span>
<span class="va">loading.gene.df</span><span class="op">=</span><span class="fu"><a href="../reference/ExportPCGenes.html">ExportPCGenes</a></span><span class="op">(</span>pca <span class="op">=</span> <span class="va">pca.res</span>, data.type <span class="op">=</span> <span class="st">"ATAC"</span>, peak.anno.key <span class="op">=</span> <span class="st">"All"</span>, pc <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, gene.num <span class="op">=</span> <span class="fl">200</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">loading.gene.df</span><span class="op">)</span></code></pre></div>
<pre><code>##    PC                                     Gene   Loadding     Type
## 1 PC1        chr2:146197847-146198364|Cfap61|I 0.03426567 Positive
## 2 PC1 chrUn_JH584304:103171-103639|Pisd-ps3|DI 0.03166168 Positive
## 3 PC1   chrUn_JH584304:98078-98476|Pisd-ps3|DI 0.03154830 Positive
## 4 PC1    chrUn_JH584304:59463-60087|Pisd-ps3|P 0.03154772 Positive
## 5 PC1        chr13:74528968-74529954|Zfp825|DI 0.03092535 Positive
## 6 PC1          chr16:97401751-97402110|Bace2|I 0.03020087 Positive</code></pre>
<hr>
</div>
<div id="go-enrichment" class="section level3">
<h3 class="hasAnchor">
<a href="#go-enrichment" class="anchor"></a>GO enrichment</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># save results to working folder</span>
<span class="co"># LoadingGO(pca.res,gene.type="ENSEMBL",go.type="BP",padj.method="BH",save = T)</span>
<span class="co"># return list</span>
<span class="va">loading.go.results</span><span class="op">=</span><span class="fu"><a href="../reference/LoadingGO.html">LoadingGO</a></span><span class="op">(</span><span class="va">pca.res</span>,gene.type<span class="op">=</span><span class="st">"SYMBOL"</span>, data.type <span class="op">=</span> <span class="st">"ATAC"</span>, peak.anno.key <span class="op">=</span> <span class="st">"All"</span>,
                             go.type<span class="op">=</span><span class="st">"BP"</span>,padj.method<span class="op">=</span><span class="st">"BH"</span>,str.width <span class="op">=</span> <span class="fl">50</span>,save <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<hr>
<div id="genes-with-positive-loaddings" class="section level4">
<h4 class="hasAnchor">
<a href="#genes-with-positive-loaddings" class="anchor"></a>Genes with positive loaddings</h4>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># positive loading genes</span>
<span class="va">positive_go_results</span><span class="op">=</span><span class="va">loading.go.results</span><span class="op">[[</span><span class="st">"Positive"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">positive_go_results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] ID          Description GeneRatio   BgRatio     pvalue      p.adjust   
## [7] qvalue      geneID      Count      
## &lt;0 è¡Œ&gt; (æˆ–0-é•¿åº¦çš„row.names)</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">positive_go_results</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="PrincipalComponentAnalysisPeak_files/figure-html/atac_loading_go_positive-1.png" width="672" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="genes-with-negative-loaddings" class="section level4">
<h4 class="hasAnchor">
<a href="#genes-with-negative-loaddings" class="anchor"></a>Genes with negative loaddings</h4>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># negative loading genes</span>
<span class="va">negative_go_results</span><span class="op">=</span><span class="va">loading.go.results</span><span class="op">[[</span><span class="st">"Negative"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">negative_go_results</span><span class="op">[[</span><span class="st">"table"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] ID          Description GeneRatio   BgRatio     pvalue      p.adjust   
## [7] qvalue      geneID      Count      
## &lt;0 è¡Œ&gt; (æˆ–0-é•¿åº¦çš„row.names)</code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">negative_go_results</span><span class="op">[[</span><span class="st">"plot"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="PrincipalComponentAnalysisPeak_files/figure-html/atac_loading_go_negative-1.png" width="672" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
<div id="d-visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#d-visualization" class="anchor"></a>3D visualization</h2>
<p>To visualize three PCs simultaneously, <code>DEbPeak</code> provides <code>PCA3D</code> to create 3D PCA plot:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PCA3D.html">PCA3D</a></span><span class="op">(</span>pca <span class="op">=</span> <span class="va">pca.res</span>, color.key <span class="op">=</span> <span class="st">"condition"</span>, main <span class="op">=</span> <span class="st">"3D PCA"</span><span class="op">)</span></code></pre></div>
<p><img src="PrincipalComponentAnalysisPeak_files/figure-html/atac_pca_3d-1.png" width="480" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-conda-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /home/softwares/anaconda3/envs/r4.0/lib/libopenblasp-r0.3.12.so
## 
## locale:
##  [1] LC_CTYPE=zh_CN.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=zh_CN.UTF-8        LC_COLLATE=zh_CN.UTF-8    
##  [5] LC_MONETARY=zh_CN.UTF-8    LC_MESSAGES=zh_CN.UTF-8   
##  [7] LC_PAPER=zh_CN.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] org.Mm.eg.db_3.12.0         AnnotationDbi_1.52.0       
##  [3] DEbPeak_0.9.0               DESeq2_1.30.1              
##  [5] SummarizedExperiment_1.20.0 Biobase_2.50.0             
##  [7] MatrixGenerics_1.2.1        matrixStats_0.58.0         
##  [9] GenomicRanges_1.42.0        GenomeInfoDb_1.26.7        
## [11] IRanges_2.24.1              S4Vectors_0.28.1           
## [13] BiocGenerics_0.42.0        
## 
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                             
##   [2] ggvenn_0.1.9                           
##   [3] apeglm_1.12.0                          
##   [4] Rsamtools_2.6.0                        
##   [5] rsvg_2.1                               
##   [6] foreach_1.5.1                          
##   [7] rprojroot_2.0.2                        
##   [8] crayon_1.4.1                           
##   [9] V8_3.4.2                               
##  [10] MASS_7.3-53.1                          
##  [11] nlme_3.1-152                           
##  [12] backports_1.2.1                        
##  [13] sva_3.38.0                             
##  [14] GOSemSim_2.25.0                        
##  [15] rlang_1.0.3                            
##  [16] XVector_0.30.0                         
##  [17] readxl_1.3.1                           
##  [18] irlba_2.3.5                            
##  [19] limma_3.46.0                           
##  [20] GOstats_2.56.0                         
##  [21] BiocParallel_1.24.1                    
##  [22] rjson_0.2.20                           
##  [23] bit64_4.0.5                            
##  [24] glue_1.6.2                             
##  [25] DiffBind_3.0.15                        
##  [26] mixsqp_0.3-43                          
##  [27] pheatmap_1.0.12                        
##  [28] parallel_4.0.3                         
##  [29] DEFormats_1.18.0                       
##  [30] base64url_1.4                          
##  [31] tcltk_4.0.3                            
##  [32] DOSE_3.23.2                            
##  [33] haven_2.3.1                            
##  [34] tidyselect_1.1.0                       
##  [35] rio_0.5.27                             
##  [36] XML_3.99-0.6                           
##  [37] tidyr_1.1.3                            
##  [38] ggpubr_0.4.0                           
##  [39] GenomicAlignments_1.26.0               
##  [40] xtable_1.8-4                           
##  [41] magrittr_2.0.1                         
##  [42] evaluate_0.14                          
##  [43] ggplot2_3.3.5                          
##  [44] cli_3.3.0                              
##  [45] zlibbioc_1.36.0                        
##  [46] hwriter_1.3.2                          
##  [47] rstudioapi_0.13                        
##  [48] bslib_0.3.1                            
##  [49] GreyListChIP_1.22.0                    
##  [50] fastmatch_1.1-3                        
##  [51] BiocSingular_1.6.0                     
##  [52] xfun_0.30                              
##  [53] askpass_1.1                            
##  [54] clue_0.3-59                            
##  [55] gson_0.0.9                             
##  [56] cluster_2.1.1                          
##  [57] caTools_1.18.2                         
##  [58] tidygraph_1.2.0                        
##  [59] tibble_3.1.0                           
##  [60] ggrepel_0.9.1                          
##  [61] Biostrings_2.58.0                      
##  [62] png_0.1-7                              
##  [63] withr_2.4.1                            
##  [64] bitops_1.0-6                           
##  [65] ggforce_0.3.3                          
##  [66] RBGL_1.66.0                            
##  [67] plyr_1.8.6                             
##  [68] cellranger_1.1.0                       
##  [69] GSEABase_1.52.1                        
##  [70] pcaPP_2.0-1                            
##  [71] dqrng_0.2.1                            
##  [72] coda_0.19-4                            
##  [73] pillar_1.5.1                           
##  [74] gplots_3.1.1                           
##  [75] GlobalOptions_0.1.2                    
##  [76] cachem_1.0.4                           
##  [77] GenomicFeatures_1.42.2                 
##  [78] fs_1.5.0                               
##  [79] GetoptLong_1.0.5                       
##  [80] clusterProfiler_4.7.1                  
##  [81] DelayedMatrixStats_1.12.3              
##  [82] vctrs_0.4.1                            
##  [83] ellipsis_0.3.2                         
##  [84] generics_0.1.0                         
##  [85] plot3D_1.4                             
##  [86] tools_4.0.3                            
##  [87] foreign_0.8-81                         
##  [88] NOISeq_2.34.0                          
##  [89] munsell_0.5.0                          
##  [90] tweenr_1.0.2                           
##  [91] fgsea_1.16.0                           
##  [92] DelayedArray_0.16.3                    
##  [93] fastmap_1.1.0                          
##  [94] compiler_4.0.3                         
##  [95] abind_1.4-5                            
##  [96] rtracklayer_1.50.0                     
##  [97] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
##  [98] GenomeInfoDbData_1.2.4                 
##  [99] gridExtra_2.3                          
## [100] edgeR_3.32.1                           
## [101] lattice_0.20-45                        
## [102] ggnewscale_0.4.7                       
## [103] AnnotationForge_1.32.0                 
## [104] utf8_1.2.1                             
## [105] dplyr_1.0.5                            
## [106] BiocFileCache_1.14.0                   
## [107] jsonlite_1.7.2                         
## [108] scales_1.1.1                           
## [109] graph_1.68.0                           
## [110] carData_3.0-4                          
## [111] sparseMatrixStats_1.2.1                
## [112] genefilter_1.72.1                      
## [113] car_3.0-11                             
## [114] doParallel_1.0.16                      
## [115] latticeExtra_0.6-29                    
## [116] brew_1.0-6                             
## [117] checkmate_2.0.0                        
## [118] rmarkdown_2.14                         
## [119] openxlsx_4.2.3                         
## [120] pkgdown_1.6.1                          
## [121] cowplot_1.1.1                          
## [122] textshaping_0.1.2                      
## [123] forcats_0.5.1                          
## [124] downloader_0.4                         
## [125] BSgenome_1.58.0                        
## [126] igraph_1.3.4                           
## [127] survival_3.2-10                        
## [128] numDeriv_2016.8-1.1                    
## [129] yaml_2.2.1                             
## [130] plotrix_3.8-2                          
## [131] systemfonts_1.0.1                      
## [132] ashr_2.2-47                            
## [133] SQUAREM_2021.1                         
## [134] htmltools_0.5.2                        
## [135] memoise_2.0.0                          
## [136] VariantAnnotation_1.36.0               
## [137] locfit_1.5-9.4                         
## [138] graphlayouts_0.7.1                     
## [139] batchtools_0.9.15                      
## [140] PCAtools_2.2.0                         
## [141] viridisLite_0.4.0                      
## [142] digest_0.6.27                          
## [143] rrcov_1.7-0                            
## [144] assertthat_0.2.1                       
## [145] rappdirs_0.3.3                         
## [146] emdbook_1.3.12                         
## [147] RSQLite_2.2.5                          
## [148] amap_0.8-18                            
## [149] yulab.utils_0.0.4                      
## [150] misc3d_0.9-1                           
## [151] data.table_1.14.2                      
## [152] blob_1.2.1                             
## [153] ragg_0.4.0                             
## [154] labeling_0.4.2                         
## [155] splines_4.0.3                          
## [156] Cairo_1.5-12.2                         
## [157] ggupset_0.3.0                          
## [158] RCurl_1.98-1.3                         
## [159] broom_0.7.5                            
## [160] hms_1.0.0                              
## [161] colorspace_2.0-0                       
## [162] BiocManager_1.30.16                    
## [163] shape_1.4.6                            
## [164] sass_0.4.1                             
## [165] Rcpp_1.0.9                             
## [166] mvtnorm_1.1-2                          
## [167] circlize_0.4.15                        
## [168] enrichplot_1.10.2                      
## [169] fansi_0.4.2                            
## [170] truncnorm_1.0-8                        
## [171] ChIPseeker_1.33.0.900                  
## [172] R6_2.5.0                               
## [173] grid_4.0.3                             
## [174] lifecycle_1.0.0                        
## [175] ShortRead_1.48.0                       
## [176] zip_2.1.1                              
## [177] curl_4.3                               
## [178] ggsignif_0.6.3                         
## [179] jquerylib_0.1.3                        
## [180] robustbase_0.95-0                      
## [181] DO.db_2.9                              
## [182] Matrix_1.3-3                           
## [183] qvalue_2.22.0                          
## [184] desc_1.3.0                             
## [185] RColorBrewer_1.1-2                     
## [186] iterators_1.0.13                       
## [187] stringr_1.4.0                          
## [188] DOT_0.1                                
## [189] ggpie_0.2.1                            
## [190] beachmat_2.6.4                         
## [191] polyclip_1.10-0                        
## [192] biomaRt_2.46.3                         
## [193] purrr_0.3.4                            
## [194] shadowtext_0.0.9                       
## [195] gridGraphics_0.5-1                     
## [196] mgcv_1.8-34                            
## [197] ComplexHeatmap_2.13.1                  
## [198] openssl_1.4.3                          
## [199] patchwork_1.0.0                        
## [200] bdsmatrix_1.3-4                        
## [201] codetools_0.2-18                       
## [202] invgamma_1.1                           
## [203] GO.db_3.12.1                           
## [204] gtools_3.8.2                           
## [205] prettyunits_1.1.1                      
## [206] dbplyr_2.1.1                           
## [207] gtable_0.3.0                           
## [208] DBI_1.1.1                              
## [209] highr_0.8                              
## [210] ggfun_0.0.6                            
## [211] httr_1.4.2                             
## [212] KernSmooth_2.23-18                     
## [213] stringi_1.5.3                          
## [214] progress_1.2.2                         
## [215] reshape2_1.4.4                         
## [216] farver_2.1.0                           
## [217] annotate_1.68.0                        
## [218] viridis_0.6.1                          
## [219] Rgraphviz_2.34.0                       
## [220] xml2_1.3.2                             
## [221] bbmle_1.0.24                           
## [222] systemPipeR_1.24.3                     
## [223] boot_1.3-28                            
## [224] geneplotter_1.68.0                     
## [225] ggplotify_0.1.0                        
## [226] Category_2.56.0                        
## [227] DEoptimR_1.0-11                        
## [228] bit_4.0.4                              
## [229] scatterpie_0.1.7                       
## [230] jpeg_0.1-8.1                           
## [231] ggraph_2.0.5                           
## [232] pkgconfig_2.0.3                        
## [233] rstatix_0.7.0                          
## [234] knitr_1.37</code></pre>
<hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
