<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>QualityControl â€¢ DEbPeak</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="QualityControl">
<meta property="og:description" content="DEbPeak">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="QualityControl_files/header-attrs-2.14/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>QualityControl</h1>
            
      
      
      <div class="hidden name"><code>QualityControl.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p><code>DEbPeak</code> provides two aspects of Quality Control (QC):</p>
<ul>
<li>
<strong>QC on count matrix</strong>:
<ul>
<li>Proportion of genes detected in different samples under <strong>different CPM thresholds</strong>
</li>
<li>The <strong>saturation of the number of genes detected</strong>.</li>
</ul>
</li>
<li>
<strong>QC on samples</strong>:
<ul>
<li>
<strong>Euclidean distance</strong> and <strong>pearson correlation coefficient</strong> of samples across different conditions</li>
<li>
<strong>sample similarity</strong> on selected principal components (<strong>check batch information and conduct batch correction</strong>)</li>
<li>
<strong>outlier detection</strong> with robust PCA</li>
</ul>
</li>
</ul>
<hr>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>The data used here are RNA-seq data of the external granule layer in the cerebellum of control and conditional SnoN knockout mice, the raw data are stored in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120279">GSE120279</a>, they are used in <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03608-0">Robust principal component analysis for accurate outlier sample detection in RNA-seq data</a>.</p>
<p>First, we create <code>DESeqDataSet</code> object with above dataset:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2">DESeq2</a></span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DEbPeak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">count.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"snon_count.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">meta.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"snon_meta.txt"</span>, package <span class="op">=</span> <span class="st">"DEbPeak"</span><span class="op">)</span>
<span class="va">count.matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">count.file</span>, header <span class="op">=</span> <span class="cn">T</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">count.matrix</span><span class="op">)</span></code></pre></div>
<pre><code>##                        KO1  KO2  KO3  KO4  KO5  KO6  WT1  WT2  WT3  WT4  WT5
## ENSMUSG00000000001.4  4556 4218 3835 3718 5741 3875 4115 5074 2931 4374 5118
## ENSMUSG00000000003.15    0    0    0    0    0    0    0    0    0    0    0
## ENSMUSG00000000028.14  350  579  435  316  432  317  245  621  419  506  545
## ENSMUSG00000000031.16  268  804   66  207   46   66  336  112   60   69  137
## ENSMUSG00000000037.16  262  157  184  162  301  233  311  176   94  139  229
## ENSMUSG00000000049.11    0    0    0    0    0    0    1    0    2    0    1
##                        WT6
## ENSMUSG00000000001.4  3625
## ENSMUSG00000000003.15    0
## ENSMUSG00000000028.14  371
## ENSMUSG00000000031.16  193
## ENSMUSG00000000037.16  186
## ENSMUSG00000000049.11    0</code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta.info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">meta.file</span>, header <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">meta.info</span><span class="op">)</span></code></pre></div>
<pre><code>##     condition
## KO1        KO
## KO2        KO
## KO3        KO
## KO4        KO
## KO5        KO
## KO6        KO</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span><span class="op">(</span>countData <span class="op">=</span> <span class="va">count.matrix</span>, colData <span class="op">=</span> <span class="va">meta.info</span>, design <span class="op">=</span> <span class="op">~</span><span class="va">condition</span><span class="op">)</span>
<span class="va">dds</span></code></pre></div>
<pre><code>## class: DESeqDataSet 
## dim: 53379 12 
## metadata(1): version
## assays(1): counts
## rownames(53379): ENSMUSG00000000001.4 ENSMUSG00000000003.15 ...
##   ENSMUSG00000115849.1 ENSMUSG00000115850.1
## rowData names(0):
## colnames(12): KO1 KO2 ... WT5 WT6
## colData names(1): condition</code></pre>
<hr>
</div>
<div id="qc-on-count-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#qc-on-count-matrix" class="anchor"></a>QC on count matrix</h2>
<div id="cpm-threshold" class="section level3">
<h3 class="hasAnchor">
<a href="#cpm-threshold" class="anchor"></a>CPM threshold</h3>
<p>This plot shows proportion of genes detected in different samples at different CPM thresholds. Theoretically, all samples should be similarly distributed, so as to avoid false positive results obtained from detection problems.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/CountQC.html">CountQC</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>, group.key <span class="op">=</span> <span class="st">"condition"</span>, type <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## [1] "Warning: 25096 features with 0 counts in all samples are to be removed for this analysis."
## [1] "Count distributions are to be computed for:"
##  [1] "KO1" "KO2" "KO3" "KO4" "KO5" "KO6" "WT1" "WT2" "WT3" "WT4" "WT5" "WT6"</code></pre>
<p><img src="QualityControl_files/figure-html/qc_counts_cpm-1.png" width="672" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="sequencing-saturation" class="section level3">
<h3 class="hasAnchor">
<a href="#sequencing-saturation" class="anchor"></a>sequencing saturation</h3>
<p>This plot shows the proportion of genes detected at different sequencing depths. If the proportion is too low, it indicates that many genes are not detected and therefore some important information may be missing, and increasing the sequencing volume can effectively solve this problem.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/CountQC.html">CountQC</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>, group.key <span class="op">=</span> <span class="st">"condition"</span>, type <span class="op">=</span> <span class="st">"saturation"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="QualityControl_files/figure-html/qc_counts_saturation-1.png" width="672" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="qc-on-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#qc-on-samples" class="anchor"></a>QC on samples</h2>
<p>QC on samples including four aspects:</p>
<ul>
<li>Euclidean distance and pearson correlation coefficient: sample similarity on <strong>the entire transcriptome</strong>.</li>
<li>PCA: sample similarity on <strong>selected principal components</strong>.</li>
<li>outlier detection with robust PCA</li>
<li>batch correction</li>
</ul>
<div id="euclidean-distance-and-pearson-correlation-coefficient" class="section level3">
<h3 class="hasAnchor">
<a href="#euclidean-distance-and-pearson-correlation-coefficient" class="anchor"></a>Euclidean distance and pearson correlation coefficient</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/SampleRelation.html">SampleRelation</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span>,anno.key <span class="op">=</span> <span class="st">"condition"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<p><img src="QualityControl_files/figure-html/qc_samples_sr-1.png" width="1344" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="pca" class="section level3">
<h3 class="hasAnchor">
<a href="#pca" class="anchor"></a>PCA</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">qc.pca.res</span><span class="op">=</span><span class="fu"><a href="../reference/PCA.html">PCA</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Use all genes for PCA!</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">PCAtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/PCAtools/man/biplot.html">biplot</a></span><span class="op">(</span><span class="va">qc.pca.res</span>,x<span class="op">=</span><span class="st">"PC1"</span>,y<span class="op">=</span><span class="st">"PC2"</span>,colby<span class="op">=</span><span class="st">"condition"</span>,legendPosition<span class="op">=</span><span class="st">"bottom"</span><span class="op">)</span></code></pre></div>
<p><img src="QualityControl_files/figure-html/qc_samples_pca-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="outlier-detection-with-robust-pca" class="section level3">
<h3 class="hasAnchor">
<a href="#outlier-detection-with-robust-pca" class="anchor"></a>outlier detection with robust PCA</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outlier.res</span><span class="op">=</span><span class="fu"><a href="../reference/OutlierDetection.html">OutlierDetection</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,var.genes <span class="op">=</span> <span class="cn">NULL</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span><span class="op">)</span></code></pre></div>
<pre><code>## Differential expression analysis with DESeq2!</code></pre>
<pre><code>## Use all genes for PCA!</code></pre>
<pre><code>## Detecting 2 outlier(s): KO3,WT1</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outlier.res</span><span class="op">$</span><span class="va">outlier</span></code></pre></div>
<pre><code>## [1] "KO3" "WT1"</code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outlier.res</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="QualityControl_files/figure-html/qc_samples_outlier-1.png" width="672" style="display: block; margin: auto;"></p>
<p>For batch correction and outlier detection, you can use the following codes:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># batch effect correction</span>
<span class="co"># this is a demo scripts</span>
<span class="va">batch.res</span><span class="op">=</span><span class="fu"><a href="../reference/QCPCA.html">QCPCA</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,var.genes <span class="op">=</span> <span class="cn">NULL</span>,remove.sample<span class="op">=</span><span class="cn">NULL</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span>,batch <span class="op">=</span> <span class="st">"cell"</span>,
                colby <span class="op">=</span> <span class="st">"dex"</span>, outlier.detection <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">batch.res</span><span class="op">$</span><span class="va">plot</span>
<span class="co"># outlier detection</span>
<span class="va">outlier.res</span><span class="op">=</span><span class="fu"><a href="../reference/QCPCA.html">QCPCA</a></span><span class="op">(</span>deobj <span class="op">=</span> <span class="va">dds</span>,var.genes <span class="op">=</span> <span class="cn">NULL</span>,remove.sample<span class="op">=</span><span class="cn">NULL</span>,transform.method <span class="op">=</span> <span class="st">"rlog"</span>,
                  outlier.detection <span class="op">=</span> <span class="cn">T</span>,rpca.method <span class="op">=</span> <span class="st">"PcaGrid"</span><span class="op">)</span>
<span class="va">outlier.res</span><span class="op">$</span><span class="va">plot</span>
<span class="co"># the final dds</span>
<span class="va">dds</span><span class="op">=</span><span class="va">outlier.res</span><span class="op">$</span><span class="va">deobj</span> <span class="co"># or outlier.res$plot</span></code></pre></div>
<hr>
</div>
</div>
<div id="save-results" class="section level2">
<h2 class="hasAnchor">
<a href="#save-results" class="anchor"></a>Save results</h2>
<p>Save results for downstream analysis:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/save.html">save.image</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/home/songyabing/R/learn/tmp/DEbPeak/example.RData"</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-conda-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /home/softwares/anaconda3/envs/r4.0/lib/libopenblasp-r0.3.12.so
## 
## locale:
##  [1] LC_CTYPE=zh_CN.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=zh_CN.UTF-8        LC_COLLATE=zh_CN.UTF-8    
##  [5] LC_MONETARY=zh_CN.UTF-8    LC_MESSAGES=zh_CN.UTF-8   
##  [7] LC_PAPER=zh_CN.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] DEbPeak_0.9.0               DESeq2_1.30.1              
##  [3] SummarizedExperiment_1.20.0 Biobase_2.50.0             
##  [5] MatrixGenerics_1.2.1        matrixStats_0.58.0         
##  [7] GenomicRanges_1.42.0        GenomeInfoDb_1.26.7        
##  [9] IRanges_2.24.1              S4Vectors_0.28.1           
## [11] BiocGenerics_0.42.0        
## 
## loaded via a namespace (and not attached):
##   [1] rsvd_1.0.3                             
##   [2] ggvenn_0.1.9                           
##   [3] apeglm_1.12.0                          
##   [4] Rsamtools_2.6.0                        
##   [5] rsvg_2.1                               
##   [6] foreach_1.5.1                          
##   [7] rprojroot_2.0.2                        
##   [8] crayon_1.4.1                           
##   [9] V8_3.4.2                               
##  [10] MASS_7.3-53.1                          
##  [11] nlme_3.1-152                           
##  [12] backports_1.2.1                        
##  [13] sva_3.38.0                             
##  [14] GOSemSim_2.25.0                        
##  [15] rlang_1.0.3                            
##  [16] XVector_0.30.0                         
##  [17] readxl_1.3.1                           
##  [18] irlba_2.3.5                            
##  [19] limma_3.46.0                           
##  [20] GOstats_2.56.0                         
##  [21] BiocParallel_1.24.1                    
##  [22] rjson_0.2.20                           
##  [23] bit64_4.0.5                            
##  [24] glue_1.6.2                             
##  [25] DiffBind_3.0.15                        
##  [26] mixsqp_0.3-43                          
##  [27] pheatmap_1.0.12                        
##  [28] parallel_4.0.3                         
##  [29] AnnotationDbi_1.52.0                   
##  [30] DEFormats_1.18.0                       
##  [31] base64url_1.4                          
##  [32] tcltk_4.0.3                            
##  [33] DOSE_3.23.2                            
##  [34] haven_2.3.1                            
##  [35] tidyselect_1.1.0                       
##  [36] rio_0.5.27                             
##  [37] XML_3.99-0.6                           
##  [38] tidyr_1.1.3                            
##  [39] ggpubr_0.4.0                           
##  [40] GenomicAlignments_1.26.0               
##  [41] xtable_1.8-4                           
##  [42] magrittr_2.0.1                         
##  [43] evaluate_0.14                          
##  [44] ggplot2_3.3.5                          
##  [45] cli_3.3.0                              
##  [46] zlibbioc_1.36.0                        
##  [47] hwriter_1.3.2                          
##  [48] rstudioapi_0.13                        
##  [49] bslib_0.3.1                            
##  [50] GreyListChIP_1.22.0                    
##  [51] fastmatch_1.1-3                        
##  [52] BiocSingular_1.6.0                     
##  [53] xfun_0.30                              
##  [54] askpass_1.1                            
##  [55] clue_0.3-59                            
##  [56] gson_0.0.9                             
##  [57] cluster_2.1.1                          
##  [58] caTools_1.18.2                         
##  [59] tidygraph_1.2.0                        
##  [60] tibble_3.1.0                           
##  [61] ggrepel_0.9.1                          
##  [62] Biostrings_2.58.0                      
##  [63] png_0.1-7                              
##  [64] withr_2.4.1                            
##  [65] bitops_1.0-6                           
##  [66] ggforce_0.3.3                          
##  [67] RBGL_1.66.0                            
##  [68] plyr_1.8.6                             
##  [69] cellranger_1.1.0                       
##  [70] GSEABase_1.52.1                        
##  [71] pcaPP_2.0-1                            
##  [72] dqrng_0.2.1                            
##  [73] coda_0.19-4                            
##  [74] pillar_1.5.1                           
##  [75] gplots_3.1.1                           
##  [76] GlobalOptions_0.1.2                    
##  [77] cachem_1.0.4                           
##  [78] GenomicFeatures_1.42.2                 
##  [79] fs_1.5.0                               
##  [80] GetoptLong_1.0.5                       
##  [81] clusterProfiler_4.7.1                  
##  [82] DelayedMatrixStats_1.12.3              
##  [83] vctrs_0.4.1                            
##  [84] ellipsis_0.3.2                         
##  [85] generics_0.1.0                         
##  [86] plot3D_1.4                             
##  [87] tools_4.0.3                            
##  [88] foreign_0.8-81                         
##  [89] NOISeq_2.34.0                          
##  [90] munsell_0.5.0                          
##  [91] tweenr_1.0.2                           
##  [92] fgsea_1.16.0                           
##  [93] DelayedArray_0.16.3                    
##  [94] fastmap_1.1.0                          
##  [95] compiler_4.0.3                         
##  [96] abind_1.4-5                            
##  [97] rtracklayer_1.50.0                     
##  [98] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
##  [99] GenomeInfoDbData_1.2.4                 
## [100] gridExtra_2.3                          
## [101] edgeR_3.32.1                           
## [102] lattice_0.20-45                        
## [103] ggnewscale_0.4.7                       
## [104] AnnotationForge_1.32.0                 
## [105] utf8_1.2.1                             
## [106] dplyr_1.0.5                            
## [107] BiocFileCache_1.14.0                   
## [108] jsonlite_1.7.2                         
## [109] scales_1.1.1                           
## [110] graph_1.68.0                           
## [111] carData_3.0-4                          
## [112] sparseMatrixStats_1.2.1                
## [113] genefilter_1.72.1                      
## [114] car_3.0-11                             
## [115] doParallel_1.0.16                      
## [116] latticeExtra_0.6-29                    
## [117] brew_1.0-6                             
## [118] checkmate_2.0.0                        
## [119] rmarkdown_2.14                         
## [120] openxlsx_4.2.3                         
## [121] pkgdown_1.6.1                          
## [122] cowplot_1.1.1                          
## [123] textshaping_0.1.2                      
## [124] forcats_0.5.1                          
## [125] downloader_0.4                         
## [126] BSgenome_1.58.0                        
## [127] igraph_1.3.4                           
## [128] survival_3.2-10                        
## [129] numDeriv_2016.8-1.1                    
## [130] yaml_2.2.1                             
## [131] plotrix_3.8-2                          
## [132] systemfonts_1.0.1                      
## [133] ashr_2.2-47                            
## [134] SQUAREM_2021.1                         
## [135] htmltools_0.5.2                        
## [136] memoise_2.0.0                          
## [137] VariantAnnotation_1.36.0               
## [138] locfit_1.5-9.4                         
## [139] graphlayouts_0.7.1                     
## [140] batchtools_0.9.15                      
## [141] PCAtools_2.2.0                         
## [142] viridisLite_0.4.0                      
## [143] digest_0.6.27                          
## [144] rrcov_1.7-0                            
## [145] assertthat_0.2.1                       
## [146] rappdirs_0.3.3                         
## [147] emdbook_1.3.12                         
## [148] RSQLite_2.2.5                          
## [149] amap_0.8-18                            
## [150] yulab.utils_0.0.4                      
## [151] misc3d_0.9-1                           
## [152] data.table_1.14.2                      
## [153] blob_1.2.1                             
## [154] ragg_0.4.0                             
## [155] labeling_0.4.2                         
## [156] splines_4.0.3                          
## [157] Cairo_1.5-12.2                         
## [158] ggupset_0.3.0                          
## [159] RCurl_1.98-1.3                         
## [160] broom_0.7.5                            
## [161] hms_1.0.0                              
## [162] colorspace_2.0-0                       
## [163] BiocManager_1.30.16                    
## [164] shape_1.4.6                            
## [165] sass_0.4.1                             
## [166] Rcpp_1.0.9                             
## [167] mvtnorm_1.1-2                          
## [168] circlize_0.4.15                        
## [169] enrichplot_1.10.2                      
## [170] fansi_0.4.2                            
## [171] truncnorm_1.0-8                        
## [172] ChIPseeker_1.33.0.900                  
## [173] R6_2.5.0                               
## [174] grid_4.0.3                             
## [175] lifecycle_1.0.0                        
## [176] ShortRead_1.48.0                       
## [177] zip_2.1.1                              
## [178] curl_4.3                               
## [179] ggsignif_0.6.3                         
## [180] jquerylib_0.1.3                        
## [181] robustbase_0.95-0                      
## [182] DO.db_2.9                              
## [183] Matrix_1.3-3                           
## [184] qvalue_2.22.0                          
## [185] desc_1.3.0                             
## [186] RColorBrewer_1.1-2                     
## [187] iterators_1.0.13                       
## [188] stringr_1.4.0                          
## [189] DOT_0.1                                
## [190] ggpie_0.2.1                            
## [191] beachmat_2.6.4                         
## [192] polyclip_1.10-0                        
## [193] biomaRt_2.46.3                         
## [194] purrr_0.3.4                            
## [195] shadowtext_0.0.9                       
## [196] gridGraphics_0.5-1                     
## [197] mgcv_1.8-34                            
## [198] ComplexHeatmap_2.13.1                  
## [199] openssl_1.4.3                          
## [200] patchwork_1.0.0                        
## [201] bdsmatrix_1.3-4                        
## [202] codetools_0.2-18                       
## [203] invgamma_1.1                           
## [204] GO.db_3.12.1                           
## [205] gtools_3.8.2                           
## [206] prettyunits_1.1.1                      
## [207] dbplyr_2.1.1                           
## [208] gtable_0.3.0                           
## [209] DBI_1.1.1                              
## [210] highr_0.8                              
## [211] ggfun_0.0.6                            
## [212] httr_1.4.2                             
## [213] KernSmooth_2.23-18                     
## [214] stringi_1.5.3                          
## [215] progress_1.2.2                         
## [216] reshape2_1.4.4                         
## [217] farver_2.1.0                           
## [218] annotate_1.68.0                        
## [219] viridis_0.6.1                          
## [220] Rgraphviz_2.34.0                       
## [221] xml2_1.3.2                             
## [222] bbmle_1.0.24                           
## [223] systemPipeR_1.24.3                     
## [224] boot_1.3-28                            
## [225] geneplotter_1.68.0                     
## [226] ggplotify_0.1.0                        
## [227] Category_2.56.0                        
## [228] DEoptimR_1.0-11                        
## [229] bit_4.0.4                              
## [230] scatterpie_0.1.7                       
## [231] jpeg_0.1-8.1                           
## [232] ggraph_2.0.5                           
## [233] pkgconfig_2.0.3                        
## [234] rstatix_0.7.0                          
## [235] knitr_1.37</code></pre>
<hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
