<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Conduct Differential Analysis with DESeq2. — ConductDESeq2 • DEbPeak</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Conduct Differential Analysis with DESeq2. — ConductDESeq2" />
<meta property="og:description" content="Conduct Differential Analysis with DESeq2." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ParseGEO.html">Parse GEO</a>
    </li>
    <li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPDE.html">Integrate RNA-seq with ChIP-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Conduct Differential Analysis with DESeq2.</h1>
    
    <div class="hidden name"><code>ConductDESeq2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Conduct Differential Analysis with DESeq2.</p>
    </div>

    <pre class="usage"><span class='fu'>ConductDESeq2</span><span class='op'>(</span>
  <span class='va'>counts.folder</span>,
  count.matrix.file <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>meta.file</span>,
  group.key <span class='op'>=</span> <span class='cn'>NULL</span>,
  count.type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"htseq-count"</span>, <span class='st'>"featurecounts"</span><span class='op'>)</span>,
  min.count <span class='op'>=</span> <span class='fl'>10</span>,
  ref.group <span class='op'>=</span> <span class='cn'>NULL</span>,
  out.folder <span class='op'>=</span> <span class='cn'>NULL</span>,
  data.type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"RNA"</span>, <span class='st'>"ChIP"</span>, <span class='st'>"ATAC"</span><span class='op'>)</span>,
  peak.anno.key <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Promoter"</span>, <span class='st'>"5' UTR"</span>, <span class='st'>"3' UTR"</span>, <span class='st'>"Exon"</span>, <span class='st'>"Intron"</span>, <span class='st'>"Downstream"</span>,
    <span class='st'>"Distal Intergenic"</span>, <span class='st'>"All"</span><span class='op'>)</span>,
  qc.ndepth <span class='op'>=</span> <span class='fl'>10</span>,
  transform.method <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"rlog"</span>, <span class='st'>"vst"</span>, <span class='st'>"ntd"</span><span class='op'>)</span>,
  var.genes <span class='op'>=</span> <span class='cn'>NULL</span>,
  batch <span class='op'>=</span> <span class='cn'>NULL</span>,
  outlier.detection <span class='op'>=</span> <span class='cn'>T</span>,
  rpca.method <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"PcaGrid"</span>, <span class='st'>"PcaHubert"</span><span class='op'>)</span>,
  k <span class='op'>=</span> <span class='fl'>2</span>,
  pca.x <span class='op'>=</span> <span class='st'>"PC1"</span>,
  pca.y <span class='op'>=</span> <span class='st'>"PC2"</span>,
  pca.z <span class='op'>=</span> <span class='st'>"PC3"</span>,
  loding.pc <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>5</span>,
  loading.gene.num <span class='op'>=</span> <span class='fl'>10</span>,
  loading.ncol <span class='op'>=</span> <span class='fl'>2</span>,
  enrich.loading.pc <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>5</span>,
  enrich.loading.gene <span class='op'>=</span> <span class='fl'>200</span>,
  gene.type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ENSEMBL"</span>, <span class='st'>"ENTREZID"</span>, <span class='st'>"SYMBOL"</span><span class='op'>)</span>,
  enrich.type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ALL"</span>, <span class='st'>"GO"</span>, <span class='st'>"KEGG"</span><span class='op'>)</span>,
  go.type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ALL"</span>, <span class='st'>"BP"</span>, <span class='st'>"MF"</span>, <span class='st'>"CC"</span><span class='op'>)</span>,
  enrich.pvalue <span class='op'>=</span> <span class='fl'>0.05</span>,
  enrich.qvalue <span class='op'>=</span> <span class='fl'>0.05</span>,
  org.db <span class='op'>=</span> <span class='st'>"org.Mm.eg.db"</span>,
  organism <span class='op'>=</span> <span class='st'>"mmu"</span>,
  padj.method <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"BH"</span>, <span class='st'>"holm"</span>, <span class='st'>"hochberg"</span>, <span class='st'>"hommel"</span>, <span class='st'>"bonferroni"</span>, <span class='st'>"BY"</span>, <span class='st'>"fdr"</span>,
    <span class='st'>"none"</span><span class='op'>)</span>,
  show.term <span class='op'>=</span> <span class='fl'>15</span>,
  str.width <span class='op'>=</span> <span class='fl'>30</span>,
  signif <span class='op'>=</span> <span class='st'>"padj"</span>,
  signif.threshold <span class='op'>=</span> <span class='fl'>0.05</span>,
  l2fc.threshold <span class='op'>=</span> <span class='fl'>1</span>,
  gene.map <span class='op'>=</span> <span class='cn'>NULL</span>,
  gtf.file <span class='op'>=</span> <span class='cn'>NULL</span>,
  norm.type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"DESeq2"</span>, <span class='st'>"TMM"</span>, <span class='st'>"CPM"</span>, <span class='st'>"RPKM"</span>, <span class='st'>"TPM"</span><span class='op'>)</span>,
  log.counts <span class='op'>=</span> <span class='cn'>TRUE</span>,
  deg.label.df <span class='op'>=</span> <span class='cn'>NULL</span>,
  deg.label.key <span class='op'>=</span> <span class='cn'>NULL</span>,
  deg.label.num <span class='op'>=</span> <span class='fl'>2</span>,
  deg.label.color <span class='op'>=</span> <span class='cn'>NULL</span>,
  fe.gene.key <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>gmt.file</span>,
  gene.sets <span class='op'>=</span> <span class='cn'>NULL</span>,
  minGSSize <span class='op'>=</span> <span class='fl'>10</span>,
  maxGSSize <span class='op'>=</span> <span class='fl'>500</span>,
  gsea.pvalue <span class='op'>=</span> <span class='fl'>0.05</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>counts.folder</th>
      <td><p>Folder contains all sample's count file. Count file should be SampleName.txt.</p></td>
    </tr>
    <tr>
      <th>count.matrix.file</th>
      <td><p>File contains count matrix, if provided, use this instead of <code>counts.folder</code>. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>meta.file</th>
      <td><p>File contains sample metadata.</p></td>
    </tr>
    <tr>
      <th>group.key</th>
      <td><p>Column in <code>meta.file</code> that represents sample group information. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>count.type</th>
      <td><p>The source of count file, chosen from htseq-count, featurecounts. Default: htseq-count.</p></td>
    </tr>
    <tr>
      <th>min.count</th>
      <td><p>A feature is considered to be detected if the corresponding number of read counts is &gt; <code>min.count</code>. By default, <code>min.count</code> = 10.</p></td>
    </tr>
    <tr>
      <th>ref.group</th>
      <td><p>Reference group name. When set NULL, select first element of groups. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>out.folder</th>
      <td><p>Folder to save enrichment results. Default: wording directory.</p></td>
    </tr>
    <tr>
      <th>data.type</th>
      <td><p>Input data type, choose from RNA, ChIP, ATAC. Default: RNA.</p></td>
    </tr>
    <tr>
      <th>peak.anno.key</th>
      <td><p>Peak location, chosen from "Promoter", "5' UTR", "3' UTR", "Exon", "Intron", "Downstream", "Distal Intergenic","All". Default: "Promoter".</p></td>
    </tr>
    <tr>
      <th>qc.ndepth</th>
      <td><p>Number of different sequencing depths to be simulated and plotted apart from the real depth. Default: 10. This parameter is only used by type "saturation".</p></td>
    </tr>
    <tr>
      <th>transform.method</th>
      <td><p>Data transformation methods, chosen from rlog, vst and ntd. Default: rlog.</p></td>
    </tr>
    <tr>
      <th>var.genes</th>
      <td><p>Select genes with larger variance for PCA analysis. Default: all genes.</p></td>
    </tr>
    <tr>
      <th>batch</th>
      <td><p>Batch column to conduct batch correction. Default value is NULL, do not conduct batch correction.</p></td>
    </tr>
    <tr>
      <th>outlier.detection</th>
      <td><p>Logical value. If TRUE, conduct outlier detection with robust PCA.</p></td>
    </tr>
    <tr>
      <th>rpca.method</th>
      <td><p>robust PCA method, chosen from <code>PcaGrid</code>, <code>PcaHubert</code>. Default: PcaGrid.</p></td>
    </tr>
    <tr>
      <th>k</th>
      <td><p>number of principal components to compute, for <code>PcaGrid</code>, <code>PcaHubert</code>. Default: 2.</p></td>
    </tr>
    <tr>
      <th>pca.x</th>
      <td><p>The principal component to display on the x axis. Default: PC1.</p></td>
    </tr>
    <tr>
      <th>pca.y</th>
      <td><p>The principal component to display on the y axis. Default: PC2.</p></td>
    </tr>
    <tr>
      <th>pca.z</th>
      <td><p>The principal component to display on the z axis. Default: PC3.</p></td>
    </tr>
    <tr>
      <th>loding.pc</th>
      <td><p>Specify PC to create loding plot. Default: 1:5.</p></td>
    </tr>
    <tr>
      <th>loading.gene.num</th>
      <td><p>Specify gene number of PC to create loding plot. Default: 10.</p></td>
    </tr>
    <tr>
      <th>loading.ncol</th>
      <td><p>The columns of loading bar or heatmap. Default: 2.</p></td>
    </tr>
    <tr>
      <th>enrich.loading.pc</th>
      <td><p>Specify PC to conduct enrichment analysis. Default: 1:5.</p></td>
    </tr>
    <tr>
      <th>enrich.loading.gene</th>
      <td><p>Specify gene number of PC to conduct enrichment analysis. Default: 200.</p></td>
    </tr>
    <tr>
      <th>gene.type</th>
      <td><p>Gene name type. Chosen from ENSEMBL, ENTREZID,SYMBOL. Default: ENSEMBL.</p></td>
    </tr>
    <tr>
      <th>enrich.type</th>
      <td><p>Enrichment type, chosen from ALL, GO, KEGG. Default: ALL.</p></td>
    </tr>
    <tr>
      <th>go.type</th>
      <td><p>GO enrichment type, chosen from ALL, BP, MF, CC. Default: ALL.</p></td>
    </tr>
    <tr>
      <th>enrich.pvalue</th>
      <td><p>Cutoff value of pvalue. Default: 0.05.</p></td>
    </tr>
    <tr>
      <th>enrich.qvalue</th>
      <td><p>Cutoff value of qvalue. Default: 0.05.</p></td>
    </tr>
    <tr>
      <th>org.db</th>
      <td><p>Organism database. Default: org.Mm.eg.db.</p></td>
    </tr>
    <tr>
      <th>organism</th>
      <td><p>Supported organism listed in 'http://www.genome.jp/kegg/catalog/org_list.html'. Default: mmu.</p></td>
    </tr>
    <tr>
      <th>padj.method</th>
      <td><p>One of "holm", "hochberg", "hommel", "bonferroni", "BH", "BY", "fdr", "none". Default: BH.</p></td>
    </tr>
    <tr>
      <th>show.term</th>
      <td><p>Number of enrichment term to show. Default: 15.</p></td>
    </tr>
    <tr>
      <th>str.width</th>
      <td><p>Length of enrichment term in plot. Default: 30.</p></td>
    </tr>
    <tr>
      <th>signif</th>
      <td><p>Significance criterion. For DESeq2 results, can be chosen from padj, pvalue. For edgeR results, can be chosen from FDR, PValue. Default: padj.</p></td>
    </tr>
    <tr>
      <th>signif.threshold</th>
      <td><p>Significance threshold to get differentially expressed genes or accessible/binding peaks. Default: 0.05.</p></td>
    </tr>
    <tr>
      <th>l2fc.threshold</th>
      <td><p>Log2 fold change threshold to get differentially expressed genes or accessible/binding peaks. Default: 1.</p></td>
    </tr>
    <tr>
      <th>gene.map</th>
      <td><p>Use data frame instead of <code>org.db</code> to conduct gene id conversion, first column should be Gene. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>gtf.file</th>
      <td><p>Gene annotation file used to get gene length, used if <code>norm.type=="RPKM"</code> or <code>norm.type=="TPM"</code>. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>norm.type</th>
      <td><p>Normalization method, chosen from DESeq2, TMM, CPM, RPKM, TPM. Default: DESeq2.</p></td>
    </tr>
    <tr>
      <th>log.counts</th>
      <td><p>Logical value, if TRUE, export log2(normalized.counts + 1), else export normalized.counts. Default: TRUE.</p></td>
    </tr>
    <tr>
      <th>deg.label.df</th>
      <td><p>Label data frame, at least contains Gene column. Default: NULL. When set NULL, use <code>deg.label.num</code>. When provided,
the second column should not be SYMBOL, ENSEMBL, ENTREZID.</p></td>
    </tr>
    <tr>
      <th>deg.label.key</th>
      <td><p>Which column to use as label. Default: NULL (use Gene column of <code>deg.label.df</code>).</p></td>
    </tr>
    <tr>
      <th>deg.label.num</th>
      <td><p>Gene number to label, choose according to log2FoldChange. When <code>deg.label.df</code> is set NULL, use this to determine genes to label. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>deg.label.color</th>
      <td><p>Color vector for labels. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>fe.gene.key</th>
      <td><p>Column name to conduct enrichment analysis. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>gmt.file</th>
      <td><p>Gene Matrix Transposed file format.</p></td>
    </tr>
    <tr>
      <th>gene.sets</th>
      <td><p>Gene sets information, containing two columns: gs_name, entrez_gene. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>minGSSize</th>
      <td><p>Minimal size of each geneSet for analyzing. Default: 10.</p></td>
    </tr>
    <tr>
      <th>maxGSSize</th>
      <td><p>Maximal size of genes annotated for testing. Default: 500.</p></td>
    </tr>
    <tr>
      <th>gsea.pvalue</th>
      <td><p>Cutoff value of pvalue. Default: 0.05.</p></td>
    </tr>
    </table>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># library(DESeq2)</span>
<span class='co'># library(DEbPeak)</span>
<span class='co'># count.file &lt;- system.file("extdata", "snon_count.txt", package = "DEbPeak")</span>
<span class='co'># meta.file &lt;- system.file("extdata", "snon_meta.txt", package = "DEbPeak")</span>
<span class='co'># gmt.file &lt;- system.file("extdata", "m5.go.bp.v2022.1.Mm.entrez.gmt", package = "DEbPeak")</span>
<span class='co'># ConductDESeq2(count.matrix.file = count.file, meta.file = meta.file, group.key = "condition",</span>
<span class='co'>#               count.type = "htseq-count", ref.group = "WT", signif = "pvalue", l2fc.threshold = 0.3, gmt.file = gmt.file)</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


