<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Integrate Differential Expression Results with Peak Annotation/Differential Expression Results. — DEbPeak • DEbPeak</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Integrate Differential Expression Results with Peak Annotation/Differential Expression Results. — DEbPeak" />
<meta property="og:description" content="Integrate Differential Expression Results with Peak Annotation/Differential Expression Results." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.4.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ParseGEO.html">Parse GEO</a>
    </li>
    <li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifAnalysis.html">Motif Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IdentifyRegulator.html">Identify regulators (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPDE.html">Integrate RNA-seq with ChIP-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Integrate Differential Expression Results with Peak Annotation/Differential Expression Results.</h1>
    
    <div class="hidden name"><code>DEbPeak.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Integrate Differential Expression Results with Peak Annotation/Differential Expression Results.</p>
    </div>

    <pre class="usage"><span class='fu'>DEbPeak</span><span class='op'>(</span>
  <span class='va'>de.res</span>,
  <span class='va'>peak.res</span>,
  peak.mode <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"consensus"</span>, <span class='st'>"diff"</span><span class='op'>)</span>,
  peak.anno.key <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Promoter"</span>, <span class='st'>"5' UTR"</span>, <span class='st'>"3' UTR"</span>, <span class='st'>"Exon"</span>, <span class='st'>"Intron"</span>, <span class='st'>"Downstream"</span>,
    <span class='st'>"Distal Intergenic"</span>, <span class='st'>"All"</span><span class='op'>)</span>,
  signif <span class='op'>=</span> <span class='st'>"padj"</span>,
  signif.threshold <span class='op'>=</span> <span class='fl'>0.05</span>,
  l2fc.threshold <span class='op'>=</span> <span class='fl'>1</span>,
  label.key <span class='op'>=</span> <span class='cn'>NULL</span>,
  peak.signif <span class='op'>=</span> <span class='st'>"padj"</span>,
  peak.signif.threshold <span class='op'>=</span> <span class='fl'>0.05</span>,
  peak.l2fc.threshold <span class='op'>=</span> <span class='fl'>1</span>,
  merge.key <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"geneId"</span>, <span class='st'>"ENSEMBL"</span>, <span class='st'>"SYMBOL"</span><span class='op'>)</span>,
  species <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Human"</span>, <span class='st'>"Mouse"</span>, <span class='st'>"Rat"</span>, <span class='st'>"Fly"</span>, <span class='st'>"Arabidopsis"</span>, <span class='st'>"Yeast"</span>, <span class='st'>"Zebrafish"</span>,
    <span class='st'>"Worm"</span>, <span class='st'>"Bovine"</span>, <span class='st'>"Pig"</span>, <span class='st'>"Chicken"</span>, <span class='st'>"Rhesus"</span>, <span class='st'>"Canine"</span>, <span class='st'>"Xenopus"</span>, <span class='st'>"Anopheles"</span>,
    <span class='st'>"Chimp"</span>, <span class='st'>"E coli strain Sakai"</span>, <span class='st'>"Myxococcus xanthus DK 1622"</span><span class='op'>)</span>,
  enhancer <span class='op'>=</span> <span class='cn'>FALSE</span>,
  seq.style <span class='op'>=</span> <span class='st'>"Ensembl"</span>,
  gtf.file <span class='op'>=</span> <span class='cn'>NULL</span>,
  dis.threshold <span class='op'>=</span> <span class='fl'>250000</span>,
  n.cores <span class='op'>=</span> <span class='fl'>1</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>de.res</th>
      <td><p>Dataframe contains all genes of differential expression analysis.</p></td>
    </tr>
    <tr>
      <th>peak.res</th>
      <td><p>Dataframe contains all peak annotation (<code>peak.mode</code> is consensus) or
differential analysis results of peak-related data (<code>peak.mode</code> is diff).</p></td>
    </tr>
    <tr>
      <th>peak.mode</th>
      <td><p>The source of peak results, choose from consensus (peak annotation) and diff (differential analysis).
Default: consensus.</p></td>
    </tr>
    <tr>
      <th>peak.anno.key</th>
      <td><p>Peak location, chosen from "Promoter", "5' UTR", "3' UTR", "Exon", "Intron", "Downstream", "Distal Intergenic","All".
Used when <code>peak.mode</code> is consensus. Default: "Promoter".</p></td>
    </tr>
    <tr>
      <th>signif</th>
      <td><p>Significance criterion for RNA-seq results. For DESeq2 results, can be chosen from padj, pvalue.
For edgeR results, can be chosen from FDR, PValue. Default: padj.</p></td>
    </tr>
    <tr>
      <th>signif.threshold</th>
      <td><p>Significance threshold for RNA-seq to get differentially expressed genes. Default: 0.05.</p></td>
    </tr>
    <tr>
      <th>l2fc.threshold</th>
      <td><p>Log2 fold change threshold for RNA-seq to get differentially expressed genes. Default: 1.</p></td>
    </tr>
    <tr>
      <th>label.key</th>
      <td><p>Which column to use as label. Default: NULL (use rownames of <code>de.res</code>).</p></td>
    </tr>
    <tr>
      <th>peak.signif</th>
      <td><p>Significance criterion for peak-associated results. For DESeq2 results, can be chosen from padj, pvalue.
For edgeR results, can be chosen from FDR, PValue. Default: padj.</p></td>
    </tr>
    <tr>
      <th>peak.signif.threshold</th>
      <td><p>Significance threshold for peak-associated results to get differentially accessible/binding peaks. Default: 0.05.</p></td>
    </tr>
    <tr>
      <th>peak.l2fc.threshold</th>
      <td><p>Log2 fold change threshold for peak-associated results to get differentially accessible/binding peaks. Default: 1.</p></td>
    </tr>
    <tr>
      <th>merge.key</th>
      <td><p>The columns used for merging, chosen from geneId (ENTREZID), ENSEMBL, SYMBOL. Default: geneId.</p></td>
    </tr>
    <tr>
      <th>species</th>
      <td><p>Species used, chosen from "Human","Mouse","Rat","Fly","Arabidopsis",
"Yeast","Zebrafish","Worm","Bovine","Pig","Chicken","Rhesus",
"Canine","Xenopus","Anopheles","Chimp","E coli strain Sakai","Myxococcus xanthus DK 1622". Default: "Human".</p></td>
    </tr>
    <tr>
      <th>enhancer</th>
      <td><p>Logical value. Whether the data is enhancer-related data. Default: FALSE.</p></td>
    </tr>
    <tr>
      <th>seq.style</th>
      <td><p>The style of sequence, chosen from UCSC, NCBI, Ensembl, None. This should be compatible with the genome and gtf file you used
to generate count matrix and peak files. Default: "UCSC".</p></td>
    </tr>
    <tr>
      <th>gtf.file</th>
      <td><p>GTF file used to create TxDb object. Useful when specie you used is not available in <code>species</code>. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>dis.threshold</th>
      <td><p>Distance threshold. Default: 250000.</p></td>
    </tr>
    <tr>
      <th>n.cores</th>
      <td><p>The number of cores to be used for this job. Default:1.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Dataframe contains integrated results. The results will contain five categories: UPbPeak, DOWNbPeak, UP, DOWN, Peak when <code>peak.mode</code> is consensus
and Down_Up, Up_Up, Down_Down, Up_Down when <code>peak.mode</code> is diff. RNA in front.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>DEbPeak</span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/mikelove/DESeq2'>DESeq2</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://ycphs.github.io/openxlsx/index.html'>openxlsx</a></span><span class='op'>)</span>
<span class='co'>### consensus mode</span>
<span class='co'># ChIP-Seq data</span>
<span class='va'>peak.file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"debchip_peaks.bed"</span>, package <span class='op'>=</span> <span class='st'>"DEbPeak"</span><span class='op'>)</span>
<span class='va'>peak.df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='GetConsensusPeak.html'>GetConsensusPeak</a></span><span class='op'>(</span>peak.file <span class='op'>=</span> <span class='va'>peak.file</span><span class='op'>)</span>
<span class='va'>peak.profile</span> <span class='op'>&lt;-</span> <span class='fu'><a href='PeakProfile.html'>PeakProfile</a></span><span class='op'>(</span><span class='va'>peak.df</span>, species <span class='op'>=</span> <span class='st'>"Mouse"</span>, by <span class='op'>=</span> <span class='st'>"gene"</span>, region.type <span class='op'>=</span> <span class='st'>"body"</span>, nbin <span class='op'>=</span> <span class='fl'>800</span><span class='op'>)</span>
</div><div class='output co'>#&gt; &gt;&gt; preparing promoter regions...	 2023-07-02 17时12分20秒 
#&gt; &gt;&gt; preparing tag matrix...		 2023-07-02 17时12分20秒 
#&gt; &gt;&gt; preparing start_site regions by ... 2023-07-02 17时12分20秒
#&gt; &gt;&gt; preparing tag matrix...  2023-07-02 17时12分20秒 
#&gt; &gt;&gt; generating figure...		 2023-07-02 17时12分28秒 </div><div class='output co'>#&gt; &gt;&gt; done...			 2023-07-02 17时12分28秒 </div><div class='output co'>#&gt; &gt;&gt; binning method is used...2023-07-02 17时12分28秒
#&gt; &gt;&gt; preparing start_site regions by gene... 2023-07-02 17时12分28秒
#&gt; &gt;&gt; preparing tag matrix by binning...  2023-07-02 17时12分28秒 
#&gt; &gt;&gt; Running bootstrapping for tag matrix...		 2023-07-02 17时12分37秒 
#&gt; &gt;&gt; binning method is used...2023-07-02 17时12分37秒
#&gt; &gt;&gt; preparing body regions by gene... 2023-07-02 17时12分37秒
#&gt; &gt;&gt; preparing tag matrix by binning...  2023-07-02 17时12分37秒 
#&gt; &gt;&gt; preparing matrix with extension from (TSS-20%)~(TTS+20%)... 2023-07-02 17时12分37秒
#&gt; &gt;&gt; 1 peaks(0.1536098%), having lengths smaller than 800bp, are filtered... 2023-07-02 17时12分40秒
#&gt; &gt;&gt; Running bootstrapping for tag matrix...		 2023-07-02 17时13分21秒 </div><div class='input'><span class='va'>peak.anno</span> <span class='op'>&lt;-</span> <span class='fu'><a href='AnnoPeak.html'>AnnoPeak</a></span><span class='op'>(</span>
  peak.df <span class='op'>=</span> <span class='va'>peak.df</span>, species <span class='op'>=</span> <span class='st'>"Mouse"</span>,
  seq.style <span class='op'>=</span> <span class='st'>"UCSC"</span>, up.dist <span class='op'>=</span> <span class='fl'>20000</span>, down.dist <span class='op'>=</span> <span class='fl'>20000</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; &gt;&gt; preparing features information...		 2023-07-02 17时13分22秒 
#&gt; &gt;&gt; identifying nearest features...		 2023-07-02 17时13分22秒 
#&gt; &gt;&gt; calculating distance from peak to TSS...	 2023-07-02 17时13分22秒 
#&gt; &gt;&gt; assigning genomic annotation...		 2023-07-02 17时13分22秒 
#&gt; &gt;&gt; adding gene annotation...			 2023-07-02 17时13分25秒 </div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; &gt;&gt; assigning chromosome lengths			 2023-07-02 17时13分25秒 
#&gt; &gt;&gt; done...					 2023-07-02 17时13分25秒 </div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 6 rows containing non-finite values (`stat_count()`).</span></div><div class='input'><span class='co'># RNA-Seq data</span>
<span class='va'>count.file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"debchip_count.txt"</span>, package <span class='op'>=</span> <span class='st'>"DEbPeak"</span><span class='op'>)</span>
<span class='va'>meta.file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"debchip_meta.txt"</span>, package <span class='op'>=</span> <span class='st'>"DEbPeak"</span><span class='op'>)</span>
<span class='va'>count.matrix</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span>file <span class='op'>=</span> <span class='va'>count.file</span>, header <span class='op'>=</span> <span class='cn'>TRUE</span>, sep <span class='op'>=</span> <span class='st'>"\t"</span><span class='op'>)</span>
<span class='va'>meta.info</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span>file <span class='op'>=</span> <span class='va'>meta.file</span>, header <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='co'># create DESeqDataSet object</span>
<span class='va'>dds</span> <span class='op'>&lt;-</span> <span class='fu'>DESeq2</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html'>DESeqDataSetFromMatrix</a></span><span class='op'>(</span>
  countData <span class='op'>=</span> <span class='va'>count.matrix</span>, colData <span class='op'>=</span> <span class='va'>meta.info</span>,
  design <span class='op'>=</span> <span class='op'>~</span><span class='va'>condition</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: some variables in design formula are characters, converting to factors</span></div><div class='input'><span class='co'># set control level</span>
<span class='va'>dds</span><span class='op'>$</span><span class='va'>condition</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/relevel.html'>relevel</a></span><span class='op'>(</span><span class='va'>dds</span><span class='op'>$</span><span class='va'>condition</span>, ref <span class='op'>=</span> <span class='st'>"NF"</span><span class='op'>)</span>
<span class='co'># conduct differential expressed genes analysis</span>
<span class='va'>dds</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/DESeq2/man/DESeq.html'>DESeq</a></span><span class='op'>(</span><span class='va'>dds</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>estimating size factors</span></div><div class='output co'>#&gt; <span class='message'>estimating dispersions</span></div><div class='output co'>#&gt; <span class='message'>gene-wise dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>mean-dispersion relationship</span></div><div class='output co'>#&gt; <span class='message'>final dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>fitting model and testing</span></div><div class='input'><span class='co'># extract results</span>
<span class='va'>dds.results</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/DESeq2/man/results.html'>results</a></span><span class='op'>(</span><span class='va'>dds</span>, contrast <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"condition"</span>, <span class='st'>"RX"</span>, <span class='st'>"NF"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>dds.results.ordered</span> <span class='op'>&lt;-</span> <span class='va'>dds.results</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/order.html'>order</a></span><span class='op'>(</span><span class='va'>dds.results</span><span class='op'>$</span><span class='va'>log2FoldChange</span>, decreasing <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='co'># Integrated with RNA-Seq</span>
<span class='va'>debchip.res</span> <span class='op'>&lt;-</span> <span class='fu'>DEbPeak</span><span class='op'>(</span>
  de.res <span class='op'>=</span> <span class='va'>dds.results.ordered</span>, peak.res <span class='op'>=</span> <span class='va'>peak.anno</span><span class='op'>[[</span><span class='st'>"df"</span><span class='op'>]</span><span class='op'>]</span>,
  peak.anno.key <span class='op'>=</span> <span class='st'>"Promoter"</span>, merge.key <span class='op'>=</span> <span class='st'>"SYMBOL"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Differential expression analysis with DESeq2!</span></div><div class='input'><span class='co'>### diff mode</span>
<span class='co'># ATAC-Seq data</span>
<span class='va'>peak.matrix.file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"RA_ATAC_count.txt"</span>, package <span class='op'>=</span> <span class='st'>"DEbPeak"</span><span class='op'>)</span>
<span class='va'>peak.meta.file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"RA_ATAC_meta.txt"</span>, package <span class='op'>=</span> <span class='st'>"DEbPeak"</span><span class='op'>)</span>
<span class='va'>peak.matrix</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span>file <span class='op'>=</span> <span class='va'>peak.matrix.file</span>, header <span class='op'>=</span> <span class='cn'>TRUE</span>, sep <span class='op'>=</span> <span class='st'>"\t"</span><span class='op'>)</span>
<span class='va'>peak.meta</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span>file <span class='op'>=</span> <span class='va'>peak.meta.file</span>, header <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>dds.peak</span> <span class='op'>&lt;-</span> <span class='fu'>DESeq2</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html'>DESeqDataSetFromMatrix</a></span><span class='op'>(</span>
  countData <span class='op'>=</span> <span class='va'>peak.matrix</span>, colData <span class='op'>=</span> <span class='va'>peak.meta</span>,
  design <span class='op'>=</span> <span class='op'>~</span><span class='va'>condition</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: some variables in design formula are characters, converting to factors</span></div><div class='input'><span class='co'># set control level</span>
<span class='va'>dds.peak</span><span class='op'>$</span><span class='va'>condition</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/relevel.html'>relevel</a></span><span class='op'>(</span><span class='va'>dds.peak</span><span class='op'>$</span><span class='va'>condition</span>, ref <span class='op'>=</span> <span class='st'>"WT"</span><span class='op'>)</span>
<span class='co'># conduct differential expressed genes analysis</span>
<span class='va'>dds.peak</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/DESeq2/man/DESeq.html'>DESeq</a></span><span class='op'>(</span><span class='va'>dds.peak</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>estimating size factors</span></div><div class='output co'>#&gt; <span class='message'>estimating dispersions</span></div><div class='output co'>#&gt; <span class='message'>gene-wise dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>mean-dispersion relationship</span></div><div class='output co'>#&gt; <span class='message'>-- note: fitType='parametric', but the dispersion trend was not well captured by the</span>
#&gt; <span class='message'>   function: y = a/x + b, and a local regression fit was automatically substituted.</span>
#&gt; <span class='message'>   specify fitType='local' or 'mean' to avoid this message next time.</span></div><div class='output co'>#&gt; <span class='message'>final dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>fitting model and testing</span></div><div class='input'><span class='co'># extract results</span>
<span class='va'>dds.peak.results</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/DESeq2/man/results.html'>results</a></span><span class='op'>(</span><span class='va'>dds.peak</span>, contrast <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"condition"</span>, <span class='st'>"KO"</span>, <span class='st'>"WT"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>dds.peak.results.ordered</span> <span class='op'>&lt;-</span> <span class='va'>dds.peak.results</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/order.html'>order</a></span><span class='op'>(</span><span class='va'>dds.peak.results</span><span class='op'>$</span><span class='va'>log2FoldChange</span>, decreasing <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>, <span class='op'>]</span>
<span class='co'># RNA-seq results</span>
<span class='co'># rna.diff.file &lt;- system.file("extdata", "RA_RNA_diff.xlsx", package = "DEbPeak")</span>
<span class='va'>rna.diff.file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"RA_RNA_diff.txt"</span>, package <span class='op'>=</span> <span class='st'>"DEbPeak"</span><span class='op'>)</span>
<span class='co'># rna.diff &lt;- openxlsx::read.xlsx(xlsxFile = rna.diff.file, rowNames = T, check.names = FALSE)</span>
<span class='va'>rna.diff</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span>file <span class='op'>=</span> <span class='va'>rna.diff.file</span>, header <span class='op'>=</span> <span class='cn'>TRUE</span>, sep <span class='op'>=</span> <span class='st'>"\t"</span><span class='op'>)</span>
<span class='co'># integrate differential expression results of RNA-seq and ATAC-seq</span>
<span class='va'>debatac.res</span> <span class='op'>&lt;-</span> <span class='fu'>DEbPeak</span><span class='op'>(</span>
  de.res <span class='op'>=</span> <span class='va'>rna.diff</span>, peak.res <span class='op'>=</span> <span class='va'>dds.peak.results.ordered</span>, peak.mode <span class='op'>=</span> <span class='st'>"diff"</span>, peak.anno.key <span class='op'>=</span> <span class='st'>"All"</span>, l2fc.threshold <span class='op'>=</span> <span class='fl'>0</span>,
  peak.l2fc.threshold <span class='op'>=</span> <span class='fl'>0</span>, species <span class='op'>=</span> <span class='st'>"Mouse"</span>, merge.key <span class='op'>=</span> <span class='st'>"SYMBOL"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Differential expression analysis with DESeq2!</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='warning'>Warning: 1.63% of input gene IDs are fail to map...</span></div><div class='output co'>#&gt; <span class='message'>Differential expression analysis with DESeq2!</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='warning'>Warning: 13.86% of input gene IDs are fail to map...</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


