<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Integrate Two Peak Annotation/Differential Analysis Results. — PeakbPeak • DEbPeak</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Integrate Two Peak Annotation/Differential Analysis Results. — PeakbPeak" />
<meta property="og:description" content="Integrate Two Peak Annotation/Differential Analysis Results." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Integrate Two Peak Annotation/Differential Analysis Results.</h1>
    
    <div class="hidden name"><code>PeakbPeak.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Integrate Two Peak Annotation/Differential Analysis Results.</p>
    </div>

    <pre class="usage"><span class='fu'>PeakbPeak</span><span class='op'>(</span>
  <span class='va'>peak1.res</span>,
  <span class='va'>peak2.res</span>,
  peak.mode <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"consensus"</span>, <span class='st'>"diff"</span><span class='op'>)</span>,
  peak.anno.key <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Promoter"</span>, <span class='st'>"5' UTR"</span>, <span class='st'>"3' UTR"</span>, <span class='st'>"Exon"</span>, <span class='st'>"Intron"</span>, <span class='st'>"Downstream"</span>,
    <span class='st'>"Distal Intergenic"</span>, <span class='st'>"All"</span><span class='op'>)</span>,
  peak1.signif <span class='op'>=</span> <span class='st'>"padj"</span>,
  peak1.signif.threshold <span class='op'>=</span> <span class='fl'>0.05</span>,
  peak1.l2fc.threshold <span class='op'>=</span> <span class='fl'>1</span>,
  peak2.signif <span class='op'>=</span> <span class='st'>"padj"</span>,
  peak2.signif.threshold <span class='op'>=</span> <span class='fl'>0.05</span>,
  peak2.l2fc.threshold <span class='op'>=</span> <span class='fl'>1</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>peak1.res</th>
      <td><p>Peak1 dataframe contains all peak annotation (<code>peak.mode</code> is consensus) or
differential analysis results of peak-related data (<code>peak.mode</code> is diff).</p></td>
    </tr>
    <tr>
      <th>peak2.res</th>
      <td><p>Peak2 dataframe contains all peak annotation (<code>peak.mode</code> is consensus) or
differential analysis results of peak-related data (<code>peak.mode</code> is diff).</p></td>
    </tr>
    <tr>
      <th>peak.mode</th>
      <td><p>The source of peak results, choose from consensus (peak annotation) and diff (differential analysis).
Default: consensus.</p></td>
    </tr>
    <tr>
      <th>peak.anno.key</th>
      <td><p>Peak location, chosen from "Promoter", "5' UTR", "3' UTR", "Exon", "Intron", "Downstream", "Distal Intergenic","All".
Used when <code>peak.mode</code> is consensus. Default: "Promoter".</p></td>
    </tr>
    <tr>
      <th>peak1.signif</th>
      <td><p>Used when <code>peak.mode</code> is diff. Significance criterion for peak-associated results for Peak1. For DESeq2 results, can be chosen from padj, pvalue.
For edgeR results, can be chosen from FDR, PValue. Default: padj.</p></td>
    </tr>
    <tr>
      <th>peak1.signif.threshold</th>
      <td><p>Used when <code>peak.mode</code> is diff. Significance threshold for peak-associated results to get differentially accessible/binding peaks for Peak1. Default: 0.05.</p></td>
    </tr>
    <tr>
      <th>peak1.l2fc.threshold</th>
      <td><p>Used when <code>peak.mode</code> is diff. Log2 fold change threshold for peak-associated results to get differentially accessible/binding peaks for Peak1. Default: 1.</p></td>
    </tr>
    <tr>
      <th>peak2.signif</th>
      <td><p>Used when <code>peak.mode</code> is diff. Significance criterion for peak-associated results for Peak2. For DESeq2 results, can be chosen from padj, pvalue.
For edgeR results, can be chosen from FDR, PValue. Default: padj.</p></td>
    </tr>
    <tr>
      <th>peak2.signif.threshold</th>
      <td><p>Used when <code>peak.mode</code> is diff. Significance threshold for peak-associated results to get differentially accessible/binding peaks for Peak2. Default: 0.05.</p></td>
    </tr>
    <tr>
      <th>peak2.l2fc.threshold</th>
      <td><p>Used when <code>peak.mode</code> is diff. Log2 fold change threshold for peak-associated results to get differentially accessible/binding peaks for Peak2. Default: 1.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Dataframe contains integration results. When <code>peak.mode</code> is 'diff', the 'Type' column contains "Down_Up", "Up_Up", "Down_Down", "Up_Down", "Peak1_Up", "Peak1_Down", "Peak2_Up", "Peak2_Down".
When <code>peak.mode</code> is 'consensus', the 'Type' column contains "Common", "Peak1", "Peak2".</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>DEbPeak</span><span class='op'>)</span>
<span class='co'># ChIP-seq data</span>
<span class='va'>chip.file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"debchip_peaks.bed"</span>, package <span class='op'>=</span> <span class='st'>"DEbPeak"</span><span class='op'>)</span>
<span class='va'>chip.df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='GetConsensusPeak.html'>GetConsensusPeak</a></span><span class='op'>(</span>peak.file <span class='op'>=</span> <span class='va'>chip.file</span><span class='op'>)</span>
<span class='va'>chip.anno</span> <span class='op'>&lt;-</span> <span class='fu'><a href='AnnoPeak.html'>AnnoPeak</a></span><span class='op'>(</span>
  peak.df <span class='op'>=</span> <span class='va'>chip.df</span>, species <span class='op'>=</span> <span class='st'>"Mouse"</span>,
  seq.style <span class='op'>=</span> <span class='st'>"UCSC"</span>, up.dist <span class='op'>=</span> <span class='fl'>20000</span>, down.dist <span class='op'>=</span> <span class='fl'>20000</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; &gt;&gt; preparing features information...		 2023-04-10 22时53分23秒 
#&gt; &gt;&gt; identifying nearest features...		 2023-04-10 22时53分23秒 
#&gt; &gt;&gt; calculating distance from peak to TSS...	 2023-04-10 22时53分23秒 
#&gt; &gt;&gt; assigning genomic annotation...		 2023-04-10 22时53分23秒 
#&gt; &gt;&gt; adding gene annotation...			 2023-04-10 22时53分25秒 </div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; &gt;&gt; assigning chromosome lengths			 2023-04-10 22时53分25秒 
#&gt; &gt;&gt; done...					 2023-04-10 22时53分25秒 </div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 6 rows containing non-finite values (stat_count).</span></div><div class='input'><span class='co'># ATAC-seq data</span>
<span class='va'>atac.file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"debatac_peaks.bed"</span>, package <span class='op'>=</span> <span class='st'>"DEbPeak"</span><span class='op'>)</span>
<span class='va'>atac.df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='GetConsensusPeak.html'>GetConsensusPeak</a></span><span class='op'>(</span>peak.file <span class='op'>=</span> <span class='va'>atac.file</span><span class='op'>)</span>
<span class='va'>atac.anno</span> <span class='op'>&lt;-</span> <span class='fu'><a href='AnnoPeak.html'>AnnoPeak</a></span><span class='op'>(</span>
  peak.df <span class='op'>=</span> <span class='va'>atac.df</span>, species <span class='op'>=</span> <span class='st'>"Mouse"</span>,
  seq.style <span class='op'>=</span> <span class='st'>"UCSC"</span>, up.dist <span class='op'>=</span> <span class='fl'>20000</span>, down.dist <span class='op'>=</span> <span class='fl'>20000</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; &gt;&gt; preparing features information...		 2023-04-10 22时53分26秒 
#&gt; &gt;&gt; identifying nearest features...		 2023-04-10 22时53分26秒 
#&gt; &gt;&gt; calculating distance from peak to TSS...	 2023-04-10 22时53分27秒 
#&gt; &gt;&gt; assigning genomic annotation...		 2023-04-10 22时53分27秒 
#&gt; &gt;&gt; adding gene annotation...			 2023-04-10 22时53分28秒 </div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; &gt;&gt; assigning chromosome lengths			 2023-04-10 22时53分29秒 
#&gt; &gt;&gt; done...					 2023-04-10 22时53分29秒 </div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 23 rows containing non-finite values (stat_count).</span></div><div class='input'><span class='co'># integrate</span>
<span class='va'>chip.atac</span> <span class='op'>&lt;-</span> <span class='fu'>PeakbPeak</span><span class='op'>(</span>peak1.res <span class='op'>=</span> <span class='va'>chip.anno</span><span class='op'>$</span><span class='va'>df</span>, peak2.res <span class='op'>=</span> <span class='va'>atac.anno</span><span class='op'>$</span><span class='va'>df</span>, peak.mode <span class='op'>=</span> <span class='st'>"consensus"</span>, peak.anno.key <span class='op'>=</span> <span class='st'>"Promoter"</span><span class='op'>)</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


