<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Prepare Count Matrix and Sample Metadata for Peak-related data. — PeakMatrix • DEbPeak</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Prepare Count Matrix and Sample Metadata for Peak-related data. — PeakMatrix" />
<meta property="og:description" content="Prepare Count Matrix and Sample Metadata for Peak-related data." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEbPeak</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEbPeak_All.html">Getting Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ParseGEO.html">Parse GEO</a>
    </li>
    <li>
      <a href="../articles/QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysis.html">Principal Component Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/PrincipalComponentAnalysisPeak.html">Principal Component Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysis.html">Differential Expression Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/DifferentialExpressionAnalysisPeak.html">Differential Expression Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysis.html">Functional Enrichment Analysis (RNA-seq)</a>
    </li>
    <li>
      <a href="../articles/FunctionalEnrichmentAnalysisPeak.html">Functional Enrichment Analysis (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/MotifEnrichment.html">Motif Enrichment (Peak-related)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIP.html">Integrate RNA-seq with ChIP-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATAC.html">Integrate RNA-seq with ATAC-seq (consensus peak)</a>
    </li>
    <li>
      <a href="../articles/IntegrateATACDE.html">Integrate RNA-seq with ATAC-seq (differential analysis)</a>
    </li>
    <li>
      <a href="../articles/IntegrateChIPATAC.html">Integrate RNA-seq with ChIP-seq and ATAC-seq</a>
    </li>
    <li>
      <a href="../articles/Utils.html">Utils</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Change log</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/showteeth/DEbPeak">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prepare Count Matrix and Sample Metadata for Peak-related data.</h1>
    
    <div class="hidden name"><code>PeakMatrix.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Prepare Count Matrix and Sample Metadata for Peak-related data.</p>
    </div>

    <pre class="usage"><span class='fu'>PeakMatrix</span><span class='op'>(</span>
  <span class='va'>meta.file</span>,
  min.overlap <span class='op'>=</span> <span class='fl'>2</span>,
  submits <span class='op'>=</span> <span class='fl'>200</span>,
  use.summarizeOverlaps <span class='op'>=</span> <span class='cn'>TRUE</span>,
  blacklist <span class='op'>=</span> <span class='cn'>TRUE</span>,
  sub.control <span class='op'>=</span> <span class='cn'>TRUE</span>,
  used.cols <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"SampleID"</span>, <span class='st'>"Condition"</span><span class='op'>)</span>,
  out.folder <span class='op'>=</span> <span class='cn'>NULL</span>,
  species <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Human"</span>, <span class='st'>"Mouse"</span>, <span class='st'>"Rat"</span>, <span class='st'>"Fly"</span>, <span class='st'>"Arabidopsis"</span>, <span class='st'>"Yeast"</span>, <span class='st'>"Zebrafish"</span>,
    <span class='st'>"Worm"</span>, <span class='st'>"Bovine"</span>, <span class='st'>"Pig"</span>, <span class='st'>"Chicken"</span>, <span class='st'>"Rhesus"</span>, <span class='st'>"Canine"</span>, <span class='st'>"Xenopus"</span>, <span class='st'>"Anopheles"</span>,
    <span class='st'>"Chimp"</span>, <span class='st'>"E coli strain Sakai"</span>, <span class='st'>"Myxococcus xanthus DK 1622"</span><span class='op'>)</span>,
  seq.style <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"UCSC"</span>, <span class='st'>"NCBI"</span>, <span class='st'>"Ensembl"</span>, <span class='st'>"None"</span><span class='op'>)</span>,
  gtf.file <span class='op'>=</span> <span class='cn'>NULL</span>,
  up.dist <span class='op'>=</span> <span class='fl'>3000</span>,
  down.dist <span class='op'>=</span> <span class='fl'>3000</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>meta.file</th>
      <td><p>Sample metadata contains peak related information (eg: sample, peakPath, bamPath, condition).</p></td>
    </tr>
    <tr>
      <th>min.overlap</th>
      <td><p>Only include peaks in at least this many peaksets in the main binding matrix. Default: 2.
Parameter of <code>dba</code>.</p></td>
    </tr>
    <tr>
      <th>submits</th>
      <td><p>If the value is greater than zero, all consensus peaks will be re-centered around a consensus summit,
with the value of summits indicating how many base pairs to include upstream and downstream of the
summit (so all consensus peaks will be of the same width, namely <code>2 * summits + 1</code>).
Default: 200. Parameter of <code>dba.count</code>.</p></td>
    </tr>
    <tr>
      <th>use.summarizeOverlaps</th>
      <td><p>Logical value, indicating that <code>summarizeOverlaps</code> should be used for counting
instead of the built-in counting code. Default: TRUE. Parameter of <code>dba.count</code>.</p></td>
    </tr>
    <tr>
      <th>blacklist</th>
      <td><p>Species-specific abnormal regions to be removed. Choose from "DBA_BLACKLIST_HG19", "DBA_BLACKLIST_HG38", "DBA_BLACKLIST_GRCH37",
"DBA_BLACKLIST_GRCH38", "DBA_BLACKLIST_MM9", "DBA_BLACKLIST_MM10", "DBA_BLACKLIST_CE10", "DBA_BLACKLIST_CE11", "DBA_BLACKLIST_DM3", "DBA_BLACKLIST_DM6",
TRUE (auto-detection genome), a GRanges object containing the blacklisted regions. Default:TRUE. Parameter of <code>dba.blacklist</code>.</p></td>
    </tr>
    <tr>
      <th>sub.control</th>
      <td><p>Logical value, whether Control read counts are subtracted for each site in each sample.
Default: TRUE. Parameter of <code>dba.count</code>.</p></td>
    </tr>
    <tr>
      <th>used.cols</th>
      <td><p>Used columns used to create sample metadata. If specified, sampleID should be placed first. Default: c("SampleID", "Condition").</p></td>
    </tr>
    <tr>
      <th>out.folder</th>
      <td><p>Output folder to save created count matrix and sample metadata. Default: NULL (current working directory).</p></td>
    </tr>
    <tr>
      <th>species</th>
      <td><p>Species used, chosen from "Human","Mouse","Rat","Fly","Arabidopsis","Yeast","Zebrafish","Worm","Bovine","Pig","Chicken","Rhesus",
"Canine","Xenopus","Anopheles","Chimp","E coli strain Sakai","Myxococcus xanthus DK 1622". Default: "Human".</p></td>
    </tr>
    <tr>
      <th>seq.style</th>
      <td><p>The style of sequence, chosen from UCSC, NCBI, Ensembl, None. This should be compatible with the genome and gtf file you used
to generate count matrix and peak files. Default: "UCSC".</p></td>
    </tr>
    <tr>
      <th>gtf.file</th>
      <td><p>GTF file used to create TxDb object. Useful when specie you used is not available in <code>species</code>. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>up.dist</th>
      <td><p>The upstream distance from the TSS. Default: 3000bp.</p></td>
    </tr>
    <tr>
      <th>down.dist</th>
      <td><p>The downstream distance from the TSS. Default: 3000bp.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Parameters for <code>annotatePeak</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A dataframe contains count matrix, peak annotation and sample metadata (if provided <code>used.cols</code>).</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># library(DEbPeak)</span>
<span class='co'># library(DESeq2)</span>
<span class='co'># metadata file contains peak and bam information</span>
<span class='co'># meta.file = 'path/to/metadata'</span>
<span class='co'># PeakMatrix(meta.file = meta.file, species = "Human",  seq.style = "UCSC",</span>
<span class='co'>#            up.dist = 20000, down.dist = 20000)</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


